Chemical systems contain an enormous number atoms, which, in most cases makes it simply impossible for us to understand anything by monitoring the atom postions directly. Consquentially, we introduce Collective variables (C\+Vs) that describe the chemical processes we are interested in and monitor these simpler quantities instead. These C\+Vs are used in many of the methods implemented in P\+L\+U\+M\+E\+D -\/ there values can be monitored using \hyperlink{PRINT}{P\+R\+I\+N\+T}, \hyperlink{Function}{Functions} of them can be calculated or they can be analyzed or biased using the \hyperlink{_analysis}{Analysis} and \hyperlink{_bias}{Biasing} methods implemented in P\+L\+U\+M\+E\+D. Before doing any of these things however we first have to tell P\+L\+U\+M\+E\+D how to calculate them.

The simplest collective variables that are implemented in P\+L\+U\+M\+E\+D 2 take in a set of atomic positions and output one or multiple scalar C\+V values. Information on these variables is given on the page entitled \hyperlink{Colvar}{C\+V Documentation} while information as to how sets of atoms can be selected can be found in the pages on \hyperlink{Group}{Groups and Virtual Atoms}. Please be aware that P\+L\+U\+M\+E\+D contains implementations of many other collective variables but that the input for these variables may be less transparent when it is first encourntered. In particular, the page on \hyperlink{dists}{Distances from reference configurations} describes the various ways that you can calculate the distance from a particular reference configuration. So you will find instructions on how to calculate the R\+M\+S\+D distance from the folded state of a protein here. Meanwhile, the page on \hyperlink{Function}{Functions} describes the various functions of collective variables that can be used in the code. This is a very powerful feature of P\+L\+U\+M\+E\+D as you can use the \hyperlink{Function}{Functions} commands to calculate any function or combination of the simple collective variables listed on the page \hyperlink{Colvar}{C\+V Documentation}. Lastly the page on \hyperlink{mcolv}{Multi\+Colvar Documentation} describes Multi\+Colvars. Multi\+Colvars allow you to use many different colvars and allow us to implement all these collective variables without implementing having an unmanigiably large ammount of code. For some things (e.\+g. \hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S} G\+R\+O\+U\+P\+A=1 G\+R\+O\+U\+P\+B=2-\/100 L\+E\+S\+S\+\_\+\+T\+H\+A\+N=\{R\+A\+T\+I\+O\+N\+A\+L R\+\_\+0=3\}) there are more computationally efficient options available in plumed (e.\+g. \hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N}). However, Multi\+Colvars are worth investigating as they provide a flexible syntax for many quite-\/complex C\+Vs.


\begin{DoxyItemize}
\item \hyperlink{Group}{Groups and Virtual Atoms}
\item \hyperlink{Colvar}{C\+V Documentation}
\item \hyperlink{dists}{Distances from reference configurations}
\item \hyperlink{Function}{Functions}
\item \hyperlink{mcolv}{Multi\+Colvar Documentation} 
\end{DoxyItemize}\hypertarget{Group}{}\section{Groups and Virtual Atoms}\label{Group}
\hypertarget{_group_atomSpecs}{}\subsection{Specifying Atoms}\label{_group_atomSpecs}
The vast majority of the C\+Vs implemented in P\+L\+U\+M\+E\+D are calculated from a list of atom positions. Within P\+L\+U\+M\+E\+D atoms are specified using their numerical indices in the molecular dynamics input file.

In P\+L\+U\+M\+E\+D lists of atoms can be either provided directly inside the definition of each collective variable, or predefined as a \hyperlink{GROUP}{G\+R\+O\+U\+P} that can be reused multiple times. Lists of atoms can be written as\+:


\begin{DoxyItemize}
\item comma separated lists of numbers (G\+R\+O\+U\+P A\+T\+O\+M\+S=10,11,15,20 L\+A\+B\+E\+L=g1)
\item numerical ranges. So G\+R\+O\+U\+P A\+T\+O\+M\+S=10-\/20 L\+A\+B\+E\+L=g2 is equivalent to G\+R\+O\+U\+P A\+T\+O\+M\+S=10,11,12,13,14,15,16,17,18,19,20 L\+A\+B\+E\+L=g2
\item numerical ranges with a stride. So G\+R\+O\+U\+P A\+T\+O\+M\+S=10-\/100\+:10 L\+A\+B\+E\+L=g3 is equivalent to G\+R\+O\+U\+P A\+T\+O\+M\+S=10,20,30,40,50,60,70,80,90,100 L\+A\+B\+E\+L=g3
\item atoms ranges with a negative stride. So G\+R\+O\+U\+P A\+T\+O\+M\+S=100-\/10\+:-\/10 L\+A\+B\+E\+L=g4 is equivalent to G\+R\+O\+U\+P A\+T\+O\+M\+S=100,90,80,70,60,50,40,30,20,10 L\+A\+B\+E\+L=g4
\item all the above methods together. For example G\+R\+O\+U\+P A\+T\+O\+M\+S=1,2,10-\/20,40-\/60\+:5,100-\/70\+:-\/2 L\+A\+B\+E\+L=g5.
\end{DoxyItemize}

Some collective variable must accept a fixed number of atoms, for example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} is calculated using two atoms only, an \hyperlink{ANGLE}{A\+N\+G\+L\+E} is calcuated using either 3 or 4 atoms and \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} is calculated using 4 atoms.

Additional material and examples can be also found in the tutorial \hyperlink{belfast-1}{Belfast tutorial\+: Analyzing C\+Vs}.\hypertarget{_group_mols}{}\subsubsection{Molecules}\label{_group_mols}
In addition, for certain colvars, pdb files can be read in using the following keywords and used to select A\+T\+O\+M\+S\+:

\begin{TabularC}{2}
\hline
\hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}  &This command is used to provide information on the molecules that are present in your system.  \\\cline{1-2}
\end{TabularC}


The information on the molecules in your system can either be provided in the form of a pdb file or as a set of lists of atoms that describe the various chains in your system using \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}. If a pdb file is used plumed the M\+O\+L\+I\+N\+F\+O command will endeavor to recognize the various chains and residues that make up the molecules in your system using the chain\+I\+Ds and resnumbers from the pdb file. You can then use this information in commands where this has been implemented to specify atom lists. One place where this is particularly useful is when using the commands \hyperlink{ALPHARMSD}{A\+L\+P\+H\+A\+R\+M\+S\+D}, \hyperlink{ANTIBETARMSD}{A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D} and \hyperlink{PARABETARMSD}{P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D}.

M\+O\+L\+I\+N\+F\+O also introduces special groups that can be used in atom selection. These special groups always begin with a @ symbol. The following special groups are currently available in P\+L\+U\+M\+E\+D\+:

\begin{TabularC}{3}
\hline
{\bfseries  Symbol }  &{\bfseries  Topology type }  &{\bfseries  Despription }   \\\cline{1-3}
@phi-\/\#  &protein  &The torsional angle defined by the C, C\+A, N and C atoms of the protein backbone in the \#th residue. See \href{http://en.wikipedia.org/wiki/Ramachandran_plot}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Ramachandran\+\_\+plot}   \\\cline{1-3}
@psi-\/\#  &protein  &The torsional angle defined by the N, C, C\+A and N atoms of the protein backbone in the \#th residue. See \href{http://en.wikipedia.org/wiki/Ramachandran_plot}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Ramachandran\+\_\+plot}   \\\cline{1-3}
@omega-\/\#  &protein  &The torsional angle defined by the C\+A, N, C and C\+A atoms of the protein backbone in the \#th residue. See \href{http://en.wikipedia.org/wiki/Ramachandran_plot}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Ramachandran\+\_\+plot}   \\\cline{1-3}
@chi1-\/\#  &protein  &The first torsional angle of the sidechain of the \#th residue. Be aware that this angle is not defined for G\+L\+Y or A\+L\+A residues. See \href{http://en.wikipedia.org/wiki/Ramachandran_plot}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Ramachandran\+\_\+plot}   \\\cline{1-3}
\end{TabularC}


The following example shows how to use \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} with \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} to calculate the torsion angles phi and psi for the first and fourth residue of the protein\+:

\begin{DoxyVerb}MOLINFO MOLTYPE=protein STRUCTURE=myprotein.pdb
t1: TORSION ATOMS=@phi-3
t2: TORSION ATOMS=@psi-4
PRINT ARG=t1,t2 FILE=colvar STRIDE=10
\end{DoxyVerb}
\hypertarget{_group_pbc}{}\subsubsection{Broken Molecules and P\+B\+C}\label{_group_pbc}
P\+L\+U\+M\+E\+D is designed so that for the majority of the C\+Vs implemented the periodic boundary conditions are treated in the same manner as they would be treated in the host code. In some codes this can be problematic when the colvars you are using involve some property of a molecule. These codes allow the atoms in the molecules to become separated by periodic boundaries, a fact which P\+L\+U\+M\+E\+D could only deal with were the topology passed from the M\+D code to P\+L\+U\+M\+E\+D. Making this work would involve a lot laborious programming and goes against our original aim of having a general patch that can be implemented in a wide variety of M\+D codes. Consequentially, we have implemented a more pragmatic solution to this probem -\/ the user specifies in input any molecules (or parts of molecules) that must be kept in tact throughout the simulation run. In P\+L\+U\+M\+E\+D 1 this was done using the A\+L\+I\+G\+N\+\_\+\+A\+T\+O\+M\+S keyword. In P\+L\+U\+M\+E\+D 2 the same effect can be acchieved using the \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} command.

The following input computes the end-\/to-\/end distance for a polymer of 100 atoms and keeps it at a value around 5.

\begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1-100
e2e: DISTANCE ATOMS=1,100 NOPBC
RESTRAINT ARG=e2e KAPPA=1 AT=5
\end{DoxyVerb}


Notice that N\+O\+P\+B\+C is used to be sure in \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} that if the end-\/to-\/end distance is larger than half the simulation box the distance is compute properly. Also notice that, since many M\+D codes break molecules across cell boundary, it might be necessary to use the \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} keyword (also notice that it should be before distance).

Notice that most expressions are invariant with respect to a change in the order of the atoms, but some of them depend on that order. E.\+g., with \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} it could be useful to specify atom lists in a reversed order.

\begin{DoxyVerb}# to see the effect, one could dump the atoms as they were before molecule reconstruction:
# DUMPATOMS FILE=dump-broken.xyz ATOMS=1-20
WHOLEMOLECULES STRIDE=1 ENTITY0=1-20
DUMPATOMS FILE=dump.xyz ATOMS=1-20
\end{DoxyVerb}


Notice that since P\+L\+U\+M\+E\+D 2.\+1 it is also possible to shift coordinates stored within P\+L\+U\+M\+E\+D so as to align them to a template structure, using the \hyperlink{FIT_TO_TEMPLATE}{F\+I\+T\+\_\+\+T\+O\+\_\+\+T\+E\+M\+P\+L\+A\+T\+E} keyword.\hypertarget{_group_vatoms}{}\subsection{Virtual Atoms}\label{_group_vatoms}
Sometimes, when calculating a colvar, you may not want to use the positions of a number of atoms directly. Instead you may wish to use the position of a virtual atom whose position is generated based on the positions of a collection of other atoms. For example you might want to use the center of mass of a group of atoms. Plumed has a number of routines for calculating the positions of these virtual atoms from lists of atoms\+:

\begin{TabularC}{2}
\hline
\hyperlink{CENTER}{C\+E\+N\+T\+E\+R}  &Calculate the center for a group of atoms, with arbitrary weights.  \\\cline{1-2}
\hyperlink{COM}{C\+O\+M}  &Calculate the center of mass for a group of atoms.  \\\cline{1-2}
\hyperlink{GHOST}{G\+H\+O\+S\+T}  &Calculate the absolute position of a ghost atom with fixed coordinatesin the local reference frame formed by three atoms. The computed ghost atom is stored as a virtual atom that can be accessed inan atom list through the the label for the G\+H\+O\+S\+T action that creates it.  \\\cline{1-2}
\end{TabularC}


To specify to a colvar that you want to use the position of a virtual atom to calculate a colvar rather than one of the atoms in your system you simply use the label for your virtual atom in place of the usual numerical index. Virtual atoms and normal atoms can be mixed together in the input to colvars as shown below\+:

\begin{DoxyVerb}COM ATOMS=1,10 LABEL=com1
DISTANCE ATOMS=11,com1
\end{DoxyVerb}


If you don't want to calculate C\+Vs from the virtual atom. That is to say you just want to monitor the position of a virtual atom (or any set of atoms) over the course of your trajectory you can do this using \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S}. \hypertarget{GROUP}{}\subsection{G\+R\+O\+U\+P}\label{GROUP}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the generic \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Define a group of atoms so that a particular list of atoms can be referenced with a single label in definitions of C\+Vs or virtual atoms.

Atoms can be listed as comma separated numbers (i.\+e. 1,2,3,10,45,7,9,..) , simple positive ranges (i.\+e. 20-\/40), ranges with a stride either positive or negative (i.\+e. 20-\/40\+:2 or 80-\/50\+:-\/2) or as combinations of all the former methods (1,2,4,5,10-\/20,21-\/40\+:2,80-\/50\+:-\/2).

Finally, lists can be imported from ndx files (G\+R\+O\+M\+A\+C\+S format). Use N\+D\+X\+\_\+\+F\+I\+L\+E to set the name of the index file and N\+D\+X\+\_\+\+G\+R\+O\+U\+P to set the name of the group to be imported (default is first one).

Notice that this command just creates a shortcut, and does not imply any real calculation. It is just convenient to better organize input files. Might be used in combination with the \hyperlink{INCLUDE}{I\+N\+C\+L\+U\+D\+E} command so as to store long group definitions in a separate file.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the numerical indexes for the set of atoms in the group. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+D\+X\+\_\+\+F\+I\+L\+E } &the name of index file (gromacs syntax)   \\\cline{1-2}
{\bfseries  N\+D\+X\+\_\+\+G\+R\+O\+U\+P } &the name of the group to be imported (gromacs syntax) -\/ first group found is used by default  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
This command create a group of atoms containing atoms 1,4,7,11 and 14 (labeled 'o'), and another containing atoms 2,3,5,6,8,9,12,13 (labeled 'h')\+: \begin{DoxyVerb}o: GROUP ATOMS=1,4,7,11,14
h: GROUP ATOMS=2,3,5,6,8,9,12,13
# compute the coordination among the two groups
c: COORDINATION GROUPA=o GROUPB=h R_0=0.3
# same could have been obtained without GROUP, just writing:
# c: COORDINATION GROUPA=1,4,7,11,14 GROUPB=2,3,5,6,8,9,12,13

# print the coordination on file 'colvar'
PRINT ARG=c FILE=colvar
\end{DoxyVerb}
 (see also \hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N} and \hyperlink{PRINT}{P\+R\+I\+N\+T})

Groups can be conveniently stored in a separate file. E.\+g. one could create a file named 'groups.\+dat' which reads \begin{DoxyVerb}o: GROUP ATOMS=1,4,7,11,14
h: GROUP ATOMS=2,3,5,6,8,9,12,13
\end{DoxyVerb}
 and then include it in the main 'plumed.\+dat' file \begin{DoxyVerb}INCLUDE FILE=groups.dat
# compute the coordination among the two groups
c: COORDINATION GROUPA=o GROUPB=h R_0=0.3
# print the coordination on file 'colvar'
PRINT ARG=c FILE=colvar
\end{DoxyVerb}
 (see also \hyperlink{INCLUDE}{I\+N\+C\+L\+U\+D\+E}, \hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}). The groups.\+dat file could be very long and include lists of thousand atoms without cluttering the main plumed.\+dat file.

A G\+R\+O\+M\+A\+C\+S index file can also be imported \begin{DoxyVerb}# import group named 'protein' from file index.ndx
pro: GROUP NDX_FILE=index.ndx NDX_GROUP=protein
# dump all the atoms of the protein on a trajectory file
DUMPATOMS ATOMS=pro FILE=traj.gro
\end{DoxyVerb}
 (see also \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S}) \hypertarget{MOLINFO}{}\subsection{M\+O\+L\+I\+N\+F\+O}\label{MOLINFO}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the setup \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This command is used to provide information on the molecules that are present in your system.

The information on the molecules in your system can either be provided in the form of a pdb file or as a set of lists of atoms that describe the various chains in your system. If a pdb file is used plumed the M\+O\+L\+I\+N\+F\+O command will endeavor to recognize the various chains and residues that make up the molecules in your system using the chain\+I\+Ds and resnumbers from the pdb file. You can then use this information in later commands to specify atom lists in terms residues. For example using this command you can find the backbone atoms in your structure automatically.

Please be aware that the pdb parser in plumed is far from perfect. You should thus check the log file and examine what plumed is actually doing whenenver you use the M\+O\+L\+I\+N\+F\+O action.

Using M\+O\+L\+I\+N\+F\+O with a protein's pdb extend the possibility of atoms selection using the @ special symbol. Current registered keywords are\+:

\begin{DoxyVerb}@phi-#
@psi-#
@omega-#
@chi1-#
\end{DoxyVerb}


that select the appropriate atoms that define each dihedral angle for residue \#.

\begin{DoxyRefDesc}{Bug}
\item[\hyperlink{bug__bug000002}{Bug}]At the moment the H\+A1 atoms in a G\+L\+Y residues are treated as if they are the C\+B atoms. This may or may not be true -\/ G\+L\+Y is problematic for secondary structure residues as it is achiral. \end{DoxyRefDesc}


\begin{DoxyRefDesc}{Bug}
\item[\hyperlink{bug__bug000003}{Bug}]If you use W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S R\+E\+S\+I\+D\+U\+E\+S=1-\/10 for a 18 amino acid protein ( 18 amino acids + 2 terminal groups = 20 residues ) the code will fail as it will not be able to interpret termnal residue 1.\end{DoxyRefDesc}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  C\+H\+A\+I\+N } &(for masochists ( mostly Davide Branduardi ) ) The atoms involved in each of the chains of interest in the structure.. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+T\+R\+U\+C\+T\+U\+R\+E } &a file in pdb format containing a reference structure. This is used to defines the atoms in the various residues, chains, etc . For more details on the P\+D\+B file format visit \href{http://www.wwpdb.org/docs.html}{\tt http\+://www.\+wwpdb.\+org/docs.\+html}   \\\cline{1-2}
{\bfseries  M\+O\+L\+T\+Y\+P\+E } &( default=protein ) what kind of molecule is contained in the pdb file   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
In the following example the M\+O\+L\+I\+N\+F\+O command is used to provide the information on which atoms are in the backbone of a protein to the A\+L\+P\+H\+A\+R\+M\+S\+D C\+V.

\begin{DoxyVerb}MOLINFO STRUCTURE=reference.pdb
ALPHARMSD BACKBONE=all TYPE=DRMSD LESS_THAN={RATIONAL R_0=0.08 NN=8 MM=12} LABEL=a 
\end{DoxyVerb}
 (see also \hyperlink{ALPHARMSD}{A\+L\+P\+H\+A\+R\+M\+S\+D}) \hypertarget{WHOLEMOLECULES}{}\subsection{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}\label{WHOLEMOLECULES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the generic \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This action is used to rebuild molecules that can become split by the periodic boundary conditions.

It is similar to the A\+L\+I\+G\+N\+\_\+\+A\+T\+O\+M\+S keyword of plumed1, and is needed since some M\+D dynamics code (e.\+g. G\+R\+O\+M\+A\+C\+S) can break molecules during the calculation.

Running some C\+Vs without this command can cause there to be discontinuities changes in the C\+V value and artifacts in the calculations. This command can be applied more than once. To see what effect is has use a variable without pbc or use the \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S} directive to output the atomic positions.

\begin{DoxyAttention}{Attention}
This directive modifies the stored position at the precise moment it is executed. This means that only collective variables which are below it in the input script will see the corrected positions. As a general rule, put it at the top of the input file. Also, unless you know exactly what you are doing, leave the default stride (1), so that this action is performed at every M\+D step.
\end{DoxyAttention}
\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  E\+N\+T\+I\+T\+Y } &the atoms that make up a molecule that you wish to align. To specify multiple molecules use a list of E\+N\+T\+I\+T\+Y keywords\+: E\+N\+T\+I\+T\+Y1, E\+N\+T\+I\+T\+Y2,... You can use multiple instances of this keyword i.\+e. E\+N\+T\+I\+T\+Y1, E\+N\+T\+I\+T\+Y2, E\+N\+T\+I\+T\+Y3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+S\+I\+D\+U\+E\+S } &this command specifies that the backbone atoms in a set of residues all must be aligned. It must be used in tandem with the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} action and the M\+O\+L\+T\+Y\+P\+E keyword. If you wish to use all the residues from all the chains in your system you can do so by specifying all. Alternatively, if you wish to use a subset of the residues you can specify the particular residues you are interested in as a list of numbers   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+T\+R\+I\+D\+E } &( default=1 ) the frequency with which molecules are reassembled. Unless you are completely certain about what you are doing leave this set equal to 1!   \\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  M\+O\+L\+T\+Y\+P\+E } &the type of molecule that is under study. This is used to define the backbone atoms  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
This command instructs plumed to reconstruct the molecule containing atoms 1-\/20 at every step of the calculation and dump them on a file.

\begin{DoxyVerb}# to see the effect, one could dump the atoms as they were before molecule reconstruction:
# DUMPATOMS FILE=dump-broken.xyz ATOMS=1-20
WHOLEMOLECULES STRIDE=1 ENTITY0=1-20
DUMPATOMS FILE=dump.xyz ATOMS=1-20
\end{DoxyVerb}
 (see also \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S})

This command instructs plumed to reconstruct two molecules containing atoms 1-\/20 and 30-\/40

\begin{DoxyVerb}WHOLEMOLECULES STRIDE=1 ENTITY0=1-20 ENTITY1=30-40
DUMPATOMS FILE=dump.xyz ATOMS=1-20,30-40
\end{DoxyVerb}
 (see also \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S})

This command instructs plumed to reconstruct the chain of backbone atoms in a protein

\begin{DoxyVerb}MOLINFO STRUCTURE=helix.pdb
WHOLEMOLECULES STRIDE=1 RESIDUES=all MOLTYPE=protein 
\end{DoxyVerb}
 (See also \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}) \hypertarget{FIT_TO_TEMPLATE}{}\subsection{F\+I\+T\+\_\+\+T\+O\+\_\+\+T\+E\+M\+P\+L\+A\+T\+E}\label{FIT_TO_TEMPLATE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the generic \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This action is used to align a molecule to a template.

This can be used to move the coordinates stored in plumed so as to be aligned with a provided template in pdb format. Pdb should contain also weights for alignment (see the format of pdb files used e.\+g. for \hyperlink{RMSD}{R\+M\+S\+D}). Weights for displacement are ignored, since no displacement is computed here. Notice that all atoms (not only those in the template) are aligned. To see what effect try the \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S} directive to output the atomic positions.

Also notice that P\+L\+U\+M\+E\+D propagate forces correctly so that you can add a bias on a C\+V computed after alignment. For many C\+Vs this has no effect, but in some case the alignment can change the result. Examples are\+:
\begin{DoxyItemize}
\item \hyperlink{POSITION}{P\+O\+S\+I\+T\+I\+O\+N} C\+V since it is affected by a rigid shift of the system.
\item \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} C\+V with C\+O\+M\+P\+O\+N\+E\+N\+T\+S. Since the alignment could involve a rotation (with T\+Y\+P\+E=O\+P\+T\+I\+M\+A\+L) the actual components could be different from the original ones.
\item \hyperlink{CELL}{C\+E\+L\+L} components for a similar reason.
\end{DoxyItemize}

In the present implementation only T\+Y\+P\+E=S\+I\+M\+P\+L\+E is implemented. As a consequence, only \hyperlink{POSITION}{P\+O\+S\+I\+T\+I\+O\+N} C\+V can be affected by the fit.

\begin{DoxyAttention}{Attention}
This directive modifies the stored position at the precise moment it is executed. This means that only collective variables which are below it in the input script will see the corrected positions. As a general rule, put it at the top of the input file. Also, unless you know exactly what you are doing, leave the default stride (1), so that this action is performed at every M\+D step.
\end{DoxyAttention}
\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+T\+R\+I\+D\+E } &( default=1 ) the frequency with which molecules are reassembled. Unless you are completely certain about what you are doing leave this set equal to 1!   \\\cline{1-2}
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a file in pdb format containing the reference structure and the atoms involved in the C\+V.   \\\cline{1-2}
{\bfseries  T\+Y\+P\+E } &( default=S\+I\+M\+P\+L\+E ) the manner in which R\+M\+S\+D alignment is performed. Should be O\+P\+T\+I\+M\+A\+L or S\+I\+M\+P\+L\+E. Currently only S\+I\+M\+P\+L\+E is implemented   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
Align the atomic position to a template then print them \begin{DoxyVerb}# to see the effect, one could dump the atoms before alignment
DUMPATOMS FILE=dump-before.xyz ATOMS=1-20
FIT_TO_TEMPLATE STRIDE=1 REFERENCE=ref.pdb TYPE=SIMPLE
DUMPATOMS FILE=dump-after.xyz ATOMS=1-20
\end{DoxyVerb}
 (see also \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S}) \hypertarget{CENTER}{}\subsection{C\+E\+N\+T\+E\+R}\label{CENTER}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the vatom \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the center for a group of atoms, with arbitrary weights.

The computed center is stored as a virtual atom that can be accessed in an atom list through the label for the C\+E\+N\+T\+E\+R action that creates it. Notice that the generated virtual atom has charge equal to the sum of the charges and mass equal to the sum of the masses. If used with the M\+A\+S\+S flag, then it provides a result identical to \hyperlink{COM}{C\+O\+M}.

When running with periodic boundary conditions, the user should take care that the atoms in the C\+O\+M group actually are in the proper periodic image. This is typically achieved using the \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} action before C\+O\+M calculation.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the list of atoms which are involved the virtual atom's definition. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  M\+A\+S\+S } &( default=off ) If set center is mass weighted  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  W\+E\+I\+G\+H\+T\+S } &Center is computed as a weighted average.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\begin{DoxyVerb}# a point which is on the line connecting atoms 1 and 10, so that its distance
# from 10 is twice its distance from 1:
c1: CENTER ATOMS=1,1,10
# this is another way of stating the same:
c1bis: CENTER ATOMS=1,10 WEIGHTS=2,1

# center of mass among these atoms:
c2: CENTER ATOMS=2,3,4,5 MASS

d1: DISTANCE ATOMS=c1,c2

PRINT ARG=d1
\end{DoxyVerb}
 (See also \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}, \hyperlink{COM}{C\+O\+M} and \hyperlink{PRINT}{P\+R\+I\+N\+T}). \hypertarget{COM}{}\subsection{C\+O\+M}\label{COM}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the vatom \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the center of mass for a group of atoms.

The computed center of mass is stored as a virtual atom that can be accessed in an atom list through the label for the C\+O\+M action that creates it.

For arbitrary weights (e.\+g. geometric center) see \hyperlink{CENTER}{C\+E\+N\+T\+E\+R}.

When running with periodic boundary conditions, the user should take care that the atoms in the C\+O\+M group actually are in the proper periodic image. This is typically achieved using the \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} action before C\+O\+M calculation.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the list of atoms which are involved the virtual atom's definition. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input instructs plumed to print the distance between the center of mass for atoms 1,2,3,4,5,6,7 and that for atoms 15,20\+: \begin{DoxyVerb}COM ATOMS=1-7         LABEL=c1
COM ATOMS=15,20       LABEL=c2
DISTANCE ATOMS=c1,c2  LABEL=d1
PRINT ARG=d1
\end{DoxyVerb}
 (See also \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} and \hyperlink{PRINT}{P\+R\+I\+N\+T}). \hypertarget{GHOST}{}\subsection{G\+H\+O\+S\+T}\label{GHOST}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the vatom \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the absolute position of a ghost atom with fixed coordinates in the local reference frame formed by three atoms. The computed ghost atom is stored as a virtual atom that can be accessed in an atom list through the the label for the G\+H\+O\+S\+T action that creates it.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the list of atoms which are involved the virtual atom's definition. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
{\bfseries  C\+O\+O\+R\+D\+I\+N\+A\+T\+E\+S } &coordinates of the ghost atom in the local reference frame. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following input instructs plumed to print the distance between the ghost atom and the center of mass for atoms 15,20\+: \begin{DoxyVerb}GHOST ATOMS=1,5,10 COORDINATES=10.0,10.0,10.0 LABEL=c1
COM ATOMS=15,20       LABEL=c2
DISTANCE ATOMS=c1,c2  LABEL=d1
PRINT ARG=d1
\end{DoxyVerb}
 (See also \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} and \hyperlink{PRINT}{P\+R\+I\+N\+T}).
\end{DoxyParagraph}
\begin{DoxyWarning}{Warning}
If a force is added to a ghost atom, the contribution to the virial could contain small artifacts. It is therefore discouraged to use G\+H\+O\+S\+T in a constant pressure simulation. 
\end{DoxyWarning}
\hypertarget{Colvar}{}\section{C\+V Documentation}\label{Colvar}
The following list contains descriptions of a number of the colvars that are currently implemented in P\+L\+U\+M\+E\+D 2.

\begin{TabularC}{2}
\hline
\hyperlink{ALPHABETA}{A\+L\+P\+H\+A\+B\+E\+T\+A}  &Measures a distance including pbc between the instantaneous values of a set of torsional angles and set of reference values.  \\\cline{1-2}
\hyperlink{ALPHARMSD}{A\+L\+P\+H\+A\+R\+M\+S\+D}  &Probe the alpha helical content of a protein structure.  \\\cline{1-2}
\hyperlink{ANGLE}{A\+N\+G\+L\+E}  &Calculate an angle.  \\\cline{1-2}
\hyperlink{ANTIBETARMSD}{A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D}  &Probe the antiparallel beta sheet content of your protein structure.  \\\cline{1-2}
\hyperlink{CELL}{C\+E\+L\+L}  &Calculate the components of the simulation cell  \\\cline{1-2}
\hyperlink{CH3SHIFTS}{C\+H3\+S\+H\+I\+F\+T\+S}  &This collective variable calculates a scoring function based on the comparison of calculated andexperimental methyl chemical shifts.   \\\cline{1-2}
\hyperlink{CONSTANT}{C\+O\+N\+S\+T\+A\+N\+T}  &Return a constant quantity.  \\\cline{1-2}
\hyperlink{CONTACTMAP}{C\+O\+N\+T\+A\+C\+T\+M\+A\+P}  &Calculate the distances between a number of pairs of atoms and transform each distance by a switching function.\+The transformed distance can be compared with a reference value in order to calculate the squared distancebetween two contact maps. Each distance can also be weighted for a given value. C\+O\+N\+T\+A\+C\+T\+M\+A\+P can be used togetherwith \hyperlink{FUNCPATHMSD}{F\+U\+N\+C\+P\+A\+T\+H\+M\+S\+D} to define a path in the contactmap space.  \\\cline{1-2}
\hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N}  &Calculate coordination numbers.  \\\cline{1-2}
\hyperlink{CS2BACKBONE}{C\+S2\+B\+A\+C\+K\+B\+O\+N\+E}  &This collective variable calculates a scoring function based on the comparison of backcalculated andexperimental backbone chemical shifts for a protein (C\+A, C\+B, C', H, H\+A, N).  \\\cline{1-2}
\hyperlink{DHENERGY}{D\+H\+E\+N\+E\+R\+G\+Y}  &Calculate Debye-\/\+Huckel interaction energy among G\+R\+O\+U\+P\+A and G\+R\+O\+U\+P\+B.  \\\cline{1-2}
\hyperlink{DIHCOR}{D\+I\+H\+C\+O\+R}  &Measures the degree of similarity between dihedral angles.  \\\cline{1-2}
\hyperlink{DIPOLE}{D\+I\+P\+O\+L\+E}  &Calculate the dipole moment for a group of atoms.  \\\cline{1-2}
\hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}  &Calculate the distance between a pair of atoms.  \\\cline{1-2}
\hyperlink{ENERGY}{E\+N\+E\+R\+G\+Y}  &Calculate the total energy of the simulation box.  \\\cline{1-2}
\hyperlink{FAKE}{F\+A\+K\+E}  &This is a fake colvar container used by cltools or various other actionsand just support input and period definition  \\\cline{1-2}
\hyperlink{GPROPERTYMAP}{G\+P\+R\+O\+P\+E\+R\+T\+Y\+M\+A\+P}  &Property maps but with a more flexible framework for the distance metric being used.   \\\cline{1-2}
\hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N}  &Calculate the radius of gyration, or other properties related to it.  \\\cline{1-2}
\hyperlink{NOE}{N\+O\+E}  &Calculates the deviation of current distances from experimental N\+O\+E derived distances.  \\\cline{1-2}
\hyperlink{PARABETARMSD}{P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D}  &Probe the parallel beta sheet content of your protein structure.  \\\cline{1-2}
\hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D}  &This Colvar calculates path collective variables.   \\\cline{1-2}
\hyperlink{PATH}{P\+A\+T\+H}  &Path collective variables with a more flexible framework for the distance metric being used.   \\\cline{1-2}
\hyperlink{POSITION}{P\+O\+S\+I\+T\+I\+O\+N}  &Calculate the components of the position of an atom.  \\\cline{1-2}
\hyperlink{PROPERTYMAP}{P\+R\+O\+P\+E\+R\+T\+Y\+M\+A\+P}  &Calculate generic property maps.  \\\cline{1-2}
\hyperlink{RDC}{R\+D\+C}  &Calculates the Residual Dipolar Coupling between two atoms.   \\\cline{1-2}
\hyperlink{TEMPLATE}{T\+E\+M\+P\+L\+A\+T\+E}  &This file provides a template for if you want to introduce a new C\+V.  \\\cline{1-2}
\hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}  &Calculate a torsional angle.  \\\cline{1-2}
\hyperlink{VOLUME}{V\+O\+L\+U\+M\+E}  &Calculate the volume of the simulation box.  \\\cline{1-2}
\end{TabularC}
\hypertarget{ALPHABETA}{}\subsection{A\+L\+P\+H\+A\+B\+E\+T\+A}\label{ALPHABETA}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Measures a distance including pbc between the instantaneous values of a set of torsional angles and set of reference values.

This colvar calculates the following quantity.

\[ s = \frac{1}{2} \sum_i \left[ 1 + \cos( \phi_i - \phi_i^{\textrm{Ref}} ) \right] \]

where the $\phi_i$ values are the instantaneous values for the \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} angles of interest. The $\phi_i^{\textrm{Ref}}$ values are the user-\/specified reference values for the torsional angles.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &the reference values for each of the torsional angles. If you use a single R\+E\+F\+E\+R\+E\+N\+C\+E value the same reference value is used for all torsions You can use multiple instances of this keyword i.\+e. R\+E\+F\+E\+R\+E\+N\+C\+E1, R\+E\+F\+E\+R\+E\+N\+C\+E2, R\+E\+F\+E\+R\+E\+N\+C\+E3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following provides an example of the input for an alpha beta similarity.

\begin{DoxyVerb}ALPHABETA ...
ATOMS1=168,170,172,188 REFERENCE1=3.14 
ATOMS2=170,172,188,190 REFERENCE2=3.14 
ATOMS3=188,190,192,230 REFERENCE3=3.14
LABEL=ab
... ALPHABETA
PRINT ARG=ab FILE=colvar STRIDE=10
\end{DoxyVerb}


Because all the reference values are the same we can calculate the same quantity using

\begin{DoxyVerb}ALPHABETA ...
ATOMS1=168,170,172,188 REFERENCE=3.14 
ATOMS2=170,172,188,190 
ATOMS3=188,190,192,230 
LABEL=ab
... ALPHABETA
PRINT ARG=ab FILE=colvar STRIDE=10
\end{DoxyVerb}


Writing out the atoms involved in all the torsions in this way can be rather tedious. Thankfully if you are working with protein you can avoid this by using the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command. P\+L\+U\+M\+E\+D uses the pdb file that you provide to this command to learn about the topology of the protein molecule. This means that you can specify torsion angles using the following syntax\+:

\begin{DoxyVerb}MOLINFO MOLTYPE=protein STRUCTURE=myprotein.pdb
ALPHABETA ...
ATOMS1=@phi-3 REFERENCE=3.14
ATOMS2=@psi-3
ATOMS3=@phi-4
LABEL=ab
... ALPHABETA 
PRINT ARG=ab FILE=colvar STRIDE=10
\end{DoxyVerb}


Here, @phi-\/3 tells plumed that you would like to calculate the $\phi$ angle in the third residue of the protein. Similarly @psi-\/4 tells plumed that you want to calculate the $\psi$ angle of the 4th residue of the protein. \hypertarget{ALPHARMSD}{}\subsection{A\+L\+P\+H\+A\+R\+M\+S\+D}\label{ALPHARMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the secondarystructure \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Probe the alpha helical content of a protein structure.

Any chain of six contiguous residues in a protein chain can form an alpha helix. This colvar thus generates the set of all possible six residue sections and calculates the R\+M\+S\+D distance between the configuration in which the residues find themselves and an idealized alpha helical structure. These distances can be calculated by either aligning the instantaneous structure with the reference structure and measuring each atomic displacement or by calculating differences between the set of interatomic distances in the reference and instantaneous structures.

This colvar is based on the following reference \cite{pietrucci09jctc}. The authors of this paper use the set of distances from the alpha helix configurations to measure the number of segments that have an alpha helical configuration. This is done by calculating the following sum of functions of the rmsd distances\+:

\[ s = \sum_i \frac{ 1 - \left(\frac{r_i-d_0}{r_0}\right)^n } { 1 - \left(\frac{r_i-d_0}{r_0}\right)^m } \]

where the sum runs over all possible segments of alpha helix. By default the N\+N, M\+M and D\+\_\+0 parameters are set equal to those used in \cite{pietrucci09jctc}. The R\+\_\+0 parameter must be set by the user -\/ the value used in \cite{pietrucci09jctc} was 0.\+08 nm.

If you change the function in the above sum you can calculate quantities such as the average distance from a purely the alpha helical configuration or the distance between the set of residues that is closest to an alpha helix and the reference configuration. To do these sorts of calculations you can use the A\+V\+E\+R\+A\+G\+E and M\+I\+N keywords. In addition you can use the L\+E\+S\+S\+\_\+\+T\+H\+A\+N keyword if you would like to change the form of the switching function. If you use any of these options you no longer need to specify N\+N, R\+\_\+0, M\+M and D\+\_\+0.

Please be aware that for codes like gromacs you must ensure that plumed reconstructs the chains involved in your C\+V when you calculate this C\+V using anthing other than T\+Y\+P\+E=D\+R\+M\+S\+D. For more details as to how to do this see \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the number of structural units that are within a certain distance of a idealised secondary structure element. This quantity can then be referenced elsewhere in the input by using the label of the action. However, thes Action can also be used to calculate the following quantities by using the keywords as described below. The quantities then calculated can be referened using the label of the action followed by a dot and then the name from the table below. Please note that you can use the L\+E\+S\+S\+\_\+\+T\+H\+A\+N keyword more than once. The resulting components will be labelled {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 and so on unless you exploit the fact that these labels are customizable. In particular, by using the L\+A\+B\+E\+L keyword in the description of you L\+E\+S\+S\+\_\+\+T\+H\+A\+N function you can set name of the component that you are calculating

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+S\+I\+D\+U\+E\+S } &this command is used to specify the set of residues that could conceivably form part of the secondary structure. It is possible to use residues numbers as the various chains and residues should have been identified else using an instance of the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} action. If you wish to use all the residues from all the chains in your system you can do so by specifying all. Alternatively, if you wish to use a subset of the residues you can specify the particular residues you are interested in as a list of numbers. Please be aware that to form secondary structure elements your chain must contain at least N residues, where N is dependent on the particular secondary structure you are interested in. As such if you define portions of the chain with fewer than N residues the code will crash.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+Y\+P\+E } &( default=D\+R\+M\+S\+D ) the manner in which R\+M\+S\+D alignment is performed. Should be O\+P\+T\+I\+M\+A\+L, S\+I\+M\+P\+L\+E or D\+R\+M\+S\+D. For more details on the O\+P\+T\+I\+M\+A\+L and S\+I\+M\+P\+L\+E methods see \hyperlink{RMSD}{R\+M\+S\+D}. For more details on the D\+R\+M\+S\+D method see \hyperlink{DRMSD}{D\+R\+M\+S\+D}.   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function.   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  N\+N } &( default=8 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input calculates the number of six residue segments of protein that are in an alpha helical configuration.

\begin{DoxyVerb}MOLINFO STRUCTURE=helix.pdb
ALPHARMSD RESIDUES=all TYPE=DRMSD LESS_THAN={RATIONAL R_0=0.08 NN=8 MM=12} LABEL=a
\end{DoxyVerb}
 (see also \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}) \hypertarget{ANGLE}{}\subsection{A\+N\+G\+L\+E}\label{ANGLE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate an angle.

This command can be used to compute the angle between three atoms. Alternatively if four atoms appear in the atom specification it calculates the angle between two vectors identified by two pairs of atoms.

If {\itshape three} atoms are given, the angle is defined as\+: \[ \theta=\arccos\left(\frac{ {\bf r}_{21}\cdot {\bf r}_{23}}{ |{\bf r}_{21}| |{\bf r}_{23}|}\right) \] Here $ {\bf r}_{ij}$ is the distance vector among the i-\/th and the j-\/th listed atom.

If {\itshape four} atoms are given, the angle is defined as\+: \[ \theta=\arccos\left(\frac{ {\bf r}_{21}\cdot {\bf r}_{34}}{ |{\bf r}_{21}| |{\bf r}_{34}|}\right) \]

Notice that angles defined in this way are non-\/periodic variables and their value is limited by definition between 0 and $\pi$.

The vectors $ {\bf r}_{ij}$ are by default evaluated taking periodic boundary conditions into account. This behavior can be changed with the N\+O\+P\+B\+C flag.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the list of atoms involved in this collective variable (either 3 or 4 atoms). For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
This command tells plumed to calculate the angle between the vector connecting atom 1 to atom 2 and the vector connecting atom 2 to atom 3 and to print it on file C\+O\+L\+V\+A\+R1. At the same time, the angle between vector connecting atom 1 to atom 2 and the vector connecting atom 3 to atom 4 is printed on file C\+O\+L\+V\+A\+R2. \begin{DoxyVerb}a: ANGLE ATOMS=1,2,3 
# equivalently one could state:
# a: ANGLE ATOMS=1,2,2,3

b: ANGLE ATOMS=1,2,3,4

PRINT ARG=a FILE=COLVAR1
PRINT ARG=b FILE=COLVAR2
\end{DoxyVerb}
 (see also \hyperlink{PRINT}{P\+R\+I\+N\+T}) \hypertarget{ANTIBETARMSD}{}\subsection{A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D}\label{ANTIBETARMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the secondarystructure \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Probe the antiparallel beta sheet content of your protein structure.

Two protein segments containing three continguous residues can form an antiparallel beta sheet. Although if the two segments are part of the same protein chain they must be separated by a minimum of 2 residues to make room for the turn. This colvar thus generates the set of all possible six residue sections that could conceivably form an antiparallel beta sheet and calculates the R\+M\+S\+D distance between the configuration in which the residues find themselves and an idealized antiparallel beta sheet structure. These distances can be calculated by either aligning the instantaneous structure with the reference structure and measuring each atomic displacement or by calculating differences between the set of interatomic distances in the reference and instantaneous structures.

This colvar is based on the following reference \cite{pietrucci09jctc}. The authors of this paper use the set of distances from the anti parallel beta sheet configurations to measure the number of segments that have an configuration that resemebles an anti paralel beta sheet. This is done by calculating the following sum of functions of the rmsd distances\+:

\[ s = \sum_i \frac{ 1 - \left(\frac{r_i-d_0}{r_0}\right)^n } { 1 - \left(\frac{r_i-d_0}{r_0}\right)^m } \]

where the sum runs over all possible segments of antiparallel beta sheet. By default the N\+N, M\+M and D\+\_\+0 parameters are set equal to those used in \cite{pietrucci09jctc}. The R\+\_\+0 parameter must be set by the user -\/ the value used in \cite{pietrucci09jctc} was 0.\+08 nm.

If you change the function in the above sum you can calculate quantities such as the average distance from a purely configuration composed of pure anti-\/parallel beta sheets or the distance between the set of residues that is closest to an anti-\/parallel beta sheet and the reference configuration. To do these sorts of calculations you can use the A\+V\+E\+R\+A\+G\+E and M\+I\+N keywords. In addition you can use the L\+E\+S\+S\+\_\+\+T\+H\+A\+N keyword if you would like to change the form of the switching function. If you use any of these options you no longer need to specify N\+N, R\+\_\+0, M\+M and D\+\_\+0.

Please be aware that for codes like gromacs you must ensure that plumed reconstructs the chains involved in your C\+V when you calculate this C\+V using anthing other than T\+Y\+P\+E=D\+R\+M\+S\+D. For more details as to how to do this see \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the number of structural units that are within a certain distance of a idealised secondary structure element. This quantity can then be referenced elsewhere in the input by using the label of the action. However, thes Action can also be used to calculate the following quantities by using the keywords as described below. The quantities then calculated can be referened using the label of the action followed by a dot and then the name from the table below. Please note that you can use the L\+E\+S\+S\+\_\+\+T\+H\+A\+N keyword more than once. The resulting components will be labelled {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 and so on unless you exploit the fact that these labels are customizable. In particular, by using the L\+A\+B\+E\+L keyword in the description of you L\+E\+S\+S\+\_\+\+T\+H\+A\+N function you can set name of the component that you are calculating

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+S\+I\+D\+U\+E\+S } &this command is used to specify the set of residues that could conceivably form part of the secondary structure. It is possible to use residues numbers as the various chains and residues should have been identified else using an instance of the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} action. If you wish to use all the residues from all the chains in your system you can do so by specifying all. Alternatively, if you wish to use a subset of the residues you can specify the particular residues you are interested in as a list of numbers. Please be aware that to form secondary structure elements your chain must contain at least N residues, where N is dependent on the particular secondary structure you are interested in. As such if you define portions of the chain with fewer than N residues the code will crash.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+Y\+P\+E } &( default=D\+R\+M\+S\+D ) the manner in which R\+M\+S\+D alignment is performed. Should be O\+P\+T\+I\+M\+A\+L, S\+I\+M\+P\+L\+E or D\+R\+M\+S\+D. For more details on the O\+P\+T\+I\+M\+A\+L and S\+I\+M\+P\+L\+E methods see \hyperlink{RMSD}{R\+M\+S\+D}. For more details on the D\+R\+M\+S\+D method see \hyperlink{DRMSD}{D\+R\+M\+S\+D}.   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function.   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  N\+N } &( default=8 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  S\+T\+Y\+L\+E } &( default=all ) Antiparallel beta sheets can either form in a single chain or from a pair of chains. If S\+T\+Y\+L\+E=all all chain configuration with the appropriate geometry are counted. If S\+T\+Y\+L\+E=inter only sheet-\/like configurations involving two chains are counted, while if S\+T\+Y\+L\+E=intra only sheet-\/like configurations involving a single chain are counted   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  S\+T\+R\+A\+N\+D\+S\+\_\+\+C\+U\+T\+O\+F\+F } &If in a segment of protein the two strands are further apart then the calculation of the actual R\+M\+S\+D is skipped as the structure is very far from being beta-\/sheet like. This keyword speeds up the calculation enormously when you are using the L\+E\+S\+S\+\_\+\+T\+H\+A\+N option. However, if you are using some other option, then this cannot be used  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input calculates the number of six residue segments of protein that are in an antiparallel beta sheet configuration.

\begin{DoxyVerb}MOLINFO STRUCTURE=helix.pdb
ANTIBETARMSD RESIDUES=all TYPE=DRMSD LESS_THAN={RATIONAL R_0=0.08 NN=8 MM=12} LABEL=a
\end{DoxyVerb}
 (see also \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}) \hypertarget{CELL}{}\subsection{C\+E\+L\+L}\label{CELL}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the components of the simulation cell

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the following quantities. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  ax } &the ax component of the cell matrix   \\\cline{1-2}
{\bfseries  ay } &the ay component of the cell matrix   \\\cline{1-2}
{\bfseries  az } &the az component of the cell matrix   \\\cline{1-2}
{\bfseries  bx } &the bx component of the cell matrix   \\\cline{1-2}
{\bfseries  by } &the by component of the cell matrix   \\\cline{1-2}
{\bfseries  bz } &the bz component of the cell matrix   \\\cline{1-2}
{\bfseries  cx } &the cx component of the cell matrix   \\\cline{1-2}
{\bfseries  cy } &the cy component of the cell matrix   \\\cline{1-2}
{\bfseries  cz } &the cz component of the cell matrix   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following input tells plumed to print the squared modulo of each of the three lattice vectors \begin{DoxyVerb}cell: CELL
aaa:    COMBINE ARG=cell.ax,cell.ay,cell.az POWERS=2,2,2 PERIODIC=NO
bbb:    COMBINE ARG=cell.bx,cell.by,cell.bz POWERS=2,2,2 PERIODIC=NO
ccc:    COMBINE ARG=cell.cx,cell.cy,cell.cz POWERS=2,2,2 PERIODIC=NO
PRINT ARG=aaa,bbb,ccc
\end{DoxyVerb}
 (See also \hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E} and \hyperlink{PRINT}{P\+R\+I\+N\+T}). 
\end{DoxyParagraph}
\hypertarget{CH3SHIFTS}{}\subsection{C\+H3\+S\+H\+I\+F\+T\+S}\label{CH3SHIFTS}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This collective variable calculates a scoring function based on the comparison of calculated and experimental methyl chemical shifts.

C\+H3\+Shift \cite{Sahakyan:2011bn} is employed to back calculate the chemical shifts of methyl groups (A\+L\+A\+:H\+B; I\+L\+E\+:H\+D,H\+G2; L\+E\+U\+:H\+D1,H\+D2; T\+H\+R\+:H\+G2; V\+A\+L\+:H\+G1,H\+G2) that are then compared with a set of experimental values to generate a score \cite{Robustelli:2010dn} \cite{Granata:2013dk}.

It is also possible to backcalculate the chemical shifts from multiple replicas and then average them to perform Replica-\/\+Averaged Restrained M\+D simulations \cite{Camilloni:2012je} \cite{Camilloni:2013hs}.

In general the system for which chemical shifts are to be calculated must be completly included in A\+T\+O\+M\+S. It should also be made whole \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} before the the back-\/calculation.

H\+O\+W T\+O C\+O\+M\+P\+I\+L\+E I\+T

\hyperlink{_installation_installingalmost}{Installing P\+L\+U\+M\+E\+D with A\+L\+M\+O\+S\+T} on how to compile P\+L\+U\+M\+E\+D with A\+L\+M\+O\+S\+T.

H\+O\+W T\+O U\+S\+E I\+T

C\+H3\+Shift reads from a text file the experimental chemical shifts\+:

\begin{DoxyVerb}CH3shifts.dat:
1.596 28
0.956 46
0.576 3 HG2
0.536 3 HD1
0.836 13 HG2
0.666 13 HD1
0.716 23 HG2
0.506 23 HD1
\end{DoxyVerb}


A template.\+pdb file is needed to the generate a topology of the protein within A\+L\+M\+O\+S\+T. For histidines in protonation states different from D the H\+I\+E/\+H\+I\+P name should be used in the template.\+pdb. G\+L\+H and A\+S\+H can be used for the alternative protonation of G\+L\+U and A\+S\+P. Non-\/standard amino acids and other molecules are not yet supported! If multiple chains are present the chain identifier must be in the standard P\+D\+B format, together with the T\+E\+R keyword at the end of each chain. Residues numbering should always go from 1 to N in both the chemical shifts files as well as in the template.\+pdb file. Two more keywords can be used to setup the topology\+: C\+Y\+S-\/\+D\+I\+S\+U to tell A\+L\+M\+O\+S\+T to look for disulphide bridges and T\+E\+R\+M\+I\+N\+I to define the protonation state of the the chain's termini (currently only D\+E\+F\+A\+U\+L\+T (N\+H3+, C\+O2-\/) and N\+O\+N\+E (N\+H, C\+O)).

One more standard file is also needed, that is an A\+L\+M\+O\+S\+T force-\/field file, corresponding to the force-\/field family used in the M\+D code, (a03\+\_\+cs2\+\_\+gromacs.\+mdb for the amber family or all22\+\_\+gromacs.\+mdb for the charmm family).

All the above files must be in a single folder that must be specified with the keyword D\+A\+T\+A (multiple definition of the C\+V can point to different folders).

\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
case 1\+:

\begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1-174
cs: CH3SHIFTS ATOMS=1-174 DATA=data/ FF=a03_gromacs.mdb FLAT=0.0 NRES=13 ENSEMBLE
cse: RESTRAINT ARG=cs SLOPE=24 KAPPA=0 AT=0.
PRINT ARG=cs,cse.bias
\end{DoxyVerb}


case 2\+:

\begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1-174
cs: CH3SHIFTS ATOMS=1-174 DATA=data/ FF=a03_gromacs.mdb FLAT=1.0 NRES=13 TERMINI=DEFAULT,NONE CYS-DISU WRITE_CS=1000
PRINT ARG=cs
\end{DoxyVerb}


(See also \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}, \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T} and \hyperlink{PRINT}{P\+R\+I\+N\+T}) \hypertarget{CONSTANT}{}\subsection{C\+O\+N\+S\+T\+A\+N\+T}\label{CONSTANT}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Return a constant quantity.

Useful in combination with functions.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  V\+A\+L\+U\+E } &The value of the constant   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input instructs plumed to compute the distance between atoms 1 and 2. If this distance is between 1.\+0 and 2.\+0, it is printed. If it is lower than 1.\+0 (larger than 2.\+0), 1.\+0 (2.\+0) is printed

\begin{DoxyVerb}one: CONSTANT VALUE=1.0
two: CONSTANT VALUE=2.0
dis: DISTANCE ATOMS=1,2
sss: SORT ARG=one,dis,two
PRINT ARG=sss.2
\end{DoxyVerb}
 (See also \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}, \hyperlink{SORT}{S\+O\+R\+T}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}). \hypertarget{CONTACTMAP}{}\subsection{C\+O\+N\+T\+A\+C\+T\+M\+A\+P}\label{CONTACTMAP}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the distances between a number of pairs of atoms and transform each distance by a switching function. The transformed distance can be compared with a reference value in order to calculate the squared distance between two contact maps. Each distance can also be weighted for a given value. C\+O\+N\+T\+A\+C\+T\+M\+A\+P can be used together with \hyperlink{FUNCPATHMSD}{F\+U\+N\+C\+P\+A\+T\+H\+M\+S\+D} to define a path in the contactmap space.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default the value of the calculated quantity can be referenced elsewhere in the input file by using the label of the action. Alternatively this Action can be used to be used to calculate the following quantities by employing the keywords listed below. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  contact } &By not using S\+U\+M or C\+M\+D\+I\+S\+T each contact will be stored in a component   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the contacts you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one contact will be calculated for each A\+T\+O\+M keyword you specify. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+W\+I\+T\+C\+H } &The switching functions to use for each of the contacts in your map. You can either specify a global switching function using S\+W\+I\+T\+C\+H or one switching function for each contact. Details of the various switching functions you can use are provided on \hyperlink{switchingfunction}{switchingfunction}. You can use multiple instances of this keyword i.\+e. S\+W\+I\+T\+C\+H1, S\+W\+I\+T\+C\+H2, S\+W\+I\+T\+C\+H3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+U\+M } &( default=off ) calculate the sum of all the contacts in the input   \\\cline{1-2}
{\bfseries  C\+M\+D\+I\+S\+T } &( default=off ) calculate the distance with respect to the provided reference contant map   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) Perform the calculation in serial -\/ for debug purpose  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &A reference value for a given contact, by default is 0.\+0 You can either specify a global reference value using R\+E\+F\+E\+R\+E\+N\+C\+E or one reference value for each contact. You can use multiple instances of this keyword i.\+e. R\+E\+F\+E\+R\+E\+N\+C\+E1, R\+E\+F\+E\+R\+E\+N\+C\+E2, R\+E\+F\+E\+R\+E\+N\+C\+E3...   \\\cline{1-2}
{\bfseries  W\+E\+I\+G\+H\+T } &A weight value for a given contact, by default is 1.\+0 You can either specify a global weight value using W\+E\+I\+G\+H\+T or one weight value for each contact. You can use multiple instances of this keyword i.\+e. W\+E\+I\+G\+H\+T1, W\+E\+I\+G\+H\+T2, W\+E\+I\+G\+H\+T3...  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following example calculates switching functions based on the distances between atoms 1 and 2, 3 and 4 and 4 and 5. The values of these three switching functions are then output to a file named colvar.

\begin{DoxyVerb}CONTACTMAP ATOMS1=1,2 ATOMS2=3,4 ATOMS3=4,5 ATOMS4=5,6 SWITCH=(RATIONAL R_0=1.5) LABEL=f1
PRINT ARG=f1.* FILE=colvar
\end{DoxyVerb}


The following example calculates the difference of the current contact map with respect to a reference provided.

\begin{DoxyVerb}CONTACTMAP ...
ATOMS1=1,2 REFERENCE1=0.1 WEIGHT1=0.5 
ATOMS2=3,4 REFERENCE2=0.5 WEIGHT2=1.0 
ATOMS3=4,5 REFERENCE3=0.25 WEIGHT3=1.0 
ATOMS4=5,6 REFERENCE4=0.0 WEIGHT4=0.5 
SWITCH=(RATIONAL R_0=1.5) 
LABEL=cmap
CMDIST
... CONTACTMAP

PRINT ARG=cmap FILE=colvar
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}) \hypertarget{COORDINATION}{}\subsection{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N}\label{COORDINATION}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate coordination numbers.

This keyword can be used to calculate the number of contacts between two groups of atoms and is defined as \[ \sum_{i\in A} \sum_{i\in B} s_{ij} \] where $s_{ij}$ is 1 if the contact between atoms $i$ and $j$ is formed, zero otherwise. In practise, $s_{ij}$ is replaced with a switching function to make it differentiable. The default switching function is\+: \[ s_{ij} = \frac{ 1 - \left(\frac{{\bf r}_{ij}-d_0}{r_0}\right)^n } { 1 - \left(\frac{{\bf r}_{ij}-d_0}{r_0}\right)^m } \] but it can be changed using the optional S\+W\+I\+T\+C\+H option.

To make your calculation faster you can use a neighbor list, which makes it that only a relevant subset of the pairwise distance are calculated at every step.

If G\+R\+O\+U\+P\+B is empty, it will sum the $\frac{N(N-1)}{2}$ pairs in G\+R\+O\+U\+P\+A. This avoids computing twice permuted indexes (e.\+g. pair (i,j) and (j,i)) thus running at twice the speed.

Notice that if there are common atoms between G\+R\+O\+U\+P\+A and G\+R\+O\+U\+P\+B the switching function should be equal to one. These \char`\"{}self contacts\char`\"{} are discarded by plumed (since version 2.\+1), so that they actually count as \char`\"{}zero\char`\"{}.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &First list of atoms. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &Second list of atoms (if empty, N$\ast$(N-\/1)/2 pairs in G\+R\+O\+U\+P\+A are counted). For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) Perform the calculation in serial -\/ for debug purpose   \\\cline{1-2}
{\bfseries  P\+A\+I\+R } &( default=off ) Pair only 1st element of the 1st group with 1st element in the second, etc   \\\cline{1-2}
{\bfseries  N\+L\+I\+S\+T } &( default=off ) Use a neighbour list to speed up the calculation  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+L\+\_\+\+C\+U\+T\+O\+F\+F } &The cutoff for the neighbour list   \\\cline{1-2}
{\bfseries  N\+L\+\_\+\+S\+T\+R\+I\+D\+E } &The frequency with which we are updating the atoms in the neighbour list   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following example instructs plumed to calculate the total coordination number of the atoms in group 1-\/10 with the atoms in group 20-\/100. For atoms 1-\/10 coordination numbers are calculated that count the number of atoms from the second group that are within 0.\+3 nm of the central atom. A neighbour list is used to make this calculation faster, this neighbour list is updated every 100 steps. \begin{DoxyVerb}COORDINATION GROUPA=1-10 GROUPB=20-100 R_0=0.3 NLIST NL_CUTOFF=0.5 NL_STRIDE=100 
\end{DoxyVerb}


The following is a dummy example which should compute the value 0 because the self interaction of atom 1 is skipped. Notice that in plumed 2.\+0 \char`\"{}self interactions\char`\"{} were not skipped, and the same calculation should return 1. \begin{DoxyVerb}c: COORDINATION GROUPA=1 GROUPB=1 R_0=0.3
PRINT ARG=c STRIDE=10
\end{DoxyVerb}


\begin{DoxyVerb}c1: COORDINATION GROUPA=1-10 GROUPB=1-10 R_0=0.3
x: COORDINATION GROUPA=1-10 R_0=0.3
c2: COMBINE ARG=x COEFFICIENTS=2
# the two variables c1 and c2 should be identical, but the calculation of c2 is twice faster
# since it runs on half of the pairs. Notice that to get the same result you
# should double it
PRINT ARG=c1,c2 STRIDE=10
\end{DoxyVerb}
 See also \hyperlink{PRINT}{P\+R\+I\+N\+T} and \hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E} \hypertarget{CS2BACKBONE}{}\subsection{C\+S2\+B\+A\+C\+K\+B\+O\+N\+E}\label{CS2BACKBONE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This collective variable calculates a scoring function based on the comparison of backcalculated and experimental backbone chemical shifts for a protein (C\+A, C\+B, C', H, H\+A, N).

Cam\+Shift \cite{Kohlhoff:2009us} is employed to back calculate the chemical shifts that are then compared with a set of experimental values to generate a score \cite{Robustelli:2010dn} \cite{Granata:2013dk}.

It is also possible to back-\/calculate the chemical shifts from multiple replicas and then average them to perform Replica-\/\+Averaged Restrained M\+D simulations \cite{Camilloni:2012je} \cite{Camilloni:2013hs}.

In general the system for which chemical shifts are to be calculated must be completly included in A\+T\+O\+M\+S. It should also be made whole \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} before the the back-\/calculation.

H\+O\+W T\+O C\+O\+M\+P\+I\+L\+E I\+T

\hyperlink{_installation_installingalmost}{Installing P\+L\+U\+M\+E\+D with A\+L\+M\+O\+S\+T} on how to compile P\+L\+U\+M\+E\+D with A\+L\+M\+O\+S\+T.

H\+O\+W T\+O U\+S\+E I\+T

To use Cam\+Shift a set of experimental data is needed. Cam\+Shift uses backbone and Cb chemical shifts that must be provided as text files\+:

\begin{DoxyVerb}CAshifts.dat:
CBshifts.dat:
Cshifts.dat:
Hshifts.dat:
HAshifts.dat:
Nshifts.dat:
#1 0.0
2 55.5
3 58.4
.
.
#last 0.0
#last+1 (first) of second chain
.
#last of second chain
\end{DoxyVerb}


All of them must always be there. If a chemical shift for an atom of a residue is not available 0.\+0 must be used. So if for example all the Cb are not available all the chemical shifts for all the residues should be set as 0.\+0.

A template.\+pdb file is needed to the generate a topology of the protein within A\+L\+M\+O\+S\+T. For histidines in protonation states different from D the H\+I\+E/\+H\+I\+P name should be used in the template.\+pdb. G\+L\+H and A\+S\+H can be used for the alternative protonation of G\+L\+U and A\+S\+P. Non-\/standard amino acids and other molecules are not yet supported! If multiple chains are present the chain identifier must be in the standard P\+D\+B format, together with the T\+E\+R keyword at the end of each chain. Residues numbering should always go from 1 to N in both the chemical shifts files as well as in the template.\+pdb file. Two more keywords can be used to setup the topology\+: C\+Y\+S-\/\+D\+I\+S\+U to tell A\+L\+M\+O\+S\+T to look for disulphide bridges and T\+E\+R\+M\+I\+N\+I to define the protonation state of the the chain's termini (currently only D\+E\+F\+A\+U\+L\+T (N\+H3+, C\+O2-\/) and N\+O\+N\+E (N\+H, C\+O)).

Two more standard files are also needed\+: an A\+L\+M\+O\+S\+T force-\/field file, corresponding to the force-\/field family used in the M\+D code, (a03\+\_\+cs2\+\_\+gromacs.\+mdb for the amber family or all22\+\_\+gromacs.\+mdb for the charmm family) and camshift.\+db, both these files can be copied from almost/branches/almost-\/2.\+1/toppar.

All the above files must be in a single folder that must be specified with the keyword D\+A\+T\+A.

Additional material and examples can be also found in the tutorial \hyperlink{belfast-9}{Belfast tutorial\+: N\+M\+R constraints}

\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
case 1\+:

\begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1-174
cs: CS2BACKBONE ATOMS=1-174 DATA=data/ FF=a03_gromacs.mdb FLAT=0.0 NRES=13 ENSEMBLE
cse: RESTRAINT ARG=cs SLOPE=24 KAPPA=0 AT=0.
PRINT ARG=cs,cse.bias
\end{DoxyVerb}


case 2\+:

\begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1-174
cs: CS2BACKBONE ATOMS=1-174 DATA=data/ FF=a03_gromacs.mdb FLAT=1.0 NRES=13 TERMINI=DEFAULT,NONE CYS-DISU WRITE_CS=1000
PRINT ARG=cs
\end{DoxyVerb}


(See also \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}, \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T} and \hyperlink{PRINT}{P\+R\+I\+N\+T}) \hypertarget{DHENERGY}{}\subsection{D\+H\+E\+N\+E\+R\+G\+Y}\label{DHENERGY}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate Debye-\/\+Huckel interaction energy among G\+R\+O\+U\+P\+A and G\+R\+O\+U\+P\+B.

This variable calculates the electrostatic interaction among G\+R\+O\+U\+P\+A and G\+R\+O\+U\+P\+B using a Debye-\/\+Huckel approximation defined as \[ \frac{1}{4\pi\epsilon_r\epsilon_0} \sum_{i\in A} \sum_{j \in B} q_i q_j \frac{e^{-\kappa |{\bf r}_{ij}|}}{|{\bf r}_{ij}|} \]

This collective variable can be used to analyze or induce electrostatically driven reactions \cite{do13jctc}. Notice that the value of the D\+H\+E\+N\+E\+R\+G\+Y is returned in plumed units (see \hyperlink{UNITS}{U\+N\+I\+T\+S}).

If G\+R\+O\+U\+P\+B is empty, it will sum the N$\ast$(N-\/1)/2 pairs in G\+R\+O\+U\+P\+A. This avoids computing twice permuted indexes (e.\+g. pair (i,j) and (j,i)) thus running at twice the speed.

Notice that if there are common atoms between G\+R\+O\+U\+P\+A and G\+R\+O\+U\+P\+B their interaction is discarded.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &First list of atoms. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &Second list of atoms (if empty, N$\ast$(N-\/1)/2 pairs in G\+R\+O\+U\+P\+A are counted). For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  I } &( default=1.\+0 ) Ionic strength (M)   \\\cline{1-2}
{\bfseries  T\+E\+M\+P } &( default=300.\+0 ) Simulation temperature (K)   \\\cline{1-2}
{\bfseries  E\+P\+S\+I\+L\+O\+N } &( default=80.\+0 ) Dielectric constant of solvent   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) Perform the calculation in serial -\/ for debug purpose   \\\cline{1-2}
{\bfseries  P\+A\+I\+R } &( default=off ) Pair only 1st element of the 1st group with 1st element in the second, etc   \\\cline{1-2}
{\bfseries  N\+L\+I\+S\+T } &( default=off ) Use a neighbour list to speed up the calculation  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+L\+\_\+\+C\+U\+T\+O\+F\+F } &The cutoff for the neighbour list   \\\cline{1-2}
{\bfseries  N\+L\+\_\+\+S\+T\+R\+I\+D\+E } &The frequency with which we are updating the atoms in the neighbour list  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
\begin{DoxyVerb}# this is printing the electrostatic interaction between two groups of atoms
dh: DHENERGY GROUPA=1-10 GROUPB=11-20 EPSILON=80.0 I=0.1 TEMP=300.0
PRINT ARG=dh
\end{DoxyVerb}
 (see also \hyperlink{PRINT}{P\+R\+I\+N\+T}) 
\end{DoxyParagraph}
\hypertarget{DIHCOR}{}\subsection{D\+I\+H\+C\+O\+R}\label{DIHCOR}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Measures the degree of similarity between dihedral angles.

This colvar calculates the following quantity.

\[ s = \frac{1}{2} \sum_i \left[ 1 + \cos( \phi_i - \psi_i ) \right] \]

where the $\phi_i$ and $\psi$ values and the instantaneous values for the \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} angles of interest.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following provides an example input for the dihcor action

\begin{DoxyVerb}DIHCOR ...
  ATOMS1=1,2,3,4,5,6,7,8
  ATOMS2=5,6,7,8,9,10,11,12
  LABEL=dih
... DIHCOR
PRINT ARG=dih FILE=colvar STRIDE=10
\end{DoxyVerb}


In the above input we are calculating the correation between the torsion angle involving atoms 1, 2, 3 and 4 and the torsion angle involving atoms 5, 6, 7 and 8. This is then added to the correlation betwene the torsion angle involving atoms 5, 6, 7 and 8 and the correlation angle involving atoms 9, 10, 11 and 12.

Writing out the atoms involved in all the torsions in this way can be rather tedious. Thankfully if you are working with protein you can avoid this by using the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command. P\+L\+U\+M\+E\+D uses the pdb file that you provide to this command to learn about the topology of the protein molecule. This means that you can specify torsion angles using the following syntax\+:

\begin{DoxyVerb}MOLINFO MOLTYPE=protein STRUCTURE=myprotein.pdb
DIHCOR ...
ATOMS1=@phi-3,@psi-3
ATOMS2=@psi-3,@phi-4
ATOMS4=@phi-4,@psi-4
... DIHCOR
PRINT ARG=dih FILE=colvar STRIDE=10
\end{DoxyVerb}


Here, @phi-\/3 tells plumed that you would like to calculate the $\phi$ angle in the third residue of the protein. Similarly @psi-\/4 tells plumed that you want to calculate the $\psi$ angle of the 4th residue of the protein. \hypertarget{DIPOLE}{}\subsection{D\+I\+P\+O\+L\+E}\label{DIPOLE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the dipole moment for a group of atoms.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P } &the group of atoms we are calculating the dipole moment for. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following tells plumed to calculate the dipole of the group of atoms containing the atoms from 1-\/10 and print it every 5 steps \begin{DoxyVerb}d: DIPOLE GROUP=1-10
PRINT FILE=output STRIDE=5 ARG=5
\end{DoxyVerb}
 (see also \hyperlink{PRINT}{P\+R\+I\+N\+T})
\end{DoxyParagraph}
\begin{DoxyAttention}{Attention}
If the total charge Q of the group in non zero, then a charge Q/\+N will be subtracted to every atom, where N is the number of atoms. This implies that the dipole (which for a charged system depends on the position) is computed on the geometric center of the group. 
\end{DoxyAttention}
\hypertarget{DISTANCE}{}\subsection{D\+I\+S\+T\+A\+N\+C\+E}\label{DISTANCE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the distance between a pair of atoms.

By default the distance is computed taking into account periodic boundary conditions. This behavior can be changed with the N\+O\+P\+B\+C flag. Moreover, single components in cartesian space (x,y, and z, with C\+O\+M\+P\+O\+N\+E\+N\+T\+S) or single components projected to the three lattice vectors (a,b, and c, with S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S) can be also computed.

Notice that Cartesian components will not have the proper periodicity! If you have to study e.\+g. the permeation of a molecule across a membrane, better to use S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default the value of the calculated quantity can be referenced elsewhere in the input file by using the label of the action. Alternatively this Action can be used to be used to calculate the following quantities by employing the keywords listed below. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  x } &{\bfseries  C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the x-\/component of the vector connecting the two atoms   \\\cline{1-3}
{\bfseries  y } &{\bfseries  C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the y-\/component of the vector connecting the two atoms   \\\cline{1-3}
{\bfseries  z } &{\bfseries  C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the z-\/component of the vector connecting the two atoms   \\\cline{1-3}
{\bfseries  a } &{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the normalized projection on the first lattice vector of the vector connecting the two atoms   \\\cline{1-3}
{\bfseries  b } &{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the normalized projection on the second lattice vector of the vector connecting the two atoms   \\\cline{1-3}
{\bfseries  c } &{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the normalized projection on the third lattice vector of the vector connecting the two atoms   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the pair of atom that we are calculating the distance between. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  C\+O\+M\+P\+O\+N\+E\+N\+T\+S } &( default=off ) calculate the x, y and z components of the distance separately and store them as label.\+x, label.\+y and label.\+z   \\\cline{1-2}
{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S } &( default=off ) calculate the a, b and c scaled components of the distance separately and store them as label.\+a, label.\+b and label.\+c  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to print the distance between atoms 3 and 5, the distance between atoms 2 and 4 and the x component of the distance between atoms 2 and 4. \begin{DoxyVerb}d1:  DISTANCE ATOMS=3,5
d2:  DISTANCE ATOMS=2,4
d2c: DISTANCE ATOMS=2,4 COMPONENTS
PRINT ARG=d1,d2,d2c.x
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The following input computes the end-\/to-\/end distance for a polymer of 100 atoms and keeps it at a value around 5. \begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1-100
e2e: DISTANCE ATOMS=1,100 NOPBC
RESTRAINT ARG=e2e KAPPA=1 AT=5
\end{DoxyVerb}
 (See also \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} and \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T}).

Notice that N\+O\+P\+B\+C is used to be sure that if the end-\/to-\/end distance is larger than half the simulation box the distance is compute properly. Also notice that, since many M\+D codes break molecules across cell boundary, it might be necessary to use the \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} keyword (also notice that it should be {\itshape before} distance). The list of atoms provided to W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S here contains all the atoms between 1 and 100. Strictly speaking, this is not necessary. If you know for sure that atoms with difference in the index say equal to 10 are {\itshape not} going to be farther than half cell you can e.\+g. use \begin{DoxyVerb}WHOLEMOLECULES ENTITY0=1,10,20,30,40,50,60,70,80,90,100
e2e: DISTANCE ATOMS=1,100 NOPBC
RESTRAINT ARG=e2e KAPPA=1 AT=5
\end{DoxyVerb}
 Just be sure that the ordered list provide to W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S has the following properties\+:
\begin{DoxyItemize}
\item Consecutive atoms should be closer than half-\/cell throughout the entire simulation.
\item Atoms required later for the distance (e.\+g. 1 and 100) should be included in the list
\end{DoxyItemize}

The following example shows how to take into account periodicity e.\+g. in z-\/component of a distance \begin{DoxyVerb}# this is a center of mass of a large group
c: COM ATOMS=1-100
# this is the distance between atom 101 and the group
d: DISTANCE ATOMS=c,101 COMPONENTS
# this makes a new variable, dd, equal to d and periodic, with domain -10,10
# this is the right choise if e.g. the cell is orthorombic and its size in
# z direction is 20.
dz: COMBINE ARG=d.z PERIODIC=-10,10
# metadynamics on dd
METAD ARG=dz SIGMA=0.1 HEIGHT=0.1 PACE=200
\end{DoxyVerb}
 (see also \hyperlink{COM}{C\+O\+M}, \hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E}, and \hyperlink{METAD}{M\+E\+T\+A\+D})

Using S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S this problem should not arise because they are always periodic with domain (-\/0.\+5,+0.5). \hypertarget{ENERGY}{}\subsection{E\+N\+E\+R\+G\+Y}\label{ENERGY}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the total energy of the simulation box.

Total energy can be biased with umbrella sampling \cite{bart-karp98jpcb} or with well tempered metadynamics \cite{Bonomi:2009p17935}.

Notice that this C\+V could be unavailable with some M\+D code. When it is available, and when also replica exchange is available, metadynamics applied to E\+N\+E\+R\+G\+Y can be used to decrease the number of required replicas.

\begin{DoxyRefDesc}{Bug}
\item[\hyperlink{bug__bug000001}{Bug}]Acceptance for replica exchange when \hyperlink{ENERGY}{E\+N\+E\+R\+G\+Y} is biased is computed correctly only of all the replicas has the same potential energy function. This is for instance not true when using G\+R\+O\+M\+A\+C\+S with lambda replica exchange of with plumed-\/hrex branch.\end{DoxyRefDesc}


\begin{DoxyParagraph}{Examples}
The following input instructs plumed to print the energy of the system \begin{DoxyVerb}ENERGY LABEL=ene
PRINT ARG=ene
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}). 
\end{DoxyParagraph}
\hypertarget{FAKE}{}\subsection{F\+A\+K\+E}\label{FAKE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This is a fake colvar container used by cltools or various other actions and just support input and period definition

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the fake atom index, a number is enough. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  P\+E\+R\+I\+O\+D\+I\+C } &if the output of your function is periodic then you should specify the periodicity of the function. If the output is not periodic you must state this using P\+E\+R\+I\+O\+D\+I\+C=N\+O,N\+O (one for the lower and the other for the upper boundary). For multicomponents then it is P\+E\+R\+I\+O\+D\+I\+C=mincomp1,maxcomp1,mincomp2,maxcomp2 etc   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  C\+O\+M\+P\+O\+N\+E\+N\+T\+S } &additional componnets that this variable is supposed to have. Periodicity is ruled by P\+E\+R\+I\+O\+D\+I\+C keyword  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\begin{DoxyVerb}FAKE ATOMS=1 PERIODIC=-3.14,3.14   LABEL=d2
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}). \hypertarget{GPROPERTYMAP}{}\subsection{G\+P\+R\+O\+P\+E\+R\+T\+Y\+M\+A\+P}\label{GPROPERTYMAP}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the mapping \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Property maps but with a more flexible framework for the distance metric being used.

This colvar calculates a property map using the formalism developed by Spiwork \cite{Spiwok:2011ce}. In essence if you have the value of some property, $X_i$, that it takes at a set of high-\/dimensional positions then you calculate the value of the property at some arbitrary point in the high-\/dimensional space using\+:

\[ X=\frac{\sum_i X_i*\exp(-\lambda D_i(x))}{\sum_i \exp(-\lambda D_i(x))} \]

Within P\+L\+U\+M\+E\+D there are multiple ways to define the distance from a high-\/dimensional configuration, $D_i$. You could calculate the R\+M\+S\+D distance or you could calculate the ammount by which a set of collective variables change. As such this implementation of the propertymap allows one to use all the different distance metric that are discussed in \hyperlink{dists}{Distances from reference configurations}. This is as opposed to the alternative implementation \hyperlink{PROPERTYMAP}{P\+R\+O\+P\+E\+R\+T\+Y\+M\+A\+P} which is a bit faster but which only allows one to use the R\+M\+S\+D distance.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a pdb file containing the set of reference configurations   \\\cline{1-2}
{\bfseries  P\+R\+O\+P\+E\+R\+T\+Y } &the property to be used in the index. This should be in the R\+E\+M\+A\+R\+K of the reference   \\\cline{1-2}
{\bfseries  T\+Y\+P\+E } &( default=O\+P\+T\+I\+M\+A\+L ) the manner in which distances are calculated. More information on the different metrics that are available in P\+L\+U\+M\+E\+D can be found in the section of the manual on \hyperlink{dists}{Distances from reference configurations}   \\\cline{1-2}
{\bfseries  L\+A\+M\+B\+D\+A } &the value of the lambda parameter for paths   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  D\+I\+S\+A\+B\+L\+E\+\_\+\+C\+H\+E\+C\+K\+S } &( default=off ) disable checks on reference input structures.   \\\cline{1-2}
{\bfseries  N\+O\+Z\+P\+A\+T\+H } &( default=off ) do not calculate the zpath position   \\\cline{1-2}
{\bfseries  N\+O\+M\+A\+P\+P\+I\+N\+G } &( default=off ) do not calculate the position on the manifold  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\hypertarget{GYRATION}{}\subsection{G\+Y\+R\+A\+T\+I\+O\+N}\label{GYRATION}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the radius of gyration, or other properties related to it.

The different properties can be calculated and selected by the T\+Y\+P\+E keyword\+: the Radius of Gyration (R\+A\+D\+I\+U\+S); the Trace of the Gyration Tensor (T\+R\+A\+C\+E); the Largest Principal Moment of the Gyration Tensor (G\+T\+P\+C\+\_\+1); the middle Principal Moment of the Gyration Tensor (G\+T\+P\+C\+\_\+2); the Smallest Principal Moment of the Gyration Tensor (G\+T\+P\+C\+\_\+3); the Asphericiry (A\+S\+P\+H\+E\+R\+I\+C\+I\+T\+Y); the Acylindricity (A\+C\+Y\+L\+I\+N\+D\+R\+I\+C\+I\+T\+Y); the Relative Shape Anisotropy (K\+A\+P\+P\+A2); the Smallest Principal Radius Of Gyration (G\+Y\+R\+A\+T\+I\+O\+N\+\_\+3); the Middle Principal Radius of Gyration (G\+Y\+R\+A\+T\+I\+O\+N\+\_\+2); the Largest Principal Radius of Gyration (G\+Y\+R\+A\+T\+I\+O\+N\+\_\+1). A derivation of all these different variants can be found in \cite{Vymetal:2011gv}

The radius of gyration is calculated using\+:

\[ s_{\rm Gyr}=\Big ( \frac{\sum_i^{n} m_i \vert {r}_i -{r}_{\rm COM} \vert ^2 }{\sum_i^{n} m_i} \Big)^{1/2} \]

with the position of the center of mass ${r}_{\rm COM}$ given by\+:

\[ {r}_{\rm COM}=\frac{\sum_i^{n} {r}_i\ m_i }{\sum_i^{n} m_i} \]

The radius of gyration is calculated without applying periodic boundary conditions so the atoms used for the calculation should all be part of the same molecule that should be made whole before calculating the cv, \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the group of atoms that you are calculating the Gyration Tensor for. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+Y\+P\+E } &( default=R\+A\+D\+I\+U\+S ) The type of calculation relative to the Gyration Tensor you want to perform   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+T\+\_\+\+M\+A\+S\+S\+\_\+\+W\+E\+I\+G\+H\+T\+E\+D } &( default=off ) set the masses of all the atoms equal to one  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to print the radius of gyration of the chain containing atoms 10 to 20. \begin{DoxyVerb}WHOLEMOLECULES ENTITY0=10-20
GYRATION TYPE=RADIUS ATOMS=10-20 LABEL=rg
PRINT ARG=rg STRIDE=1 FILE=colvar 
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}) \hypertarget{NOE}{}\subsection{N\+O\+E}\label{NOE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculates the deviation of current distances from experimental N\+O\+E derived distances.

N\+O\+E distances are calculated between couple of atoms, averaging over equivalent couples, and compared with a set of reference distances. Distances can also be averaged over multiple replicas to perform replica-\/averaged simulations. Each N\+O\+E is defined by two groups containing the same number of atoms and by a reference distance, distances are calculated in pairs.

\[ NOE() = \sum_i^{noes}((\frac{1}{N_{eq}}\sum_j^{N_{eq}} (\frac{1}{r_j^6}))^{\frac{-1}{6}} - d_i^{exp})^2 \]

Reference distances can also be considered as upper limits only, in this case the sum is over a half parabola.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &the atoms involved in each of the contacts you wish to calculate. Keywords like G\+R\+O\+U\+P\+A1, G\+R\+O\+U\+P\+A2, G\+R\+O\+U\+P\+A3,... should be listed and one contact will be calculated for each A\+T\+O\+M keyword you specify. You can use multiple instances of this keyword i.\+e. G\+R\+O\+U\+P\+A1, G\+R\+O\+U\+P\+A2, G\+R\+O\+U\+P\+A3...   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &the atoms involved in each of the contacts you wish to calculate. Keywords like G\+R\+O\+U\+P\+B1, G\+R\+O\+U\+P\+B2, G\+R\+O\+U\+P\+B3,... should be listed and one contact will be calculated for each A\+T\+O\+M keyword you specify. You can use multiple instances of this keyword i.\+e. G\+R\+O\+U\+P\+B1, G\+R\+O\+U\+P\+B2, G\+R\+O\+U\+P\+B3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  W\+R\+I\+T\+E\+\_\+\+N\+O\+E } &( default=0 ) Write the back-\/calculated chemical shifts every \# steps.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  U\+P\+P\+E\+R\+\_\+\+L\+I\+M\+I\+T\+S } &( default=off ) Set to T\+R\+U\+E if you want to consider the reference distances as upper limits.   \\\cline{1-2}
{\bfseries  E\+N\+S\+E\+M\+B\+L\+E } &( default=off ) Set to T\+R\+U\+E if you want to average over multiple replicas.   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) Perform the calculation in serial -\/ for debug purpose  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+E\+D\+I\+S\+T } &A compulsory reference distance for a given N\+O\+E\+You can either specify a global reference value using N\+O\+E\+D\+I\+S\+T or one reference value for each contact. You can use multiple instances of this keyword i.\+e. N\+O\+E\+D\+I\+S\+T1, N\+O\+E\+D\+I\+S\+T2, N\+O\+E\+D\+I\+S\+T3...  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
In the following examples three noes are defined, the first is calculated based on the distances of atom 1-\/2 and 3-\/2; the second is defined by the distance 5-\/7 and the third by the distances 4-\/15,4-\/16,8-\/15,8-\/16.
\end{DoxyParagraph}
\begin{DoxyVerb}NOE ...
GROUPA1=1,3 GROUPB1=2,2 NOEDIST1=0.5
GROUPA2=5 GROUPB2=7 NOEDIST2=0.4
GROUPA3=4,4,8,8 GROUPB3=15,16,15,16 NOEDIST3=0.3
LABEL=noes
... NOE

PRINT ARG=noes FILE=colvar
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}) \hypertarget{PARABETARMSD}{}\subsection{P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D}\label{PARABETARMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the secondarystructure \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Probe the parallel beta sheet content of your protein structure.

Two protein segments containing three continguous residues can form a parallel beta sheet. Although if the two segments are part of the same protein chain they must be separated by a minimum of 3 residues to make room for the turn. This colvar thus generates the set of all possible six residue sections that could conceivably form a parallel beta sheet and calculates the R\+M\+S\+D distance between the configuration in which the residues find themselves and an idealized parallel beta sheet structure. These distances can be calculated by either aligning the instantaneous structure with the reference structure and measuring each atomic displacement or by calculating differences between the set of interatomic distances in the reference and instantaneous structures.

This colvar is based on the following reference \cite{pietrucci09jctc}. The authors of this paper use the set of distances from the parallel beta sheet configurations to measure the number of segments whose configuration resembles a parallel beta sheet. This is done by calculating the following sum of functions of the rmsd distances\+:

\[ s = \sum_i \frac{ 1 - \left(\frac{r_i-d_0}{r_0}\right)^n } { 1 - \left(\frac{r_i-d_0}{r_0}\right)^m } \]

where the sum runs over all possible segments of parallel beta sheet. By default the N\+N, M\+M and D\+\_\+0 parameters are set equal to those used in \cite{pietrucci09jctc}. The R\+\_\+0 parameter must be set by the user -\/ the value used in \cite{pietrucci09jctc} was 0.\+08 nm.

If you change the function in the above sum you can calculate quantities such as the average distance from a structure composed of only parallel beta sheets or the distance between the set of residues that is closest to a parallel beta sheet and the reference configuration. To do these sorts of calculations you can use the A\+V\+E\+R\+A\+G\+E and M\+I\+N keywords. In addition you can use the L\+E\+S\+S\+\_\+\+T\+H\+A\+N keyword if you would like to change the form of the switching function. If you use any of these options you no longer need to specify N\+N, R\+\_\+0, M\+M and D\+\_\+0.

Please be aware that for codes like gromacs you must ensure that plumed reconstructs the chains involved in your C\+V when you calculate this C\+V using anthing other than T\+Y\+P\+E=D\+R\+M\+S\+D. For more details as to how to do this see \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the number of structural units that are within a certain distance of a idealised secondary structure element. This quantity can then be referenced elsewhere in the input by using the label of the action. However, thes Action can also be used to calculate the following quantities by using the keywords as described below. The quantities then calculated can be referened using the label of the action followed by a dot and then the name from the table below. Please note that you can use the L\+E\+S\+S\+\_\+\+T\+H\+A\+N keyword more than once. The resulting components will be labelled {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 and so on unless you exploit the fact that these labels are customizable. In particular, by using the L\+A\+B\+E\+L keyword in the description of you L\+E\+S\+S\+\_\+\+T\+H\+A\+N function you can set name of the component that you are calculating

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+S\+I\+D\+U\+E\+S } &this command is used to specify the set of residues that could conceivably form part of the secondary structure. It is possible to use residues numbers as the various chains and residues should have been identified else using an instance of the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} action. If you wish to use all the residues from all the chains in your system you can do so by specifying all. Alternatively, if you wish to use a subset of the residues you can specify the particular residues you are interested in as a list of numbers. Please be aware that to form secondary structure elements your chain must contain at least N residues, where N is dependent on the particular secondary structure you are interested in. As such if you define portions of the chain with fewer than N residues the code will crash.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+Y\+P\+E } &( default=D\+R\+M\+S\+D ) the manner in which R\+M\+S\+D alignment is performed. Should be O\+P\+T\+I\+M\+A\+L, S\+I\+M\+P\+L\+E or D\+R\+M\+S\+D. For more details on the O\+P\+T\+I\+M\+A\+L and S\+I\+M\+P\+L\+E methods see \hyperlink{RMSD}{R\+M\+S\+D}. For more details on the D\+R\+M\+S\+D method see \hyperlink{DRMSD}{D\+R\+M\+S\+D}.   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function.   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  N\+N } &( default=8 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  S\+T\+Y\+L\+E } &( default=all ) Parallel beta sheets can either form in a single chain or from a pair of chains. If S\+T\+Y\+L\+E=all all chain configuration with the appropriate geometry are counted. If S\+T\+Y\+L\+E=inter only sheet-\/like configurations involving two chains are counted, while if S\+T\+Y\+L\+E=intra only sheet-\/like configurations involving a single chain are counted   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  S\+T\+R\+A\+N\+D\+S\+\_\+\+C\+U\+T\+O\+F\+F } &If in a segment of protein the two strands are further apart then the calculation of the actual R\+M\+S\+D is skipped as the structure is very far from being beta-\/sheet like. This keyword speeds up the calculation enormously when you are using the L\+E\+S\+S\+\_\+\+T\+H\+A\+N option. However, if you are using some other option, then this cannot be used  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input calculates the number of six residue segments of protein that are in an parallel beta sheet configuration.

\begin{DoxyVerb}MOLINFO STRUCTURE=helix.pdb
PARABETARMSD RESIDUES=all TYPE=DRMSD LESS_THAN={RATIONAL R_0=0.08 NN=8 MM=12} LABEL=a
\end{DoxyVerb}
 (see also \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}) \hypertarget{PATHMSD}{}\subsection{P\+A\+T\+H\+M\+S\+D}\label{PATHMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This Colvar calculates path collective variables.

This is the Path Collective Variables implementation ( see \cite{brand07} ). This variable computes the progress along a given set of frames that is provided in input (\char`\"{}sss\char`\"{} component) and the distance from them (\char`\"{}zzz\char`\"{} component). (see below).

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the following quantities. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  sss } &the position on the path   \\\cline{1-2}
{\bfseries  zzz } &the distance from the path   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  L\+A\+M\+B\+D\+A } &the lambda parameter is needed for smoothing, is in the units of plumed   \\\cline{1-2}
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &the pdb is needed to provide the various milestones   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+E\+I\+G\+H\+\_\+\+S\+I\+Z\+E } &size of the neighbor list   \\\cline{1-2}
{\bfseries  N\+E\+I\+G\+H\+\_\+\+S\+T\+R\+I\+D\+E } &how often the neighbor list needs to be calculated in time units  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
Here below is a case where you have defined three frames and you want to calculate the progress along the path and the distance from it in p1

\begin{DoxyVerb}p1: PATHMSD REFERENCE=file.pdb  LAMBDA=500.0 NEIGH_STRIDE=4 NEIGH_SIZE=8 
PRINT ARG=p1.sss,p1.zzz STRIDE=1 FILE=colvar FMT=%8.4f
\end{DoxyVerb}


note that N\+E\+I\+G\+H\+\_\+\+S\+T\+R\+I\+D\+E=4 N\+E\+I\+G\+H\+\_\+\+S\+I\+Z\+E=8 control the neighborlist parameter (optional but recommended for performance) and states that the neighbor list will be calculated every 4 timesteps and consider only the closest 8 member to the actual md snapshots.

In the R\+E\+F\+E\+R\+E\+N\+C\+E P\+D\+B file the frames must be separated either using E\+N\+D or E\+N\+D\+M\+D\+L.

\begin{DoxyNote}{Note}
The implementation of this collective variable and of \hyperlink{PROPERTYMAP}{P\+R\+O\+P\+E\+R\+T\+Y\+M\+A\+P} is shared, as well as most input options. 
\end{DoxyNote}
\hypertarget{PATH}{}\subsection{P\+A\+T\+H}\label{PATH}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the mapping \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Path collective variables with a more flexible framework for the distance metric being used.

The Path Collective Variables developed by Branduardi and co-\/workers \cite{brand07} allow one to compute the progress along a high-\/dimensional path and the distance from the high-\/dimensional path. The progress along the path (s) is computed using\+:

\[ s = \frac{ \sum_{i=1}^N i \exp( -\lambda R[X - X_i] ) }{ \sum_{i=1}^N \exp( -\lambda R[X - X_i] ) } \]

while the distance from the path (z) is measured using\+:

\[ z = -\frac{1}{\lambda} \ln\left[ \sum_{i=1}^N \exp( -\lambda R[X - X_i] ) \right] \]

In these expressions $N$ high-\/dimensional frames ( $X_i$) are used to describe the path in the high-\/dimensional space. The two expressions above are then functions of the distances from each of the high-\/dimensional frames $R[X - X_i]$. Within P\+L\+U\+M\+E\+D there are multiple ways to define the distance from a high-\/dimensional configuration. You could calculate the R\+M\+S\+D distance or you could calculate the ammount by which a set of collective variables change. As such this implementation of the path cv allows one to use all the difference distance metrics that are discussed in \hyperlink{dists}{Distances from reference configurations}. This is as opposed to the alternative implementation of path (\hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D}) which is a bit faster but which only allows one to use the R\+M\+S\+D distance.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a pdb file containing the set of reference configurations   \\\cline{1-2}
{\bfseries  T\+Y\+P\+E } &( default=O\+P\+T\+I\+M\+A\+L ) the manner in which distances are calculated. More information on the different metrics that are available in P\+L\+U\+M\+E\+D can be found in the section of the manual on \hyperlink{dists}{Distances from reference configurations}   \\\cline{1-2}
{\bfseries  L\+A\+M\+B\+D\+A } &the value of the lambda parameter for paths   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  D\+I\+S\+A\+B\+L\+E\+\_\+\+C\+H\+E\+C\+K\+S } &( default=off ) disable checks on reference input structures.   \\\cline{1-2}
{\bfseries  N\+O\+Z\+P\+A\+T\+H } &( default=off ) do not calculate the zpath position   \\\cline{1-2}
{\bfseries  N\+O\+S\+P\+A\+T\+H } &( default=off ) do not calculate the spath position  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\hypertarget{POSITION}{}\subsection{P\+O\+S\+I\+T\+I\+O\+N}\label{POSITION}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the components of the position of an atom.

Notice that single components will not have the proper periodicity! If you need the values to be consistent through P\+B\+C you should use S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S, which defines values that by construction are in the -\/0.\+5,0.\+5 domain. This is similar to the equivalent flag for \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}. Also notice that by default the minimal image distance from the origin is considered (can be changed with N\+O\+P\+B\+C).

\begin{DoxyAttention}{Attention}
This variable should be used with extreme care since it allows to easily go into troubles. See comments below.
\end{DoxyAttention}
This variable can be safely used only if Hamiltonian is not invariant for translation (i.\+e. there are other absolute positions which are biased, e.\+g. by position restraints) and cell size and shapes are fixed through the simulation.

If you are not in this situation and still want to use the absolute position of an atom you should first fix the reference frame. This can be done e.\+g. using \hyperlink{FIT_TO_TEMPLATE}{F\+I\+T\+\_\+\+T\+O\+\_\+\+T\+E\+M\+P\+L\+A\+T\+E}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the following quantities. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  x } &the x-\/component of the atom position   \\\cline{1-2}
{\bfseries  y } &the y-\/component of the atom position   \\\cline{1-2}
{\bfseries  z } &the z-\/component of the atom position   \\\cline{1-2}
\end{TabularC}


In addition the following quantities can be calculated by employing the keywords listed below

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  a } &{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the normalized projection on the first lattice vector of the atom position   \\\cline{1-3}
{\bfseries  b } &{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the normalized projection on the second lattice vector of the atom position   \\\cline{1-3}
{\bfseries  c } &{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S }  &the normalized projection on the third lattice vector of the atom position   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M } &the atom number. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+C\+A\+L\+E\+D\+\_\+\+C\+O\+M\+P\+O\+N\+E\+N\+T\+S } &( default=off ) calculate the a, b and c scaled components of the position separately and store them as label.\+a, label.\+b and label.\+c  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\begin{DoxyVerb}# align to a template
FIT_TO_TEMPLATE REFERENCE=ref.pdb
p: POSITION ATOM=3
PRINT ARG=p.x,p.y,p.z
\end{DoxyVerb}
 (see also \hyperlink{FIT_TO_TEMPLATE}{F\+I\+T\+\_\+\+T\+O\+\_\+\+T\+E\+M\+P\+L\+A\+T\+E}) \hypertarget{PROPERTYMAP}{}\subsection{P\+R\+O\+P\+E\+R\+T\+Y\+M\+A\+P}\label{PROPERTYMAP}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate generic property maps.

This Colvar calculates the property maps according to the work of Spiwok \cite{Spiwok:2011ce}.

Basically it calculates \begin{eqnarray} X=\frac{\sum_i X_i*\exp(-\lambda D_i(x))}{\sum_i \exp(-\lambda D_i(x))} \\ Y=\frac{\sum_i Y_i*\exp(-\lambda D_i(x))}{\sum_i \exp(-\lambda D_i(x))} \\ \cdots\\ zzz=-\frac{1}{\lambda}\log(\sum_i \exp(-\lambda D_i(x))) \end{eqnarray}

where the parameters $X_i$ and $Y_i$ are provided in the input pdb (allv.\+pdb in this case) and $D_i(x)$ is the M\+S\+D after optimal alignment calculated on the pdb frames you input (see Kearsley).

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
The names of the components in this action can be customized by the user in the actions input file. However, in addition to these customizable components the following quantities will always be output

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  zzz } &the minimum distance from the reference points   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  L\+A\+M\+B\+D\+A } &the lambda parameter is needed for smoothing, is in the units of plumed   \\\cline{1-2}
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &the pdb is needed to provide the various milestones   \\\cline{1-2}
{\bfseries  P\+R\+O\+P\+E\+R\+T\+Y } &the property to be used in the indexing\+: this goes in the R\+E\+M\+A\+R\+K field of the reference   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+E\+I\+G\+H\+\_\+\+S\+I\+Z\+E } &size of the neighbor list   \\\cline{1-2}
{\bfseries  N\+E\+I\+G\+H\+\_\+\+S\+T\+R\+I\+D\+E } &how often the neighbor list needs to be calculated in time units  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
\begin{DoxyVerb}p3: PROPERTYMAP REFERENCE=../../trajectories/path_msd/allv.pdb PROPERTY=X,Y LAMBDA=69087 NEIGH_SIZE=8 NEIGH_STRIDE=4
PRINT ARG=p3.X,p3.Y,p3.zzz STRIDE=1 FILE=colvar FMT=%8.4f
\end{DoxyVerb}

\end{DoxyParagraph}
note that N\+E\+I\+G\+H\+\_\+\+S\+T\+R\+I\+D\+E=4 N\+E\+I\+G\+H\+\_\+\+S\+I\+Z\+E=8 control the neighborlist parameter (optional but recommended for performance) and states that the neighbor list will be calculated every 4 timesteps and consider only the closest 8 member to the actual md snapshots.

In this case the input line instructs plumed to look for two properties X and Y with attached values in the R\+E\+M\+A\+R\+K line of the reference pdb (Note\+: No spaces from X and = and 1 !!!!). e.\+g.

\begin{DoxyVerb}REMARK X=1 Y=2 
ATOM      1  CL  ALA     1      -3.171   0.295   2.045  1.00  1.00
ATOM      5  CLP ALA     1      -1.819  -0.143   1.679  1.00  1.00
.......
END
REMARK X=2 Y=3 
ATOM      1  CL  ALA     1      -3.175   0.365   2.024  1.00  1.00
ATOM      5  CLP ALA     1      -1.814  -0.106   1.685  1.00  1.00
....
END
\end{DoxyVerb}


\begin{DoxyNote}{Note}
The implementation of this collective variable and of \hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D} is shared, as well as most input options. 
\end{DoxyNote}
\hypertarget{RDC}{}\subsection{R\+D\+C}\label{RDC}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculates the Residual Dipolar Coupling between two atoms.

The R\+D\+C between two atomic nuclei depends on the $\theta$ angle between the inter-\/nuclear vector and the external magnetic field. In isotropic media R\+D\+Cs average to zero because of the orientational averaging, but when the rotational symmetry is broken, either through the introduction of an alignment medium or for molecules with highly anisotropic paramagnetic susceptibility, R\+D\+Cs become measurable.

\[ D=D_{max}0.5(3\cos^2(\theta)-1) \]

where

\[ D_{max}=-\mu_0\gamma_1\gamma_2h/(8\pi^3r^3) \]

that is the maximal value of the dipolar coupling for the two nuclear spins with gyromagnetic ratio $\gamma$. $\mu$ is the magnetic constant and h is the Planck constant.

Common Gyromagnetic Ratios (C.\+G.\+S)
\begin{DoxyItemize}
\item H(1) 26.\+7513
\item C(13) 6.\+7261
\item N(15) -\/2.\+7116
\item N\+H -\/72.\+5388
\item C\+H 179.\+9319
\item C\+N -\/18.\+2385
\item C\+C 45.\+2404
\end{DoxyItemize}

This collective variable calculates the Residual Dipolar Coupling for a set of couple of atoms using the above definition. From the calculated R\+D\+Cs and a set of experimental values it calculates either their correlation or the squared quality factor

\[ Q^2=\frac{\sum_i(D_i-D^{exp}_i)^2}{\sum_i(D^{exp}_i)^2} \]

R\+D\+Cs report only on the fraction of molecules that is aligned, this means that comparing the R\+D\+Cs from a single structure in a M\+D simulation to the experimental values is not particularly meaningfull, from this point of view it is better to compare their correlation. The fraction of aligned molecules can be obtained by maximising the correlation between the calculated and the experimental R\+D\+Cs. This fraction can be used as a scaling factor in the calculation of the R\+D\+Cs in order to compare their values. The averaging of the R\+D\+Cs calculated with the above definition from a standard M\+D should result to 0 because of the rotational diffusion, but this variable can be used to break the rotational symmetry.

R\+D\+Cs can also be calculated using a Single Value Decomposition approach, in this case the code rely on the a set of function from the G\+N\+U Scientific Library (G\+S\+L). (With S\+V\+D forces are not currently implemented).

Replica-\/\+Averaged restrained simulations can be performed with this C\+V using the E\+N\+S\+E\+M\+B\+L\+E flag.

Additional material and examples can be also found in the tutorial \hyperlink{belfast-9}{Belfast tutorial\+: N\+M\+R constraints}

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the couple of atoms involved in each of the bonds for which you wish to calculate the R\+D\+C. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one dipolar coupling will be calculated for each A\+T\+O\+M\+S keyword you specify. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  W\+R\+I\+T\+E\+\_\+\+D\+C } &( default=0 ) Write the back-\/calculated dipolar couplings every \# steps.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  E\+N\+S\+E\+M\+B\+L\+E } &( default=off ) Set to T\+R\+U\+E if you want to average over multiple replicas.   \\\cline{1-2}
{\bfseries  C\+O\+R\+R\+E\+L\+A\+T\+I\+O\+N } &( default=off ) Set to T\+R\+U\+E if you want to kept constant the bonds distances.   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) Set to T\+R\+U\+E if you want to run the C\+V in serial.   \\\cline{1-2}
{\bfseries  S\+V\+D } &( default=off ) Set to T\+R\+U\+E if you want to backcalculate using Single Value Decomposition (need G\+S\+L at compilation time).  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  C\+O\+U\+P\+L\+I\+N\+G } &Add an experimental value for each coupling. You can use multiple instances of this keyword i.\+e. C\+O\+U\+P\+L\+I\+N\+G1, C\+O\+U\+P\+L\+I\+N\+G2, C\+O\+U\+P\+L\+I\+N\+G3...   \\\cline{1-2}
{\bfseries  G\+Y\+R\+O\+M } &Add the product of the gyromagnetic constants for each bond. You can use multiple instances of this keyword i.\+e. G\+Y\+R\+O\+M1, G\+Y\+R\+O\+M2, G\+Y\+R\+O\+M3...   \\\cline{1-2}
{\bfseries  S\+C\+A\+L\+E } &Add a scaling factor to take into account concentration and other effects. You can use multiple instances of this keyword i.\+e. S\+C\+A\+L\+E1, S\+C\+A\+L\+E2, S\+C\+A\+L\+E3...   \\\cline{1-2}
{\bfseries  B\+O\+N\+D\+L\+E\+N\+G\+T\+H } &Set a fixed length for for the bonds distances. You can use multiple instances of this keyword i.\+e. B\+O\+N\+D\+L\+E\+N\+G\+T\+H1, B\+O\+N\+D\+L\+E\+N\+G\+T\+H2, B\+O\+N\+D\+L\+E\+N\+G\+T\+H3...  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
In the following example five N-\/\+H R\+D\+Cs are defined and their correlation with respect to a set of experimental data is calculated.
\end{DoxyParagraph}
\begin{DoxyVerb}RDC ...
GYROM=-72.5388
SCALE=1.0 
CORRELATION
ATOMS1=20,21 COUPLING1=8.17
ATOMS2=37,38 COUPLING2=-8.271
ATOMS3=56,57 COUPLING3=-10.489
ATOMS4=76,77 COUPLING4=-9.871
ATOMS5=92,93 COUPLING5=-9.152
LABEL=nh
... RDC 

rdce: RESTRAINT ARG=nh KAPPA=0. SLOPE=-25000.0 AT=1.

PRINT ARG=nh,rdce.bias FILE=colvar
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}, \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T}) \hypertarget{TEMPLATE}{}\subsection{T\+E\+M\+P\+L\+A\+T\+E}\label{TEMPLATE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This file provides a template for if you want to introduce a new C\+V.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+E\+M\+P\+L\+A\+T\+E\+\_\+\+I\+N\+P\+U\+T } &the keyword with which you specify what atoms to use should be added like this. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+E\+M\+P\+L\+A\+T\+E\+\_\+\+C\+O\+M\+P\+U\+L\+S\+O\+R\+Y } &all compulsory keywords should be added like this with a description here   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  T\+E\+M\+P\+L\+A\+T\+E\+\_\+\+D\+E\+F\+A\+U\+L\+T\+\_\+\+O\+F\+F\+\_\+\+F\+L\+A\+G } &( default=off ) flags that are by default not performed should be specified like this   \\\cline{1-2}
{\bfseries  T\+E\+M\+P\+L\+A\+T\+E\+\_\+\+D\+E\+F\+A\+U\+L\+T\+\_\+\+O\+N\+\_\+\+F\+L\+A\+G } &( default=on ) flags that are by default performed should be specified like this  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+E\+M\+P\+L\+A\+T\+E\+\_\+\+O\+P\+T\+I\+O\+N\+A\+L } &all optional keywords that have input should be added like a description here  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\hypertarget{TORSION}{}\subsection{T\+O\+R\+S\+I\+O\+N}\label{TORSION}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate a torsional angle.

This command can be used to compute the torsion between four atoms or alternatively to calculate the angle between two vectors projected on the plane orthogonal to an axis.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the four atoms involved in the torsional angle   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+X\+I\+S } &two atoms that define an axis. You can use this to find the angle in the plane perpendicular to the axis between the vectors specified using the V\+E\+C\+T\+O\+R1 and V\+E\+C\+T\+O\+R2 keywords.   \\\cline{1-2}
{\bfseries  V\+E\+C\+T\+O\+R1 } &two atoms that define a vector. You can use this in combination with V\+E\+C\+T\+O\+R2 and A\+X\+I\+S   \\\cline{1-2}
{\bfseries  V\+E\+C\+T\+O\+R2 } &two atoms that define a vector. You can use this in combination with V\+E\+C\+T\+O\+R1 and A\+X\+I\+S   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  C\+O\+S\+I\+N\+E } &( default=off ) calculate cosine instead of dihedral  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
This input tells plumed to print the torsional angle between atoms 1, 2, 3 and 4 on file C\+O\+L\+V\+A\+R. \begin{DoxyVerb}t: TORSION ATOMS=1,2,3,4
# this is an alternative, equivalent, definition:
# t: TORSION VECTOR1=2,1 AXIS=2,3 VECTOR2=3,4
PRINT ARG=t FILE=COLVAR
\end{DoxyVerb}


If you are working with a protein you can specify the special named torsion angles $\phi$, $\psi$, $\omega$ and $\chi_1$ by using T\+O\+R\+S\+I\+O\+N in combination with the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command. This can be done by using the following syntax.

\begin{DoxyVerb}MOLINFO MOLTYPE=protein STRUCTURE=myprotein.pdb
t1: TORSION ATOMS=@phi-3
t2: TORSION ATOMS=@psi-4
PRINT ARG=t1,t2 FILE=colvar STRIDE=10
\end{DoxyVerb}


Here, @phi-\/3 tells plumed that you would like to calculate the $\phi$ angle in the third residue of the protein. Similarly @psi-\/4 tells plumed that you want to calculate the $\psi$ angle of the 4th residue of the protein. \hypertarget{VOLUME}{}\subsection{V\+O\+L\+U\+M\+E}\label{VOLUME}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the volume of the simulation box.

\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following input tells plumed to print the volume of the system \begin{DoxyVerb}VOLUME LABEL=vol
PRINT ARG=vol
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}). 
\end{DoxyParagraph}
\hypertarget{dists}{}\section{Distances from reference configurations}\label{dists}
One colvar that has been shown to be very sucessful in studying protein folding is the distance between the instantaneous configuration and a reference configuration -\/ often the structure of the folded state. When the free energy of a protein is shown as a function of this collective variable there is a minima for low values of the C\+V, which is due to the folded state of the protein. There is then a second minima at higher values of the C\+V, which is the minima corresponding to the unfolded state.

A slight problem with this sort of collective variable is that there are many different ways of calculating the distance from a particular reference structure. The simplest -\/ adding together the distances by which each of the atoms has been translated in going from the reference configuration to the instantanous configuration -\/ is not particularly sensible. A distance calculated in this way does not neglect translation of the center of mass of the molecule and rotation of the frame of reference. A common practise is thus to remove these components by calculating the \hyperlink{RMSD}{R\+M\+S\+D} distance between the reference and instantaneous configurations. This is not the only way to calculate the distance, however. One could also calculate the total ammount by which a large number of collective variables change in moving from the reference to the instaneous configurations. One could even combine R\+M\+S\+D distances with the ammount the collective variables change. A full list of the ways distances can be measured in P\+L\+U\+M\+E\+D is given below\+:

\begin{TabularC}{2}
\hline
\hyperlink{DRMSD}{D\+R\+M\+S\+D}  &Calculate the distance R\+M\+S\+D with respect to a reference structure.   \\\cline{1-2}
\hyperlink{MULTI-RMSD}{M\+U\+L\+T\+I-\/\+R\+M\+S\+D}  &Calculate the R\+M\+S\+D distance moved by a number of separated domains from their positions in a reference structure.   \\\cline{1-2}
\hyperlink{RMSD}{R\+M\+S\+D}  &Calculate the R\+M\+S\+D with respect to a reference structure.   \\\cline{1-2}
\hyperlink{TARGET}{T\+A\+R\+G\+E\+T}  &This function measures the pythagorean distance from a particular structure measured in the space defined by some set of collective variables.  \\\cline{1-2}
\end{TabularC}


These options for calculating distances are re-\/used in a number of places in the code. For instance they are used in some of the analysis algorithms that are implemented in P\+L\+U\+M\+E\+D and in \hyperlink{PATH}{P\+A\+T\+H} collective variables. \hypertarget{DRMSD}{}\subsection{D\+R\+M\+S\+D}\label{DRMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the distance R\+M\+S\+D with respect to a reference structure.

To calculate the root-\/mean-\/square deviation between the atoms in two configurations you must first superimpose the two structures in some ways. Obviously, it is the internal vibrational motions of the structure -\/ i.\+e. not the translations and rotations -\/ that are interesting. However, aligning two structures by removing the translational and rotational motions is not easy. Furthermore, in some cases there can be alignment issues caused by so-\/called frame-\/fitting problems. It is thus often cheaper and easier to calculate the distances between all the pairs of atoms. The distance between the two structures, $\mathbf{X}^a$ and $\mathbf{X}^b$ can then be measured as\+:

\[ d(\mathbf{X}^A, \mathbf{X}^B) = \frac{1}{N(N-1)} \sum_{i \ne j} [ d(\mathbf{x}_i^a,\mathbf{x}_j^a) - d(\mathbf{x}_i^b,\mathbf{x}_j^b) ]^2 \]

where $N$ is the number of atoms and $d(\mathbf{x}_i,\mathbf{x}_j)$ represents the distance between atoms $i$ and $j$. Clearly, this representation of the configuration is invariant to translation and rotation. However, it can become expensive to calculate when the number of atoms is large. This can be resolved within the D\+R\+M\+S\+D colvar by setting L\+O\+W\+E\+R\+\_\+\+C\+U\+T\+O\+F\+F and U\+P\+P\+E\+R\+\_\+\+C\+U\+T\+O\+F\+F. These keywords ensure that only pairs of atoms that are within a certain range are incorporated into the above sum.

In P\+D\+B files the atomic coordinates and box lengths should be in Angstroms unless you are working with natural units. If you are working with natural units then the coordinates should be in your natural length unit. For more details on the P\+D\+B file format visit \href{http://www.wwpdb.org/docs.html}{\tt http\+://www.\+wwpdb.\+org/docs.\+html}

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a file in pdb format containing the reference structure and the atoms involved in the C\+V.   \\\cline{1-2}
{\bfseries  L\+O\+W\+E\+R\+\_\+\+C\+U\+T\+O\+F\+F } &only pairs of atoms further than L\+O\+W\+E\+R\+\_\+\+C\+U\+T\+O\+F\+F are considered in the calculation.   \\\cline{1-2}
{\bfseries  U\+P\+P\+E\+R\+\_\+\+C\+U\+T\+O\+F\+F } &only pairs of atoms closer than U\+P\+P\+E\+R\+\_\+\+C\+U\+T\+O\+F\+F are considered in the calculation.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following tells plumed to calculate the distance R\+M\+S\+D between the positions of the atoms in the reference file and their instantaneous position. Only pairs of atoms whose distance in the reference structure is within 0.\+1 and 0.\+8 nm are considered.

\begin{DoxyVerb}DRMSD REFERENCE=file.pdb LOWER_CUTOFF=0.1 UPPER_CUTOFF=0.8
\end{DoxyVerb}


... \hypertarget{MULTI-RMSD}{}\subsection{M\+U\+L\+T\+I-\/\+R\+M\+S\+D}\label{MULTI-RMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the R\+M\+S\+D distance moved by a number of separated domains from their positions in a reference structure.

When you have large proteins the calculation of the root mean squared deviation between all the atoms in a reference structure and the instantaneous configuration becomes prohibitively expensive. You may thus instead want to calculate the R\+M\+S\+D between the atoms in a set of domains of your protein and your reference structure. That is to say\+:

\[ d(X,X_r) = \sqrt{ \sum_{i} w_i\vert X_i - X_i' \vert^2 } \]

where here the sum is over the domains of the protein, $X_i$ represents the positions of the atoms in domain $i$ in the instantaneous configuration and $X_i'$ is the positions of the atoms in domain $i$ in the reference configuration. $w_i$ is an optional weight.

The distances for each of the domains in the above sum can be calculated using the \hyperlink{DRMSD}{D\+R\+M\+S\+D} or \hyperlink{RMSD}{R\+M\+S\+D} measures or using a combination of these distance. The reference configuration is specified in a pdb file like the one below\+:

\begin{DoxyVerb}ATOM      2  O   ALA     2      -0.926  -2.447  -0.497  1.00  1.00      DIA  O
ATOM      4  HNT ALA     2       0.533  -0.396   1.184  1.00  1.00      DIA  H
ATOM      6  HT1 ALA     2      -0.216  -2.590   1.371  1.00  1.00      DIA  H
ATOM      7  HT2 ALA     2      -0.309  -1.255   2.315  1.00  1.00      DIA  H
ATOM      8  HT3 ALA     2      -1.480  -1.560   1.212  1.00  1.00      DIA  H
ATOM      9  CAY ALA     2      -0.096   2.144  -0.669  1.00  1.00      DIA  C
ATOM     10  HY1 ALA     2       0.871   2.385  -0.588  1.00  1.00      DIA  H
TER
ATOM     12  HY3 ALA     2      -0.520   2.679  -1.400  1.00  1.00      DIA  H
ATOM     14  OY  ALA     2      -1.139   0.931  -0.973  1.00  1.00      DIA  O
ATOM     16  HN  ALA     2       1.713   1.021  -0.873  1.00  1.00      DIA  H
ATOM     18  HA  ALA     2       0.099  -0.774  -2.218  1.00  1.00      DIA  H
ATOM     19  CB  ALA     2       2.063  -1.223  -1.276  1.00  1.00      DIA  C
ATOM     20  HB1 ALA     2       2.670  -0.716  -2.057  1.00  1.00      DIA  H
ATOM     21  HB2 ALA     2       2.556  -1.051  -0.295  1.00  1.00      DIA  H
ATOM     22  HB3 ALA     2       2.070  -2.314  -1.490  1.00  1.00      DIA  H
END
\end{DoxyVerb}


with the T\+E\+R keyword being used to separate the various domains in you protein.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a file in pdb format containing the reference structure and the atoms involved in the C\+V.   \\\cline{1-2}
{\bfseries  T\+Y\+P\+E } &( default=M\+U\+L\+T\+I-\/\+S\+I\+M\+P\+L\+E ) the manner in which R\+M\+S\+D alignment is performed. Should be M\+U\+L\+T\+I-\/\+O\+P\+T\+I\+M\+A\+L, M\+U\+L\+T\+I-\/\+O\+P\+T\+I\+M\+A\+L-\/\+F\+A\+S\+T, M\+U\+L\+T\+I-\/\+S\+I\+M\+P\+L\+E or M\+U\+L\+T\+I-\/\+D\+R\+M\+S\+D.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+Q\+U\+A\+R\+E\+D } &( default=off ) This should be setted if you want M\+S\+D instead of R\+M\+S\+D  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following tells plumed to calculate the R\+M\+S\+D distance between the positions of the atoms in the reference file and their instantaneous position. The Kearseley algorithm for each of the domains.

\begin{DoxyVerb}MULTI-RMSD REFERENCE=file.pdb TYPE=MULTI-OPTIMAL
\end{DoxyVerb}


The following tells plumed to calculate the R\+M\+S\+D distance btween the positions of the atoms in the domains of reference the reference structure and their instantaneous positions. Here distances are calculated using the \hyperlink{DRMSD}{D\+R\+M\+S\+D} measure.

\begin{DoxyVerb}MULTI-RMSD REFERENCE=file.pdb TYPE=MULTI-DRMSD
\end{DoxyVerb}


... \hypertarget{RMSD}{}\subsection{R\+M\+S\+D}\label{RMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the colvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the R\+M\+S\+D with respect to a reference structure.

The aim with this colvar it to calculate something like\+:

\[ d(X,X') = \vert X-X' \vert \]

where $ X $ is the instantaneous position of all the atoms in the system and $ X' $ is the positions of the atoms in some reference structure provided as input. $ d(X,X') $ thus measures the distance all the atoms have moved away from this reference configuration. Oftentimes, it is only the internal motions of the structure -\/ i.\+e. not the translations of the center of mass or the rotations of the reference frame -\/ that are interesting. Hence, when calculating the the root-\/mean-\/square deviation between the atoms in two configurations you must first superimpose the two structures in some way. At present P\+L\+U\+M\+E\+D provides two distinct ways of performing this superposition. The first method is applied when you use T\+Y\+P\+E=S\+I\+M\+P\+L\+E in the input line. This instruction tells P\+L\+U\+M\+E\+D that the root mean square deviation is to be calculated after the positions of the geometric centers in the reference and instantaneous configurations are aligned. In other words $d(X,x')$ is to be calculated using\+:

\[ d(X,X') = \sqrt{ \sum_i \sum_\alpha^{x,y,z} \frac{w_i}{\sum_j w_j}( X_{i,\alpha}-com_\alpha(X)-{X'}_{i,\alpha}+com_\alpha(X') )^2 } \] with \[ com_\alpha(X)= \sum_i \frac{w'_{i}}{\sum_j w'_j}X_{i,\alpha} \] and \[ com_\alpha(X')= \sum_i \frac{w'_{i}}{\sum_j w'_j}X'_{i,\alpha} \] Obviously, $ com_\alpha(X) $ and $ com_\alpha(X') $ represent the positions of the center of mass in the reference and instantaneous configurations if the weights \$w'\$ are set equal to the atomic masses. If the weights are all set equal to one, however, $com_\alpha(X) $ and $ com_\alpha(X') $ are the positions of the geometric centers. Notice that there are sets of weights\+: $ w' $ and $ w $. The first is used to calculate the position of the center of mass (so it determines how the atoms are {\itshape aligned}). Meanwhile, the second is used when calculating how far the atoms have actually been {\itshape displaced}. These weights are assigned in the reference configuration that you provide as input (i.\+e. the appear in the input file to this action that you set using R\+E\+F\+E\+R\+E\+N\+C\+E=whatever.\+pdb). This input reference configuration consists of a simple pdb file containing the set of atoms for which you want to calculate the R\+M\+S\+D displacement and their positions in the reference configuration. It is important to note that the indices in this pdb need to be set correctly. The indices in this file determine the indices of the instantaneous atomic positions that are used by P\+L\+U\+M\+E\+D when calculating this colvar. As such if you want to calculate the R\+M\+S\+D distance moved by the 1st, 4th, 6th and 28th atoms in the M\+D codes input file then the indices of the corresponding refernece positions in this pdb file should be set equal to 1, 4, 6 and 28.

The pdb input file should also contain the values of $w$ and $w'$. In particular, the O\+C\+C\+U\+P\+A\+N\+C\+Y column (the first column after the coordinates) is used provides the values of $ w'$ that are used to calculate the position of the centre of mass. The B\+E\+T\+A column (the second column after the Cartesian coordinates) is used to provide the $ w $ values which are used in the the calculation of the displacement. Please note that it is possible to use fractional values for beta and for the occupancy. However, we recommend you only do this when you really know what you are doing however as the results can be rather strange.

In P\+D\+B files the atomic coordinates and box lengths should be in Angstroms unless you are working with natural units. If you are working with natural units then the coordinates should be in your natural length unit. For more details on the P\+D\+B file format visit \href{http://www.wwpdb.org/docs.html}{\tt http\+://www.\+wwpdb.\+org/docs.\+html}.

A different method is used to calculate the R\+M\+S\+D distance when you use T\+Y\+P\+E=O\+P\+T\+I\+M\+A\+L on the input line. In this case the root mean square deviation is calculated after the positions of geometric centers in the reference and instantaneous configurations are aligned A\+N\+D after an optimal alignment of the two frames is performed so that motion due to rotation of the reference frame between the two structures is removed. The equation for $d(X,X')$ in this case reads\+:

\[ d(X,X') = \sqrt{ \sum_i \sum_\alpha^{x,y,z} \frac{w_i}{\sum_j w_j}[ X_{i,\alpha}-com_\alpha(X)- \sum_\beta M(X,X',w')_{\alpha,\beta}({X'}_{i,\beta}-com_\beta(X')) ]^2 } \]

where $ M(X,X',w') $ is the optimal alignment matrix which is calculated using the Kearsley \cite{kearsley} algorithm. Again different sets of weights are used for the alignment ( $w'$) and for the displacement calcuations ( $w$). This gives a great deal of flexibility as it allows you to use a different sets of atoms (which may or may not overlap) for the alignment and displacement parts of the calculation. This may be very useful when you want to calculate how a ligand moves about in a protein cavity as you can use the protein as a reference system and do no alignment of the ligand.

(Note\+: when this form of R\+M\+S\+D is used to calculate the secondary structure variables (\hyperlink{ALPHARMSD}{A\+L\+P\+H\+A\+R\+M\+S\+D}, \hyperlink{ANTIBETARMSD}{A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D} and \hyperlink{PARABETARMSD}{P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D} all the atoms in the segment are assumed to be part of both the alignment and displacement sets and all weights are set equal to one)

Please note that there are a number of other methods for calculating the distance between the instantaneous configuration and a reference configuration that are available in plumed. More information on these various methods can be found in the section of the manual on \hyperlink{dists}{Distances from reference configurations}.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a file in pdb format containing the reference structure and the atoms involved in the C\+V.   \\\cline{1-2}
{\bfseries  T\+Y\+P\+E } &( default=S\+I\+M\+P\+L\+E ) the manner in which R\+M\+S\+D alignment is performed. Should be O\+P\+T\+I\+M\+A\+L or S\+I\+M\+P\+L\+E.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+Q\+U\+A\+R\+E\+D } &( default=off ) This should be setted if you want M\+S\+D instead of R\+M\+S\+D  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following tells plumed to calculate the R\+M\+S\+D distance between the positions of the atoms in the reference file and their instantaneous position. The Kearseley algorithm is used so this is done optimally.

\begin{DoxyVerb}RMSD REFERENCE=file.pdb TYPE=OPTIMAL
\end{DoxyVerb}


... \hypertarget{TARGET}{}\subsection{T\+A\+R\+G\+E\+T}\label{TARGET}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This function measures the pythagorean distance from a particular structure measured in the space defined by some set of collective variables.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  T\+Y\+P\+E } &( default=E\+U\+C\+L\+I\+D\+E\+A\+N ) the manner in which the distance should be calculated   \\\cline{1-2}
{\bfseries  R\+E\+F\+E\+R\+E\+N\+C\+E } &a file in pdb format containing the reference structure. In the P\+D\+B file the atomic coordinates and box lengths should be in Angstroms unless you are working with natural units. If you are working with natural units then the coordinates should be in your natural length unit. The charges and masses of the atoms (if required) should be inserted in the beta and occupancy columns respectively. For more details on the P\+D\+B file format visit \href{http://www.wwpdb.org/docs.html}{\tt http\+://www.\+wwpdb.\+org/docs.\+html}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\hypertarget{Function}{}\section{Functions}\label{Function}
When performing biased dynamics or analysing a trajectory you may wish to analyse/bias the value of some function of a set of collective variables rather than the values of the collective variables directly. You can do this with P\+L\+U\+M\+E\+D by using any one of the following list of functions\+:

\begin{TabularC}{2}
\hline
\hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E}  &Calculate a polynomial combination of a set of other variables.  \\\cline{1-2}
\hyperlink{ENSEMBLE}{E\+N\+S\+E\+M\+B\+L\+E}  &Calculates the replica averaging of a collective variable over multiple replicas.  \\\cline{1-2}
\hyperlink{FUNCPATHMSD}{F\+U\+N\+C\+P\+A\+T\+H\+M\+S\+D}  &This function calculates path collective variables.   \\\cline{1-2}
\hyperlink{FUNCSUMHILLS}{F\+U\+N\+C\+S\+U\+M\+H\+I\+L\+L\+S}  &This function is intended to be called by the command line tool sum\+\_\+hillsand it is meant to integrate a H\+I\+L\+L\+S file or an H\+I\+L\+L\+S file interpreted as a histogram i a variety of ways. Therefore it is not expected that you use this during your dynamics (it will crash!)  \\\cline{1-2}
\hyperlink{MATHEVAL}{M\+A\+T\+H\+E\+V\+A\+L}  &Calculate a combination of variables using a matheval expression.  \\\cline{1-2}
\hyperlink{PIECEWISE}{P\+I\+E\+C\+E\+W\+I\+S\+E}  &Compute a piecewise straight line through its arguments that passes througha set of ordered control points.   \\\cline{1-2}
\hyperlink{SORT}{S\+O\+R\+T}  &This function can be used to sort colvars according to their magnitudes.  \\\cline{1-2}
\end{TabularC}
\hypertarget{COMBINE}{}\subsection{C\+O\+M\+B\+I\+N\+E}\label{COMBINE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate a polynomial combination of a set of other variables.

The functional form of this function is \[ C=\sum_{i=1}^{N_{arg}} c_i x_i^{p_i} \]

The coefficients c and powers p are provided as vectors.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+R\+G } &the input for this action is the scalar output from one or more other actions. The particular scalars that you will use are referenced using the label of the action. If the label appears on its own then it is assumed that the Action calculates a single scalar value. The value of this scalar is thus used as the input to this new action. If $\ast$ or $\ast$.$\ast$ appears the scalars calculated by all the proceding actions in the input file are taken. Some actions have multi-\/component outputs and each component of the output has a specific label. For example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} action labelled dist may have three componets x, y and z. To take just the x component you should use dist.\+x, if you wish to take all three components then use dist.$\ast$.More information on the referencing of Actions can be found in the section of the manual on the P\+L\+U\+M\+E\+D \hyperlink{_syntax}{Getting started}. Scalar values can also be referenced using P\+O\+S\+I\+X regular expressions as detailed in the section on \hyperlink{Regex}{Regular Expressions}. To use this feature you you must compile P\+L\+U\+M\+E\+D with the appropriate flag.   \\\cline{1-2}
{\bfseries  P\+E\+R\+I\+O\+D\+I\+C } &if the output of your function is periodic then you should specify the periodicity of the function. If the output is not periodic you must state this using P\+E\+R\+I\+O\+D\+I\+C=N\+O   \\\cline{1-2}
{\bfseries  C\+O\+E\+F\+F\+I\+C\+I\+E\+N\+T\+S } &( default=1.\+0 ) the coefficients of the arguments in your function   \\\cline{1-2}
{\bfseries  P\+O\+W\+E\+R\+S } &( default=1.\+0 ) the powers to which you are raising each of the arguments in your function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+R\+M\+A\+L\+I\+Z\+E } &( default=off ) normalize all the coefficents so that in total they are equal to one  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following input tells plumed to print the distance between atoms 3 and 5 its square (as computed from the x,y,z components) and the distance again as computed from the square root of the square. \begin{DoxyVerb}DISTANCE LABEL=dist      ATOMS=3,5 COMPONENTS
COMBINE  LABEL=distance2 ARG=dist.x,dist.y,dist.z POWERS=2,2,2 PERIODIC=NO
COMBINE  LABEL=distance  ARG=distance2 POWERS=0.5 PERIODIC=NO
PRINT ARG=distance,distance2
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} and \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}). 
\end{DoxyParagraph}
\hypertarget{ENSEMBLE}{}\subsection{E\+N\+S\+E\+M\+B\+L\+E}\label{ENSEMBLE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculates the replica averaging of a collective variable over multiple replicas.

Each collective variable is averaged separately and stored in a component labelled {\itshape label}.cvlabel.

Note that in case of variables such as \hyperlink{CS2BACKBONE}{C\+S2\+B\+A\+C\+K\+B\+O\+N\+E}, \hyperlink{CH3SHIFTS}{C\+H3\+S\+H\+I\+F\+T\+S}, \hyperlink{NOE}{N\+O\+E} and \hyperlink{RDC}{R\+D\+C} it is possible to perform the replica-\/averaging inside the variable, in fact in those cases are the single experimental values that averaged before calculating the collective variable.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+R\+G } &the input for this action is the scalar output from one or more other actions. The particular scalars that you will use are referenced using the label of the action. If the label appears on its own then it is assumed that the Action calculates a single scalar value. The value of this scalar is thus used as the input to this new action. If $\ast$ or $\ast$.$\ast$ appears the scalars calculated by all the proceding actions in the input file are taken. Some actions have multi-\/component outputs and each component of the output has a specific label. For example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} action labelled dist may have three componets x, y and z. To take just the x component you should use dist.\+x, if you wish to take all three components then use dist.$\ast$.More information on the referencing of Actions can be found in the section of the manual on the P\+L\+U\+M\+E\+D \hyperlink{_syntax}{Getting started}. Scalar values can also be referenced using P\+O\+S\+I\+X regular expressions as detailed in the section on \hyperlink{Regex}{Regular Expressions}. To use this feature you you must compile P\+L\+U\+M\+E\+D with the appropriate flag.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following input tells plumed to calculate the distance between atoms 3 and 5 and the average it over the available replicas. \begin{DoxyVerb}dist: DISTANCE ATOMS=3,5 
ens: ENSEMBLE ARG=dist
PRINT ARG=dist,ens.dist
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} and \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}). 
\end{DoxyParagraph}
\hypertarget{FUNCPATHMSD}{}\subsection{F\+U\+N\+C\+P\+A\+T\+H\+M\+S\+D}\label{FUNCPATHMSD}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This function calculates path collective variables.

This is the Path Collective Variables implementation ( see \cite{brand07} ). This variable computes the progress along a given set of frames that is provided in input (\char`\"{}s\char`\"{} component) and the distance from them (\char`\"{}z\char`\"{} component). It is a function of M\+S\+D that are obtained by the joint use of M\+S\+D variable and S\+Q\+U\+A\+R\+E\+D flag (see below).

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the following quantities. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  s } &the position on the path   \\\cline{1-2}
{\bfseries  z } &the distance from the path   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+R\+G } &the input for this action is the scalar output from one or more other actions. The particular scalars that you will use are referenced using the label of the action. If the label appears on its own then it is assumed that the Action calculates a single scalar value. The value of this scalar is thus used as the input to this new action. If $\ast$ or $\ast$.$\ast$ appears the scalars calculated by all the proceding actions in the input file are taken. Some actions have multi-\/component outputs and each component of the output has a specific label. For example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} action labelled dist may have three componets x, y and z. To take just the x component you should use dist.\+x, if you wish to take all three components then use dist.$\ast$.More information on the referencing of Actions can be found in the section of the manual on the P\+L\+U\+M\+E\+D \hyperlink{_syntax}{Getting started}. Scalar values can also be referenced using P\+O\+S\+I\+X regular expressions as detailed in the section on \hyperlink{Regex}{Regular Expressions}. To use this feature you you must compile P\+L\+U\+M\+E\+D with the appropriate flag.   \\\cline{1-2}
{\bfseries  L\+A\+M\+B\+D\+A } &the lambda parameter is needed for smoothing, is in the units of plumed   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  N\+E\+I\+G\+H\+\_\+\+S\+I\+Z\+E } &size of the neighbor list   \\\cline{1-2}
{\bfseries  N\+E\+I\+G\+H\+\_\+\+S\+T\+R\+I\+D\+E } &how often the neighbor list needs to be calculated in time units  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
Here below is a case where you have defined three frames and you want to calculate the progress alng the path and the distance from it in p1

\begin{DoxyVerb}t1: RMSD REFERENCE=frame_1.dat TYPE=OPTIMAL SQUARED
t2: RMSD REFERENCE=frame_21.dat TYPE=OPTIMAL SQUARED
t3: RMSD REFERENCE=frame_42.dat TYPE=OPTIMAL SQUARED
p1: FUNCPATHMSD ARG=t1,t2,t3 LAMBDA=500.0 
PRINT ARG=t1,t2,t3,p1.s,p1.z STRIDE=1 FILE=colvar FMT=%8.4f
\end{DoxyVerb}


In this second example is shown how to define a P\+A\+T\+H in the \hyperlink{CONTACTMAP}{C\+O\+N\+T\+A\+C\+T\+M\+A\+P} space\+:

\begin{DoxyVerb}CONTACTMAP ...
ATOMS1=1,2 REFERENCE1=0.1  
ATOMS2=3,4 REFERENCE2=0.5  
ATOMS3=4,5 REFERENCE3=0.25  
ATOMS4=5,6 REFERENCE4=0.0  
SWITCH=(RATIONAL R_0=1.5) 
LABEL=c1
CMDIST
... CONTACTMAP

CONTACTMAP ...
ATOMS1=1,2 REFERENCE1=0.3  
ATOMS2=3,4 REFERENCE2=0.9  
ATOMS3=4,5 REFERENCE3=0.45  
ATOMS4=5,6 REFERENCE4=0.1  
SWITCH=(RATIONAL R_0=1.5) 
LABEL=c2
CMDIST
... CONTACTMAP

CONTACTMAP ...
ATOMS1=1,2 REFERENCE1=1.0  
ATOMS2=3,4 REFERENCE2=1.0  
ATOMS3=4,5 REFERENCE3=1.0  
ATOMS4=5,6 REFERENCE4=1.0  
SWITCH=(RATIONAL R_0=1.5) 
LABEL=c3
CMDIST
... CONTACTMAP

p1: FUNCPATHMSD ARG=c1,c2,c3 LAMBDA=500.0 
PRINT ARG=c1,c2,c3,p1.s,p1.z STRIDE=1 FILE=colvar FMT=%8.4f
\end{DoxyVerb}
 \hypertarget{FUNCSUMHILLS}{}\subsection{F\+U\+N\+C\+S\+U\+M\+H\+I\+L\+L\+S}\label{FUNCSUMHILLS}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This function is intended to be called by the command line tool sum\+\_\+hills and it is meant to integrate a H\+I\+L\+L\+S file or an H\+I\+L\+L\+S file interpreted as a histogram i a variety of ways. Therefore it is not expected that you use this during your dynamics (it will crash!)

In the future one could implement periodic integration during the metadynamics or straightforward M\+D as a tool to check convergence \hypertarget{MATHEVAL}{}\subsection{M\+A\+T\+H\+E\+V\+A\+L}\label{MATHEVAL}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate a combination of variables using a matheval expression.

This action computes an arbitrary function of one or more precomputed collective variables. Arguments are chosen with the A\+R\+G keyword, and the function is provided with the F\+U\+N\+C string. Notice that this string should contain no space. Within F\+U\+N\+C, one can refer to the arguments as x,y,z, and t (up to four variables provided as A\+R\+G). This names can be customized using the V\+A\+R keyword (see examples below).

If you want a function that depends not only on collective variables but also on time you can use the \hyperlink{TIME}{T\+I\+M\+E} action.

\begin{DoxyAttention}{Attention}
The M\+A\+T\+H\+E\+V\+A\+L object only works if libmatheval is installed on the system and P\+L\+U\+M\+E\+D has been linked to it
\end{DoxyAttention}
\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+R\+G } &the input for this action is the scalar output from one or more other actions. The particular scalars that you will use are referenced using the label of the action. If the label appears on its own then it is assumed that the Action calculates a single scalar value. The value of this scalar is thus used as the input to this new action. If $\ast$ or $\ast$.$\ast$ appears the scalars calculated by all the proceding actions in the input file are taken. Some actions have multi-\/component outputs and each component of the output has a specific label. For example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} action labelled dist may have three componets x, y and z. To take just the x component you should use dist.\+x, if you wish to take all three components then use dist.$\ast$.More information on the referencing of Actions can be found in the section of the manual on the P\+L\+U\+M\+E\+D \hyperlink{_syntax}{Getting started}. Scalar values can also be referenced using P\+O\+S\+I\+X regular expressions as detailed in the section on \hyperlink{Regex}{Regular Expressions}. To use this feature you you must compile P\+L\+U\+M\+E\+D with the appropriate flag.   \\\cline{1-2}
{\bfseries  P\+E\+R\+I\+O\+D\+I\+C } &if the output of your function is periodic then you should specify the periodicity of the function. If the output is not periodic you must state this using P\+E\+R\+I\+O\+D\+I\+C=N\+O   \\\cline{1-2}
{\bfseries  F\+U\+N\+C } &the function you wish to evaluate   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  V\+A\+R } &the names to give each of the arguments in the function. If you have up to three arguments in your function you can use x, y and z to refer to them. Otherwise you must use this flag to give your variables names.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to perform a metadynamics using as a C\+V the difference between two distances. \begin{DoxyVerb}dAB: DISTANCE ARG=10,12
dAC: DISTANCE ARG=10,15
diff: MATHEVAL ARG=dAB,dAC FUNC=y-x PERIODIC=NO
# notice: the previous line could be replaced with the following
# diff: COMBINE ARG=dAB,dAC COEFFICIENTS=-1,1
METAD ARG=diff WIDTH=0.1 HEIGHT=0.5 BIASFACTOR=10 PACE=100
\end{DoxyVerb}
 (see also \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}, \hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E}, and \hyperlink{METAD}{M\+E\+T\+A\+D}). Notice that forces applied to diff will be correctly propagated to atoms 10, 12, and 15. Also notice that since M\+A\+T\+H\+E\+V\+A\+L is used without the V\+A\+R option the two arguments should be referred to as x and y in the expression F\+U\+N\+C. For simple functions such as this one it is possible to use \hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E}, which does not require libmatheval to be installed on your system.

The following input tells plumed to print the angle between vectors identified by atoms 1,2 and atoms 2,3 its square (as computed from the x,y,z components) and the distance again as computed from the square root of the square. \begin{DoxyVerb}DISTANCE LABEL=d1 ATOMS=1,2 COMPONENTS
DISTANCE LABEL=d2 ATOMS=2,3 COMPONENTS
MATHEVAL ...
  LABEL=theta
  ARG=d1.x,d1.y,d1.z,d2.x,d2.y,d2.z
  VAR=ax,ay,az,bx,by,bz
  FUNC=acos((ax*bx+ay*by+az*bz)/sqrt((ax*ax+ay*ay+az*az)*(bx*bx+by*by+bz*bz))
  PERIODIC=NO
... MATHEVAL
PRINT ARG=theta
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} and \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}). \hypertarget{TIME}{}\subsubsection{T\+I\+M\+E}\label{TIME}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the generic \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
retrieve the time of the simulation to be used elsewere

\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\begin{DoxyVerb}TIME            LABEL=t1
PRINT ARG=t1
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}). \hypertarget{PIECEWISE}{}\subsection{P\+I\+E\+C\+E\+W\+I\+S\+E}\label{PIECEWISE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Compute a piecewise straight line through its arguments that passes through a set of ordered control points.

For variables less than the first (greater than the last) point, the value of the first (last) point is used.

\[ \frac{y_{i+1}-y_i}{x_{i+1}-x_i}(s-x_i)+y_i ; if x_i<s<x_{i+1} \] \[ y_N ; if x>x_{N-1} \] \[ y_1 ; if x<x_0 \]

Control points are passed using the P\+O\+I\+N\+T0=... P\+O\+I\+N\+T1=... syntax as in the example below

If one argument is supplied, it results in a scalar quantity. If multiple arguments are supplied, it results in a vector of values. Each value will be named as the name of the original argument with suffix \+\_\+pfunc.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the following quantities. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  \+\_\+pfunc } &one or multiple instances of this quantity will be referenceable elsewhere in the input file. These quantities will be named with the arguments of the function followed by the character string \+\_\+pfunc. These quantities tell the user the values of the piecewise functions of each of the arguments.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+R\+G } &the input for this action is the scalar output from one or more other actions. The particular scalars that you will use are referenced using the label of the action. If the label appears on its own then it is assumed that the Action calculates a single scalar value. The value of this scalar is thus used as the input to this new action. If $\ast$ or $\ast$.$\ast$ appears the scalars calculated by all the proceding actions in the input file are taken. Some actions have multi-\/component outputs and each component of the output has a specific label. For example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} action labelled dist may have three componets x, y and z. To take just the x component you should use dist.\+x, if you wish to take all three components then use dist.$\ast$.More information on the referencing of Actions can be found in the section of the manual on the P\+L\+U\+M\+E\+D \hyperlink{_syntax}{Getting started}. Scalar values can also be referenced using P\+O\+S\+I\+X regular expressions as detailed in the section on \hyperlink{Regex}{Regular Expressions}. To use this feature you you must compile P\+L\+U\+M\+E\+D with the appropriate flag.   \\\cline{1-2}
{\bfseries  P\+O\+I\+N\+T } &This keyword is used to specify the various points in the function above. You can use multiple instances of this keyword i.\+e. P\+O\+I\+N\+T1, P\+O\+I\+N\+T2, P\+O\+I\+N\+T3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
\begin{DoxyVerb}dist1: DISTANCE ATOMS=1,10
dist2: DISTANCE ATOMS=2,11

pw: PIECEWISE POINT0=1,10 POINT1=1,PI POINT2=3,10 ARG=dist1
ppww: PIECEWISE POINT0=1,10 POINT1=1,PI POINT2=3,10 ARG=dist1,dist2
PRINT ARG=pw,ppww.dist1_pfunc,ppww.dist2_pfunc
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} and \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}). 
\end{DoxyParagraph}
\hypertarget{SORT}{}\subsection{S\+O\+R\+T}\label{SORT}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the function \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This function can be used to sort colvars according to their magnitudes.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
This function sorts its arguments according to their magnitudes. The lowest argument will be labelled {\itshape label}.1, the second lowest will be labelled {\itshape label}.2 and so on.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+R\+G } &the input for this action is the scalar output from one or more other actions. The particular scalars that you will use are referenced using the label of the action. If the label appears on its own then it is assumed that the Action calculates a single scalar value. The value of this scalar is thus used as the input to this new action. If $\ast$ or $\ast$.$\ast$ appears the scalars calculated by all the proceding actions in the input file are taken. Some actions have multi-\/component outputs and each component of the output has a specific label. For example a \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} action labelled dist may have three componets x, y and z. To take just the x component you should use dist.\+x, if you wish to take all three components then use dist.$\ast$.More information on the referencing of Actions can be found in the section of the manual on the P\+L\+U\+M\+E\+D \hyperlink{_syntax}{Getting started}. Scalar values can also be referenced using P\+O\+S\+I\+X regular expressions as detailed in the section on \hyperlink{Regex}{Regular Expressions}. To use this feature you you must compile P\+L\+U\+M\+E\+D with the appropriate flag.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}
The following input tells plumed to print the distance of the closest and of the farthest atoms to atom 1, chosen among atoms from 2 to 5 \begin{DoxyVerb}d12:  DISTANCE ATOMS=1,2
d13:  DISTANCE ATOMS=1,3
d14:  DISTANCE ATOMS=1,4
d15:  DISTANCE ATOMS=1,5
sort: SORT ARG=d12,d13,d14,d15
PRINT ARG=sort.1,sort.4
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} and \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}). 
\end{DoxyParagraph}
\hypertarget{mcolv}{}\section{Multi\+Colvar Documentation}\label{mcolv}
Oftentimes, when you do not need one of the collective variables described elsewhere in the manual, what you want instead is a function of a distribution of collective variables of a particular type. For instance you might need to calculate a minimum distance or the number of coordination numbers greater than a 3.\+0. To avoid dupilcating the code to calculate an angle or distance many times and to make it easier to implement very complex collective variables P\+L\+U\+M\+E\+D provides these sort of collective variables using so-\/called Multi\+Colvars. Multi\+Colvars are named in this way because a single P\+L\+U\+M\+E\+D action can be used to calculate a number of different collective variables. For instance the \hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S} action can be used to calculate the minimum distance, the number of distances less than a certain value, the number of distances within a certain range... A more detailed introduction to multicolvars is provided in this \href{http://www.youtube.com/watch?v=iDvZmbWE5ps}{\tt 10-\/minute video}. Descriptions of the various multicolvars that are implemented in P\+L\+U\+M\+E\+D 2 are given below\+:

\begin{TabularC}{2}
\hline
\hyperlink{ANGLES}{A\+N\+G\+L\+E\+S}  &Calculate functions of the distribution of angles .  \\\cline{1-2}
\hyperlink{BRIDGE}{B\+R\+I\+D\+G\+E}  &Calculate the number of atoms that bridge two parts of a structure  \\\cline{1-2}
\hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R}  &Calculate the coordination numbers of atoms so that you can then calculate functions of the distribution ofcoordination numbers such as the minimum, the number less than a certain quantity and so on.   \\\cline{1-2}
\hyperlink{DENSITY}{D\+E\+N\+S\+I\+T\+Y}  &Calculate functions of the density of atoms as a function of the box. This allows one to calculatethe number of atoms in half the box.  \\\cline{1-2}
\hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S}  &Calculate the distances between one or many pairs of atoms. You can then calculate functions of the distribution ofdistances such as the minimum, the number less than a certain quantity and so on.   \\\cline{1-2}
\hyperlink{FCCUBIC}{F\+C\+C\+U\+B\+I\+C}  &\\\cline{1-2}
\hyperlink{MOLECULES}{M\+O\+L\+E\+C\+U\+L\+E\+S}  &Calculate the vectors connecting a pair of atoms in order to represent the orientation of a molecule.  \\\cline{1-2}
\hyperlink{Q3}{Q3}  &Calculate 3rd order Steinhardt parameters.  \\\cline{1-2}
\hyperlink{Q4}{Q4}  &Calculate 4th order Steinhardt parameters.  \\\cline{1-2}
\hyperlink{Q6}{Q6}  &Calculate 6th order Steinhardt parameters.  \\\cline{1-2}
\hyperlink{SIMPLECUBIC}{S\+I\+M\+P\+L\+E\+C\+U\+B\+I\+C}  &Calculate whether or not the coordination spheres of atoms are arranged as they would be in a simplecubic structure.  \\\cline{1-2}
\hyperlink{TETRAHEDRAL}{T\+E\+T\+R\+A\+H\+E\+D\+R\+A\+L}  &\\\cline{1-2}
\hyperlink{TORSIONS}{T\+O\+R\+S\+I\+O\+N\+S}  &Calculate whether or not a set of torsional angles are within a particular range.  \\\cline{1-2}
\hyperlink{XDISTANCES}{X\+D\+I\+S\+T\+A\+N\+C\+E\+S}  &Calculate the x components of the vectors connecting one or many pairs of atoms. You can then calculate functions of the distribution ofvalues such as the minimum, the number less than a certain quantity and so on.   \\\cline{1-2}
\hyperlink{YDISTANCES}{Y\+D\+I\+S\+T\+A\+N\+C\+E\+S}  &Calculate the y components of the vectors connecting one or many pairs of atoms. You can then calculate functions of the distribution ofvalues such as the minimum, the number less than a certain quantity and so on.  \\\cline{1-2}
\hyperlink{ZDISTANCES}{Z\+D\+I\+S\+T\+A\+N\+C\+E\+S}  &Calculate the z components of the vectors connecting one or many pairs of atoms. You can then calculate functions of the distribution ofvalues such as the minimum, the number less than a certain quantity and so on.  \\\cline{1-2}
\end{TabularC}


To instruct P\+L\+U\+M\+E\+D to calculate a multicolvar you give an instruction that looks something like this\+:

\begin{DoxyVerb}NAME <atoms involved> <parameters> <what am I calculating> TOL=0.001 LABEL=label
\end{DoxyVerb}


Oftentimes the simplest way to specify the atoms involved is to use multiple instances of the A\+T\+O\+M\+S keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... Separate instances of the quantity specified by N\+A\+M\+E are then calculated for each of the sets of atoms. For example if the command issued contains the following\+:

\begin{DoxyVerb}DISTANCES ATOMS1=1,2 ATOMS2=3,4 ATOMS3=5,6
\end{DoxyVerb}


The distances between atoms 1 and 2, atoms 3 and 4, and atoms 5 and 6 are calculated. Obviously, generating this sort of input is rather tedious so short cuts are also available many of the collective variables. These are described on the manual pages for the actions.

After specifying the atoms involved you sometimes need to specify some parameters that required in the calculation. For instance, for \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R} -\/ the number of atoms in the first coordination sphere of each of the atoms in the system -\/ you need to specify the parameters for a \hyperlink{switchingfunction}{switchingfunction} that will tell us whether or not an atom is in the first coordination sphere. Details as to how to do this are provided on the manual pages.

One of the most important keywords for multicolvars is the T\+O\+L keyword. This specifies that terms in sums that contribute less than a certain value can be ignored. In addition, it is assumed that the derivative with respect to these terms are essentially zero. By increasing the T\+O\+L parameter you can increase the speed of the calculation. Be aware, however, that this increase in speed is only possible because you are lowering the accuracy with which you are computing the quantity of interest.

Once you have specified the base quanties that are to be calculated from the atoms involved and any parameters you need to specify what function of these base quanties is to be calculated. For most multicolvars you can calculate the minimum, the number less than a target value, the number within a certain range, the number more than a target value and the average value directly.\hypertarget{mcolv_multicolvarfunction}{}\subsection{Multi\+Colvar functions}\label{mcolv_multicolvarfunction}
As well as the relatively simple quantities described above more complex functions of the distribution of values for the base colvars can be computed by employing multicolvars in conjuction with the following actions\+:

\begin{TabularC}{2}
\hline
\hyperlink{AROUND}{A\+R\+O\+U\+N\+D}  &This quantity can be used to calculate functions of the distribution of collective variables for the atoms that lie in a particular, user-\/specified part of of the cell.  \\\cline{1-2}
\hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E}  &Calculate averages over spherical regions centered on atoms  \\\cline{1-2}
\hyperlink{LOCAL_Q3}{L\+O\+C\+A\+L\+\_\+\+Q3}  &Calculate the local degree of order around an atoms by taking the average dot product between the $q_3$ vector on the central atom and the $q_3$ vectoron the atoms in the first coordination sphere.  \\\cline{1-2}
\hyperlink{LOCAL_Q4}{L\+O\+C\+A\+L\+\_\+\+Q4}  &Calculate the local degree of order around an atoms by taking the average dot product between the $q_4$ vector on the central atom and the $q_4$ vectoron the atoms in the first coordination sphere.  \\\cline{1-2}
\hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6}  &Calculate the local degree of order around an atoms by taking the average dot product between the $q_6$ vector on the central atom and the $q_6$ vectoron the atoms in the first coordination sphere.  \\\cline{1-2}
\hyperlink{NLINKS}{N\+L\+I\+N\+K\+S}  &Calculate number of pairs of atoms/molecules that are \char`\"{}linked\char`\"{}  \\\cline{1-2}
\hyperlink{SPRINT}{S\+P\+R\+I\+N\+T}  &Calculate S\+P\+R\+I\+N\+T topological variables.  \\\cline{1-2}
\end{TabularC}
\hypertarget{mcolv_multicolvarbias}{}\subsection{Multi\+Colvar bias}\label{mcolv_multicolvarbias}
There may be occasitions when you want add restraints on many collective variables. For instance if you are studying a cluster you might want to add a wall on the distances between each of the atoms and the center of mass of the cluster in order to prevent the cluster subliming. Alternatively, you may wish to insist that a particular set of atoms in your system all have a coordination number greater than 2. You can add these sorts of restraints by employing the following biases, which all act on the set of collective variable values calculated by a multicolvar. So for example the following set of commands\+:

\begin{DoxyVerb}COM ATOMS=1-20 LABEL=c1
DISTANCES GROUPA=c1 GROUPB=1-20 LABEL=d1
UWALLS DATA=d1 AT=2.5 KAPPA=0.2 LABEL=sr
\end{DoxyVerb}


creates the aforementioned set of restraints on the distances between the 20 atoms in a cluster and the center of mass of the cluster.

The list of biases of this type are as follows\+:

\begin{TabularC}{2}
\hline
\hyperlink{UWALLS}{U\+W\+A\+L\+L\+S}  &Add \hyperlink{UPPER_WALLS}{U\+P\+P\+E\+R\+\_\+\+W\+A\+L\+L\+S} restraints on all the multicolvar values  \\\cline{1-2}
\end{TabularC}


Notice that (in theory) you could also use this functionality to add additional terms to your forcefield or to implement your forcefield. \hypertarget{ANGLES}{}\subsection{A\+N\+G\+L\+E\+S}\label{ANGLES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate functions of the distribution of angles .

You can use this command to calculate functions such as\+:

\[ f(x) = \sum_{ijk} g( \theta_{ijk} ) \]

Alternatively you can use this command to calculate functions such as\+:

\[ f(x) = \sum_{ijk} s(r_{ij})s(r_{jk}) g(\theta_{ijk}) \]

where $s(r)$ is a \hyperlink{switchingfunction}{switchingfunction}. This second form means that you can use this to calculate functions of the angles in the first coordination sphere of an atom / molecule \cite{lj-recon}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P } &Calculate angles for each distinct set of three atoms in the group   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &A group of central atoms about which angles should be calculated   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &When used in conjuction with G\+R\+O\+U\+P\+A this keyword instructs plumed to calculate all distinct angles involving one atom from G\+R\+O\+U\+P\+A and two atoms from G\+R\+O\+U\+P\+B. The atom from G\+R\+O\+U\+P\+A is the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+C } &This must be used in conjuction with G\+R\+O\+U\+P\+A and G\+R\+O\+U\+P\+B. All angles involving one atom from G\+R\+O\+U\+P\+A, one atom from G\+R\+O\+U\+P\+B and one atom from G\+R\+O\+U\+P\+C are calculated. The G\+R\+O\+U\+P\+A atoms are assumed to be the central atoms   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &A switching function that ensures that only angles between atoms that are within a certain fixed cutoff are calculated. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H\+A } &A switching function on the distance between the atoms in group A and the atoms in group B   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H\+B } &A switching function on the distance between the atoms in group A and the atoms in group B  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following example instructs plumed to find the average of two angles and to print it to a file

\begin{DoxyVerb}ANGLES ATOMS1=1,2,3 ATOMS2=4,5,6 MEAN LABEL=a1
PRINT ARG=a1.mean FILE=colvar 
\end{DoxyVerb}


The following example tells plumed to calculate all angles involving at least one atom from G\+R\+O\+U\+P\+A and two atoms from G\+R\+O\+U\+P\+B in which the distances are less than 1.\+0. The number of angles between $\frac{\pi}{4}$ and $\frac{3\pi}{4}$ is then output

\begin{DoxyVerb}ANGLES GROUPA=1-10 GROUPB=11-100 BETWEEN={GAUSSIAN LOWER=0.25pi UPPER=0.75pi} SWITCH={GAUSSIAN R_0=1.0} LABEL=a1
PRINT ARG=a1.between FILE=colvar
\end{DoxyVerb}


This final example instructs plumed to calculate all the angles in the first coordination spheres of the atoms. A discretized-\/normalized histogram of the distribution is then output

\begin{DoxyVerb}ANGLES GROUP=1-38 HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=pi NBINS=20} SWITCH={GAUSSIAN R_0=1.0} LABEL=a1
PRINT ARG=a1.* FILE=colvar
\end{DoxyVerb}
 \hypertarget{BRIDGE}{}\subsection{B\+R\+I\+D\+G\+E}\label{BRIDGE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the number of atoms that bridge two parts of a structure

This quantity calculates\+:

\[ f(x) = \sum_{ijk} s_A(r_{ij})s_B(r_{ik}) \]

where the sum over $i$ is over all the ``bridging atoms" and $s_A$ and $s_B$ are \hyperlink{switchingfunction}{switchingfunction}.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  B\+R\+I\+D\+G\+I\+N\+G\+\_\+\+A\+T\+O\+M\+S } &The list of atoms that can form the bridge between the two interesting parts of the structure.   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+A } &The list of atoms that are in the first interesting part of the structure   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &The list of atoms that are in the second interesting part of the structure   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &The parameters of the two \hyperlink{switchingfunction}{switchingfunction} in the above formula   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H\+A } &The \hyperlink{switchingfunction}{switchingfunction} on the distance between bridging atoms and the atoms in group A   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H\+B } &The \hyperlink{switchingfunction}{switchingfunction} on the distance between the bridging atoms and the atoms in group B  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following example instructs plumed to calculate the number of water molecules that are bridging betweeen atoms 1-\/10 and atoms 11-\/20 and to print the value to a file

\begin{DoxyVerb}BRIDGE BRIDGING_ATOMS=100-200 GROUPA=1-10 GROUPB=11-20 LABEL=w1
PRINT ARG=a1.mean FILE=colvar 
\end{DoxyVerb}
 \hypertarget{COORDINATIONNUMBER}{}\subsection{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R}\label{COORDINATIONNUMBER}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the coordination numbers of atoms so that you can then calculate functions of the distribution of coordination numbers such as the minimum, the number less than a certain quantity and so on.

To make the calculation of coordination numbers differentiable the following function is used\+:

\[ s = \frac{ 1 - \left(\frac{r-d_0}{r_0}\right)^n } { 1 - \left(\frac{r-d_0}{r_0}\right)^m } \]

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  max } &{\bfseries  M\+A\+X }  &the maximum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S\+A } &this keyword is used for colvars such as the coordination number. In that context it species that plumed should calculate one coordination number for each of the atoms specified in S\+P\+E\+C\+I\+E\+S\+A. Each of these cooordination numbers specifies how many of the atoms specifies using S\+P\+E\+C\+I\+E\+S\+B is within the specified cutoff   \\\cline{1-2}
{\bfseries  S\+P\+E\+C\+I\+E\+S\+B } &this keyword is used for colvars such as the coordination number. It must appear with S\+P\+E\+C\+I\+E\+S\+A. For a full explanation see the documentation for that keyword   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+A\+X } &calculate the maximum value. To make this quantity continuous the maximum is calculated using $ \textrm{max} = \beta \log \sum_i \exp\left( \frac{s_i}{\beta}\right) $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.max.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to calculate the coordination numbers of atoms 1-\/100 with themselves. The minimum coordination number is then calculated. \begin{DoxyVerb}COORDINATIONNUMBER SPECIES=1-100 R_0=1.0 MIN={BETA=0.1}
\end{DoxyVerb}


The following input tells plumed to calculate how many atoms from 1-\/100 are within 3.\+0 of each of the atoms from 101-\/110. In the first 101 is the central atom, in the second 102 is the central atom and so on. The number of coordination numbers more than 6 is then computed. \begin{DoxyVerb}COORDINATIONNUMBER SPECIESA=101-110 SPECIESB=1-100 R_0=3.0 MORE_THAN={RATIONAL R_0=6.0 NN=6 MM=12 D_0=0}
\end{DoxyVerb}
 \hypertarget{DENSITY}{}\subsection{D\+E\+N\+S\+I\+T\+Y}\label{DENSITY}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate functions of the density of atoms as a function of the box. This allows one to calculate the number of atoms in half the box.

\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples }

\end{DoxyParagraph}
The following example calculates the number of atoms in one half of the simulation box.

\begin{DoxyVerb}DENSITY SPECIES=1-100 LABEL=d
SUBCELL ARG=d XLOWER=0.0 XUPPER=0.5 LABEL=d1
PRINT ARG=d1.* FILE=colvar1 FMT=%8.4f
\end{DoxyVerb}
 \hypertarget{DISTANCES}{}\subsection{D\+I\+S\+T\+A\+N\+C\+E\+S}\label{DISTANCES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the distances between one or many pairs of atoms. You can then calculate functions of the distribution of distances such as the minimum, the number less than a certain quantity and so on.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  dhenergy } &{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y }  &the Debye-\/\+Huckel interaction energy. You can calculate this quantity multiple times using different parameters   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  max } &{\bfseries  M\+A\+X }  &the maximum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P } &Calculate the distance between each distinct pair of atoms in the group   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+B.   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+A.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  M\+A\+X } &calculate the maximum value. To make this quantity continuous the maximum is calculated using $ \textrm{max} = \beta \log \sum_i \exp\left( \frac{s_i}{\beta}\right) $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.max.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y } &calculate the Debye-\/\+Huckel interaction energy. This is a alternative implementation of \hyperlink{DHENERGY}{D\+H\+E\+N\+E\+R\+G\+Y} that is particularly useful if you want to calculate the Debye-\/\+Huckel interaction energy and some other function of set of distances between the atoms in the two groups. The input for this keyword should read D\+H\+E\+N\+E\+R\+G\+Y=\{I= $I$ T\+E\+M\+P= $T$ E\+P\+S\+I\+L\+O\+N= $\epsilon$\}. You can use multiple instances of this keyword i.\+e. D\+H\+E\+N\+E\+R\+G\+Y1, D\+H\+E\+N\+E\+R\+G\+Y2, D\+H\+E\+N\+E\+R\+G\+Y3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to calculate the distances between atoms 3 and 5 and between atoms 1 and 2 and to print the minimum for these two distances. \begin{DoxyVerb}DISTANCES ATOMS1=3,5 ATOMS2=1,2 MIN={BETA=0.1} LABEL=d1
PRINT ARG=d1.min
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The following input tells plumed to calculate the distances between atoms 3 and 5 and between atoms 1 and 2 and then to calculate the number of these distances that are less than 0.\+1 nm. The number of distances less than 0.\+1nm is then printed to a file. \begin{DoxyVerb}DISTANCES ATOMS1=3,5 ATOMS2=1,2 LABEL=d1 LESS_THAN={RATIONAL R_0=0.1}
PRINT ARG=d1.lt0.1
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} \hyperlink{switchingfunction}{switchingfunction}).

The following input tells plumed to calculate all the distances between atoms 1, 2 and 3 (i.\+e. the distances between atoms 1 and 2, atoms 1 and 3 and atoms 2 and 3). The average of these distances is then calculated. \begin{DoxyVerb}DISTANCES GROUP=1-3 MEAN LABEL=d1
PRINT ARG=d1.mean
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T})

The following input tells plumed to calculate all the distances between the atoms in G\+R\+O\+U\+P\+A and the atoms in G\+R\+O\+U\+P\+B. In other words the distances between atoms 1 and 2 and the distance between atoms 1 and 3. The number of distances more than 0.\+1 is then printed to a file. \begin{DoxyVerb}DISTANCES GROUPA=1 GROUPB=2,3 MORE_THAN={RATIONAL R_0=0.1}
PRINT ARG=d1.gt0.1 
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} \hyperlink{switchingfunction}{switchingfunction}) \hypertarget{FCCUBIC}{}\subsection{F\+C\+C\+U\+B\+I\+C}\label{FCCUBIC}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
\hypertarget{MOLECULES}{}\subsection{M\+O\+L\+E\+C\+U\+L\+E\+S}\label{MOLECULES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the vectors connecting a pair of atoms in order to represent the orientation of a molecule.

At its simplest this command can be used to calculate the average length of an internal vector in a collection of different molecules. When used in conjunction with Muti\+Colvar\+Functions in can be used to do a variety of more complex tasks.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  M\+O\+L } &The numerical indices of the atoms in the molecule. The orientation of the molecule is equal to the vector connecting the first two atoms specified. If a third atom is specified its position is used to specify where the molecule is. If a third atom is not present the molecule is assumed to be at the center of the vector connecting the first two atoms. You can use multiple instances of this keyword i.\+e. M\+O\+L1, M\+O\+L2, M\+O\+L3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  V\+M\+E\+A\+N } &( default=off ) calculate the norm of the mean vector. The final value can be referenced using {\itshape label}.vmean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to calculate the distances between two of the atoms in a molecule. This is done for the same set of atoms four different molecules and the average separation is then calculated.

\begin{DoxyVerb}MOLECULES MOL1=1,2 MOL2=3,4 MOL3=5,6 MOL4=7,8 MEAN LABEL=mm
PRINT ARG=mm.mean FILE=colvar
\end{DoxyVerb}
 \hypertarget{Q3}{}\subsection{Q3}\label{Q3}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate 3rd order Steinhardt parameters.

The 3rd order Steinhardt parameters allow us to measure the degree to which the first coordination shell around an atom is ordered. The Steinhardt parameter for atom, $i$ is complex vector whose components are calculated using the following formula\+:

\[ q_{3m}(i) = \frac{\sum_j \sigma( r_{ij} ) Y_{3m}(\mathbf{r}_{ij}) }{\sum_j \sigma( r_{ij} ) } \]

where $Y_{3m}$ is one of the 3rd order spherical harmonics so $m$ is a number that runs from $-3$ to $+3$. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. The parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise.

The Steinhardt parameters can be used to measure the degree of order in the system in a variety of different ways. The simplest way of measuring whether or not the coordination sphere is ordered is to simply take the norm of the above vector i.\+e.

\[ Q_3(i) = \sqrt{ \sum_{m=-3}^3 q_{3m}(i)^{*} q_{3m}(i) } \]

This norm is small when the coordination shell is disordered and larger when the coordination shell is ordered. Furthermore, when the keywords L\+E\+S\+S\+\_\+\+T\+H\+A\+N, M\+I\+N, M\+A\+X, H\+I\+S\+T\+O\+G\+R\+A\+M, M\+E\+A\+N and so on are used with this colvar it is the distribution of these normed quantities that is investigated.

Other measures of order can be taken by averaging the components of the individual $q_3$ vectors individually or by taking dot products of the $q_{3}$ vectors on adjacent atoms. More information on these variables can be found in the documentation for \hyperlink{LOCAL_Q3}{L\+O\+C\+A\+L\+\_\+\+Q3}, \hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} and \hyperlink{NLINKS}{N\+L\+I\+N\+K\+S}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S\+A } &this keyword is used for colvars such as the coordination number. In that context it species that plumed should calculate one coordination number for each of the atoms specified in S\+P\+E\+C\+I\+E\+S\+A. Each of these cooordination numbers specifies how many of the atoms specifies using S\+P\+E\+C\+I\+E\+S\+B is within the specified cutoff   \\\cline{1-2}
{\bfseries  S\+P\+E\+C\+I\+E\+S\+B } &this keyword is used for colvars such as the coordination number. It must appear with S\+P\+E\+C\+I\+E\+S\+A. For a full explanation see the documentation for that keyword   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=12 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=24 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean   \\\cline{1-2}
{\bfseries  V\+M\+E\+A\+N } &( default=off ) calculate the norm of the mean vector. The final value can be referenced using {\itshape label}.vmean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following command calculates the average Q3 parameter for the 64 atoms in a box of Lennard Jones and prints this quantity to a file called colvar\+:

\begin{DoxyVerb}Q3 SPECIES=1-64 D_0=1.3 R_0=0.2 MEAN LABEL=q3
PRINT ARG=q3.mean FILE=colvar
\end{DoxyVerb}


The following command calculates the histogram of Q3 parameters for the 64 atoms in a box of Lennard Jones and prints these quantities to a file called colvar\+:

\begin{DoxyVerb}Q3 SPECIES=1-64 D_0=1.3 R_0=0.2 HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=1.0 NBINS=20 SMEAR=0.1} LABEL=q3
PRINT ARG=q3.* FILE=colvar
\end{DoxyVerb}


The following command could be used to measure the Q3 paramters that describe the arrangement of chlorine ions around the sodium atoms in Na\+Cl. The imagined system here is composed of 64 Na\+Cl formula units and the atoms are arranged in the input with the 64 Na $^+$ ions followed by the 64 Cl $-$ ions. Once again the average Q3 paramter is calculated and output to a file called colvar

\begin{DoxyVerb}Q3 SPECIESA=1-64 SPECIESB=65-128 D_0=1.3 R_0=0.2 MEAN LABEL=q3
PRINT ARG=q3.mean FILE=colvar
\end{DoxyVerb}
 \hypertarget{Q4}{}\subsection{Q4}\label{Q4}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate 4th order Steinhardt parameters.

The 4th order Steinhardt parameters allow us to measure the degree to which the first coordination shell around an atom is ordered. The Steinhardt parameter for atom, $i$ is complex vector whose components are calculated using the following formula\+:

\[ q_{4m}(i) = \frac{\sum_j \sigma( r_{ij} ) Y_{4m}(\mathbf{r}_{ij}) }{\sum_j \sigma( r_{ij} ) } \]

where $Y_{4m}$ is one of the 4th order spherical harmonics so $m$ is a number that runs from $-4$ to $+4$. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. The parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise.

The Steinhardt parameters can be used to measure the degree of order in the system in a variety of different ways. The simplest way of measuring whether or not the coordination sphere is ordered is to simply take the norm of the above vector i.\+e.

\[ Q_4(i) = \sqrt{ \sum_{m=-4}^4 q_{4m}(i)^{*} q_{4m}(i) } \]

This norm is small when the coordination shell is disordered and larger when the coordination shell is ordered. Furthermore, when the keywords L\+E\+S\+S\+\_\+\+T\+H\+A\+N, M\+I\+N, M\+A\+X, H\+I\+S\+T\+O\+G\+R\+A\+M, M\+E\+A\+N and so on are used with this colvar it is the distribution of these normed quantities that is investigated.

Other measures of order can be taken by averaging the components of the individual $q_4$ vectors individually or by taking dot products of the $q_{4}$ vectors on adjacent atoms. More information on these variables can be found in the documentation for \hyperlink{LOCAL_Q4}{L\+O\+C\+A\+L\+\_\+\+Q4}, \hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} and \hyperlink{NLINKS}{N\+L\+I\+N\+K\+S}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S\+A } &this keyword is used for colvars such as the coordination number. In that context it species that plumed should calculate one coordination number for each of the atoms specified in S\+P\+E\+C\+I\+E\+S\+A. Each of these cooordination numbers specifies how many of the atoms specifies using S\+P\+E\+C\+I\+E\+S\+B is within the specified cutoff   \\\cline{1-2}
{\bfseries  S\+P\+E\+C\+I\+E\+S\+B } &this keyword is used for colvars such as the coordination number. It must appear with S\+P\+E\+C\+I\+E\+S\+A. For a full explanation see the documentation for that keyword   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=12 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=24 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean   \\\cline{1-2}
{\bfseries  V\+M\+E\+A\+N } &( default=off ) calculate the norm of the mean vector. The final value can be referenced using {\itshape label}.vmean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following command calculates the average Q4 parameter for the 64 atoms in a box of Lennard Jones and prints this quantity to a file called colvar\+:

\begin{DoxyVerb}Q4 SPECIES=1-64 D_0=1.3 R_0=0.2 MEAN LABEL=q4
PRINT ARG=q4.mean FILE=colvar
\end{DoxyVerb}


The following command calculates the histogram of Q4 parameters for the 64 atoms in a box of Lennard Jones and prints these quantities to a file called colvar\+:

\begin{DoxyVerb}Q4 SPECIES=1-64 D_0=1.3 R_0=0.2 HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=1.0 NBINS=20 SMEAR=0.1} LABEL=q4
PRINT ARG=q4.* FILE=colvar
\end{DoxyVerb}


The following command could be used to measure the Q4 paramters that describe the arrangement of chlorine ions around the sodium atoms in Na\+Cl. The imagined system here is composed of 64 Na\+Cl formula units and the atoms are arranged in the input with the 64 Na $^+$ ions followed by the 64 Cl $-$ ions. Once again the average Q4 paramter is calculated and output to a file called colvar

\begin{DoxyVerb}Q4 SPECIESA=1-64 SPECIESB=65-128 D_0=1.3 R_0=0.2 MEAN LABEL=q4
PRINT ARG=q4.mean FILE=colvar
\end{DoxyVerb}
 \hypertarget{Q6}{}\subsection{Q6}\label{Q6}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate 6th order Steinhardt parameters.

The 6th order Steinhardt parameters allow us to measure the degree to which the first coordination shell around an atom is ordered. The Steinhardt parameter for atom, $i$ is complex vector whose components are calculated using the following formula\+:

\[ q_{6m}(i) = \frac{\sum_j \sigma( r_{ij} ) Y_{6m}(\mathbf{r}_{ij}) }{\sum_j \sigma( r_{ij} ) } \]

where $Y_{6m}$ is one of the 6th order spherical harmonics so $m$ is a number that runs from $-6$ to $+6$. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. The parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise.

The Steinhardt parameters can be used to measure the degree of order in the system in a variety of different ways. The simplest way of measuring whether or not the coordination sphere is ordered is to simply take the norm of the above vector i.\+e.

\[ Q_6(i) = \sqrt{ \sum_{m=-6}^6 q_{6m}(i)^{*} q_{6m}(i) } \]

This norm is small when the coordination shell is disordered and larger when the coordination shell is ordered. Furthermore, when the keywords L\+E\+S\+S\+\_\+\+T\+H\+A\+N, M\+I\+N, M\+A\+X, H\+I\+S\+T\+O\+G\+R\+A\+M, M\+E\+A\+N and so on are used with this colvar it is the distribution of these normed quantities that is investigated.

Other measures of order can be taken by averaging the components of the individual $q_6$ vectors individually or by taking dot products of the $q_{6}$ vectors on adjacent atoms. More information on these variables can be found in the documentation for \hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6}, \hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} and \hyperlink{NLINKS}{N\+L\+I\+N\+K\+S}.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S\+A } &this keyword is used for colvars such as the coordination number. In that context it species that plumed should calculate one coordination number for each of the atoms specified in S\+P\+E\+C\+I\+E\+S\+A. Each of these cooordination numbers specifies how many of the atoms specifies using S\+P\+E\+C\+I\+E\+S\+B is within the specified cutoff   \\\cline{1-2}
{\bfseries  S\+P\+E\+C\+I\+E\+S\+B } &this keyword is used for colvars such as the coordination number. It must appear with S\+P\+E\+C\+I\+E\+S\+A. For a full explanation see the documentation for that keyword   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=12 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=24 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean   \\\cline{1-2}
{\bfseries  V\+M\+E\+A\+N } &( default=off ) calculate the norm of the mean vector. The final value can be referenced using {\itshape label}.vmean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following command calculates the average Q6 parameter for the 64 atoms in a box of Lennard Jones and prints this quantity to a file called colvar\+:

\begin{DoxyVerb}Q6 SPECIES=1-64 D_0=1.3 R_0=0.2 MEAN LABEL=q6
PRINT ARG=q6.mean FILE=colvar
\end{DoxyVerb}


The following command calculates the histogram of Q6 parameters for the 64 atoms in a box of Lennard Jones and prints these quantities to a file called colvar\+:

\begin{DoxyVerb}Q6 SPECIES=1-64 D_0=1.3 R_0=0.2 HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=1.0 NBINS=20 SMEAR=0.1} LABEL=q6
PRINT ARG=q6.* FILE=colvar
\end{DoxyVerb}


The following command could be used to measure the Q6 paramters that describe the arrangement of chlorine ions around the sodium atoms in Na\+Cl. The imagined system here is composed of 64 Na\+Cl formula units and the atoms are arranged in the input with the 64 Na $^+$ ions followed by the 64 Cl $-$ ions. Once again the average Q6 paramter is calculated and output to a file called colvar

\begin{DoxyVerb}Q6 SPECIESA=1-64 SPECIESB=65-128 D_0=1.3 R_0=0.2 MEAN LABEL=q6
PRINT ARG=q6.mean FILE=colvar
\end{DoxyVerb}
 \hypertarget{SIMPLECUBIC}{}\subsection{S\+I\+M\+P\+L\+E\+C\+U\+B\+I\+C}\label{SIMPLECUBIC}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate whether or not the coordination spheres of atoms are arranged as they would be in a simple cubic structure.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  max } &{\bfseries  M\+A\+X }  &the maximum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S\+A } &this keyword is used for colvars such as the coordination number. In that context it species that plumed should calculate one coordination number for each of the atoms specified in S\+P\+E\+C\+I\+E\+S\+A. Each of these cooordination numbers specifies how many of the atoms specifies using S\+P\+E\+C\+I\+E\+S\+B is within the specified cutoff   \\\cline{1-2}
{\bfseries  S\+P\+E\+C\+I\+E\+S\+B } &this keyword is used for colvars such as the coordination number. It must appear with S\+P\+E\+C\+I\+E\+S\+A. For a full explanation see the documentation for that keyword   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+A\+X } &calculate the maximum value. To make this quantity continuous the maximum is calculated using $ \textrm{max} = \beta \log \sum_i \exp\left( \frac{s_i}{\beta}\right) $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.max.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to calculate the simple cubic parameter for the atoms 1-\/100 with themselves. The mean value is then calculated. \begin{DoxyVerb}SIMPLECUBIC SPECIES=1-100 R_0=1.0 MEAN
\end{DoxyVerb}


The following input tells plumed to look at the ways atoms 1-\/100 are within 3.\+0 are arranged about atoms from 101-\/110. The number of simple cubic parameters that are greater than 0.\+8 is then output \begin{DoxyVerb}SIMPLECUBIC SPECIESA=101-110 SPECIESB=1-100 R_0=3.0 MORE_THAN={RATIONAL R_0=0.8 NN=6 MM=12 D_0=0}
\end{DoxyVerb}
 \hypertarget{TETRAHEDRAL}{}\subsection{T\+E\+T\+R\+A\+H\+E\+D\+R\+A\+L}\label{TETRAHEDRAL}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  max } &{\bfseries  M\+A\+X }  &the maximum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S } &this keyword is used for colvars such as coordination number. In that context it specifies that plumed should calculate one coordination number for each of the atoms specified. Each of these coordination numbers specifies how many of the other specified atoms are within a certain cutoff of the central atom.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  S\+P\+E\+C\+I\+E\+S\+A } &this keyword is used for colvars such as the coordination number. In that context it species that plumed should calculate one coordination number for each of the atoms specified in S\+P\+E\+C\+I\+E\+S\+A. Each of these cooordination numbers specifies how many of the atoms specifies using S\+P\+E\+C\+I\+E\+S\+B is within the specified cutoff   \\\cline{1-2}
{\bfseries  S\+P\+E\+C\+I\+E\+S\+B } &this keyword is used for colvars such as the coordination number. It must appear with S\+P\+E\+C\+I\+E\+S\+A. For a full explanation see the documentation for that keyword   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+A\+X } &calculate the maximum value. To make this quantity continuous the maximum is calculated using $ \textrm{max} = \beta \log \sum_i \exp\left( \frac{s_i}{\beta}\right) $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.max.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
\hypertarget{TORSIONS}{}\subsection{T\+O\+R\+S\+I\+O\+N\+S}\label{TORSIONS}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate whether or not a set of torsional angles are within a particular range.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following provides an example of the input for the torsions command

\begin{DoxyVerb}TORSIONS ...
ATOMS1=168,170,172,188
ATOMS2=170,172,188,190
ATOMS3=188,190,192,230 
LABEL=ab
... TORSIONS
PRINT ARG=ab.* FILE=colvar STRIDE=10
\end{DoxyVerb}


Writing out the atoms involved in all the torsions in this way can be rather tedious. Thankfully if you are working with protein you can avoid this by using the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command. P\+L\+U\+M\+E\+D uses the pdb file that you provide to this command to learn about the topology of the protein molecule. This means that you can specify torsion angles using the following syntax\+:

\begin{DoxyVerb}MOLINFO MOLTYPE=protein STRUCTURE=myprotein.pdb
TORSIONS ...
ATOMS1=@phi-3
ATOMS2=@psi-3
ATOMS3=@phi-4
LABEL=ab
... TORSIONS 
PRINT ARG=ab FILE=colvar STRIDE=10
\end{DoxyVerb}


Here, @phi-\/3 tells plumed that you would like to calculate the $\phi$ angle in the third residue of the protein. Similarly @psi-\/4 tells plumed that you want to calculate the $\psi$ angle of the 4th residue of the protein. \hypertarget{XDISTANCES}{}\subsection{X\+D\+I\+S\+T\+A\+N\+C\+E\+S}\label{XDISTANCES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the x components of the vectors connecting one or many pairs of atoms. You can then calculate functions of the distribution of values such as the minimum, the number less than a certain quantity and so on.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  dhenergy } &{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y }  &the Debye-\/\+Huckel interaction energy. You can calculate this quantity multiple times using different parameters   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P } &Calculate the distance between each distinct pair of atoms in the group   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+B.   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+A.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y } &calculate the Debye-\/\+Huckel interaction energy. This is a alternative implementation of \hyperlink{DHENERGY}{D\+H\+E\+N\+E\+R\+G\+Y} that is particularly useful if you want to calculate the Debye-\/\+Huckel interaction energy and some other function of set of distances between the atoms in the two groups. The input for this keyword should read D\+H\+E\+N\+E\+R\+G\+Y=\{I= $I$ T\+E\+M\+P= $T$ E\+P\+S\+I\+L\+O\+N= $\epsilon$\}. You can use multiple instances of this keyword i.\+e. D\+H\+E\+N\+E\+R\+G\+Y1, D\+H\+E\+N\+E\+R\+G\+Y2, D\+H\+E\+N\+E\+R\+G\+Y3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following input tells plumed to calculate the x-\/component of the vector connecting atom 3 to atom 5 and the x-\/component of the vector connecting atom 1 to atom 2. The minimum of these two quantities is then printed \begin{DoxyVerb}XDISTANCES ATOMS1=3,5 ATOMS2=1,2 MIN={BETA=0.1} LABEL=d1
PRINT ARG=d1.min
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The following input tells plumed to calculate the x-\/component of the vector connecting atom 3 to atom 5 and the x-\/component of the vector connecting atom 1 to atom 2. The number of values that are less than 0.\+1nm is then printed to a file. \begin{DoxyVerb}XDISTANCES ATOMS1=3,5 ATOMS2=1,2 LABEL=d1 LESS_THAN={RATIONAL R_0=0.1}
PRINT ARG=d1.lt0.1
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} \hyperlink{switchingfunction}{switchingfunction}).

The following input tells plumed to calculate the x-\/components of all the distinct vectors that can be created between atoms 1, 2 and 3 (i.\+e. the vectors between atoms 1 and 2, atoms 1 and 3 and atoms 2 and 3). The average of these quantities is then calculated. \begin{DoxyVerb}XDISTANCES GROUP=1-3 AVERAGE LABEL=d1
PRINT ARG=d1.average
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T})

The following input tells plumed to calculate all the vectors connecting the the atoms in G\+R\+O\+U\+P\+A to the atoms in G\+R\+O\+U\+P\+B. In other words the vector between atoms 1 and 2 and the vector between atoms 1 and 3. The number of values more than 0.\+1 is then printed to a file. \begin{DoxyVerb}XDISTANCES GROUPA=1 GROUPB=2,3 MORE_THAN={RATIONAL R_0=0.1}
PRINT ARG=d1.gt0.1 
\end{DoxyVerb}
 (See also \hyperlink{PRINT}{P\+R\+I\+N\+T} \hyperlink{switchingfunction}{switchingfunction}) \hypertarget{YDISTANCES}{}\subsection{Y\+D\+I\+S\+T\+A\+N\+C\+E\+S}\label{YDISTANCES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the y components of the vectors connecting one or many pairs of atoms. You can then calculate functions of the distribution of values such as the minimum, the number less than a certain quantity and so on.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  dhenergy } &{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y }  &the Debye-\/\+Huckel interaction energy. You can calculate this quantity multiple times using different parameters   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P } &Calculate the distance between each distinct pair of atoms in the group   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+B.   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+A.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y } &calculate the Debye-\/\+Huckel interaction energy. This is a alternative implementation of \hyperlink{DHENERGY}{D\+H\+E\+N\+E\+R\+G\+Y} that is particularly useful if you want to calculate the Debye-\/\+Huckel interaction energy and some other function of set of distances between the atoms in the two groups. The input for this keyword should read D\+H\+E\+N\+E\+R\+G\+Y=\{I= $I$ T\+E\+M\+P= $T$ E\+P\+S\+I\+L\+O\+N= $\epsilon$\}. You can use multiple instances of this keyword i.\+e. D\+H\+E\+N\+E\+R\+G\+Y1, D\+H\+E\+N\+E\+R\+G\+Y2, D\+H\+E\+N\+E\+R\+G\+Y3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
See documentation for \hyperlink{XDISTANCES}{X\+D\+I\+S\+T\+A\+N\+C\+E\+S} for examples of how to use this command. You just need to substitute Y\+D\+I\+S\+T\+A\+N\+C\+E\+S for X\+D\+I\+S\+T\+A\+N\+C\+E\+S to investigate the y component rather than the x component. \hypertarget{ZDISTANCES}{}\subsection{Z\+D\+I\+S\+T\+A\+N\+C\+E\+S}\label{ZDISTANCES}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the z components of the vectors connecting one or many pairs of atoms. You can then calculate functions of the distribution of values such as the minimum, the number less than a certain quantity and so on.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  dhenergy } &{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y }  &the Debye-\/\+Huckel interaction energy. You can calculate this quantity multiple times using different parameters   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M\+S } &the atoms involved in each of the collective variables you wish to calculate. Keywords like A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3,... should be listed and one C\+V will be calculated for each A\+T\+O\+M keyword you specify (all A\+T\+O\+M keywords should define the same number of atoms). The eventual number of quantities calculated by this action will depend on what functions of the distribution you choose to calculate. You can use multiple instances of this keyword i.\+e. A\+T\+O\+M\+S1, A\+T\+O\+M\+S2, A\+T\+O\+M\+S3...   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P } &Calculate the distance between each distinct pair of atoms in the group   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Or alternatively by using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  G\+R\+O\+U\+P\+A } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+B.   \\\cline{1-2}
{\bfseries  G\+R\+O\+U\+P\+B } &Calculate the distances between all the atoms in G\+R\+O\+U\+P\+A and all the atoms in G\+R\+O\+U\+P\+B. This must be used in conjuction with G\+R\+O\+U\+P\+A.   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  V\+E\+R\+B\+O\+S\+E } &( default=off ) write a more detailed output   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y } &calculate the Debye-\/\+Huckel interaction energy. This is a alternative implementation of \hyperlink{DHENERGY}{D\+H\+E\+N\+E\+R\+G\+Y} that is particularly useful if you want to calculate the Debye-\/\+Huckel interaction energy and some other function of set of distances between the atoms in the two groups. The input for this keyword should read D\+H\+E\+N\+E\+R\+G\+Y=\{I= $I$ T\+E\+M\+P= $T$ E\+P\+S\+I\+L\+O\+N= $\epsilon$\}. You can use multiple instances of this keyword i.\+e. D\+H\+E\+N\+E\+R\+G\+Y1, D\+H\+E\+N\+E\+R\+G\+Y2, D\+H\+E\+N\+E\+R\+G\+Y3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
See documentation for \hyperlink{XDISTANCES}{X\+D\+I\+S\+T\+A\+N\+C\+E\+S} for examples of how to use this command. You just need to substitute Z\+D\+I\+S\+T\+A\+N\+C\+E\+S for X\+D\+I\+S\+T\+A\+N\+C\+E\+S to investigate the z component rather than the x component. \hypertarget{AROUND}{}\subsection{A\+R\+O\+U\+N\+D}\label{AROUND}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
This quantity can be used to calculate functions of the distribution of collective variables for the atoms that lie in a particular, user-\/specified part of of the cell.

Each of the base quantities calculated by a multicolvar can can be assigned to a particular point in three dimensional space. For example, if we have the coordination numbers for all the atoms in the system each coordination number can be assumed to lie on the position of the central atom. Because each base quantity can be assigned to a particular point in space we can calculate functions of the distribution of base quantities in a particular part of the box by using\+:

\[ \overline{s}_{\tau} = \frac{ \sum_i f(s_i) w(x_i,y_i,z_i) }{ \sum_i w(x_i,y_i,z_i) } \]

where the sum is over the collective variables, $s_i$, each of which can be thought to be at $ (x_i,y_i,z_i)$. The function $ w(x_i,y_i,z_i) $ measures whether or not the system is in the subregion of interest. It is equal to\+:

\[ w(x_i,y_i,z_i) = \int_{xl}^{xu} \int_{yl}^{yu} \int_{zl}^{zu} \textrm{d}x\textrm{d}y\textrm{d}z K\left( \frac{x - x_i}{\sigma} \right)K\left( \frac{y - y_i}{\sigma} \right)K\left( \frac{z - z_i}{\sigma} \right) \]

where $K$ is one of the kernel functions described on \hyperlink{histogrambead}{histogrambead} and $\sigma$ is a bandwidth parameter. The function $(s_i)$ can be any of the usual L\+E\+S\+S\+\_\+\+T\+H\+A\+N, M\+O\+R\+E\+\_\+\+T\+H\+A\+N, W\+I\+T\+H\+I\+N etc that are used in all other multicolvars.

When A\+R\+O\+U\+N\+D is used with the \hyperlink{DENSITY}{D\+E\+N\+S\+I\+T\+Y} action the number of atoms in the specified region is calculated

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
This Action can be used to calculate the following quantities by employing the keywords listed below. You must select which from amongst these quantities you wish to calculate -\/ this command cannot be run unless one of the quantities below is being calculated.\+These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2 etc.

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{The atoms involved can be specified using}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  A\+T\+O\+M } &the atom whose vicinity we are interested in examining. For more information on how to specify lists of atoms see \hyperlink{Group}{Groups and Virtual Atoms}   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &certain actions in plumed work by calculating a list of variables and summing over them. This particular action can be used to calculate functions of these base variables or prints them to a file. This keyword thus takes the label of one of those such variables as input.   \\\cline{1-2}
{\bfseries  S\+I\+G\+M\+A } &the width of the function to be used for kernel density estimation   \\\cline{1-2}
{\bfseries  K\+E\+R\+N\+E\+L } &( default=gaussian ) the type of kernel function to be used   \\\cline{1-2}
{\bfseries  X\+L\+O\+W\+E\+R } &( default=0.\+0 ) the lower boundary in x relative to the x coordinate of the atom (0 indicates use full extent of box).   \\\cline{1-2}
{\bfseries  X\+U\+P\+P\+E\+R } &( default=0.\+0 ) the upper boundary in x relative to the x coordinate of the atom (0 indicates use full extent of box).   \\\cline{1-2}
{\bfseries  Y\+L\+O\+W\+E\+R } &( default=0.\+0 ) the lower boundary in y relative to the y coordinate of the atom (0 indicates use full extent of box).   \\\cline{1-2}
{\bfseries  Y\+U\+P\+P\+E\+R } &( default=0.\+0 ) the upper boundary in y relative to the y coordinate of the atom (0 indicates use full extent of box).   \\\cline{1-2}
{\bfseries  Z\+L\+O\+W\+E\+R } &( default=0.\+0 ) the lower boundary in z relative to the z coordinate of the atom (0 indicates use full extent of box).   \\\cline{1-2}
{\bfseries  Z\+U\+P\+P\+E\+R } &( default=0.\+0 ) the upper boundary in z relative to the z coordinate of the atom (0 indicates use full extent of box).   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean   \\\cline{1-2}
{\bfseries  V\+M\+E\+A\+N } &( default=off ) calculate the norm of the mean vector. The final value can be referenced using {\itshape label}.vmean   \\\cline{1-2}
{\bfseries  O\+U\+T\+S\+I\+D\+E } &( default=off ) calculate quantities for colvars that are on atoms outside the region of interest  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following commands tell plumed to calculate the average coordination number for the atoms that have x (in fractional coordinates) within 2.\+0 nm of the com of mass c1. The final value will be labeled s.\+mean. \begin{DoxyVerb}COM ATOMS=1-100 LABEL=c1
COORDINATIONNUMBER SPECIES=1-100 R_0=1.0 LABEL=c
AROUND ARG=c ORIGIN=c1 XLOWER=-2.0 XUPPER=2.0 SIGMA=0.1 MEAN LABEL=s
\end{DoxyVerb}
 \hypertarget{LOCAL_AVERAGE}{}\subsection{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E}\label{LOCAL_AVERAGE}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate averages over spherical regions centered on atoms

As is explained in \href{http://www.youtube.com/watch?v=iDvZmbWE5ps}{\tt this video } certain multicolvars calculate one scalar quantity or one vector for each of the atoms in the system. For example \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R} measures the coordination number of each of the atoms in the system and \hyperlink{Q4}{Q4} measures the 4th order Steinhardt parameter for each of the atoms in the system. These quantities provide tell us something about the disposition of the atoms in the first coordination sphere of each of the atoms of interest. Lechner and Dellago \cite{dellago-q6} have suggested that one can probe local order in a system by taking the average value of such symmetry functions over the atoms within a spherical cutoff of each of these atoms in the systems. When this is done with Steinhardt parameters they claim this gives a coordinate that is better able to distinguish solid and liquid configurations of Lennard-\/\+Jones atoms.

You can calculate such locally averaged quantities within plumed by using the L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E command. This command calculates the following atom-\/centered quantities\+:

\[ s_i = \frac{ c_i + \sum_j \sigma(r_{ij})c_j }{ 1 + \sum_j \sigma(r_{ij}) } \]

where the $c_i$ and $c_j$ values can be for any one of the symmetry functions that can be calculated using plumed multicolvars. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. Lechner and Dellago suggest that the parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise.

The $s_i$ quantities calculated using the above command can be again thought of as atom-\/centred symmetry functions. They thus operate much like multicolvars. You can thus calculate properties of the distribution of $s_i$ values using M\+E\+A\+N, L\+E\+S\+S\+\_\+\+T\+H\+A\+N, H\+I\+S\+T\+O\+G\+R\+A\+M and so on. You can also probe the value of these averaged variables in regions of the box by using the command in tandem with the \hyperlink{AROUND}{A\+R\+O\+U\+N\+D} command.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &the labels of the action that calculates the multicolvars we are interested in   \\\cline{1-2}
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
This example input calculates the coordination numbers for all the atoms in the system. These coordination numbers are then averaged over spherical regions. The number of averaged coordination numbers that are greater than 4 is then output to a file.

\begin{DoxyVerb}COORDINATIONNUMBER SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=d1
LOCAL_AVERAGE ARG=d1 SWITCH={RATIONAL D_0=1.3 R_0=0.2} MORE_THAN={RATIONAL R_0=4} LABEL=la
PRINT ARG=la.* FILE=colvar 
\end{DoxyVerb}


This example input calculates the $q_4$ (see \hyperlink{Q4}{Q4}) vectors for each of the atoms in the system. These vectors are then averaged component by component over a spherical region. The average value for this quantity is then outputeed to a file. This calculates the quantities that were used in the paper by Lechner and Dellago \cite{dellago-q6}

\begin{DoxyVerb}Q4 SPECIES=1-64 SWITCH={RATIONAL D_0=1.3 R_0=0.2} LABEL=q4
LOCAL_AVERAGE ARG=q4 SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LABEL=la
PRINT ARG=la.* FILE=colvar
\end{DoxyVerb}
 \hypertarget{LOCAL_Q3}{}\subsection{L\+O\+C\+A\+L\+\_\+\+Q3}\label{LOCAL_Q3}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the local degree of order around an atoms by taking the average dot product between the $q_3$ vector on the central atom and the $q_3$ vector on the atoms in the first coordination sphere.

The \hyperlink{Q3}{Q3} command allows one to calculate one complex vectors for each of the atoms in your system that describe the degree of order in the coordination sphere around a particular atom. The difficulty with these vectors comes when combining the order parameters from all of the individual atoms/molecules so as to get a measure of the global degree of order for the system. The simplest way of doing this -\/ calculating the average Steinhardt parameter -\/ can be problematic. If one is examining nucleation say only the order parameters for those atoms in the nucleus will change significantly when the nucleus forms. The order parameters for the atoms in the surrounding liquid will remain pretty much the same. As such if one models a small nucleus embedded in a very large amount of solution/melt any change in the average order parameter will be negligible. Substantial changes in the value of this average can be observed in simulations of nucleation but only because the number of atoms is relatively small.

When the average \hyperlink{Q3}{Q3} parameter is used to bias the dynamics a problems can occur. These averaged coordinates cannot distinguish between the correct, single-\/nucleus pathway and a concerted pathway in which all the atoms rearrange themselves into their solid-\/like configuration simultaneously. This second type of pathway would be impossible in reality because there is a large entropic barrier that prevents concerted processes like this from happening. However, in the finite sized systems that are commonly simulated this barrier is reduced substantially. As a result in simulations where average Steinhardt parameters are biased there are often quite dramatic system size effects

If one wants to simulate nucleation using some form on biased dynamics what is really required is an order parameter that measures\+:


\begin{DoxyItemize}
\item Whether or not the coordination spheres around atoms are ordered
\item Whether or not the atoms that are ordered are clustered together in a crystalline nucleus
\end{DoxyItemize}

\hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} and \hyperlink{NLINKS}{N\+L\+I\+N\+K\+S} are variables that can be combined with the Steinhardt parameteters allow to calculate variables that satisfy these requirements. L\+O\+C\+A\+L\+\_\+\+Q3 is another variable that can be used in these sorts of calculations. The L\+O\+C\+A\+L\+\_\+\+Q3 parameter for a particular atom is a number that measures the extent to which the orientation of the atoms in the first coordination sphere of an atom match the orientation of the central atom. It does this by calculating the following quantity for each of the atoms in the system\+:

\[ s_i = \frac{ \sum_j \sigma( r_{ij} ) \sum_{m=-3}^3 q_{3m}^{*}(i)q_{3m}(j) }{ \sum_j \sigma( r_{ij} ) } \]

where $q_{3m}(i)$ and $q_{3m}(j)$ are the 3rd order Steinhardt vectors calculated for atom $i$ and atom $j$ respectively and the asterix denotes complex conjugation. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. The parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise. The sum in the numerator of this expression is the dot product of the Steinhardt parameters for atoms $i$ and $j$ and thus measures the degree to which the orientations of these adjacent atoms is correlated.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &the labels of the action that calculates the multicolvars we are interested in   \\\cline{1-2}
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following command calculates the average value of the L\+O\+C\+A\+L\+\_\+\+Q3 parameter for the 64 Lennard Jones atoms in the system under study and prints this quantity to a file called colvar.

\begin{DoxyVerb}Q3 SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=q3
LOCAL_Q3 ARG=q3 SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LABEL=lq3
PRINT ARG=lq3.mean FILE=colvar
\end{DoxyVerb}


The following input calculates the distribution of L\+O\+C\+A\+L\+\_\+\+Q3 parameters at any given time and outputs this information to a file.

\begin{DoxyVerb}Q3 SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=q3
LOCAL_Q3 ARG=q3 SWITCH={RATIONAL D_0=1.3 R_0=0.2} HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=1.0 NBINS=20 SMEAR=0.1} LABEL=lq3
PRINT ARG=lq3.* FILE=colvar
\end{DoxyVerb}


The following calculates the L\+O\+C\+A\+L\+\_\+\+Q3 parameters for atoms 1-\/5 only. For each of these atoms comparisons of the geometry of the coordination sphere are done with those of all the other atoms in the system. The final quantity is the average and is outputted to a file

\begin{DoxyVerb}Q3 SPECIESA=1-5 SPECIESB=1-64 D_0=1.3 R_0=0.2 LABEL=q3a
Q3 SPECIESA=6-64 SPECIESB=1-64 D_0=1.3 R_0=0.2 LABEL=q3b

LOCAL_Q3 ARG=q3a,q3b SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LOWMEM LABEL=w3
PRINT ARG=w3.* FILE=colvar
\end{DoxyVerb}
 \hypertarget{LOCAL_Q4}{}\subsection{L\+O\+C\+A\+L\+\_\+\+Q4}\label{LOCAL_Q4}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the local degree of order around an atoms by taking the average dot product between the $q_4$ vector on the central atom and the $q_4$ vector on the atoms in the first coordination sphere.

The \hyperlink{Q4}{Q4} command allows one to calculate one complex vectors for each of the atoms in your system that describe the degree of order in the coordination sphere around a particular atom. The difficulty with these vectors comes when combining the order parameters from all of the individual atoms/molecules so as to get a measure of the global degree of order for the system. The simplest way of doing this -\/ calculating the average Steinhardt parameter -\/ can be problematic. If one is examining nucleation say only the order parameters for those atoms in the nucleus will change significantly when the nucleus forms. The order parameters for the atoms in the surrounding liquid will remain pretty much the same. As such if one models a small nucleus embedded in a very large amount of solution/melt any change in the average order parameter will be negligible. Substantial changes in the value of this average can be observed in simulations of nucleation but only because the number of atoms is relatively small.

When the average \hyperlink{Q4}{Q4} parameter is used to bias the dynamics a problems can occur. These averaged coordinates cannot distinguish between the correct, single-\/nucleus pathway and a concerted pathway in which all the atoms rearrange themselves into their solid-\/like configuration simultaneously. This second type of pathway would be impossible in reality because there is a large entropic barrier that prevents concerted processes like this from happening. However, in the finite sized systems that are commonly simulated this barrier is reduced substantially. As a result in simulations where average Steinhardt parameters are biased there are often quite dramatic system size effects

If one wants to simulate nucleation using some form on biased dynamics what is really required is an order parameter that measures\+:


\begin{DoxyItemize}
\item Whether or not the coordination spheres around atoms are ordered
\item Whether or not the atoms that are ordered are clustered together in a crystalline nucleus
\end{DoxyItemize}

\hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} and \hyperlink{NLINKS}{N\+L\+I\+N\+K\+S} are variables that can be combined with the Steinhardt parameteters allow to calculate variables that satisfy these requirements. L\+O\+C\+A\+L\+\_\+\+Q4 is another variable that can be used in these sorts of calculations. The L\+O\+C\+A\+L\+\_\+\+Q4 parameter for a particular atom is a number that measures the extent to which the orientation of the atoms in the first coordination sphere of an atom match the orientation of the central atom. It does this by calculating the following quantity for each of the atoms in the system\+:

\[ s_i = \frac{ \sum_j \sigma( r_{ij} ) \sum_{m=-4}^4 q_{4m}^{*}(i)q_{4m}(j) }{ \sum_j \sigma( r_{ij} ) } \]

where $q_{4m}(i)$ and $q_{4m}(j)$ are the 4th order Steinhardt vectors calculated for atom $i$ and atom $j$ respectively and the asterix denotes complex conjugation. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. The parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise. The sum in the numerator of this expression is the dot product of the Steinhardt parameters for atoms $i$ and $j$ and thus measures the degree to which the orientations of these adjacent atoms is correlated.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &the labels of the action that calculates the multicolvars we are interested in   \\\cline{1-2}
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following command calculates the average value of the L\+O\+C\+A\+L\+\_\+\+Q4 parameter for the 64 Lennard Jones atoms in the system under study and prints this quantity to a file called colvar.

\begin{DoxyVerb}Q4 SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=q4
LOCAL_Q4 ARG=q4 SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LABEL=lq4
PRINT ARG=lq4.mean FILE=colvar
\end{DoxyVerb}


The following input calculates the distribution of L\+O\+C\+A\+L\+\_\+\+Q4 parameters at any given time and outputs this information to a file.

\begin{DoxyVerb}Q4 SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=q4
LOCAL_Q4 ARG=q4 SWITCH={RATIONAL D_0=1.3 R_0=0.2} HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=1.0 NBINS=20 SMEAR=0.1} LABEL=lq4
PRINT ARG=lq4.* FILE=colvar
\end{DoxyVerb}


The following calculates the L\+O\+C\+A\+L\+\_\+\+Q4 parameters for atoms 1-\/5 only. For each of these atoms comparisons of the geometry of the coordination sphere are done with those of all the other atoms in the system. The final quantity is the average and is outputted to a file

\begin{DoxyVerb}Q4 SPECIESA=1-5 SPECIESB=1-64 D_0=1.3 R_0=0.2 LABEL=q4a
Q4 SPECIESA=6-64 SPECIESB=1-64 D_0=1.3 R_0=0.2 LABEL=q4b

LOCAL_Q4 ARG=q4a,q4b SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LOWMEM LABEL=w4
PRINT ARG=w4.* FILE=colvar
\end{DoxyVerb}
 \hypertarget{LOCAL_Q6}{}\subsection{L\+O\+C\+A\+L\+\_\+\+Q6}\label{LOCAL_Q6}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the crystallization \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate the local degree of order around an atoms by taking the average dot product between the $q_6$ vector on the central atom and the $q_6$ vector on the atoms in the first coordination sphere.

The \hyperlink{Q6}{Q6} command allows one to calculate one complex vectors for each of the atoms in your system that describe the degree of order in the coordination sphere around a particular atom. The difficulty with these vectors comes when combining the order parameters from all of the individual atoms/molecules so as to get a measure of the global degree of order for the system. The simplest way of doing this -\/ calculating the average Steinhardt parameter -\/ can be problematic. If one is examining nucleation say only the order parameters for those atoms in the nucleus will change significantly when the nucleus forms. The order parameters for the atoms in the surrounding liquid will remain pretty much the same. As such if one models a small nucleus embedded in a very large amount of solution/melt any change in the average order parameter will be negligible. Substantial changes in the value of this average can be observed in simulations of nucleation but only because the number of atoms is relatively small.

When the average \hyperlink{Q6}{Q6} parameter is used to bias the dynamics a problems can occur. These averaged coordinates cannot distinguish between the correct, single-\/nucleus pathway and a concerted pathway in which all the atoms rearrange themselves into their solid-\/like configuration simultaneously. This second type of pathway would be impossible in reality because there is a large entropic barrier that prevents concerted processes like this from happening. However, in the finite sized systems that are commonly simulated this barrier is reduced substantially. As a result in simulations where average Steinhardt parameters are biased there are often quite dramatic system size effects

If one wants to simulate nucleation using some form on biased dynamics what is really required is an order parameter that measures\+:


\begin{DoxyItemize}
\item Whether or not the coordination spheres around atoms are ordered
\item Whether or not the atoms that are ordered are clustered together in a crystalline nucleus
\end{DoxyItemize}

\hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} and \hyperlink{NLINKS}{N\+L\+I\+N\+K\+S} are variables that can be combined with the Steinhardt parameteters allow to calculate variables that satisfy these requirements. L\+O\+C\+A\+L\+\_\+\+Q6 is another variable that can be used in these sorts of calculations. The L\+O\+C\+A\+L\+\_\+\+Q6 parameter for a particular atom is a number that measures the extent to which the orientation of the atoms in the first coordination sphere of an atom match the orientation of the central atom. It does this by calculating the following quantity for each of the atoms in the system\+:

\[ s_i = \frac{ \sum_j \sigma( r_{ij} ) \sum_{m=-6}^6 q_{6m}^{*}(i)q_{6m}(j) }{ \sum_j \sigma( r_{ij} ) } \]

where $q_{6m}(i)$ and $q_{6m}(j)$ are the 6th order Steinhardt vectors calculated for atom $i$ and atom $j$ respectively and the asterix denotes complex conjugation. The function $\sigma( r_{ij} )$ is a \hyperlink{switchingfunction}{switchingfunction} that acts on the distance between atoms $i$ and $j$. The parameters of this function should be set so that it the function is equal to one when atom $j$ is in the first coordination sphere of atom $i$ and is zero otherwise. The sum in the numerator of this expression is the dot product of the Steinhardt parameters for atoms $i$ and $j$ and thus measures the degree to which the orientations of these adjacent atoms is correlated.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
When the label of this action is used as the input for a second you are not referring to a scalar quantity as you are in regular collective variables. The label is used to reference the full set of quantities calculated by the action. This is usual when using \hyperlink{mcolv_multicolvarfunction}{Multi\+Colvar functions}. Generally when doing this the previously calculated multicolvar will be referenced using the D\+A\+T\+A keyword rather than A\+R\+G.

This Action can be used to calculate the following scalar quantities directly. These quantities are calculated by employing the keywords listed below. These quantities can then be referenced elsewhere in the input file by using this Action's label followed by a dot and the name of the quantity. Some amongst them can be calculated multiple times with different parameters. In this case the quantities calculated can be referenced elsewhere in the input by using the name of the quantity followed by a numerical identifier e.\+g. {\itshape label}.lessthan-\/1, {\itshape label}.lessthan-\/2 etc. When doing this and, for clarity we have made the label of the components customizable. As such by using the L\+A\+B\+E\+L keyword in the description of the keyword input you can customize the component name

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &the labels of the action that calculates the multicolvars we are interested in   \\\cline{1-2}
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements   \\\cline{1-2}
{\bfseries  M\+E\+A\+N } &( default=off ) take the mean of these variables. The final value can be referenced using {\itshape label}.mean  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.   \\\cline{1-2}
{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N } &calculate the number of variables more than a certain target value. This quantity is calculated using $\sum_i 1.0 - \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.more\+\_\+than. You can use multiple instances of this keyword i.\+e. M\+O\+R\+E\+\_\+\+T\+H\+A\+N1, M\+O\+R\+E\+\_\+\+T\+H\+A\+N2, M\+O\+R\+E\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.more\+\_\+than-\/1, {\itshape label}.more\+\_\+than-\/2, {\itshape label}.more\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N } &calculate the number of variables less than a certain target value. This quantity is calculated using $\sum_i \sigma(s_i)$, where $\sigma(s)$ is a \hyperlink{switchingfunction}{switchingfunction}. The final value can be referenced using {\itshape label}.less\+\_\+than. You can use multiple instances of this keyword i.\+e. L\+E\+S\+S\+\_\+\+T\+H\+A\+N1, L\+E\+S\+S\+\_\+\+T\+H\+A\+N2, L\+E\+S\+S\+\_\+\+T\+H\+A\+N3... The corresponding values are then referenced using {\itshape label}.less\+\_\+than-\/1, {\itshape label}.less\+\_\+than-\/2, {\itshape label}.less\+\_\+than-\/3...   \\\cline{1-2}
{\bfseries  M\+I\+N } &calculate the minimum value. To make this quantity continuous the minimum is calculated using $ \textrm{min} = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } $ The value of $\beta$ in this function is specified using (B\+E\+T\+A= $\beta$) The final value can be referenced using {\itshape label}.min.   \\\cline{1-2}
{\bfseries  B\+E\+T\+W\+E\+E\+N } &calculate the number of values that are within a certain range. These quantities are calculated using kernel density estimation as described on \hyperlink{histogrambead}{histogrambead}. The final value can be referenced using {\itshape label}.between. You can use multiple instances of this keyword i.\+e. B\+E\+T\+W\+E\+E\+N1, B\+E\+T\+W\+E\+E\+N2, B\+E\+T\+W\+E\+E\+N3... The corresponding values are then referenced using {\itshape label}.between-\/1, {\itshape label}.between-\/2, {\itshape label}.between-\/3...   \\\cline{1-2}
{\bfseries  H\+I\+S\+T\+O\+G\+R\+A\+M } &calculate a discretized histogram of the distribution of values. This shortcut allows you to calculates N\+B\+I\+N quantites like B\+E\+T\+W\+E\+E\+N.   \\\cline{1-2}
{\bfseries  M\+O\+M\+E\+N\+T\+S } &calculate the moments of the distribution of collective variables. The $m$th moment of a distribution is calculated using $\frac{1}{N} \sum_{i=1}^N ( s_i - \overline{s} )^m $, where $\overline{s}$ is the average for the distribution. The moments keyword takes a lists of integers as input or a range. Each integer is a value of $m$. The final calculated values can be referenced using moment-\/ $m$.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following command calculates the average value of the L\+O\+C\+A\+L\+\_\+\+Q6 parameter for the 64 Lennard Jones atoms in the system under study and prints this quantity to a file called colvar.

\begin{DoxyVerb}Q6 SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=q6
LOCAL_Q6 ARG=q6 SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LABEL=lq6
PRINT ARG=lq6.mean FILE=colvar
\end{DoxyVerb}


The following input calculates the distribution of L\+O\+C\+A\+L\+\_\+\+Q6 parameters at any given time and outputs this information to a file.

\begin{DoxyVerb}Q6 SPECIES=1-64 D_0=1.3 R_0=0.2 LABEL=q6
LOCAL_Q6 ARG=q6 SWITCH={RATIONAL D_0=1.3 R_0=0.2} HISTOGRAM={GAUSSIAN LOWER=0.0 UPPER=1.0 NBINS=20 SMEAR=0.1} LABEL=lq6
PRINT ARG=lq6.* FILE=colvar
\end{DoxyVerb}


The following calculates the L\+O\+C\+A\+L\+\_\+\+Q6 parameters for atoms 1-\/5 only. For each of these atoms comparisons of the geometry of the coordination sphere are done with those of all the other atoms in the system. The final quantity is the average and is outputted to a file

\begin{DoxyVerb}Q6 SPECIESA=1-5 SPECIESB=1-64 D_0=1.3 R_0=0.2 LABEL=q6a
Q6 SPECIESA=6-64 SPECIESB=1-64 D_0=1.3 R_0=0.2 LABEL=q6b

LOCAL_Q6 ARG=q4a,q4b SWITCH={RATIONAL D_0=1.3 R_0=0.2} MEAN LOWMEM LABEL=w4
PRINT ARG=w6.* FILE=colvar
\end{DoxyVerb}
 \hypertarget{NLINKS}{}\subsection{N\+L\+I\+N\+K\+S}\label{NLINKS}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate number of pairs of atoms/molecules that are \char`\"{}linked\char`\"{}

In its simplest guise this coordinate calculates a coordination number. Each pair of atoms is assumed \char`\"{}linked\char`\"{} if they are within some cutoff of each other. In more complex applications each entity is a vector and this quantity measures whether pairs of vectors are (a) within a certain cutoff and (b) if the two vectors have similar orientations. The vectors on individual atoms could be Steinhardt parameters (see \hyperlink{Q3}{Q3}, \hyperlink{Q4}{Q4} and \hyperlink{Q6}{Q6}) or they could describe some internal vector in a molecule.

\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &the labels of the action that calculates the multicolvars we are interested in   \\\cline{1-2}
{\bfseries  N\+N } &( default=6 ) The n parameter of the switching function   \\\cline{1-2}
{\bfseries  M\+M } &( default=12 ) The m parameter of the switching function   \\\cline{1-2}
{\bfseries  D\+\_\+0 } &( default=0.\+0 ) The d\+\_\+0 parameter of the switching function   \\\cline{1-2}
{\bfseries  R\+\_\+0 } &The r\+\_\+0 parameter of the switching function   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords.  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following calculates how many bonds there are in a system containing 64 atoms and outputs this quantity to a file.

\begin{DoxyVerb}DENSITY SPECIES=1-64 LABEL=d1
NLINKS ARG=d1 SWITCH={RATIONAL D_0=1.3 R_0=0.2} LABEL=dd
PRINT ARG=dd FILE=colvar
\end{DoxyVerb}


The following calculates how many pairs of neighbouring atoms in a system containg 64 atoms have similar dispositions for the atoms in their coordination sphere. This calculation uses the dot product of the Q6 vectors on adjacent atoms to measure whether or not two atoms have the same ``orientation"

\begin{DoxyVerb}Q6 SPECIES=1-64 SWITCH={RATIONAL D_0=1.3 R_0=0.2} LABEL=q6
NLINKS ARG=q6 SWITCH={RATIONAL D_0=1.3 R_0=0.2} LABEL=dd
PRINT ARG=dd FILE=colvar
\end{DoxyVerb}
 \hypertarget{SPRINT}{}\subsection{S\+P\+R\+I\+N\+T}\label{SPRINT}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the multicolvar \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Calculate S\+P\+R\+I\+N\+T topological variables.

The S\+P\+R\+I\+N\+T topological variables are calculated from the largest eigenvalue, $\lambda$ of an $n\times n$ adjacency matrix and its corresponding eigenvector, $\mathbf{V}$, using\+:

\[ s_i = \sqrt{n} \lambda v_i \]

You can use different quantities to measure whether or not two given atoms/molecules are adjacent or not in the adjacency matrix. The simplest measure of adjacency is is whether two atoms/molecules are within some cutoff of each other. Further complexity can be added by insisting that two molecules are adjacent if they are within a certain distance of each other and if they have similar orientations.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default this Action calculates the following quantities. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  coord } &all \$n\$ sprint coordinates are calculated and then stored in increasing order. the smallest sprint coordinate will be labelled {\itshape label}.coord-\/1, the second smallest will be labelleled {\itshape label}.coord-\/1 and so on   \\\cline{1-2}
\end{TabularC}


In addition the following quantities can be calculated by employing the keywords listed below

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
{\bfseries  spath } &{\bfseries  S\+P\+A\+T\+H }  &the position on the path   \\\cline{1-3}
{\bfseries  zpath } &{\bfseries  Z\+P\+A\+T\+H }  &the distance from the path   \\\cline{1-3}
{\bfseries  dhenergy } &{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y }  &the Debye-\/\+Huckel interaction energy. You can calculate this quantity multiple times using different parameters   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  max } &{\bfseries  M\+A\+X }  &the maximum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  sum } &{\bfseries  S\+U\+M }  &the sum of values   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &the labels of the action that calculates the multicolvars we are interested in   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+O\+P\+B\+C } &( default=off ) ignore the periodic boundary conditions when calculating distances   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{TabularC}{2}
\hline
{\bfseries  T\+O\+L } &this keyword can be used to speed up your calculation. When accumulating sums in which the individual terms are numbers inbetween zero and one it is assumed that terms less than a certain tolerance make only a small contribution to the sum. They can thus be safely ignored as can the the derivatives wrt these small quantities.   \\\cline{1-2}
{\bfseries  S\+W\+I\+T\+C\+H } &This keyword is used if you want to employ an alternative to the continuous swiching function defined above. The following provides information on the \hyperlink{switchingfunction}{switchingfunction} that are available. When this keyword is present you no longer need the N\+N, M\+M, D\+\_\+0 and R\+\_\+0 keywords. You can use multiple instances of this keyword i.\+e. S\+W\+I\+T\+C\+H1, S\+W\+I\+T\+C\+H2, S\+W\+I\+T\+C\+H3...  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
This example input calculates the 7 S\+P\+R\+I\+N\+T coordinates for a 7 atom cluster of Lennard-\/\+Jones atoms and prints their values to a file. In this input the S\+P\+R\+I\+N\+T coordinates are calculated in the manner described in ?? so two atoms are adjacent if they are within a cutoff\+:

\begin{DoxyVerb}DENSITY SPECIES=1-7 LABEL=d1
SPRINT ARG=d1 SWITCH={RATIONAL R_0=0.1} LABEL=ss
PRINT ARG=ss.* FILE=colvar 
\end{DoxyVerb}


This example input calculates the 14 S\+P\+R\+I\+N\+T coordinates foa a molecule composed of 7 hydrogen and 7 carbon atoms. Once again two atoms are adjacent if they are within a cutoff\+:

\begin{DoxyVerb}DENSITY SPECIES=1-7 LABEL=c
DENSITY SPECIES=8-14 LABEL=h

SPRINT ...
 ARG=c,h
 SWITCH11={RATIONAL R_0=2.6 NN=6 MM=12}
 SWITCH12={RATIONAL R_0=2.2 NN=6 MM=12}
 SWITCH22={RATIONAL R_0=2.2 NN=6 MM=12}
 LABEL=ss
... SPRINT

PRINT ARG=ss.* FILE=colvar
\end{DoxyVerb}
 \hypertarget{UWALLS}{}\subsection{U\+W\+A\+L\+L\+S}\label{UWALLS}
\begin{TabularC}{2}
\hline
&{\bfseries  This is part of the manyrestraints \hyperlink{mymodules}{module }}   \\\cline{1-2}
\end{TabularC}
Add \hyperlink{UPPER_WALLS}{U\+P\+P\+E\+R\+\_\+\+W\+A\+L\+L\+S} restraints on all the multicolvar values

This action takes the set of values calculated by the colvar specified by label in the D\+A\+T\+A keyword and places a restraint on each quantity, $x$, with the following functional form\+:

$ k((x-a+o)/s)^e $

$k$ (K\+A\+P\+P\+A) is an energy constant in internal unit of the code, $s$ (E\+P\+S) a rescaling factor and $e$ (E\+X\+P) the exponent determining the power law. By default\+: E\+X\+P = 2, E\+P\+S = 1.\+0, O\+F\+F = 0.

\begin{DoxyParagraph}{Description of components}

\end{DoxyParagraph}
By default the value of the calculated quantity can be referenced elsewhere in the input file by using the label of the action. Alternatively this Action can be used to be used to calculate the following quantities by employing the keywords listed below. These quanties can be referenced elsewhere in the input by using this Action's label followed by a dot and the name of the quantity required from the list below.

\begin{TabularC}{2}
\hline
{\bfseries  Quantity }  &{\bfseries  Description }   \\\cline{1-2}
{\bfseries  bias } &the instantaneous value of the bias potentials   \\\cline{1-2}
\end{TabularC}


In addition the following quantities can be calculated by employing the keywords listed below

\begin{TabularC}{3}
\hline
{\bfseries  Quantity }  &{\bfseries  Keyword }  &{\bfseries  Description }   \\\cline{1-3}
{\bfseries  vmean } &{\bfseries  V\+M\+E\+A\+N }  &the norm of the mean vector. The output component can be refererred to elsewhere in the input file by using the label.\+vmean   \\\cline{1-3}
{\bfseries  spath } &{\bfseries  S\+P\+A\+T\+H }  &the position on the path   \\\cline{1-3}
{\bfseries  zpath } &{\bfseries  Z\+P\+A\+T\+H }  &the distance from the path   \\\cline{1-3}
{\bfseries  dhenergy } &{\bfseries  D\+H\+E\+N\+E\+R\+G\+Y }  &the Debye-\/\+Huckel interaction energy. You can calculate this quantity multiple times using different parameters   \\\cline{1-3}
{\bfseries  between } &{\bfseries  B\+E\+T\+W\+E\+E\+N }  &the number/fraction of values within a certain range. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  lessthan } &{\bfseries  L\+E\+S\+S\+\_\+\+T\+H\+A\+N }  &the number of values less than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  max } &{\bfseries  M\+A\+X }  &the maximum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  mean } &{\bfseries  M\+E\+A\+N }  &the mean value. The output component can be refererred to elsewhere in the input file by using the label.\+mean   \\\cline{1-3}
{\bfseries  min } &{\bfseries  M\+I\+N }  &the minimum value. This is calculated using the formula described in the description of the keyword so as to make it continuous.   \\\cline{1-3}
{\bfseries  moment } &{\bfseries  M\+O\+M\+E\+N\+T\+S }  &the central moments of the distribution of values. The second moment would be referenced elsewhere in the input file using {\itshape label}.moment-\/2, the third as {\itshape label}.moment-\/3, etc.   \\\cline{1-3}
{\bfseries  morethan } &{\bfseries  M\+O\+R\+E\+\_\+\+T\+H\+A\+N }  &the number of values more than a target value. This is calculated using one of the formula described in the description of the keyword so as to make it continuous. You can calculate this quantity multiple times using different parameters.   \\\cline{1-3}
{\bfseries  sum } &{\bfseries  S\+U\+M }  &the sum of values   \\\cline{1-3}
\end{TabularC}


\begin{DoxyParagraph}{Compulsory keywords}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  D\+A\+T\+A } &certain actions in plumed work by calculating a list of variables and summing over them. This particular action can be used to calculate functions of these base variables or prints them to a file. This keyword thus takes the label of one of those such variables as input.   \\\cline{1-2}
{\bfseries  A\+T } &the radius of the sphere   \\\cline{1-2}
{\bfseries  K\+A\+P\+P\+A } &the force constant for the wall. The k\+\_\+i in the expression for a wall.   \\\cline{1-2}
{\bfseries  O\+F\+F\+S\+E\+T } &( default=0.\+0 ) the offset for the start of the wall. The o\+\_\+i in the expression for a wall.   \\\cline{1-2}
{\bfseries  E\+X\+P } &( default=2.\+0 ) the powers for the walls. The e\+\_\+i in the expression for a wall.   \\\cline{1-2}
{\bfseries  E\+P\+S } &( default=1.\+0 ) the values for s\+\_\+i in the expression for a wall   \\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Options}

\end{DoxyParagraph}
\begin{TabularC}{2}
\hline
{\bfseries  N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S } &( default=off ) calculate the derivatives for these quantities numerically   \\\cline{1-2}
{\bfseries  S\+E\+R\+I\+A\+L } &( default=off ) do the calculation in serial. Do not parallelize   \\\cline{1-2}
{\bfseries  L\+O\+W\+M\+E\+M } &( default=off ) lower the memory requirements  

\\\cline{1-2}
\end{TabularC}


\begin{DoxyParagraph}{Examples}

\end{DoxyParagraph}
The following set of commands can be used to stop a cluster composed of 20 atoms subliming. The position of the centre of mass of the cluster is calculated by the \hyperlink{COM}{C\+O\+M} command labelled c1. The \hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S} command labelled d1 is then used to calculate the distance between each of the 20 atoms in the cluster and the center of mass of the cluster. These distances are then passed to the U\+W\+A\+L\+L\+S command, which adds a \hyperlink{UPPER_WALLS}{U\+P\+P\+E\+R\+\_\+\+W\+A\+L\+L\+S} restraint on each of them and thereby prevents each of them from moving very far from the centre of mass of the cluster.

\begin{DoxyVerb}COM ATOMS=1-20 LABEL=c1
DISTANCES GROUPA=c1 GROUPB=1-20 LABEL=d1
UWALLS DATA=d1 AT=2.5 KAPPA=0.2 LABEL=sr
\end{DoxyVerb}
 