The following pages describe how to perform a variety of tasks using P\+L\+U\+M\+E\+D 2

\begin{TabularC}{2}
\hline
\hyperlink{belfast-1}{Belfast tutorial\+: Analyzing C\+Vs}  &This tutorial explains how to use plumed to analyze C\+Vs   \\\cline{1-2}
\hyperlink{belfast-2}{Belfast tutorial\+: Adaptive variables I}  &How to use path C\+Vs   \\\cline{1-2}
\hyperlink{belfast-3}{Belfast tutorial\+: Adaptive variables I\+I}  &Dimensionality reduction and sketch maps   \\\cline{1-2}
\hyperlink{belfast-4}{Belfast tutorial\+: Umbrella sampling}  &Umbrella sampling, reweighting, and weighted histogram   \\\cline{1-2}
\hyperlink{belfast-5}{Belfast tutorial\+: Out of equilibrium dynamics}  &How to run a steered M\+D simulations and how to estimate the free energy   \\\cline{1-2}
\hyperlink{belfast-6}{Belfast tutorial\+: Metadynamics}  &How to run a metadynamics simulation   \\\cline{1-2}
\hyperlink{belfast-7}{Belfast tutorial\+: Replica exchange I}  &Parallel tempering and Metadynamics, Well-\/\+Tempered Ensemble   \\\cline{1-2}
\hyperlink{belfast-8}{Belfast tutorial\+: Replica exchange I\+I and Multiple walkers}  &Bias exchange and multiple walkers   \\\cline{1-2}
\hyperlink{belfast-9}{Belfast tutorial\+: N\+M\+R constraints}  &N\+M\+R constraints   \\\cline{1-2}
\hyperlink{belfast-10}{Belfast tutorial\+: Steinhardt Parameters}  &Steinhardt Parameters   \\\cline{1-2}
\hyperlink{mindist}{Calculating a minimum distance}  &This tutorial explains how to calculate the minimum distance between groups of atoms and serves as an introduction to Multi\+Colvars   \\\cline{1-2}
\hyperlink{moving}{Moving from Plumed 1 to Plumed 2}  &This tutorial explains how plumed 1 input files can be translated into the new plumed 2 syntax.   \\\cline{1-2}
\hyperlink{munster}{Munster tutorial}  &A short 3 hours tutorial   \\\cline{1-2}
\end{TabularC}


In addition, the following websites contain resources that might be helpful

\begin{TabularC}{2}
\hline
\href{http://www.youtube.com/watch?v=iDvZmbWE5ps}{\tt http\+://www.\+youtube.\+com/watch?v=i\+Dv\+Zmb\+W\+E5ps}  &A short video introduction to the use of multicolvars in P\+L\+U\+M\+E\+D 2   \\\cline{1-2}
\href{http://www.youtube.com/watch?v=PxJP16qNCYs}{\tt http\+://www.\+youtube.\+com/watch?v=\+Px\+J\+P16q\+N\+C\+Ys}  &A short video introduction to the syntax of the P\+L\+U\+M\+E\+D 2 input file   \\\cline{1-2}
\href{http://en.wikipedia.org/wiki/Metadynamics}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Metadynamics}  &A wikipedia article on metadynamics   \\\cline{1-2}
\end{TabularC}
\hypertarget{belfast-1}{}\section{Belfast tutorial\+: Analyzing C\+Vs}\label{belfast-1}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
The aim of this tutorial is to introduce the users to the plumed syntax. We will go through the writing of simple collective variable and we will use them to analyse a trajectory in terms of probabilty distributions and free energy.\hypertarget{belfast-1_belfast-1-lo}{}\subsection{Learning Outcomes}\label{belfast-1_belfast-1-lo}
Once this tutorial is completed students will\+:


\begin{DoxyItemize}
\item Know how to write a simple plumed input file
\item Know how to analyse a trajectory using plumed
\end{DoxyItemize}\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
The \href{tutorial-resources/belfast-1.tar.gz}{\tt tarball } for this project contains the following files\+:


\begin{DoxyItemize}
\item trajectory-\/short.\+xyz \+: a (short) trajectory for a 16 residue protein in xyz format. All calculations with plumed driver use this trajectory.
\item template.\+pdb \+: a single frame from the trajectory that can be used in conjuction with the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command
\end{DoxyItemize}\hypertarget{belfast-10_Instructions}{}\subsection{Instructions}\label{belfast-10_Instructions}
P\+L\+U\+M\+E\+D2 is a library that can be accessed by multiple codes adding a relatively simple and well documented interface. Once P\+L\+U\+M\+E\+D is installed you can run a plumed executable that can be used for multiple purposes\+:

\begin{DoxyVerb}plumed --help 
\end{DoxyVerb}


some of the listed options report about the plumed available functionalities, other can be used to tell plumed to do something\+: from analysing a trajectory to patch the source code of a M\+D code and so on. All the commands have further options that can be seen using plumed command --help, i.\+e.\+:

\begin{DoxyVerb}plumed driver --help
\end{DoxyVerb}


In the following we are going to see how to write an input file for plumed2 that can be used to analyse a trajectory.\hypertarget{belfast-1_Units}{}\subsubsection{A note on units}\label{belfast-1_Units}
By default the P\+L\+U\+M\+E\+D inputs and outputs quantities in the following units\+:


\begin{DoxyItemize}
\item Energy -\/ k\+J/mol
\item Length -\/ nanometers
\item Time -\/ picoseconds
\end{DoxyItemize}

If you want to change these units you can do this using the \hyperlink{UNITS}{U\+N\+I\+T\+S} keyword.\hypertarget{belfast-1_introinput}{}\subsubsection{Introduction to the P\+L\+U\+M\+E\+D input file}\label{belfast-1_introinput}
A typical input file for P\+L\+U\+M\+E\+D input is composed by specification of one or more C\+Vs, the printout frequency and a termination line. Comments are denoted with a \# and the termination of the input for P\+L\+U\+M\+E\+D is marked with the keyword E\+N\+D\+P\+L\+U\+M\+E\+D. Whatever it follows is ignored by P\+L\+U\+M\+E\+D. You can introduce blank lines. They are not interpreted by P\+L\+U\+M\+E\+D.

In the following input we will analyse the \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} between the two terminal carbons of a 16 residues peptide, and we will \hyperlink{PRINT}{P\+R\+I\+N\+T} the results in file named C\+O\+L\+V\+A\+R.

\begin{DoxyVerb}#my first plumed input:
DISTANCE ATOMS=2,253 LABEL=e2edist

#printout frequency
PRINT ARG=e2edist STRIDE=1 FILE=COLVAR 

#endofinput 
ENDPLUMED
here I can write what I want it won't be read.
\end{DoxyVerb}


Now we can use this simple input file to analyse the trajectory included in the R\+E\+S\+O\+U\+R\+C\+E\+S\+:

\begin{DoxyVerb}plumed driver --plumed plumed.dat --ixyz trajectory-short.xyz --length-units 0.1
\end{DoxyVerb}


N\+O\+T\+E\+: --length-\/units 0.\+1, xyz files, as well as pdb files, are in Angstrom.

You should have a file C\+O\+L\+V\+A\+R, if you look at it (i.\+e. more C\+O\+L\+V\+A\+R) the first two lines should be\+:

\begin{DoxyVerb}#! FIELDS time e2edist 
 0.000000 2.5613161
\end{DoxyVerb}


N\+O\+T\+E\+: the first line of the file C\+O\+L\+V\+A\+R tells you what is the content of each column.

In P\+L\+U\+M\+E\+D2 the commands defined in the input files are executed in the same order in which they are written, this means that the following input file is wrong\+:

\begin{DoxyVerb}#printout frequency
PRINT ARG=cvdist STRIDE=1 FILE=COLVAR 
#my first plumed input:
DISTANCE ATOMS=2,253 LABEL=e2edist
#endofinput 
ENDPLUMED
here I can write what I want it won't be read.
\end{DoxyVerb}


Try to run it.

Sometimes, when calculating a collective variable, you may not want to use the positions of a number of atoms directly. Instead you may wish to use the position of a virtual atom whose position is generated based on the positions of a collection of other atoms. For example you might want to use the center of mass of a group of atoms (\hyperlink{COM}{C\+O\+M})\+:

Since P\+L\+U\+M\+E\+D executes the input in order you need to define the new Virtual Atom before using it\+:

\begin{DoxyVerb}first: COM ATOMS=1,2,3,4,5,6
last: COM ATOMS=251-256

e2edist: DISTANCE ATOMS=2,253
comdist: DISTANCE ATOMS=first,last

PRINT ARG=e2edist,comdist STRIDE=1 FILE=COLVAR 

ENDPLUMED
\end{DoxyVerb}


N\+O\+T\+E\+: an action (i.\+e. C\+O\+M or D\+I\+S\+T\+A\+N\+C\+E here) can be either label using L\+A\+B\+E\+L as we did before or as label\+: A\+C\+T\+I\+O\+N as we have just done here.

With the above input this is what happen inside P\+L\+U\+M\+E\+D with a S\+T\+R\+I\+D\+E=1\+:


\begin{DoxyEnumerate}
\item calculates the position of the Virtual Atom 'first' as the \hyperlink{COM}{C\+O\+M} of atoms from 1 to 6;
\item calculates the position of the Virtual Atom 'last' as the \hyperlink{COM}{C\+O\+M} of atoms from 251 to 256;
\item calculates the distance between atoms 2 and 253 and saves it in 'e2edist';
\item calculates the distance between the two atoms 'first' and 'last' and saves it in 'comdist';
\item print the content of 'e2edist' and 'comdist' in the file C\+O\+L\+V\+A\+R
\end{DoxyEnumerate}

In the above input we have used to different ways of writing the atoms used in \hyperlink{COM}{C\+O\+M} calculation\+:


\begin{DoxyEnumerate}
\item A\+T\+O\+M\+S=1,2,3,4,5,6 is the explicit list of the atoms we need
\item A\+T\+O\+M\+S=251-\/256 is the range of atoms needed
\end{DoxyEnumerate}

ranges of atoms can be defined with a stride which can also be negative\+:


\begin{DoxyEnumerate}
\item A\+T\+O\+M\+S=from,to\+:by (i.\+e.\+: 251-\/256\+:2)
\item A\+T\+O\+M\+S=to,from\+:-\/by (i.\+e.\+: 256-\/251\+:-\/2)
\end{DoxyEnumerate}

Now by plotting the content of the C\+O\+L\+V\+A\+R file we can compare the behaviour in this trajectory of both the terminal carbons as well as of the centre of masses of the terminal residues.

\begin{DoxyVerb}gnuplot
\end{DoxyVerb}


What do you expect to see now by looking at the trajectory? Let's have a look at it

\begin{DoxyVerb}vmd template.pdb trajectory-short.xyz 
\end{DoxyVerb}


Virtual atoms can be used in place of standard atoms everywhere an atom can be given as input, they can also be used together with standard atoms. So for example we can analyse the \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} angle for a set of Virtual and Standard atoms\+:

\begin{DoxyVerb}first: COM ATOMS=1-6
last: COM ATOMS=251-256
cvtor: TORSION ATOMS=first,102,138,last

PRINT ARG=cvtor STRIDE=1 FILE=COLVAR 

ENDPLUMED
\end{DoxyVerb}


The above C\+V don't look smart to learn something about the system we are looking at. In principle C\+V are used to reduce the complexity of a system by looking at a small number of properties that could be enough to rationalise its behaviour.

Now try to write a collective variable that measures the Radius of Gyration of the system\+: \hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N}.

N\+O\+T\+E\+: if what you need for one or more variables is a long list of atoms and not a virtual atom one can use the keyword \hyperlink{GROUP}{G\+R\+O\+U\+P}. A G\+R\+O\+U\+P can be defined using A\+T\+O\+M\+S in the same way we saw before, in addition it is also possible to define a G\+R\+O\+U\+P by reading a G\+R\+O\+M\+A\+C\+S index file.

\begin{DoxyVerb}ca: GROUP ATOMS=9,16,31,55,69,90,102,114,124,138,160,174,194,208,224,238
\end{DoxyVerb}


Now 'ca' is not a virtual atom but a simple list of atoms.\hypertarget{belfast-1_multicol}{}\subsubsection{M\+U\+L\+T\+I\+C\+O\+L\+V\+A\+R}\label{belfast-1_multicol}
Sometimes it can be useful to calculate properties of many similar collective variables at the same time, for example one can be interested in calculating the properties of the distances between a group of atoms, or properties linked to the distribution of the dihedral angles of a chain and so on. In P\+L\+U\+M\+E\+D2 this kind of collective variables fall under the name of M\+U\+L\+T\+I\+C\+O\+L\+V\+A\+R (cf. \hyperlink{mcolv}{Multi\+Colvar Documentation}.) Here we are going to analyse the distances between C\+A carbons along the chain\+:

\begin{DoxyVerb}ca: GROUP ATOMS=9,16,31,55,69,90,102,114,124,138,160,174,194,208,224,238
dd: DISTANCES GROUP=ca MEAN MIN={BETA=50} MAX={BETA=0.02} MOMENTS=2

PRINT ARG=dd.mean,dd.min,dd.max,dd.moment-2 STRIDE=1 FILE=COLVAR 

ENDPLUMED
\end{DoxyVerb}


The above input tells P\+L\+U\+M\+E\+D to calculate all the distances between C\+A carbons and then look for the mean distance, the minimum distance, the maximum distance and the variance. In this way we have defined four collective variables that are calculated using the distances. These four collective variables are stored as components of the defined action 'dd'\+: dd.\+mean, dd.\+min, dd.\+max, dd.\+moment-\/2.

The infrastracture of multicolvar has been used to develop many P\+L\+U\+M\+E\+D2 collective variables as for example the set of Secondary Structure C\+Vs (\hyperlink{ANTIBETARMSD}{A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D}, \hyperlink{PARABETARMSD}{P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D} and \hyperlink{ALPHARMSD}{A\+L\+P\+H\+A\+R\+M\+S\+D}).

\begin{DoxyVerb}MOLINFO STRUCTURE=template.pdb
abeta: ANTIBETARMSD RESIDUES=all TYPE=DRMSD LESS_THAN={RATIONAL R_0=0.08 NN=8 MM=12} STRANDS_CUTOFF=1

PRINT ARG=abeta.lessthan STRIDE=1 FILE=COLVAR 

ENDPLUMED
\end{DoxyVerb}


We have now seen how to write the input some of the many C\+Vs available in P\+L\+U\+M\+E\+D. More complex C\+Vs will be discussed in the next workshop, \hyperlink{belfast-2}{Belfast tutorial\+: Adaptive variables I}.\hypertarget{belfast-1_analysis}{}\subsubsection{Analysis of Collective Variables}\label{belfast-1_analysis}
Collective variables are usually used to visualize the Free Energy of a system. Given a system evolving at fixed temperature, fixed number of particles and fixed volume, it will explore different conformations with a probability

\[ P(q)\propto e^{-\frac{U(q)}{kb_BT}} \] where $ q $ are the microscopic coordinates and $ k_B $ is the Boltzmann constant.

It is possible to analyse the above probabilty as a function of one or more collective variable $ s(q)$\+:

\[ P(s)\propto \int dq e^{-\frac{U(q)}{kb_BT}} \delta(s-s(q)) \]

where the $ \delta $ function means that to for a given value $ s$ of the collective variable are counted only those conformations for which the C\+V is $ s$. The probability can be recast to a free energy by taking its logarithm\+: \[ F(s)=-k_B T \log P(s) \]

This means that by estimating the probability distribution of a C\+V it is possible to know the free energy of a system along that C\+V. Estimating the probability distribution of the conformations of a system is what is called 'sampling'.

In order to estimate a probability distribution one needs to make \hyperlink{HISTOGRAM}{H\+I\+S\+T\+O\+G\+R\+A\+M} from the calculated C\+Vs. P\+L\+U\+M\+E\+D2 includes the possibility of histogramming data both on the fly as well as a posteriori as we are going to do now.

\begin{DoxyVerb}MOLINFO STRUCTURE=template.pdb
abeta: ANTIBETARMSD RESIDUES=all TYPE=DRMSD LESS_THAN={RATIONAL R_0=0.08 NN=8 MM=12} STRANDS_CUTOFF=1
ca: GROUP ATOMS=9,16,31,55,69,90,102,114,124,138,160,174,194,208,224,238
DISTANCES ...
GROUP=ca MEAN MIN={BETA=50} MAX={BETA=0.02} MOMENTS=2 LABEL=dd
... DISTANCES 

PRINT ARG=abeta.lessthan,dd.mean,dd.min,dd.max,dd.moment-2 STRIDE=1 FILE=COLVAR 

HISTOGRAM ...
ARG=abeta.lessthan,dd.mean
USE_ALL_DATA
KERNEL=discrete
GRID_MIN=0,0.8
GRID_MAX=4,1.2
GRID_BIN=40,40
GRID_WFILE=histo
... HISTOGRAM

ENDPLUMED
\end{DoxyVerb}


N\+O\+T\+E\+: H\+I\+S\+T\+O\+G\+R\+A\+M ... means that what follow is part of the \hyperlink{HISTOGRAM}{H\+I\+S\+T\+O\+G\+R\+A\+M} function, the same can be done for any action in P\+L\+U\+M\+E\+D.

The above input tells P\+L\+U\+M\+E\+D to accumulate the two collective variables on a G\+R\+I\+D. In addition the probability can be converted to a free-\/energy using the flag F\+R\+E\+E-\/\+E\+N\+E\+R\+G\+Y and setting the temperature using T\+E\+M\+P (i.\+e. 300\+K). Histograms can be accumulated in a smoother way by using a K\+E\+R\+N\+E\+L function, a kernel is a normalised function, for example a normalised gaussian is the default kernel in P\+L\+U\+M\+E\+D, that is added to the histogram centered in the position of the data. Estimating a probability density using kernels can in principle give more accurate results, on the other hand in addition to the choice of the binning one has to choose a parameter that is the W\+I\+D\+T\+H of the kernel function. As a rule of thumb\+: the grid spacing should be smaller (i.\+e. one half or less) than the B\+A\+N\+D\+W\+I\+D\+T\+H and the B\+A\+N\+D\+W\+I\+D\+T\+H should be smaller (i.\+e. one order of magnitude) than the variance observed/expected for the variable.

\begin{DoxyVerb}HISTOGRAM ...
ARG=abeta.lessthan,dd.mean
USE_ALL_DATA
GRID_MIN=0,0.8
GRID_MAX=4,1.2
GRID_SPACING=0.04,0.004
BANDWIDTH=0.08,0.008
GRID_WFILE=histo
... HISTOGRAM

ENDPLUMED
\end{DoxyVerb}


If you have time less at the end of the session read the manual and look for alternative collective variables to analyse the trajectory. Furthemore try to play with the \hyperlink{HISTOGRAM}{H\+I\+S\+T\+O\+G\+R\+A\+M} parameters to see the effect of using K\+E\+R\+N\+E\+L in analysing data. \hypertarget{belfast-2}{}\section{Belfast tutorial\+: Adaptive variables I}\label{belfast-2}
\hypertarget{belfast-2_belfast-2-aim}{}\subsection{Aim}\label{belfast-2_belfast-2-aim}
In this section we want to introduce the concept of adaptive collective variables. These are special variables that are knowledge-\/based in that are built from a pre-\/existing notion of the mechanism of the transition under study\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
Here is the \href{tutorial-resources/belfast-2.tar.gz}{\tt tarball with the files referenced in the following}.\hypertarget{belfast-2_belfast-2-problem}{}\subsection{What happens when in a complex reaction?}\label{belfast-2_belfast-2-problem}
When you deal with a complex conformational transition that you want to analyze (or bias), very often you cannot just describe it with a single order parameter.

As an example in Figure \hyperlink{belfast-2_belfast-2-cdk-fig}{belfast-\/2-\/cdk-\/fig} I consider a large conformational transition like those involved in activating the kinase via open-\/close transition of the activation loop. In sticks you see the part involved in the large conformational change, the rest is either keeping the structure and just moving a bit or is a badly resolved region in the X-\/ray structure. This is a complex transition and it is hard to tell which is the order parameter that best describes the transition.

\label{belfast-2_belfast-2-cdk-fig}%
\hypertarget{belfast-2_belfast-2-cdk-fig}{}%


One could identify a distance that can distinguish open from close but
\begin{DoxyItemize}
\item the plasicity of the loop is such that the same distance can correspond to an almost closed loop and almost open loop. One would like to completely divide these two situations with something which is discriminating what intuitively one would think as open and closed
\item the transition state is an important point where one would like to see a certain crucial interaction forming/breaking so to better explain what is really happening. If you capture then hypothetically you would be able to say what is dictating the rate of this conformational transition. A generic distance is a very hybrid measure that is unlikely to capture a salt-\/bridge formation and a concerted change of many dihedral change or desolvation contribution which are happening while the transition is happening. All these things are potentially important in such transition but none of them is explaining the whole thing.
\end{DoxyItemize}

So basically in these cases you have to deal with an intrinsic multidimensional collective variable where you would need many dimensions. How would you visualize a 10 dimensional C\+V where you use many distances, coordinations and dihedrals (ouch, they're periodic too!) ?

Another typical case is the docking of a small molecule in a protein cleft or gorge, which is the mechanism of drug action. This involves specific conformational transition from both the small molecule and the protein as the small molecule approaches the protein cavity. This also might imply a specific desolvation pattern.

Other typical examples are chemical reactions. Nucleophiloic attacks typically happen with a role from the solvent (see some nice paper from Nobel-\/prize winner Arieh Warshel) and sizeable geometric distortions of the neighboring groups.\hypertarget{belfast-2_belfast-2-pcvs-general}{}\subsection{Path collective variables}\label{belfast-2_belfast-2-pcvs-general}
One possibility to describe many different thing that happen in a single reaction is to use a dimensional reduction technique and in plumed the simplest example that may show its usefulness can be considered that of the path collective variables.

In a nutshell, your reaction might be very complex and happening in many degree of freedom but intuitively is a sort of track along which the reaction proceeds. So what we need is a coordinate that, given a conformation, just tells which point along the \char`\"{}reactive track\char`\"{} is closest.

\label{belfast-2_belfast-2-ab-fig}%
\hypertarget{belfast-2_belfast-2-ab-fig}{}%


For example, in Fig. \hyperlink{belfast-2_belfast-2-ab-fig}{belfast-\/2-\/ab-\/fig}, you see a typical chemical reaction (hydrolisys of methylphosphate) with the two end-\/points denoted by A and B. If you are given a third point, just by looking at it, you might find that this is more resemblant to the reactant than the product, so, hypothetically, if you would intuitively give a parameter that would be 1 for a configuration in the A state and 2 for a configuration in the B state, you probably would give it something like 1.\+3, right?

Path collective variables are the extension to this concept in the case you have many conformation that describe your path, and therefore, instead of an index that goes from 1 to 2 you have an index that goes from 1 to $N$ , where $N$ is the number of conformation that you use in input to describe your path.

From a mathematical point of view, that's rather simple. The progress along the path is calculated with the following equation\+:

\label{belfast-2_belfast-2-s-eq}%
\hypertarget{belfast-2_belfast-2-s-eq}{}%
 \[ S(X)=\frac{\sum_{i=1}^{N} i\ \exp^{-\lambda \vert X-X_i \vert }}{ \sum_{i=1}^{N} \exp^{-\lambda \vert X-X_i \vert } } \]

where in \hyperlink{belfast-2_belfast-2-s-eq}{belfast-\/2-\/s-\/eq} the $ \vert X-X_i \vert $ represents a distance between one configuration $ X $ which is analyzed and another from the set that compose the path $ X_i $. The parameter $ \lambda $ is a positive value that is tuned in a way explaned later. here are a number of things to note to make you think that this is exactely what you want.
\begin{DoxyItemize}
\item The negative exponential function is something that is 1 whenever the value at the exponent is zero, and is progressively smaller when the value is larger than zero (trivially, the case with the value at the exponent larger than zero never occurs since lambda is a positive quantity and the distance is by definition positive).
\item Whenever you sit exactly on a specific images $ X_j $ then all the other terms in the sum disappear (if $ \lambda $ is large enough) and only the value $ j $ survives returning exactely $ S(X)=j $.
\end{DoxyItemize}

In order to provide a value which is continuous, the parameter $ \lambda $ should be correclty tuned. As a rule of thumb I use the following formula

\[ \lambda=\frac{2.3 (N-1) }{\sum_{i=1}^{N-1} \vert X_i-X_{i+1} \vert } \]

which imply that one should calculate the average distance between consecutive frames composing the path. Note also that this distance should be more or less similar between the frames. Generally I tolerate fluctuation of the order of 10/15 percent tops. If you have larger, then it is better to have a smaller value of $ \lambda $.

It is important to note that in principle one could even have a specific $ \lambda $ value associated to each frame of the path but this would provide some distortion in the diffucion coefficient which could potentially harm a straightforward interpretation of the free energy landscape.

So, at this point is better to understand what is meant with \char`\"{}distance\char`\"{} since a distance between two conformations can be calculated in very many ways. The way we refer here is by using mean square deviation after optimal alignment. This means that at each step in which the analisys is performed, a number N of optimal alignments is performed. Namely what is calculated is $ \vert X-X_i \vert = d(X,X_i)^2 $ where $ d(X,X_i) $ is the R\+M\+S\+D as defined in what you see here \hyperlink{RMSD}{R\+M\+S\+D}.

Using the M\+S\+D instead of R\+M\+S\+D is sometimes more convenient and more stable (you do not have a denominator that gies to zero in the derivatives when biasing.

Anyway this is a matter of choice. Potentially one could equally employ other metrics like a set of coordinations (this was done in the past), and then you would avoid the problem of rototranslations (well, which is not a problem since you have it already in plumed) but for some applications that might become appealing. So in path collective variables (and in all the dimensional reduction based collective variables) the problem is converted from the side of choosing the collective variable in choosing the right way to calculate distances, also called \char`\"{}metrics\char`\"{}.

The discussion of this issue is well beyond the topic of this tutorial, so we can move forward in how to tell plumed to calculate the progress along the path whenever the M\+S\+D after optimal alignment is used as distance.

\begin{DoxyVerb}p1: PATHMSD REFERENCE=all.pdb LAMBDA=50.0
PRINT ARG=p1.sss,p1.zzz STRIDE=100 FILE=colvar FMT=%8.4f
\end{DoxyVerb}


Note that reference contains a set of P\+D\+B, appended one after the other, with a E\+N\+D field. Note that there is no need to place all the atoms of the system in the P\+D\+B reference file you provide. Just put the atoms that you think might be needed. You can leave out big domains, solvent and ions if you think that is not important for your use.

Additionally, note that the measure units of L\+A\+M\+B\+D\+A are in the units of the code. In gromacs they are in nm$^\wedge$2 while N\+A\+M\+D is Ang$^\wedge$2. \hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D} produces two arguments that can be printed or used in other Action\+With\+Arguments. One is the progress along the path of \hyperlink{belfast-2_belfast-2-s-eq}{belfast-\/2-\/s-\/eq}, the other is the distance from the closest point along the path, which is denoted with the zzz component. This is defined as

\label{belfast-2_belfast-2-s-eq}%
\hypertarget{belfast-2_belfast-2-s-eq}{}%
 \[ Z(X)=-\frac{1}{\lambda}\log (\sum_{i=1}^{N} \ \exp^{-\lambda \vert X-X_i \vert }) \]

It is easy to understand that in case of perfect match of $ X=X_i $ this equation gives back the value of $ \vert X-X_i \vert $ provided that the lambda is adjusted correctly.

So, the two variables, put together can be visualized as \label{belfast-2_belfast-2-ab-sz-fig}%
\hypertarget{belfast-2_belfast-2-ab-sz-fig}{}%
 This variable is important because whenever your simulation is running close to the path (low Z values), then you know that you are reproducing reliably the path you provided in input but if by chance you find some other path that goes, say, from $ S=1, Z=0 $ to $ S=N, Z=0 $ via large Z values, then it might well be that you have just discovered a good alternative pathway. If your path indeed is going from $ S=1, Z=large $ to $ S=N, Z=large $ then it might well be that you do not have your reaction accomplished, since your reaction, by definition should go from the reactant which is located at $ S=1, Z=0 $ to the product, which is located at $ S=1, Z=N $ so you should pay attention. This case is exemplified in Fig. \hyperlink{belfast-2_belfast-2-ab-sz-nowhere-fig}{belfast-\/2-\/ab-\/sz-\/nowhere-\/fig}

\label{belfast-2_belfast-2-ab-sz-nowhere-fig}%
\hypertarget{belfast-2_belfast-2-ab-sz-nowhere-fig}{}%
 \hypertarget{belfast-2_pcvs-topo}{}\subsection{A note on the path topology}\label{belfast-2_pcvs-topo}
A truly important point is that if you get a trajectory from some form of accelerated dynamics (e.\+g. simply by heating) this cannot simply be converted into a path. Since it is likely that your trajectory is going stochastically back and forth (not in the case of S\+M\+D or U\+S, discussed later), your trajectory might be not topologically suitable. To understand that, suppose you simply collect a reactive trajectory of 100 ps into the reference path you give to the \hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D} and simply you assign the frame of 1 ps to index 1 (first frame occurring in the reference file provided to \hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D}), the frame of 2 ps to index 2 and so on \+: it might be that you have two points which are really similar but one is associated to step, say 5 and the other is associated with frame 12. When you analyse the same trajectory, when you are sitting on any of those points then the calculation of S will be an average like $ S(X)=(5+12)/2=8.5 $ which is an average of the two indexes and is completely misleading sinec it let you think that you are moving between point 8 and 9, which is not the case. So this evidences that your reference frames should be \char`\"{}topologically consecutive\char`\"{}. This means that frame 1 should be the closest to frame 2 and all the other frames should be farther apart. Similarly frame 2 should be equally close (in an \hyperlink{RMSD}{R\+M\+S\+D} sense) to 1 and 3 while all the others should be farther apart. Same for frame 3\+: this should be closest to frame 2 and 4 and farther apart from all the others and so on. This is equivalent to calculate an \char`\"{}\+R\+M\+S\+D matrix\char`\"{} which can be conveniently done in vmd (this is a good exercise for a number of reasons) with R\+M\+S\+D Trajectory tools, by choosing different reference system along the set of reference frames.

\label{belfast-2_belfast-2-good-matrix-fig}%
\hypertarget{belfast-2_belfast-2-good-matrix-fig}{}%
 This is shown in Fig. \hyperlink{belfast-2_belfast-2-good-matrix-fig}{belfast-\/2-\/good-\/matrix-\/fig} where the matrix has a typical gullwing shape.

On the contrary, whenever you extract the frames from a pdb that you produced via free M\+D or some biased methods (S\+M\+D or Targeted M\+D for example) then your frame-\/to-\/frame distance is rather inhomogeneous and looks something like

\label{belfast-2_belfast-2-bad-matrix-fig}%
\hypertarget{belfast-2_belfast-2-bad-matrix-fig}{}%
 Aside from the general shape, which is important to keep the conformation-\/to-\/index relation (this, as we will see in the next part is crucial in the multidimensional scaling) the crucial thing is the distance between neighbors.

\label{belfast-2_belfast-2-good-vs-bad-fig}%
\hypertarget{belfast-2_belfast-2-good-vs-bad-fig}{}%
 As a matter of fact, this is not much important in the analysis but is truly crucial in the bias. When biasing a simulation, you generally want to introduce a force that push your system somewhere. In particular, when you add a bias which is based on a path collective variable, most likely you want that your system goes back and forth along your path. The bias is generally applied by an additional term in the hamiltonian, this can be a spring term for Umbrella Sampling, a Gaussian function for Metadynamics or whatever term which is a function of the collective variable $ s $. Therefore the Hamiltonian $ H (X) $ where $ X $ is the point of in the configurational phase space where your system is takes the following form \[ H'(X)=H(X)+U(S(X)) \] where $ U(S(X)) $ is the force term which depends on the collective variable that ultimately is a function of the $ X $. Now, when you use biased dynamics you need to evolve according the forces that this term produces (this only holds for M\+D, while not in M\+C) and therefore you need \[ F_i=-\frac{d H'(X) }{d x_i} = -\frac{d H'(X) }{d x_i} -\frac{\partial U(S(X)) }{ \partial S}\frac{\partial S(X)}{\partial x_i} \]

This underlines the fact that, whenever $ \frac{\partial S(X)}{\partial x_i} $ is zero, then you have no force on the system. Now the derivative of the progress along the path is \[ \frac{\partial S(X) }{\partial x_i} =\frac{\sum_{i=1}^{N} -\lambda\ i\ \frac{\partial \vert X-X_i \vert}{ \partial x_i} \exp^{-\lambda \vert X-X_i \vert }}{ \sum_{i=1}^{N} \exp^{-\lambda \vert X-X_i \vert } } - \frac{ (\sum_{i=1}^{N} i\ \exp^{-\lambda \vert X-X_i \vert } ) (\sum_{j=1}^{N} -\lambda \frac{\partial \vert X-X_j \vert}{ \partial x_i} \exp^{-\lambda \vert X-X_j \vert } ) }{ ( \sum_{i=1}^{N} \exp^{-\lambda \vert X-X_i \vert } )^2} = \frac{\sum_{i=1}^{N} -\lambda\ i\ \frac{\partial \vert X-X_i \vert}{ \partial x_i} \exp^{-\lambda \vert X-X_i \vert }}{ \sum_{i=1}^{N} \exp^{-\lambda \vert X-X_i \vert } } -s(X) \frac{ (\sum_{j=1}^{N} -\lambda \frac{\partial \vert X-X_j \vert}{ \partial x_i} \exp^{-\lambda \vert X-X_j \vert } ) } {\sum_{i=1}^{N} \exp^{-\lambda \vert X-X_i \vert } } \] which can be rewritten as \label{belfast-2_belfast-2-sder-eq}%
\hypertarget{belfast-2_belfast-2-sder-eq}{}%
 \[ \frac{\partial S(X) }{\partial x_i} = \lambda \frac{\sum_{i=1}^{N} \frac{\partial \vert X-X_i \vert}{ \partial x_i} \exp^{-\lambda \vert X-X_i \vert } [ s(X) - i ] } { \sum_{i=1}^{N} \exp^{-\lambda \vert X-X_i \vert } } \]

It is interesting to note that whenever the $ \lambda $ is too small the force will vanish. Additionally, when $ \lambda $ is too large, then it one single exponential term will dominate over the other on a large part of phase space while the other will vanish. This means that the $ S(X) $ will assume almost discrete values that produce zero force. Funny, isn't it?\hypertarget{belfast-2_belfast-2-howmany}{}\subsection{How many frames do I need?}\label{belfast-2_belfast-2-howmany}
A very common question that comes is the following\+: \char`\"{}\+I have my reaction or a model of it. how many frames do I need to properly define a path collective variable?\char`\"{} This is a very important point that requires a bit of thinking. It all depends on the limiting scale in your reaction. For example, if in your process you have a torsion, as the smallest event that you want to capture with path collective variable, then it is important that you mimick that torsion in the path and that this does not contain simply the initial and final point but also some intermediate. Similarly, if you have a concerted bond breaking, it might be that all takes place in the range of an Angstrom or so. In this case you should have intermediate frames that cover the sub-\/\+Angstrom scale. If you have both in the same path, then the smallest dominates and you have to mimick also the torsion with sub-\/\+Angstrom accuracy.\hypertarget{belfast-2_belfast-2-pcvs-tricks}{}\subsection{Some tricks of the trade\+: the neighbors list.}\label{belfast-2_belfast-2-pcvs-tricks}
If it happens that you have a very complex and detailed path to use, say that it contains 100 frames with 200 atoms each, then the calculation of a 100 alignment is required every time you need the C\+V. This can be quite expensive but you can use a trick. If your trajectory is continuous and you are sure that your trajectory does not show jumps where your system suddedly move from the reactant to the product, then you can use a so-\/called neighbor list. The plumed input shown before then becomes

\begin{DoxyVerb}p1: PATHMSD REFERENCE=all.pdb LAMBDA=50.0 NEIGH_STRIDE=100 NEIGH_SIZE=10 
PRINT ARG=p1.sss,p1.zzz STRIDE=100 FILE=colvar FMT=%8.4f
\end{DoxyVerb}


and in this case only the closest 10 frames from the path will be used for the C\+V. Then the list of the frames to use is updated every 100 steps. If you are using a biased dynamics this may introduce sudden change in the derivatives, therefore it is better to check the stability of the setup before running production-\/quality calculations.\hypertarget{belfast-2_belfast-2-ala}{}\subsection{The molecule of the day\+: alanine dipeptide}\label{belfast-2_belfast-2-ala}
Here and probably in other parts of the tutorial a simple molecule is used as a test case. This is alanine dipeptide in vacuum. This rather simple molecule is useful to make many benchmark that are around for data analysis and free energy methods. It is a rather nice example since it presents two states separated by a high (free) energy barrier.

In Fig. \hyperlink{belfast-2_belfast-2-ala-fig}{belfast-\/2-\/ala-\/fig} its structure is shown.

\label{belfast-2_belfast-2-ala-fig}%
\hypertarget{belfast-2_belfast-2-ala-fig}{}%
 The two main metastable states are called $ C_7eq $ and $ C_7ax $.

\label{belfast-2_belfast-2-transition-fig}%
\hypertarget{belfast-2_belfast-2-transition-fig}{}%
 Here metastable states are intended as states which have a relatively low free energy compared to adjacent conformation. At this stage it is not really useful to know what is the free energy, just think in term of internal energy. This is almost the same for such a small system whith so few degrees of freedom.

It is conventional use to show the two states in terms of Ramachandran dihedral angles, which are denoted with $ \Phi $ and $ \Psi $ in Fig. \hyperlink{belfast-2_belfast-2-transition-fig}{belfast-\/2-\/transition-\/fig} .

\label{belfast-2_belfast-2-rama-fig}%
\hypertarget{belfast-2_belfast-2-rama-fig}{}%
\hypertarget{belfast-2_belfast-2-examples}{}\subsection{Examples}\label{belfast-2_belfast-2-examples}
Now as a simple example, I want to show you that plotting some free dynamics trajectories shoot from the saddle point, you get a different plot in the path collective variables if you use the right path or if you use the wrong path.

In Fig. \hyperlink{belfast-2_belfast-2-good-bad-path-fig}{belfast-\/2-\/good-\/bad-\/path-\/fig} I show you two example of possible path that join the $ C_7eq $ and $ C_7ax $ metastable states in alanine dipeptide. You might clearly expect that real (rare) trajectories that move from one basin to the other would rather move along the black line than on the red line.

\label{belfast-2_belfast-2-good-bad-path-fig}%
\hypertarget{belfast-2_belfast-2-good-bad-path-fig}{}%
 So, in this example we do a sort of \char`\"{}commmittor analysis\char`\"{} where we start shooting a number of free molecular dynamics from the saddle point located at $ \Phi=0 $ and $ \Psi=-1 $ and we want to see which way do they go. Intuitively, by assigning random velocities every time we should find part of the trajectories that move woward $ C_7eq $ and part that move towards $ C_7ax $.

I provided you with two directories, each containing a bash script script.\+sh whose core (it is a bit more complicated in practice...) consists in\+:

\begin{DoxyVerb}#
# set how many runs you want to do
#
ntests=50
for i in `seq 1 $ntests`
do
        #
        # assign a random velocity at each timestep by initializing the
        #
        sed s/SEED/$RANDOM/ md.mdp >newmd.mdp
        #
        # do the topology: this should write a topol.tpr
        #
        $GROMPP -c start.gro -p topol.top -f newmd.mdp
        $GROMACS_BIN/$MDRUN -plumed plumed.dat
        mv colvar colvar_$i
done
\end{DoxyVerb}


This runs 50 short M\+D runs (few hundreds steps) and just saves the colvar file into a labeled colvar file. In each mdrun plumed is used to plot the collective variables and it is something that reads like the follwing\+: \begin{DoxyVerb}# Phi
t1: TORSION ATOMS=5,7,9,15
# Psi
t2: TORSION ATOMS=7,9,15,17
# The right path
p1: PATHMSD REFERENCE=right_path.dat LAMBDA=15100.
# The wrong path
p2: PATHMSD REFERENCE=wrong_path.dat  LAMBDA=8244.4
# Just a printout of all the stuff calculated so far
PRINT ARG=* STRIDE=2 FILE=colvar FMT=%12.8f
\end{DoxyVerb}


where I just want to plot $ \Phi $ , $ \Psi $ and the two path collective variables. Note that each path has a different L\+A\+M\+B\+D\+A parameters. Here the Ramachandran angles are plotted so you can realize which path is the system walking in a more confortable projection. This is of course fine in such a small system but whenever you have to deal with larger systems and control hundreds of C\+Vs at the same time, I think that path collective variables produce a more intuituve description for what you want to do.

If you run the script simply with

\begin{DoxyVerb}pd@plumed:~> ./script.sh
\end{DoxyVerb}


then after a minute or so, you should have a directory which is full of colvar files. Let's revise together how the colvar file is formatted\+:

\begin{DoxyVerb}#! FIELDS time t1 t2 p1.sss p1.zzz p2.sss p2.zzz
#! SET min_t1 -pi
#! SET max_t1 pi
#! SET min_t2 -pi
#! SET max_t2 pi
 0.000000  -0.17752998  -1.01329788  13.87216908   0.00005492  12.00532256   0.00233905
 0.004000  -0.13370142  -1.10611136  13.87613508   0.00004823  12.03390658   0.00255806
 0.008000  -0.15633049  -1.14298481  13.88290617   0.00004511  12.07203319   0.00273764
 0.012000  -0.23856451  -1.12343958  13.89969608   0.00004267  12.12872544   0.00284883
...
\end{DoxyVerb}


In first column you have the time, then t1 ( $ \Phi $) , then t2 ( $ \Psi $ ) and the path collective variables p1 and p2. Note that the action P\+A\+T\+H\+M\+S\+D calculates both the progress along the path (p1.\+sss) and the distance from it (p1.\+zzz) . In P\+L\+U\+M\+E\+D jargon, these are called \char`\"{}components\char`\"{}. So a single Action (a line in plumed input) can calculate many components at the same time. This is not always the case\+: sometimes by default you have one component but specific flags may enable more components to be calculated (see \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} for example). Note that the header (all the part of a colvar file that contains \# as first character) is telling already what it inside the file and eventually also tells you if a variable is contained in boundaries (for example torsions, are periodic and their codomain is defined in -\/\+Pi and Pi ).

At the end of the script, you also have two additional scripts. One is named script\+\_\+rama.\+gplt and the other is named script\+\_\+path.\+gplt. They contain some gnuplot commands that are very handy to visualize all the colvar files without making you load one by one, that would be a pain.

Now, let's visualize the result from the wrong path directory. In order to do so, after having run the calculation, then do

\begin{DoxyVerb}pd@plumed:~>gnuplot 
gnuplot> load "script_rama.gplt"
\end{DoxyVerb}


what you see is that all the trajectories join the reactant and the product state along the low free energy path depicted before.

Now if you try to load the same bunch of trajectories as a function of the progress along the path and the distance from the path in the case of the wrong path then simply do

\begin{DoxyVerb}gnuplot> load "script_path_wrong.gplt"
\end{DoxyVerb}


What do you see? A number of trajectories move from the middle towards the right bottom side at low distance from the path. In the middle of the progress along the path, you have higher distance. This is expected since the distance zero corresponds to a unlikely, highly-\/energetic path which is unlikely to occur. Differently, if you now do

\begin{DoxyVerb}gnuplot> load "script_path_right.gplt"
\end{DoxyVerb}


You see that the path, compared to what happened before, run much closer to small distance from the path. This means that the provided path is highlt resemblant and representative of what hapens in the reactive path.

Note that going at high distances can be also beneficial. It might help you to explore alternative paths that you have not explored before. But beware, the more you get far from the path, the more states are accessible, in a similar way as the fact that the surface of a sphere increase with its radius. The surface ramps up much faster than the radius therefore you have a lots of states there. This means also high entropy, so many systems actually tend to drift to high distances while, on the contrary, zero distance is never reached in practice (zero entropy system is impossible to reach at finite temperature). So you can see by yourself that this can be a big can of worms. In particular, my experience with path collective variables and biological systems tells me that most of time is hopeless to go to high distances to find new path in many cases (for example, in folding). While this is worth whenever you think that the paths are not too many (alternative routes in chemical reaction or enzymatic catalysis).\hypertarget{belfast-2_belfast-2-pcvs-format}{}\subsection{How to format my input?}\label{belfast-2_belfast-2-pcvs-format}
Very often it is asked how to format a set of pdb to be suitably used with path collective variables. Here are some tricks.
\begin{DoxyItemize}
\item When you dump the files with vmd or (for gromacs users, using trjcat), the pdb you obtain is reindexed from 1. This is also the case when you select a subensemble of atoms of the path (e.\+g. the heavy atoms only or the backbone atoms). This is rather unfortunate and you have to fix is someway. My preference is to dump the whole pdb but water (when I do not need it) and use some awk script to select the atoms I am interested in.
\item Pay attention to the last two column. These are occupancy and beta. With the first (occupancy) you set the atoms which are used to perform the alignment. The atoms which have zero occupancy will not be used in the alignment. The second column is beta and controls which atoms are used for the calculation of the distance after having performed the alignment on the set of atoms which have nonzero occupancy column. In this way you can align all your system by using a part of the system and calculate the distance respect to another set. This is handy in case of protein-\/ligand. You set the alignment of the protein and you calculate the distance based on the ligand and the part of the protein which is in contact with the protein. This is done for example in \href{http://pubs.acs.org/doi/abs/10.1021/jp911689r}{\tt this article}.
\item \href{http://www.multiscalelab.org/utilities/PlumedGUI}{\tt Plumed-\/\+G\+U\+I} (version $>$ 2.\+0) provides the {\itshape Structure-\/$>$Build reference structure...} function to generate inputs that conform to the above rules from within V\+M\+D.
\item Note that all the atoms contained in the R\+E\+F\+E\+R\+E\+N\+C\+E must be the same. You cannot have a variable number of atoms in each pdb contained in the reference.
\item The reference is composed as a set of concatenated P\+D\+Bs that are interrupted by a T\+E\+R/\+E\+N\+D/\+E\+N\+D\+M\+D\+L card. Both H\+E\+T\+A\+T\+M and A\+T\+O\+M cards denote the atoms of the set.
\item Include in the reference frames only the needed atoms. For example, if you have a methyl group involved in a conformational transition, it might be that you do not want to include the hydrogen atoms of the methyl since these rotate fast and probably they do not play ant relevant role.
\end{DoxyItemize}\hypertarget{belfast-2_belfast-2-pcvs-metad-on-path}{}\subsection{Fast forward\+: metadynamics on the path}\label{belfast-2_belfast-2-pcvs-metad-on-path}
This section is actually set a bit foward but I included here for completeness now. It is recommended to be read after you have an introduction on Metadynamics and to well-\/tempered Metadynamics in particular. Here I want to show a couple of concept together.
\begin{DoxyItemize}
\item Path collective variables can be used for exploring alternative routes. It is effective in highly structure molecules, while it is tricky on complex molecules whenever you have many competing routes
\item Path collective variables suffer from problems at the endpoints (as the higly popular coordinates \hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N} for example) that can be cured with flexible hills and an appropriate reweighting procedure within the well-\/tempered Metadynamics scheme.
\end{DoxyItemize}

Let's go to the last problem. All comes from the derivative \hyperlink{belfast-2_belfast-2-sder-eq}{belfast-\/2-\/sder-\/eq}. Whenever you have a point of phase space which is similar to one of the endpoint than one of the points in the center then you get a $ s(X) $ which is 1 or N (where N is the number of frames composing the path collective variable). In this case that exponential will dominate the others and you are left with a constant (since the derivative of R\+M\+S\+D is a constant since it is linear in space). This means that, no matter what happens here, you have small force. Additionally you have small motion in the C\+V space. You can move a lot in configuration space but if the closest point is one of the endpoint, your C\+V value will always be one of the endpoint itself. So, if you use a fixed width of your C\+V which you retrieve from a middle point in your path, this is not suitable at all at the endpoints where your C\+V flucutates much less. On the contrary if you pick the hills width by making a free dynamics on the end states you might pick some sigmas that are smaller than what you might use in the middle of the path. This might give a rough free energy profile and definitely more time to converge. A possible solution is to use the adaptive gaussian width scheme. In this scheme you adapt the hills to their fluctuation in time. You find more details in \cite{Branduardi:2012dl} Additionally you also adopt a non spherical shape taking into account variable correlation. So in this scheme you do not have to fix one sigma per variable sigma, but just the time in which you calculate this correlation (another possibility is to calculate it from the compression of phase space but will not be covered here). The input of metadynamics might become something like this

\begin{DoxyVerb}t1: TORSION ATOMS=5,7,9,15
t2: TORSION ATOMS=7,9,15,17
p1: PATHMSD REFERENCE=right_path.dat LAMBDA=15100.
p2: PATHMSD REFERENCE=wrong_path.dat  LAMBDA=8244.4
#
# do a metadynamics on the right path, use adaptive hills whose decay time is 125 steps (250 fs)
# and rather standard WT parameters
#
meta: METAD ARG=p1.sss,p1.zzz  ADAPTIVE=DIFF SIGMA=125 HEIGHT=2.4 TEMP=300 BIASFACTOR=12 PACE=125
PRINT ARG=* STRIDE=10 FILE=colvar FMT=%12.8f
\end{DoxyVerb}


You can find this example in the directory B\+I\+A\+S\+E\+D\+\_\+\+D\+Y\+N\+A\+M\+I\+C\+S. After you run for a while it is interesting to have a feeling for the change in shape of the hills. That you can do with

\begin{DoxyVerb}pd@plumed:~> gnuplot 
gnuplot>  p "<head -400 HILLS" u 2:3:4:5 w xyer 
\end{DoxyVerb}


that plots the hills in the progress along the path and the distance from the path and add an error bar which reflects the diagonal width of the flexible hills for the first 400 hills (Hey note the funny trick in gnuplot in which you can manipulate the data like in a bash script directly in gnuplot. That's very handy!).

\label{belfast-2_belfast-2-metadpath-fig}%
\hypertarget{belfast-2_belfast-2-metadpath-fig}{}%


There are a number of things to observe\+: first that the path explores high distance since the metadynamics is working also in the distance from the path thus accessing the paths that were not explored before, namely the one that goes from the upper left corner of the ramachandran plot and the one that passes through the lower left corner. So in this way one can also explore other paths. Additionally you can see that the hills are changing size rather considerably. This helps the system to travel faster since at each time you use something that has a nonzero gradient and your forces act on your system in an effective way. Another point is that you can see broad hills covering places which you have not visited in practice. For example you see that hills extend so wide to cover point that have negative progress along the path, which is impossible by using the present definition of the progress along the path. This introduced a problem in calculating the free energy. You actually have to correct for the point that you visited in reality.

You can actually use \hyperlink{sum_hills}{sum\+\_\+hills} to this purpose in a two-\/step procedure. First you calculate the negative bias on a given range\+:

\begin{DoxyVerb}pd@plumed:~> plumed sum_hills --hills HILLS --negbias  --outfile negative_bias.dat --bin 100,100 --min -5,-0.005 --max 25,0.05
\end{DoxyVerb}


and then calculate the correction. You can use the same hills file for that purpose. The initial transient time should not matter if your simulation is long enough to see many recrossing and, secondly, you should check that the hills heigh in the welltempered are small compared to the beginning.

\begin{DoxyVerb}pd@plumed:~> plumed sum_hills --histo HILLS --bin 100,100 --min -5,-0.005 --max 25,0.05 --kt 2.5 --sigma 0.5,0.001 --outhisto correction.dat
\end{DoxyVerb}


Note that in the correction you should assign a sigma, that is a \char`\"{}trust radius\char`\"{} in which you think that whenever you have a point somewhere, there in the neighborhood you assign a bin (it is done with Gaussian in reality, but this does not matter much). This is the important point that compenstates for the issues you might encounter putting excessive large hills in places that you have not visited. It is nice to have a look to the correction and compare with the hills in the same range.

\begin{DoxyVerb}gnuplot> set pm3d
gnuplot> spl "correction.dat" u 1:2:3 w l
gnuplot> set contour
gnuplot> set cntrp lev incremental -20,4.,1000.
gnuplot> set view map
gnuplot> unset clabel 
gnuplot> replot 
\end{DoxyVerb}


You might notice that there are no contour in the unrealistic range, this means that the free energy correction is impossible to calculate since it is too high (see Fig. \hyperlink{belfast-2_belfast-2-metadpath-correction-fig}{belfast-\/2-\/metadpath-\/correction-\/fig} ).

\label{belfast-2_belfast-2-metadpath-correction-fig}%
\hypertarget{belfast-2_belfast-2-metadpath-correction-fig}{}%


Now the last thing that one has to do to have the most plausible free energy landscape is to sum both the correction and the negative bias produced. This can be easily done in gnuplot as follows\+:

\begin{DoxyVerb}gnuplot> set pm3d
gnuplot> spl "<paste negative_bias.dat correction.dat " u 1:2:($3+$8) w pm3d
gnuplot> set view map
gnuplot> unset key
gnuplot> set xr [-2:23]
gnuplot> set contour
gnuplot> unset clabel
gnuplot> set cbrange [-140:-30] 
gnuplot> set cntrp lev incr -140,6,-30
\end{DoxyVerb}


\label{belfast-2_belfast-2-metadpath-free-fig}%
\hypertarget{belfast-2_belfast-2-metadpath-free-fig}{}%
 So now we can comment a bit on the free energy surface obtained and note that there is a free energy path that connects the two endpoints and runs very close to zero distance from the path. This means that our input path is actually resemblant of what is really happening in the system. Additionally you can see that there are many comparable routes different from the straight path. Can you make a sense of it just by looking at the free energy on the Ramachandran plot? \hypertarget{belfast-3}{}\section{Belfast tutorial\+: Adaptive variables I\+I}\label{belfast-3}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
The aim of this tutorial is to consolidate the material that was covered during \hyperlink{belfast-1}{Belfast tutorial\+: Analyzing C\+Vs} and \hyperlink{belfast-2}{Belfast tutorial\+: Adaptive variables I} on analysing trajectories using collective variables and path collective variables. We will then build on this material by showing how you can use the multidimensional scaling algorithm to automate the process of finding collective variables.\hypertarget{belfast-3_belfast-3-lo}{}\subsection{Learning Outcomes}\label{belfast-3_belfast-3-lo}
Once this tutorial is completed students will\+:


\begin{DoxyItemize}
\item Know how to load colvar data into the G\+I\+S\+M\+O plugin
\item Know how to run the multidimensional scaling algorithms on a trajectory
\item Be able to explain how we can automate the process of finding collective variables by seeking out an isometry between a high-\/dimensional and low-\/dimensional space
\end{DoxyItemize}\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
The \href{tutorial-resources/belfast-3.tar.gz}{\tt tarball } for this project contains the following files\+:


\begin{DoxyItemize}
\item trajectory-\/short.\+xyz \+: a (short) trajectory for a 16 residue protein in xyz format. All calculations with plumed driver use this trajectory.
\item trajectory-\/short.\+pdb \+: the same trajectory in pdb format, this can be loaded with V\+M\+D
\item template.\+pdb \+: a single frame from the trajectory that can be used in conjuction with the \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command
\end{DoxyItemize}\hypertarget{belfast-10_Instructions}{}\subsection{Instructions}\label{belfast-10_Instructions}
\hypertarget{belfast-3_vis-traj}{}\subsubsection{Visualising the trajectory}\label{belfast-3_vis-traj}
The aim of this tutorial is to understand the data contained in the trajectory called trajectory-\/short.\+pdb. This file contains some frames from a simulation from a 16 residue protein. As a start point then let load this trajectory with vmd and have a look at it. Type the following command into the command line\+:

\begin{DoxyVerb}vmd trajectory-short.pdb
\end{DoxyVerb}


Look at it with the various representations that vmd offers. If you at are at the plumed tutorial try typing the letter m on the keyboard -\/ we have made the new cartoon representation will update automatically for each frame of the trajectory -\/ cool huh! What are your impressions about this trajectory based on looking at it with V\+M\+D? How many basins in the free energy landscape is this trajectory sampling from? What can we tell from looking at this trajectory that we could perhaps put in a paper?

If your answers to the questions at the end of the above paragraph are I don't know that is good. We can tell very little by just looking at a trajectory. In fact the whole point of today has been to find ways of analyzing trajectories precisely so that we are not put in this position of staring at trajetories mystified!\hypertarget{belfast-10_cvs}{}\subsubsection{Finding collective variables}\label{belfast-10_cvs}
Right so lets come up with some C\+Vs to analyse this trajectory with. As some of you may know we can understand the conformation of proteins by looking at the Ramachandran angles. For those of you who don't know here is a Wikkepedia article\+:

\href{http://en.wikipedia.org/wiki/Ramachandran_plot}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Ramachandran\+\_\+plot}

Our protein has 32 ramachandran angles. We'll come back to that. For the time being pick out a couple that you think might be useful and construct a plumed input to calculate them and print them to a file. You will need to use the \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} and \hyperlink{PRINT}{P\+R\+I\+N\+T} commands in order to do this. Once you have created your plumed input use driver to calculate the torsional angles using the following command\+:

\begin{DoxyVerb}plumed driver --plumed plumed.dat --ixyz trajectory.xyz
\end{DoxyVerb}


If you have done this correctly you should have an output file containing your torsional angles. We can use vmd+\+G\+I\+S\+M\+O to visualise the relationship between the ramachandran angles and the atomic configurations. To do this first load the trajectory in V\+M\+D\+:

\begin{DoxyVerb}vmd trajectory-short.pdb
\end{DoxyVerb}


Then click on Extensions$>$Analysis$>$G\+I\+S\+M\+O. A new window should open in this window click on File$>$Load colvars. You will be asked to select a colvar file. Select the file that was output by the plumed calculation above. Once the file is loaded you should be able to select the labels that you gave to the Ramachandran angles you calculated with plumed. If you do so you will see that this data is plotted in the G\+I\+S\+M\+O window so that you can interact with it and the trajectory.

What can you conclude from this exercise. Do the C\+V values of the various frames appear in clusters in the plane? Do points in different clusters correspond to structures that look the same or different? Are there similar looking structures clustered together or are they always far apart? What can we conclude about the various basins in the free energy landscape that have been explored in this trajectory? How many are there? Would your estimate be the same if you tried the above estimate with a different pair of ramachandran angles?\hypertarget{belfast-3_dim-red}{}\subsubsection{Dimensionality reduction}\label{belfast-3_dim-red}
What we have done for most of today is seek out a function that takes as input the position of all the atoms in the system -\/ a $3N$ dimensional vector, where $N$ is the number of atoms. This function then outputs a single number -\/ the value of the collective variable -\/ that tells us where in a low dimensional space we should project that configuration. Problems can arise because this collective-\/variable function is many-\/to-\/one. As you have hopefully seen in the previous exercise markedly different confifgurations of the protein can actually have quite similar values of a particular ramachandran angle.

We are going to spend the rest of this session introducing an alternative approach to this bussiness of finding collective variables. In this alternative approach we are going to stop trying to seek out a function that can take any configuration of the atoms (any $3N$-\/dimensional vector) and find it's low dimensional proejection on the collective variable axis. Instead we are going to take a set of configurations of the atoms (a set of $3N$-\/dimensional vectors of atom positions) and try to find a sensible set of projections for these configurations. We already touched on this idea earlier when we looked at paths. Our assumption, when we introduced this idea, was that we could find an ordered set of high-\/dimensional configurations that represented the transtion pathway the system takes as it crossed a barrier and changed between two particularly interesting configurations. Lets say we have a path defined by four reference configurations -\/ this implies that to travel between the configurations at the start and the end of this path you have to pass through configuration 1, then configuration 2, then configuration 3 and then configuration 4. This ordering means that the numbers 1 through 4 constitute sensible projections of these high-\/dimensional configurations. The numbers 1 through 4 all lie on a single cartesian axis -\/ a low-\/dimensional space.

The problem when it comes to applying this idea to the data that we have in the trajectory-\/short trajectory is that we have no information on the ``order" of these points. We have not been told that this trajectory represents the transition between two interesting points in phase space and thus we cannot apply the logic of paths. Hence, to seek out a low dimensional representation we are going to try and find a representation of this data we are going to seek out \href{http://en.wikipedia.org/wiki/Isometry}{\tt an isometry } between the space containing the $3N$-\/dimensional vectors of atom positions and some lower-\/dimensional space. This idea is explained in more detail in the following \href{https://www.youtube.com/watch?v=ofC2qz0_9_A&feature=youtu.be}{\tt video }.

Let's now generate our isometric embedding. You will need to create a plumed input file that contains the following instructions\+:

\begin{DoxyVerb}CLASSICAL_MDS ...
  ATOMS=1-256
  METRIC=OPTIMAL-FAST
  USE_ALL_DATA
  NLOW_DIM=2
  OUTPUT_FILE=rmsd-embed
... CLASSICAL_MDS
\end{DoxyVerb}


You should then run this calculation using the following command\+:

\begin{DoxyVerb}plumed driver --ixyz trajectory-short.xyz --plumed plumed.dat
\end{DoxyVerb}


This should generate an output file called rmsd-\/embed. You should now be able to use V\+M\+D+\+G\+I\+S\+M\+O to visualise this output. Do the C\+V values of the various frames appear in clusters in the plane? Do points in different clusters correspond to structures that look the same or different? Are there similar looking structures clustered together or are they always far apart? What can we conclude about the various basins in the free energy landscape that have been explored in this trajectory? How many are there? Do you think this gives you a fuller picture of the trajectory than the ones you obtained by considering ramachandran angles?\hypertarget{belfast-3_extensions}{}\subsection{Extensions}\label{belfast-3_extensions}
As discussed in the previous section this approach to trajectory analysis works by calcalating distances between pairs of atomic configurations. Projections corresponding to these configurations are then generated in the low dimensional space in a way that tries to preserve these pairwise distances. There are, however, an infinite number of ways of calculating the distance between two high-\/dimensional configurations. In the previous section we used the R\+M\+S\+D distance but you could equally use the D\+R\+M\+S\+D distance. You could even try calculating a large number of collective variables for each of the high-\/dimensional points and seeing how much these all changed. You can use these different types of distances with the \hyperlink{CLASSICAL_MDS}{C\+L\+A\+S\+S\+I\+C\+A\+L\+\_\+\+M\+D\+S} action that was introduced in the previous section. If you have time less at the end of the session read the manual for the \hyperlink{CLASSICAL_MDS}{C\+L\+A\+S\+S\+I\+C\+A\+L\+\_\+\+M\+D\+S} action and see if you can calculate an M\+D\+S projection using an alternative defintion of the distances between configurations. Some suggestions to try in order of increasing difficulty\+: D\+R\+M\+S\+D, how much the full set of 32 ramachandran angles change and the change in the contact map\hypertarget{belfast-10_further}{}\subsection{Further Reading}\label{belfast-10_further}
There is a growing community of people using these ideas to analyse trajectory data. Some start points for investigating their work in more detail are\+:


\begin{DoxyItemize}
\item \href{http://sketchmap.berlios.de}{\tt http\+://sketchmap.\+berlios.\+de}
\item \href{http://www.annualreviews.org/doi/abs/10.1146/annurev-physchem-040412-110006}{\tt http\+://www.\+annualreviews.\+org/doi/abs/10.\+1146/annurev-\/physchem-\/040412-\/110006} 
\end{DoxyItemize}\hypertarget{belfast-4}{}\section{Belfast tutorial\+: Umbrella sampling}\label{belfast-4}
\hypertarget{belfast-4_belfast-4-aims}{}\subsection{Aims}\label{belfast-4_belfast-4-aims}
In the previous lectures we learned how to compute collective variables (C\+Vs) from atomic positions. We will now learn how one can add a bias potential to enforce the exploration of a particular region of the space. We will also see how it is possible to bias C\+Vs so as to enhance the sampling of events hindered by large free-\/energy barriers and how to analyze this kind of simulation. This technique is known as \char`\"{}umbrella sampling\char`\"{} and can be used in combination with the weighted-\/histogram analysis method to compute free-\/energy landscapes. In this tutorial we will use simple collective variables, but the very same approach can be used with any kind of collective variable.\hypertarget{belfast-4_belfast-4-theory}{}\subsection{Summary of theory}\label{belfast-4_belfast-4-theory}
\hypertarget{belfast-4_belfast-4-theory-biased-sampling}{}\subsubsection{Biased sampling}\label{belfast-4_belfast-4-theory-biased-sampling}
A system at temperature $ T$ samples conformations from the canonical ensemble\+: \[ P(q)\propto e^{-\frac{U(q)}{k_BT}} \]. Here $ q $ are the microscopic coordinates and $ k_B $ is the Boltzmann constant. Since $ q $ is a highly dimensional vector, it is often convenient to analyze it in terms of a few collective variables (see \hyperlink{belfast-1}{Belfast tutorial\+: Analyzing C\+Vs} , \hyperlink{belfast-2}{Belfast tutorial\+: Adaptive variables I} , and \hyperlink{belfast-3}{Belfast tutorial\+: Adaptive variables I\+I} ). The probability distribution for a C\+V $ s$ is \[ P(s)\propto \int dq e^{-\frac{U(q)}{k_BT}} \delta(s-s(q)) \] This probability can be expressed in energy units as a free energy landscape $ F(s) $\+: \[ F(s)=-k_B T \log P(s) \].

Now we would like to modify the potential by adding a term that depends on the C\+V only. That is, instead of using $ U(q) $, we use $ U(q)+V(s(q))$. There are several reasons why one would like to introduce this potential. One is to avoid that the system samples some un-\/desired portion of the conformational space. As an example, imagine that you want to study dissociation of a complex of two molecules. If you perform a very long simulation you will be able to see association and dissociation. However, the typical time required for association will depend on the size of the simulation box. It could be thus convenient to limit the exploration to conformations where the distance between the two molecules is lower than a given threshold. This could be done by adding a bias potential on the distance between the two molecules. Another example is the simulation of a portion of a large molecule taken out from its initial context. The fragment alone could be unstable, and one might want to add additional potentials to keep the fragment in place. This could be done by adding a bias potential on some measure of the distance from the experimental structure (e.\+g. on root-\/mean-\/square deviation).

Whatever C\+V we decide to bias, it is very important to recognize which is the effect of this bias and, if necessary, remove it a posteriori. The biased distribution of the C\+V will be \[ P'(s)\propto \int dq e^{-\frac{U(q)+V(s(q))}{k_BT}} \delta(s-s(q))\propto e^{-\frac{V(s(q))}{k_BT}}P(s) \] and the biased free energy landscape \[ F'(s)=-k_B T \log P'(s)=F(s)+V(s)+C \] Thus, the effect of a bias potential on the free energy is additive. Also notice the presence of an undetermined constant $ C $. This constant is irrelevant for what concerns free-\/energy differences and barriers, but will be important later when we will learn the weighted-\/histogram method. Obviously the last equation can be inverted so as to obtain the original, unbiased free-\/energy landscape from the biased one just subtracting the bias potential \[ F(s)=F'(s)-V(s)+C \]

Additionally, one might be interested in recovering the distribution of an arbitrary observable. E.\+g., one could add a bias on the distance between two molecules and be willing to compute the unbiased distribution of some torsional angle. In this case there is no straightforward relationship that can be used, and one has to go back to the relationship between the microscopic probabilities\+: \[ P(q)\propto P'(q) e^{\frac{V(s(q))}{k_BT}} \] The consequence of this expression is that one can obtained any kind of unbiased information from a biased simulation just by weighting every sampled conformation with a weight \[ w\propto e^{\frac{V(s(q))}{k_BT}} \] That is, frames that have been explored in spite of a high (disfavoring) bias potential $ V $ will be counted more than frames that has been explored with a less disfavoring bias potential.\hypertarget{belfast-4_belfast-4-theory-us}{}\subsubsection{Umbrella sampling}\label{belfast-4_belfast-4-theory-us}
Often in interesting cases the free-\/energy landscape has several local minima. If these minima have free-\/energy differences that are on the order of a few times $k_BT$ they might all be relevant. However, if they are separated by a high saddle point in the free-\/energy landscape (i.\+e. a low probability region) than the transition between one and the other will take a lot of time and these minima will correspond to metastable states. The transition between one minimum and the other could require a time scale which is out of reach for molecular dynamics. In these situations, one could take inspiration from catalysis and try to favor in a controlled manner the conformations corresponding to the transition state.

Imagine that you know since the beginning the shape of the free-\/energy landscape $ F(s) $ as a function of one C\+V $ s $. If you perform a molecular dynamics simulation using a bias potential which is exactly equal to $ -F(s) $, the biased free-\/energy landscape will be flat and barrierless. This potential acts as an \char`\"{}umbrella\char`\"{} that helps you to safely cross the transition state in spite of its high free energy.

It is however difficult to have an a priori guess of the free-\/energy landscape. We will see later how adaptive techniques such as metadynamics (\hyperlink{belfast-6}{Belfast tutorial\+: Metadynamics}) can be used to this aim. Because of this reason, umbrella sampling is often used in a slightly different manner.

Imagine that you do not know the exact height of the free-\/energy barrier but you have an idea of where the barrier is located. You could try to just favor the sampling of the transition state by adding a harmonic restraint on the C\+V, e.\+g. in the form \[ V(s)=\frac{k}{2} (s-s_0)^2 \]. The sampled distribution will be \[ P'(q)\propto P(q) e^{\frac{-k(s(q)-s_0)^2}{2k_BT}} \] For large values of $ k $, only points close to $ s_0 $ will be explored. It is thus clear how one can force the system to explore only a predefined region of the space adding such a restraint. By combining simulations performed with different values of $ s_0 $, one could obtain a continuous set of simulations going from one minimum to the other crossing the transition state. In the next section we will see how to combine the information from these simulations.\hypertarget{belfast-4_belfast-4-theory-wham}{}\subsubsection{Weighted histogram analysis method}\label{belfast-4_belfast-4-theory-wham}
Let's now consider multiple simulations performed with restraints located in different positions. In particular, we will consider the $i$-\/th bias potential as $V_i$. The probability to observe a given value of the collective variable $s$ is\+: \[ P_i({s})=\frac{P({s})e^{-\frac{V_i({s})}{k_BT}}}{\int ds' P({s}') e^{-\frac{V_i({s}')}{k_BT}}}= \frac{P({s})e^{-\frac{V_i({s})}{k_BT}}}{Z_i} \] where \[ Z_i=\sum_{q}e^{-\left(U(q)+V_i(q)\right)} \] The likelyhood for the observation of a sequence of snapshots $q_i(t)$ (where $i$ is the index of the trajectory and $t$ is time) is just the product of the probability of each of the snapshots. We use here the minus-\/logarithm of the likelihood (so that the product is converted to a sum) that can be written as \[ \mathcal{L}=-\sum_i \int dt \log P_i({s}_i(t))= \sum_i \int dt \left( -\log P({s}_i(t)) +\frac{V_i({s}_i(t))}{k_BT} +\log Z_i \right) \] One can then maximize the likelyhood by setting $\frac{\delta \mathcal{L}}{\delta P({\bf s})}=0$. After some boring algebra the following expression can be obtained \[ 0=\sum_{i}\int dt\left(-\frac{\delta_{{\bf s}_{i}(t),{\bf s}}}{P({\bf s})}+\frac{e^{-\frac{V_{i}({\bf s})}{k_{B}T}}}{Z_{i}}\right) \] In this equation we aim at finding $P(s)$. However, also the list of normalization factors $Z_i$ is unknown, and they should be found selfconsistently. Thus one can find the solution as \[ P({\bf s})\propto \frac{N({\bf s})}{\sum_i\int dt\frac{e^{-\frac{V_{i}({\bf s})}{k_{B}T}}}{Z_{i}} } \] where $Z$ is selfconsistently determined as \[ Z_i\propto\int ds' P({\bf s}') e^{-\frac{V_i({\bf s}')}{k_BT}} \]

These are the W\+H\+A\+M equations that are traditionally solved to derive the unbiased probability $P(s)$ by the combination of multiple restrained simulations. To make a slightly more general implementation, one can compute the weights that should be assigned to each snapshot, that turn out to be\+: \[ w_i(t)\propto \frac{1}{\sum_j\int dt\frac{e^{-\beta V_{j}({\bf s}_i(t))}}{Z_{j}} } \] The normalization factors can in turn be found from the weights as \[ Z_i\propto\frac{\sum_j \int dt e^{-\beta V_i({\bf s}_j(t))} w_j(t)}{ \sum_j \int dt w_j(t)} \]

This allows to straighforwardly compute averages related to other, non-\/biased degrees of freedom, and it is thus a bit more flexible. It is sufficient to precompute this factors $w$ and use them to weight every single frame in the trajectory.\hypertarget{belfast-4_belfast-4-learning-outcomes}{}\subsection{Learning Outcomes}\label{belfast-4_belfast-4-learning-outcomes}
Once this tutorial is completed students will know how to\+:


\begin{DoxyItemize}
\item Setup simulations with restraints.
\item Use multiple-\/restraint umbrella sampling simulations to enhance the transition across a free-\/energy barrier.
\item Analyze the results and compute weighted averages and free-\/energy profiles.
\end{DoxyItemize}\hypertarget{belfast-4_belfast-4-resources}{}\subsection{Resources}\label{belfast-4_belfast-4-resources}
The \href{tutorial-resources/belfast-4.tar.gz}{\tt tarball} for this project contains the following files\+:
\begin{DoxyItemize}
\item A gromacs topology (topol.\+top), configuration (conf.\+gro), and control file (grompp.\+mdp). They should not be needed.
\item A gromacs binary file (topol.\+tpr). This is enough for running this system.
\item A small C++ program that computes W\+H\+A\+M (wham.\+cpp) and a script that can be used to feed it (wham.\+sh)
\end{DoxyItemize}

By working in the directory where the topol.\+tpr file is stored, one can launch gromacs with the command \begin{DoxyVerb}mdrun_mpi -plumed plumed.dat -nsteps 100000
\end{DoxyVerb}
 (notice that the -\/nsteps flag allows the number of steps to be changed).\hypertarget{belfast-4_belfast-4-instructions}{}\subsection{Instructions}\label{belfast-4_belfast-4-instructions}
\hypertarget{belfast-4_belfast-4-system}{}\subsubsection{The model system}\label{belfast-4_belfast-4-system}
We here use a a model system alanine dipeptide with C\+H\+A\+R\+M27 all atom force field already seen in the previous section.\hypertarget{belfast-4_belfast-4-restrained-simulations}{}\subsubsection{Restrained simulations}\label{belfast-4_belfast-4-restrained-simulations}
The simplest way in which one might influence a C\+V is by forcing the system to stay close to a chosen value during the simulation. This is achieved with a restraining potential that P\+L\+U\+M\+E\+D provides via the directive \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T}. In the umbrella sampling method a bias potential is added so as to favor the exploration of some regions of the conformational space and to disfavor the exploration of other regions \cite{torrie-valleau} . A properly chosen bias potential could allow for example to favor the transition state sampling thus enhancing the transition state for a conformational transition. However, choosing such a potential is not trivial. In a later section we will see how metadynamics can be used to this aim. The simplest way to use umbrella sampling is that to apply harmonic constraints to one or more C\+Vs.

We will now see how to enforce the exploration of a the neighborhood of a selected point the C\+V space using a \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T} potential.

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Impose an umbrella potential on CV 1 and CV 2
# with a spring constant of 500 kjoule/mol
# at fixed points on the Ramachandran plot
#
restraint-phi: RESTRAINT ARG=phi KAPPA=500 AT=-0.3
restraint-psi: RESTRAINT ARG=psi KAPPA=500 AT=+0.3

# monitor the two variables and the bias potential from the two restraints
PRINT STRIDE=10 ARG=phi,psi,restraint-phi.bias,restraint-psi.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The syntax for the command \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T} is rather trivial. The directive is followed by a keyword A\+R\+G followed by the label of the C\+V on which the umbrella potential has to act. The keyword K\+A\+P\+P\+A determines the hardness of the spring constant and its units are \mbox{[}Energy units\mbox{]}/\mbox{[}Units of the C\+V \mbox{]}. The additional potential introduced by the U\+M\+B\+R\+E\+L\+L\+A takes the form of a simple Hooke’s law\+: \[ U(s)=\frac{k}{2} (x-x_0)^2 \].

where $ x_0 $ is the value specified following the A\+T keyword. The choice of A\+T ( $ x_0 $) is obviously depending on the specific case. K\+A\+P\+P\+A ( $ k $) is typically chosen not to affect too much the intrinsic fluctuations of the system. A typical recipe is $ k \approx \frac{k_BT}{\sigma^2} $, where $ \sigma^2 $ is the variance of the C\+V in a free simulation). In real applications, one must be careful with values of $ k $ larger than $ \frac{k_BT}{\sigma^2} $ because they could break down the molecular dynamics integrator.

The C\+Vs as well as the two bias potentials are shown in the C\+O\+L\+V\+A\+R file. For this specific input the C\+O\+L\+V\+A\+R file has in first column the time, in the second the value of $\phi$, in the third the value of $\psi$, in the fourth the the additional potential introduced by the restraint on $\phi$ and in the fifth the additional potential introduced by the restraint on $\psi$.

It may happen that one wants that a given C\+V just stays within a given range of values. This is achieved in plumed through the directives \hyperlink{UPPER_WALLS}{U\+P\+P\+E\+R\+\_\+\+W\+A\+L\+L\+S} and \hyperlink{LOWER_WALLS}{L\+O\+W\+E\+R\+\_\+\+W\+A\+L\+L\+S} that act on specific collective variables and limit the exploration within given ranges.\hypertarget{belfast-4_belfast-4-reweighting}{}\subsubsection{Reweighting the results}\label{belfast-4_belfast-4-reweighting}
Now consider a simulation performed restraining the variable $\phi $\+: \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
restraint-phi: RESTRAINT ARG=phi KAPPA=10.0 AT=-2
PRINT STRIDE=10 ARG=phi,psi,restraint-phi.bias FILE=COLVAR10\end{DoxyVerb}


and compare the result with the one from a single simulation with no restraint \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
# we use a "dummy" restraint with strength zero here
restraint-phi: RESTRAINT ARG=phi KAPPA=0.0 AT=-2
PRINT STRIDE=10 ARG=phi,psi,restraint-phi.bias FILE=COLVAR0
\end{DoxyVerb}


Plot the time dependence of $\phi $ in the two cases and try to understand the difference.

Now let's try to compute the probability that $\psi $ falls within a given range, say between 1 and 2. This can be done e.\+g. with this shell script \begin{DoxyVerb}> grep -v \# COLVAR0 | tail -n 80000 |
  awk '{if($3>1 && $3<2)a++; else b++;}END{print a/(a+b)}'
\end{DoxyVerb}
 Notice that we here considered only the last 80000 frames in the average. Look at the time series for $\psi $ and guess why. Also notice that the script is removing the initial comments. After this trivial preprocessing, the script is just counting how many times the third column ( $ \psi $) lies between 1 and 2 and how many times it doesn't. At the end it prints the number of times the variable is between 1 and 2 divided by the total count. The result should be something around 0.\+40. Now try to do it on trajectories generated with different values of A\+T. Does the result depend on A\+T?

We can now try to reweight the result so as to get rid of the bias introduced by the restraint. Since the reweighting factor is just $\exp(\frac{V}{k_BT} $ the script should be modified as \begin{DoxyVerb}> grep -v \# COLVAR10 | tail -n 80000 |
awk '{w=exp($4/2.5); if($3>1 && $3<2)a+=w; else b+=w;}END{print a/(a+b)}'
\end{DoxyVerb}
 Notice that 2.\+5 is just $k_BT$ in kj/mol units.

Repeat this calculation for different values of A\+T. Does the result depend on A\+T?\hypertarget{belfast-4_belfast-4-fes}{}\subsubsection{A free-\/energy landscape}\label{belfast-4_belfast-4-fes}
One can also count the probability of an angle to be in a precise bin. The logarithm of this quantity, in kb\+T units, is the free-\/energy associated to that bin. There are several ways to compute histograms, either with P\+L\+U\+M\+E\+D or with external programs. Here I decided to use awk.

\begin{DoxyVerb}grep -v \# COLVAR10 | tail -n 80000 |
awk 'BEGIN{
  min1=-3.14159265358979
  max1=+3.14159265358979
  min2=-3.14159265358979
  max2=+3.14159265358979
  nb1=100;
  nb2=100;
  for(i1=0;i1<nb1;i1++) for(i2=0;i2<nb2;i2++) f[i1,i2]=0.0;
}{
  i1=int(($2-min1)*nb1/(max1-min1));
  i2=int(($3-min2)*nb2/(max2-min2));
# we assume the potential is in the last column, and kbT=2.5 kj/mol
  w=exp($NF/2.5);
  f[i1,i2]+=w;
}
END{
  for(i1=0;i1<nb1;i1++){
  for(i2=0;i2<nb2;i2++) print min1+i1/100.0*(max1-min1), min2+i2/100.0*(max2-min2), -2.5*log(f[i1,i2]);
  print "";
}}' > plotme
\end{DoxyVerb}


You can then plot the \char`\"{}plotme\char`\"{} file with \begin{DoxyVerb}gnuplot> set pm3d map
gnuplot> splot "plotme"
\end{DoxyVerb}
\hypertarget{belfast-4_belfast-4-wham}{}\subsubsection{Combining multiple restraints}\label{belfast-4_belfast-4-wham}
In the last paragraph you have seen how to reweight simulations done with restraints in different positions to obtain virtually the same result. Let's now see how to combine data from multiple restraint simulations. A possible choice is to download and use the W\+H\+A\+M software \href{http://membrane.urmc.rochester.edu/content/wham}{\tt here}, which is well documented. This is probably the best idea for analyzing a real simulation.

For the sake of learning a bit, we will use a different approach here, namely we will use a short C++ program that implements the weight calculation. Notice that whereas people typically use harmonic restraints in this framework, P\+L\+U\+M\+E\+D offers a very large variety of bias potentials. For this reason we will keep things as general as possible and use an approach that can be in principle used also to combine simulation with restraint on different variables or with complicated bias potential.

The first step is to generate several simulations with different positions of the restraint, gradually going from say -\/2 to +2. You can obtain them using e.\+g. the following script\+: \begin{DoxyVerb}for AT in -2.0 -1.5 -1.0 -0.5 +0.0 +0.5 +1.0 +1.5 +2.0
do

cat >plumed.dat << EOF
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Impose an umbrella potential on CV 1 and CV 2
# with a spring constant of 500 kjoule/mol
# at fixed points on the Ramachandran plot
#
restraint-phi: RESTRAINT ARG=phi KAPPA=40.0 AT=$AT
# monitor the two variables and the bias potential from the two restraints
PRINT STRIDE=10 ARG=phi,psi,restraint-phi.bias FILE=COLVAR$AT
EOF

mdrun_mpi -plumed plumed.dat -nsteps 100000 -x traj$AT.xtc

done\end{DoxyVerb}


Notice that we are here saving separate trajectories for the separate simulation, as well as separate colvar files. In each simulation the restraint is located in a different position. Have a look at the plot of (phi,psi) for the different simulations to understand what is happening.

An often misunderstood fact about W\+H\+A\+M is that data of the different trajectories can be mixed and it is not necessary to keep track of which restraint was used to produce every single frame. Let's get the concatenated trajectory

\begin{DoxyVerb}trjcat -cat -f traj*.xtc -o alltraj.xtc
\end{DoxyVerb}


Now we should compute the value of each of the bias potentials on the entire (concatenated) trajectory \begin{DoxyVerb}for AT in -2.0 -1.5 -1.0 -0.5 +0.0 +0.5 +1.0 +1.5 +2.0
do

cat >plumed.dat << EOF
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
restraint-phi: RESTRAINT ARG=phi KAPPA=40.0 AT=$AT

# monitor the two variables and the bias potential from the two restraints
PRINT STRIDE=10 ARG=phi,psi,restraint-phi.bias FILE=ALLCOLVAR$AT
EOF

plumed driver --mf_xtc alltraj.xtc --trajectory-stride=10 --plumed plumed.dat

done
\end{DoxyVerb}


It is very important that this script is consistent with the one used to generate the multiple simulations above. Now, single files named A\+L\+L\+C\+O\+L\+V\+A\+R\+X\+X will contain on the fourth column the value of the bias centered in X\+X computed on the entire concatenated trajectory.

Next step is to compile the C++ program that computes weights self-\/consistently solving the W\+H\+A\+M equations. This is named wham.\+cpp and can be compiled with \begin{DoxyVerb}g++ -O3 wham.cpp -o wham.x
\end{DoxyVerb}
 and can be then used through a wrapper script wham.\+sh as \begin{DoxyVerb}./wham.sh ALLCOLVAR* > colvar
\end{DoxyVerb}
 The resulting colvar file will contain 3 columns\+: time, phi, and psi, plus the weights obtained from W\+H\+A\+M written in logarithmic scale. That is, the file will contain $k_BT \log w $.

Try now to use this file to compute the unbiased free-\/energy landscape as a function of phi and psi. You can use the script that you used earlier to compute histogram.\hypertarget{belfast-4_belfast-4-comments}{}\subsection{Comments}\label{belfast-4_belfast-4-comments}
\hypertarget{belfast-4_belfast-4-comments-1}{}\subsubsection{How does P\+L\+U\+M\+E\+D work}\label{belfast-4_belfast-4-comments-1}
The fact that when you add a force on the collective variable P\+L\+U\+M\+E\+D can force the atoms to do something depends on the fact that the collective variables implemented in P\+L\+U\+M\+E\+D has analytical derivatives. By biasing the value of a single C\+V one turns to affect the time evolution of the system itself. Notice that some of the collective variables could be implemented without derivatives (either because the developers were lazy or because the C\+Vs cannot be derived). In this case you might want to have a look at the N\+U\+M\+E\+R\+I\+C\+A\+L\+\_\+\+D\+E\+R\+I\+V\+A\+T\+I\+V\+E\+S option.\hypertarget{belfast-4_belfast-4-further-reading}{}\subsection{Further Reading}\label{belfast-4_belfast-4-further-reading}
Umbrella sampling method is a widely used technique. You can find several resources on the web, e.\+g.\+:


\begin{DoxyItemize}
\item \href{http://en.wikipedia.org/wiki/Umbrella_sampling}{\tt http\+://en.\+wikipedia.\+org/wiki/\+Umbrella\+\_\+sampling} 
\end{DoxyItemize}\hypertarget{belfast-5}{}\section{Belfast tutorial\+: Out of equilibrium dynamics}\label{belfast-5}
In plumed you can bring a system in a specific state in a collective variable by means of the \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T} directive. This directive is very flexible and allows for a programmed series of draggings and can be used also to sample multiple events within a single simulation. Here I will explain the concepts of it and show some examples\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
Here is the \href{tutorial-resources/belfast-5.tar.gz}{\tt tarball with the files referenced in the following }.\hypertarget{belfast-5_belfast-5-SMD}{}\subsection{Steered M\+D}\label{belfast-5_belfast-5-SMD}
Steered M\+D (S\+M\+D) is often used to drag the system from an initial configuration to a final one by pulling one or more C\+Vs. Most of time the aim of such simulations is to prepare the system in a particular state or produce nice snapshots for a cool movie. All the C\+Vs present in P\+L\+U\+M\+E\+D can be used in S\+M\+D.

In S\+M\+D the Hamiltonian of the system $ H $ is modified into $H_\lambda$. This new Hamiltonian contains now another new term which now depends on time only via a Harmonic potential centered on a point which moves linear with time \begin{eqnarray*} H_\lambda(X,t)&=&H(X)+U_\lambda(X,t)\\ &=&H(X)+\frac{k(t)}{2}(s(X)-\lambda(t))^2\\ &=&H(X)+\frac{k(t)}{2}(s(X)-s_0-vt)^2. \end{eqnarray*} This means that if the $ k $ is tight enough the system will follow closely the center of the moving harmonic spring. But be careful, if the spring constant is too hard your equations of motion will now keep up since they are tuned to the fastest motion in your system so if you artificially introduce a higher freqeuncy in your system you'll screw up the dynamics. The same is true for the pulling speed $ v $. As a matter of fact I never encountered the case where I had to lower the time step and I could all the time be happy just by making a softer spring constant or a slower steering speed. Generally, integrators of equations of motion like velocity-\/\+Verlet are very tolerant. Note that one can also make the spring constant depend on time and this, as we will see later in the examples is particularly useful to prepare your state.

In simulations, it is more convenient to adopt a situation where you specify only the starting point, the final point of cvs and the time in which you want to cover the transition. That's why the plumed input is done in such a way.

For example, let's go back to the alanine dipeptide example encountered in \hyperlink{belfast-2_belfast-2-ala}{The molecule of the day\+: alanine dipeptide}. Let's say that now we want to steer from $ C_7eq $ to $ C_7ax $. If you think, just by dragging along the $ \Phi $ dihedral angle from a value of -\/1 to a value 1 should be enough to the state of interest. Additionally, it might be important to you not to stress the system too much, so you might want first to increase the $ k $ first so to lock the system in $ \Phi =-1 $, then move it gently to $ \Phi =1 $ and then release again your spring constant so to end up to an equilibrated and unconstrained state. This you can program in P\+L\+U\+M\+E\+D like this \begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles
# drag this
phi: TORSION ATOMS=5,7,9,15
# this is just to monitor that you end up in the interesting state
psi: TORSION ATOMS=7,9,15,17
# the movingrestraint
restraint: ...
        MOVINGRESTRAINT
        ARG=phi
        AT0=-1.0 STEP0=0      KAPPA0=0
        AT1=-1.0 STEP1=2000   KAPPA1=1000
        AT2=1.0  STEP2=4000   KAPPA2=1000
        AT3=1.0  STEP3=6000   KAPPA3=0
...
# monitor the two variables and various restraint outputs
PRINT STRIDE=10 ARG=* FILE=COLVAR
\end{DoxyVerb}


Please note the syntax of \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T} \+: You need one (or more) argument(s) and a set of steps denote by A\+T\+X, S\+T\+E\+P\+X, K\+A\+P\+P\+A\+X where X is a incremental starting from 0 that assign the center and the harness of the spring at step S\+T\+E\+P\+X. What happens in between is a linear interpolation of the A\+T and K\+A\+P\+P\+A parameters. If those are identical in two consecutive steps then nothing is happening to that parameter. So if you put the same K\+A\+P\+P\+A and A\+T in two different S\+T\+E\+Ps then this will give you an umbrella potential placed in the same point between the two time intervals defined by S\+T\+E\+P. Note that you need to run a bit more than 6000 steps because after this your system has no more restraints so the actual thermalization period starts here.

The C\+O\+L\+V\+A\+R file produced has the following shape \begin{DoxyVerb}#! FIELDS time phi psi restraint.bias restraint.force2 restraint.phi_cntr restraint.phi_work
#! SET min_phi -pi
#! SET max_phi pi
#! SET min_psi -pi
#! SET max_psi pi
 0.000000 -1.409958 1.246193 0.000000 0.000000 -1.000000 0.000000
 0.020000 -1.432352 1.256545 0.467321 4.673211 -1.000000 0.441499
 0.040000 -1.438652 1.278405 0.962080 19.241592 -1.000000 0.918101
 0.060000 -1.388132 1.283709 1.129846 33.895372 -1.000000 1.344595
 0.080000 -1.360254 1.275045 1.297832 51.913277 -1.000000 1.691475
 ...
\end{DoxyVerb}


So we have time, phi, psi and the bias from the moving restraint. Note that at step 0 is zero since we imposed this to start from zero and ramp up in the first 2000 steps up to a value of 2000 k\+J/mol/rad$^\wedge$2. It increases immediately since already at step 1 the harmonic potential is going to be increased in bits towards the value of 1000 which is set by K\+A\+P\+P\+A. The value of restraint.\+force2 is the squared force (which is a proxy of the force magnitude, despite the direction) on the C\+V.

\begin{eqnarray*} -\frac{\partial H_\lambda(X,t)}{\partial s}&=&-(s(X)-s_0-vt) \end{eqnarray*}

Note that the actual force on an atom of the system involved in a C\+V is instead

\begin{eqnarray*} -\frac{\partial H_\lambda(X,t)}{\partial x_i}&=& -\frac{\partial H_\lambda(X,t)}{\partial s} \frac{\partial s}{\partial x_i} \\ &=& -(s(X)-s_0-vt) \frac{\partial s}{\partial x_i} \end{eqnarray*}

This is important because in C\+Vs that have a derivative that change significantly with space then you might have regions in which no force is exerted while in others you might have an enormous force on it. Typically this is the case of sigmoids that are used in coordination numbers in which, in the tails, they are basically flat as a function of particle positions. Additionally note that this happens on any force-\/based enhanced-\/sampling algorithm so keep it in mind. Very often people miss this aspect and complain either that a C\+V or a enhanced-\/sampling method does not work. This is another good reason to use tight spring force so to compensate in the force the lack of derivative from the C\+V.

The other argument in colvar is restraint.\+phi\+\_\+cntr which is the center of the harmonic potential. This is a useful quantity so you may know how close the system is follwing the center of harmonic potential (more on this below). The last parameter is restraint.\+phi\+\_\+work. The work is defined as

\begin{eqnarray*} W=\int_0^{t_s}dt\ \frac{\partial H_\lambda(t)}{\partial t} \end{eqnarray*}

so this is changing only when the Hamiltonian is changing with time. There are two time dependent contributions in this integral\+: one can come from the fact that $ k(t) $ changes with time and another from the fact that the center of the spring potential is changing with time. \begin{eqnarray*} W&=&\int_0^{t_s}dt\ \frac{\partial H_\lambda(t)}{\partial t}\\ &=&\int_0^{t_s}dt\ \frac{\partial H_\lambda(t)}{\partial \lambda} \frac{\partial \lambda(t)}{\partial t} + \int_0^{t_s}dt\ \frac{\partial H_\lambda(t)}{\partial k} \frac{\partial k}{\partial t} \\ &=& \int_0^{t_s} -k(t)( s(X)-\lambda(t)) \frac{\partial \lambda(t)}{\partial t} dt + \int_0^{t_s} \frac{( s(X)-\lambda(t))^2}{2}\frac{\partial k}{\partial t} dt \\ &=& \int_0^{t_s} - k(t)( s(X)-\lambda(t)) v dt + \int_0^{t_s} \frac{( s(X)-\lambda(t))^2}{2}\frac{\partial k}{\partial t} dt \\ &\simeq& \sum_i - k(t_i)( s(X(t_i))-\lambda(t_i)) \Delta \lambda(t_i) + \sum_i\ \frac{( s(X(t_i))-\lambda(t_i))^2}{2} \Delta k(t_i) \end{eqnarray*} where we denoted $ \Delta \lambda(t_i) $ the difference of the center of the harmonic potential respect to the step before and $ \Delta k(t_i) $ is the difference in spring constant respect to the step before. So in the exercised proposed in the first phase you see only the second part of the work since this is the part connected with the spring constant increase. After this phase you see the increase due to the motion of the center and then you later the release of the spring constant.

\label{belfast-5_belfast-5-work-1-fig}%
\hypertarget{belfast-5_belfast-5-work-1-fig}{}%
 \begin{center} {\bfseries  The work profile as function of time when steering ala dipeptide along the $ \Phi $ variable. } \end{center} 

This you get with gnuplot\+:

\begin{DoxyVerb}pd@plumed:~>gnuplot
gnuplot>  p "COLVAR" u 1:7 w lp  
\end{DoxyVerb}


Another couple of interesting thing that you can check is
\begin{DoxyItemize}
\item Is my system finally in the $ C7ax $ ? Plot the two dihedral to have a sense if we are in the right state. You know the target position what should look like, right?
\item Is my system moving close to the center of the harmonic potential? This is important and we will see why in a while.
\end{DoxyItemize}\hypertarget{belfast-5_belfast-5-path}{}\subsection{Moving on a more complex path}\label{belfast-5_belfast-5-path}
Very often it is useful to use this movingrestraint to make a fancier schedule by using nice properties of \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T}. For example you can plan a schedule to drive multiple C\+Vs at the same time in specific point of the phase space and also to stop for a while in specific using a fixed harmonic potential. This can be handy in case of an umbrella sampling run where you might want to explore a 1-\/dimensional landscape by acquiring some statistics in one point and then moving to the next to acquire more statistics. With \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T} you can do it in only one file. To give an example of such capabilities, let's say that we want to move from $ C7eq $ vertically toward $ \Phi =-1.5 ; \Psi=-1.3 $, stop by for a while (e.\+g. to acquire a statistics that you might need for an umbrella sampling), then moving toward $ \Phi =1.3 ; \Psi=-1.3 $ which roughly corresponds to $ C7ax $.

This can be programmed conveniently with \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T} by adopting the following schedule

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
# the movingrestraint
restraint: ...
        MOVINGRESTRAINT
        ARG=phi,psi
        AT0=-1.5,1.3  STEP0=0      KAPPA0=0,0
        AT1=-1.5,1.3  STEP1=2000   KAPPA1=1000,1000
        AT2=-1.5,-1.3 STEP2=4000   KAPPA2=1000,1000
        AT3=-1.5,-1.3 STEP3=4000   KAPPA3=1000,1000
        AT4=1.3,-1.3  STEP4=6000   KAPPA4=1000,1000
        AT5=1.3,-1.3  STEP5=8000   KAPPA5=0,0
...
# monitor the two variables and various restraint outputs
PRINT STRIDE=10 ARG=* FILE=COLVAR
\end{DoxyVerb}


Note that by adding two arguments for movingrestraint, now I am allowed to put two values (separated by comma, as usual for multiple values in P\+L\+U\+M\+E\+D) and correspondingly two K\+A\+P\+P\+A values. One for each variable. Please note that no space must be used bewtween the arguments! This is a very common fault in writing the inputs.

By plotting the instataneous value of the variables and the value of the center of the harmonic potentials we can inspect the pathways that we make the system walk on the Ramachandran plot. (How to do this? Have a look to the header of C\+O\+L\+V\+A\+R file to plot the right fields)

\label{belfast-5_belfast-5-doublesteer}%
\hypertarget{belfast-5_belfast-5-doublesteer}{}%
.png  \begin{center} {\bfseries  Plot of the double steering schedule using \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T} } \end{center} \hypertarget{belfast-5_belfast-5-work}{}\subsection{Why work is important?}\label{belfast-5_belfast-5-work}
The work as we have seen is the cumulative change of the hamiltonian in time. So it is connected with the change in energy of your system while you move it around. It has also a more important connection with the free energy via the Jarzynski equation which reads \[ \Delta F=-\beta^{-1}\ln \langle \exp^{-\beta W} \rangle \] This is important and says that potentially you can calculate the free energy even by driving your system very fast and out of equilibrium between two states of your interest. Unfortunately this in practice not possible since the accurate calculation of the quantity $ \langle \exp^{-\beta W} \rangle $ has a huge associated error bar since it involves the average of a noisy quantity (the work) being exponentiated. So, before going wild with S\+M\+D, I want to make a small exercise on how tricky that is even for the smallest system.

Now we run, say 30 S\+M\+D run and we calculate the free energy difference by using Jarzynski equality and see how this differs from the average. First note that the average $ \langle \exp^{-\beta W} \rangle $ is an average over a number of steered M\+D runs which start from the same value of C\+V and reach the final value of C\+V. So it is important to create initially an ensemble of states which are compatible with a given value of C\+Vs. Let's assume that we can do this by using a restrained M\+D in a point (say at $ Phi=-1.5 $). In practice the umbrella biases a bit your distribution and the best situation would be to do this with a flat bottom potential and then choosing the snapshot that correspond to the wanted starting value and start from them.

In the directory J\+A\+R\+Z/\+M\+A\+K\+E\+\_\+\+E\+N\+S\+E\+M\+B\+L\+E you find the script to run. After you generate the constrained ensemble this needs to be translated from xtc format to something that G\+R\+O\+M\+A\+C\+S is able to read in input, typically a more convenient gro file. To do so just to \begin{DoxyVerb}pd@plumed:~> echo 0 | trjconv_mpi-dp-pl -f traj.xtc -s topol.tpr -o all.gro
pd@plumed:~> awk 'BEGIN{i=1}{if($1=="Generated"){outfile=sprintf("start_%d.gro",i);i++}print >>outfile; if(NF==3){close(outfile)}}' all.gro
\end{DoxyVerb}


This will generate a set of numbered gro files. Now copy them in the parallel directory M\+A\+K\+E\+\_\+\+S\+T\+E\+E\+R. There you will find a script (script.\+sh) where you can set the number of runs that you want to go for. Just try 20 and let it run. Will take short time. The script will also produce a script\+\_\+rama.\+gplt that you can use to visualize all the work performed in a single gnuplot session. Just do\+: \begin{DoxyVerb}pd@plumed:~>gnuplot
gnuplot> load "script_work.gplt"
\end{DoxyVerb}


What you see is something like in Fig.

\label{belfast-5_belfast-5-jarz-fig}%
\hypertarget{belfast-5_belfast-5-jarz-fig}{}%
 There are a number of interesting fact here. First you see that different starting points may end with very different work values. Not bad, one would say, since Jarzyinsi equality is saying that we can make an average and everything will be ok. So now calculate the average work by using the following bash one-\/liner\+:

\begin{DoxyVerb}pd@plumed:~> ntest=20; for i in `seq 1 $ntest` ; do tail -1 colvar_$i | awk '{print $7 }' ; done | awk '{g+=exp(-$1/kt)    ;  gg+=(exp(-$1/kt))*(exp(-$1/kt))  ; i++}END{gavg=g/i;ggavg=gg/i ;  stdev=sqrt(ggavg-gavg*gavg); print "FREE ENERGY ESTIMATE ",-kt*log(gavg)," STDEV ", kt*stdev/gavg}' kt=2.4
\end{DoxyVerb}


For my test, what I get is a value of \begin{DoxyVerb}FREE ENERGY ESTIMATE  17.482  STDEV  7.40342
\end{DoxyVerb}
 and what this is saying is that the only thing that matters is the lowest work that I sampled. This has such an enormous weight over all the other trajectories that will do so that it will be the only other to count, and all the other do not matter much. So it is a kind of a waste of time. Also the standard deviation is rather high and probably it might well be that you obtain a much better result by using a standard umbrella sampling where you can use profitably most of the statistics. Here you waste most of the statistics indeed, since only the lowest work sampled will matter.

Some important point for doing some further exercises\+:


\begin{DoxyItemize}
\item How does the work distribution change if you increase the simulation time? Note that you have to increase both the time in the md.\+mdp file and in the plumed.\+dat file.
\item How the work change if you now use a softer spring constant? And a harder one?
\item In particular, what happens when you have softer spring constant, say 10? This does not look like working? Can you guess what is going on there from an analysis of C\+O\+L\+V\+A\+R files only?
\item Have a look of the trajectories in the Ramachandran plot in case of fast simulations and slow simulation. What can you observe? Is there a correlation between steering speed and how often you can go on the low energy path?
\end{DoxyItemize}\hypertarget{belfast-5_belfast-5-target}{}\subsection{Targeted M\+D}\label{belfast-5_belfast-5-target}
Targeted M\+D can be seen as a special case of steered M\+D where the R\+M\+S\+D from a reference structure is used as a collective variable. It can be used for example if one wants to prepare the system so that the coordinates of selected atoms are as close as possible to a target pdb structure.

As an example we can take alanine dipeptide again

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles
# these variables will be just monitored to see what happens
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
# creates a CV that measures the RMSD from a reference pdb structure
# the RMSD is measured after OPTIMAL alignment with the target structure
rmsd: RMSD REFERENCE=c7ax.pdb TYPE=OPTIMAL
# the movingrestraint
restraint: ...
        MOVINGRESTRAINT
        ARG=rmsd
        AT0=0.0 STEP0=0      KAPPA0=0
        AT1=0.0 STEP1=5000   KAPPA1=10000
...
# monitor the two variables and various restraint outputs
PRINT STRIDE=10 ARG=* FILE=COLVAR
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{RMSD}{R\+M\+S\+D}, \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

Note that \hyperlink{RMSD}{R\+M\+S\+D} should be provided a reference structure in pdb format and can contain part of the system but the second column (the index) must reflect that of the full pdb so that P\+L\+U\+M\+E\+D knows specifically which atom to drag where. The \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T} bias potential here acts on the rmsd, and the other two variables (phi and psi) are untouched. Notice that whereas the force from the restraint should be applied at every step (thus rmsd is computed at every step) the two torsions are computed only every 10 steps. P\+L\+U\+M\+E\+D automatically detect which variables are used at every step, leading to better performance when complicated and computationally expensive variables are monitored -\/ this is not the case here, since the two torsions are very fast to compute. Note that here the work always increase with time and never gets lower which is somewhat surprising if you tink that we are moving in another metastable state. One would expect this to bend and give a signal of approaching a minimum like before. Nevertheless consider what you we are doing\+: we are constraining the system in one specific conformation and this is completely unnatural for a system at 300 kelvin so, even for this small system adopting a specific conformation in which all the heavy atoms are in a precise position is rather unrealistic. This means that this state is an high free energy state. \hypertarget{belfast-6}{}\section{Belfast tutorial\+: Metadynamics}\label{belfast-6}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
The aim of this tutorial is to introduce the users to running a metadynamics simulation with P\+L\+U\+M\+E\+D. We will set up a simple simulation of alanine dipeptide in vacuum, analyze the output, and estimate free energies from the simulation. We will also learn how to run a well-\/tempered metadynamics simulation and detect issues related to a bad choice of collective variables.\hypertarget{belfast-6_belfast-6-theory}{}\subsection{Summary of theory}\label{belfast-6_belfast-6-theory}
In metadynamics, an external history-\/dependent bias potential is constructed in the space of a few selected degrees of freedom $ \vec{s}({q}) $, generally called collective variables (C\+Vs) \cite{metad}. This potential is built as a sum of Gaussians deposited along the trajectory in the C\+Vs space\+:

\[ V(\vec{s},t) = \sum_{ k \tau < t} W(k \tau) \exp\left( -\sum_{i=1}^{d} \frac{(s_i-s_i({q}(k \tau)))^2}{2\sigma_i^2} \right). \]

where $ \tau $ is the Gaussian deposition stride, $ \sigma_i $ the width of the Gaussian for the ith C\+V, and $ W(k \tau) $ the height of the Gaussian. The effect of the metadynamics bias potential is to push the system away from local minima into visiting new regions of the phase space. Furthermore, in the long time limit, the bias potential converges to minus the free energy as a function of the C\+Vs\+:

\[ V(\vec{s},t\rightarrow \infty) = -F(\vec{s}) + C. \]

In standard metadynamics, Gaussians of constant height are added for the entire course of a simulation. As a result, the system is eventually pushed to explore high free-\/energy regions and the estimate of the free energy calculated from the bias potential oscillates around the real value. In well-\/tempered metadynamics \cite{Barducci:2008}, the height of the Gaussian is decreased with simulation time according to\+:

\[ W (k \tau ) = W_0 \exp \left( -\frac{V(\vec{s}({q}(k \tau)),k \tau)}{k_B\Delta T} \right ), \]

where $ W_0 $ is an initial Gaussian height, $ \Delta T $ an input parameter with the dimension of a temperature, and $ k_B $ the Boltzmann constant. With this rescaling of the Gaussian height, the bias potential smoothly converges in the long time limit, but it does not fully compensate the underlying free energy\+:

\[ V(\vec{s},t\rightarrow \infty) = -\frac{\Delta T}{T+\Delta T}F(\vec{s}) + C. \]

where $ T $ is the temperature of the system. In the long time limit, the C\+Vs thus sample an ensemble at a temperature $ T+\Delta T $ which is higher than the system temperature $ T $. The parameter $ \Delta T $ can be chosen to regulate the extent of free-\/energy exploration\+: $ \Delta T = 0$ corresponds to standard molecular dynamics, $ \Delta T \rightarrow \infty $ to standard metadynamics. In well-\/tempered metadynamics literature and in P\+L\+U\+M\+E\+D, you will often encounter the term \char`\"{}biasfactor\char`\"{} which is the ratio between the temperature of the C\+Vs ( $ T+\Delta T $) and the system temperature ( $ T $)\+:

\[ \gamma = \frac{T+\Delta T}{T}. \]

The biasfactor should thus be carefully chosen in order for the relevant free-\/energy barriers to be crossed efficiently in the time scale of the simulation.

Additional information can be found in the several review papers on metadynamics \cite{gerv-laio09review} \cite{WCMS:WCMS31} \cite{WCMS:WCMS1103}.\hypertarget{belfast-6_belfast-6-learning-outcomes}{}\subsection{Learning Outcomes}\label{belfast-6_belfast-6-learning-outcomes}
Once this tutorial is completed students will know how to\+:


\begin{DoxyItemize}
\item run a metadynamics simulation using P\+L\+U\+M\+E\+D
\item analyze the output of the simulation
\item restart a metadynamics simulation
\item calculate free energies from a metadynamics simulation
\item run a well-\/tempered metadynamics simulation using P\+L\+U\+M\+E\+D
\item detect issues with the choice of the collective variables
\end{DoxyItemize}\hypertarget{belfast-6_belfast-6-resources}{}\subsection{Resources}\label{belfast-6_belfast-6-resources}
The \href{tutorial-resources/belfast-6.tar.gz}{\tt tarball } for this project contains the following directories\+:


\begin{DoxyItemize}
\item T\+O\+P\+O\+: it contains the gromacs topology and configuration files to simulate alanine dipeptide in vacuum
\item Exercise\+\_\+1\+: run a metadynamics simulation with 2 C\+Vs, dihedrals phi and psi, and analyze the output
\item Exercise\+\_\+2\+: restart a metadynamics simulation
\item Exercise\+\_\+3\+: calculate free energies from a metadynamics simulation and monitor convergence
\item Exercise\+\_\+4\+: run a well-\/tempered metadynamics simulation with 2 C\+Vs, dihedrals phi and psi
\item Exercise\+\_\+5\+: run a well-\/tempered metadynamics simulation with 1 C\+V, dihedral psi
\end{DoxyItemize}\hypertarget{belfast-6_belfast-6-instructions}{}\subsection{Instructions}\label{belfast-6_belfast-6-instructions}
\hypertarget{belfast-6_belfast-6-system}{}\subsubsection{The model system}\label{belfast-6_belfast-6-system}
Here we use as model system alanine dipeptide in vacuum with A\+M\+B\+E\+R99\+S\+B-\/\+I\+L\+D\+N all-\/atom force field.\hypertarget{belfast-6_belfast-6-exercise-1}{}\subsubsection{Exercise 1. Setup and run a metadynamics simulation}\label{belfast-6_belfast-6-exercise-1}
In this exercise, we will run a metadynamics simulation on alanine dipeptide in vacuum, using as C\+Vs the two backbone dihedral angles phi and psi. In order to run this simulation we need to prepare the P\+L\+U\+M\+E\+D input file (plumed.\+dat) as follows.

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate metadynamics in phi and psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad for both CVs. 
#
metad: METAD ARG=phi,psi PACE=500 HEIGHT=1.2 SIGMA=0.35,0.35 FILE=HILLS 

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The syntax for the command \hyperlink{METAD}{M\+E\+T\+A\+D} is simple. The directive is followed by a keyword A\+R\+G followed by the labels of the C\+Vs on which the metadynamics potential will act. The keyword P\+A\+C\+E determines the stride of Gaussian deposition in number of time steps, while the keyword H\+E\+I\+G\+H\+T specifies the height of the Gaussian in k\+Joule/mol. For each C\+Vs, one has to specified the width of the Gaussian by using the keyword S\+I\+G\+M\+A. Gaussian will be written to the file indicated by the keyword F\+I\+L\+E.

Once the P\+L\+U\+M\+E\+D input file is prepared, one has to run Gromacs with the option to activate P\+L\+U\+M\+E\+D and read the input file\+:

\begin{DoxyVerb}mdrun_mpi -plumed
\end{DoxyVerb}


During the metadynamics simulation, P\+L\+U\+M\+E\+D will create two files, named C\+O\+L\+V\+A\+R and H\+I\+L\+L\+S. The C\+O\+L\+V\+A\+R file contains all the information specified by the P\+R\+I\+N\+T command, in this case the value of the C\+Vs every 10 steps of simulation, along with the current value of the metadynamics bias potential. The H\+I\+L\+L\+S file contains a list of the Gaussians deposited along the simulation. If we give a look at the header of this file, we can find relevant information about its content\+:

\begin{DoxyVerb}#! FIELDS time phi psi sigma_phi sigma_psi height biasf
#! SET multivariate false
#! SET min_phi -pi
#! SET max_phi pi
#! SET min_psi -pi
#! SET max_psi pi
\end{DoxyVerb}


The line starting with F\+I\+E\+L\+D\+S tells us what is displayed in the various columns of the H\+I\+L\+L\+S file\+: the time of the simulation, the value of phi and psi, the width of the Gaussian in phi and psi, the height of the Gaussian, and the biasfactor. This quantity is relevant only for well-\/tempered metadynamics simulation (see \hyperlink{belfast-6_belfast-6-exercise-4}{Exercise 4. Setup and run a well-\/tempered metadynamics simulation, part I}) and it is equal to 1 in standard metadynamics simulations. We will use the H\+I\+L\+L\+S file later to calculate free-\/energies from the metadynamics simulation and assess its convergence. For the moment, we can plot the behavior of the C\+Vs during the simulation.

\label{belfast-6_belfast-6-metad-fig}%
\hypertarget{belfast-6_belfast-6-metad-fig}{}%
 By inspecting Figure \hyperlink{belfast-6_belfast-6-metad-fig}{belfast-\/6-\/metad-\/fig}, we can see that the system is initialized in one of the two metastable states of alanine dipeptide. After a while (t=0.\+3 ns), the system is pushed by the metadynamics bias potential to visit the other local minimum. As the simulation continues, the bias potential fills the underlying free-\/energy landscape, and the system is able to diffuse in the entire phase space.

If we use the P\+L\+U\+M\+E\+D input file described above, the expense of a metadynamics simulation increases with the length of the simulation as one has to evaluate the values of a larger and larger number of Gaussians at every step. To avoid this issue you can store the bias on a grid. In order to use grids, we have to add some additional information to the line of the \hyperlink{METAD}{M\+E\+T\+A\+D} directive, as follows.

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate metadynamics in phi and psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad for both CVs.
# The bias potential will be stored on a grid
# with bin size equal to 0.1 rad for both CVs. 
# The boundaries of the grid are -pi and pi, for both CVs.
#
METAD ...
LABEL=metad
ARG=phi,psi 
PACE=500
HEIGHT=1.2
SIGMA=0.35,0.35
FILE=HILLS
GRID_MIN=-pi,-pi
GRID_MAX=pi,pi
GRID_SPACING=0.1,0.1 
... METAD

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}


The bias potential will be stored on a grid, whose boundaries are specified by the keywords G\+R\+I\+D\+\_\+\+M\+I\+N and G\+R\+I\+D\+\_\+\+M\+A\+X. Notice that you should provide either the number of bins for every collective variable (G\+R\+I\+D\+\_\+\+B\+I\+N) or the desired grid spacing (G\+R\+I\+D\+\_\+\+S\+P\+A\+C\+I\+N\+G). In case you provide both P\+L\+U\+M\+E\+D will use the most conservative choice (highest number of bins) for each dimension. In case you do not provide any information about bin size (neither G\+R\+I\+D\+\_\+\+B\+I\+N nor G\+R\+I\+D\+\_\+\+S\+P\+A\+C\+I\+N\+G) and if Gaussian width is fixed P\+L\+U\+M\+E\+D will use 1/5 of the Gaussian width as grid spacing. This default choice should be reasonable for most applications.\hypertarget{belfast-6_belfast-6-exercise-2}{}\subsubsection{Exercise 2. Restart a metadynamics simulation}\label{belfast-6_belfast-6-exercise-2}
If we try to run again a metadynamics simulation using the script above in a directory where a C\+O\+L\+V\+A\+R and H\+I\+L\+L\+S files are already present, P\+L\+U\+M\+E\+D will create a backup copy of the old files, and run a new simulation. Instead, if we want to restart a previous simulation, we have to add the keyword R\+E\+S\+T\+A\+R\+T to the P\+L\+U\+M\+E\+D input file (plumed.\+dat), as follows.

\begin{DoxyVerb}# restart previous simulation
RESTART

# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate metadynamics in phi and psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad for both CVs. 
#
metad: METAD ARG=phi,psi PACE=500 HEIGHT=1.2 SIGMA=0.35,0.35 FILE=HILLS 

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{RESTART}{R\+E\+S\+T\+A\+R\+T}, \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

In this way, P\+L\+U\+M\+E\+D will read the old Gaussians from the H\+I\+L\+L\+S file and append the new information to both C\+O\+L\+V\+A\+R and H\+I\+L\+L\+S files.\hypertarget{belfast-6_belfast-6-exercise-3}{}\subsubsection{Exercise 3. Calculate free-\/energies and monitor convergence}\label{belfast-6_belfast-6-exercise-3}
One can estimate the free energy as a function of the metadynamics C\+Vs directly from the metadynamics bias potential. In order to do so, the utility \hyperlink{sum_hills}{sum\+\_\+hills} should be used to sum the Gaussians deposited during the simulation and stored in the H\+I\+L\+L\+S file.

To calculate the two-\/dimensional free energy as a function of phi and psi, it is sufficient to use the following command line\+:

\begin{DoxyVerb}plumed sum_hills --hills HILLS
\end{DoxyVerb}


The command above generates a file called fes.\+dat in which the free-\/energy surface as function of phi and psi is calculated on a regular grid. One can modify the default name for the free energy file, as well as the boundaries and bin size of the grid, by using the following options of \hyperlink{sum_hills}{sum\+\_\+hills} \+:

\begin{DoxyVerb}--outfile - specify the outputfile for sumhills
--min - the lower bounds for the grid
--max - the upper bounds for the grid
--bin - the number of bins for the grid
--spacing - grid spacing, alternative to the number of bins
\end{DoxyVerb}


It is also possible to calculate one-\/dimensional free energies from the two-\/dimensional metadynamics simulation. For example, if one is interested in the free energy as a function of the phi dihedral alone, the following command line should be used\+:

\begin{DoxyVerb}plumed sum_hills --hills HILLS --idw phi --kt 2.5
\end{DoxyVerb}


The result should look like this\+:

\label{belfast-6_belfast-6-phifes-fig}%
\hypertarget{belfast-6_belfast-6-phifes-fig}{}%
 To assess the convergence of a metadynamics simulation, one can calculate the estimate of the free energy as a function of simulation time. At convergence, the reconstructed profiles should be similar, apart from a constant offset. The option --stride should be used to give an estimate of the free energy every N Gaussians deposited, and the option --mintozero can be used to align the profiles by setting the global minimum to zero. If we use the following command line\+:

\begin{DoxyVerb}plumed sum_hills --hills HILLS --idw phi --kt 2.5 --stride 500 --mintozero
\end{DoxyVerb}


one free energy is calculated every 500 Gaussians deposited, and the global minimum is set to zero in all profiles. The resulting plot should look like the following\+:

\label{belfast-6_belfast-6-phifest-fig}%
\hypertarget{belfast-6_belfast-6-phifest-fig}{}%
 To assess the convergence of the simulation more quantitatively, we can calculate the free-\/energy difference between the two local minima in the one-\/dimensional free energy along phi as a function of simulation time. We can use the bash script analyze\+\_\+\+F\+E\+S.\+sh to integrate the multiple free-\/energy profiles in the two basins defined by the following intervals in phi space\+: basin A, -\/3$<$phi$<$-\/1, basin B, 0.\+5$<$phi$<$1.\+5.

\begin{DoxyVerb}./analyze_FES.sh NFES -3.0 -1.0 0.5 1.5 KBT 
\end{DoxyVerb}


where N\+F\+E\+S is the number of profiles (free-\/energy estimates at different times of the simulation) generated by the option --stride of \hyperlink{sum_hills}{sum\+\_\+hills}, and K\+B\+T is the temperature in energy units (in this case K\+B\+T=2.\+5).

\label{belfast-6_belfast-6-difft-fig}%
\hypertarget{belfast-6_belfast-6-difft-fig}{}%
 This analysis, along with the observation of the diffusive behavior in the C\+Vs space, suggest that the simulation is converged.\hypertarget{belfast-6_belfast-6-exercise-4}{}\subsubsection{Exercise 4. Setup and run a well-\/tempered metadynamics simulation, part I}\label{belfast-6_belfast-6-exercise-4}
In this exercise, we will run a well-\/tempered metadynamics simulation on alanine dipeptide in vacuum, using as C\+Vs the two backbone dihedral angles phi and psi. To activate well-\/tempered metadynamics, we need to add two keywords to the line of \hyperlink{METAD}{M\+E\+T\+A\+D}, which specifies the biasfactor and temperature of the simulation. For the first example, we will try a biasfactor equal to 6. Here how the P\+L\+U\+M\+E\+D input file (plumed.\+dat) should look like\+:

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate metadynamics in phi and psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad for both CVs.
# Well-tempered metadynamics is activated,
# and the biasfactor is set to 6.0
#
metad: METAD ARG=phi,psi PACE=500 HEIGHT=1.2 SIGMA=0.35,0.35 FILE=HILLS BIASFACTOR=6.0 TEMP=300.0

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

After running the simulation using the instruction described above, we can have a look at the H\+I\+L\+L\+S file. At variance with standard metadynamics, the last two columns of the H\+I\+L\+L\+S file report more useful information. The last column contains the value of the biasfactor used, while the last but one the height of the Gaussian, which is rescaled during the simulation following the well-\/tempered recipe.

\begin{DoxyVerb}#! FIELDS time phi psi sigma_phi sigma_psi height biasf
#! SET multivariate false
#! SET min_phi -pi
#! SET max_phi pi
#! SET min_psi -pi
#! SET max_psi pi
      1.0000     -1.3100    0.0525          0.35            0.35      1.4400      6
      2.0000     -1.4054    1.9742          0.35            0.35      1.4400      6
      3.0000     -1.9997    2.5177          0.35            0.35      1.4302      6
      4.0000     -2.2256    2.1929          0.35            0.35      1.3622      6
\end{DoxyVerb}


If we carefully look at the height column, we will notice that in the beginning the value reported is higher than the initial height specified in the input file, which should be 1.\+2 k\+Joule/mol. In fact, this column reports the height of the Gaussian rescaled by the pre-\/factor that in well-\/tempered metadynamics relates the bias potential to the free energy. In this way, when we will use \hyperlink{sum_hills}{sum\+\_\+hills}, the sum of the Gaussians deposited will directly provide the free-\/energy, without further rescaling needed.

We can plot the time evolution of the C\+Vs along with the height of the Gaussian.

\label{belfast-6_belfast-6-wtb6-fig}%
\hypertarget{belfast-6_belfast-6-wtb6-fig}{}%
 The system is initialized in one of the local minimum where it starts accumulating bias. As the simulation progresses and the bias added grows, the Gaussian height is progressively reduced. After a while (t=0.\+8 ns), the system is able to escape the local minimum and explore a new region of the phase space. As soon as this happens, the Gaussian height is restored to the initial value and starts to decrease again. In the long time, the Gaussian height becomes smaller and smaller while the system diffuses in the entire C\+Vs space.

We can now try a different biasfactor and see the effect on the simulation. If we choose a biasfactor equal to 1.\+5, we can notice a faster decrease of the Gaussian height with simulation time, as expected by the well-\/tempered recipe. We will also conclude from the plot below that this biasfactor is not large enough to allow for the system to escape from the initial local minimum in the time scale of this simulation.

\label{belfast-6_belfast-6-wtb15-fig}%
\hypertarget{belfast-6_belfast-6-wtb15-fig}{}%
 Following the procedure described for standard metadynamics in the previous example, we can estimate the free energy as a function of time and monitor the convergence of the simulations using the analyze\+\_\+\+F\+E\+S.\+sh script. We will do this for the simulation in which the biasfactor was set to 6.\+0. In this case we will notice that the oscillations observed in standard metadynamics are here damped, and the bias potential converges more smoothly to the underlying free-\/energy landscape, provided that the biasfactor is sufficiently high for the free-\/energy barriers of the system under study to be crossed.

\label{belfast-6_belfast-6-wtdifft-fig}%
\hypertarget{belfast-6_belfast-6-wtdifft-fig}{}%
\hypertarget{belfast-6_belfast-6-exercise-5}{}\subsubsection{Exercise 5. Setup and run a well-\/tempered metadynamics simulation, part I\+I}\label{belfast-6_belfast-6-exercise-5}
In this exercise, we will study the effect of neglecting a relavant degree of freedom in the choice of metadynamics C\+Vs. We are going to run a well-\/tempered metadynamics simulation with the psi dihedral alone as C\+V, using the following P\+L\+U\+M\+E\+D input file (plumed.\+dat)\+:

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate metadynamics in psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad.
# Well-tempered metadynamics is activated,
# and the biasfactor is set to 10.0
#
metad: METAD ARG=psi PACE=500 HEIGHT=1.2 SIGMA=0.35 FILE=HILLS BIASFACTOR=10.0 TEMP=300.0

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

Let's look at the H\+I\+L\+L\+S file, in particular at the time serie of the C\+V psi and of the Gaussian height.

\label{belfast-6_belfast-6-phialone-fig}%
\hypertarget{belfast-6_belfast-6-phialone-fig}{}%
 From this plot, we observe a nice diffusive behavior of the C\+V psi when the Gaussian height is already quite small. This happens until t=3 ns, when the C\+V seems to be stuck for a while in a small region of the C\+V space. This behavior is typical of a situation in which a slow variable is not included in the set of C\+V. When something happens in this hidden degree of freedom, the biased C\+Vs typically cannot access anymore regions of the phase space previously visited. To understand this behavior, we need to visualize the time evolution of both phi and psi stored in the C\+O\+L\+V\+A\+R file.

\label{belfast-6_belfast-6-hidden-fig}%
\hypertarget{belfast-6_belfast-6-hidden-fig}{}%
 It is clear from the plot above that what happened around t=3 ns is a jump of the neglected, slow degree of freedom phi from one free-\/energy basin to another. The dynamics of phi is not biased by any potential, so we need to equilibrate this degree of freedom, i.\+e. to observe multiple transitions from the two basins, before declaring convergence of our simulation. Or alternatively we can add phi to the set of C\+Vs. This example demonstrates how to declare convergence of a well-\/tempered metadynamics simulation it is necessary but not sufficient to observe\+: 1) Gaussians with very small height, 2) a diffusive behavior in the C\+V space (as in the first 3 ns of this example). What we should do is repeating the simulation multiple times starting from different initial conformations. If in all simulations, we observe a diffusive behavior in the biased C\+V when the Gaussian height is very small, and we obtain very similar free-\/energy surfaces, then we can be quite confident that our simulations are converged to the right value. If this is not the case, a manual inspection of the runs can help us identifying the missing slow degrees of freedom to add to the set of biased C\+Vs. \hypertarget{belfast-7}{}\section{Belfast tutorial\+: Replica exchange I}\label{belfast-7}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
The aim of this tutorial is to introduce the users to running a parallel tempering (P\+T) simulation using P\+L\+U\+M\+E\+D. We will set up a simple simulation of alanine dipeptide in vacuum, analyze the output, calculate free energies from the simulation, and detect problems. We will also learn how to run a combined P\+T-\/metadynamics simulation (P\+T\+Meta\+D) and introduce the users to the Well-\/\+Tempered Ensemble (W\+T\+E).\hypertarget{belfast-7_belfast-7-theory}{}\subsection{Summary of theory}\label{belfast-7_belfast-7-theory}
In Replica Exchange Methods \cite{sugi-okam99cpl} (R\+E\+M), sampling is accelerated by modifying the original Hamiltonian of the system. This goal is achieved by simulating N non-\/interacting replicas of the system, each evolving in parallel according to a different Hamiltonian. At fixed intervals, an exchange of configurations between two replicas is attempted. One popular case of R\+E\+M is P\+T, in which replicas are simulated using the same potential energy function, but different temperatures. By accessing high temperatures, replicas are prevented from being trapped in local minima. In P\+T, exchanges are usually attempted between adjacent temperatures with the following acceptance probability\+:

\[ p(i \rightarrow j) = min \{ 1,\Delta_{i,j}^{PT} \}, \]

with

\[ \Delta_{i,j}^{PT} = \left ( \frac{1}{k_B T_i}-\frac{1}{k_B T_j} \right ) \left( U(R_i) - U(R_j) \right ), \]

where $ R_i $ and $ R_j $ are the configurations at temperature $ T_i $ and $ T_j $, respectively. The equation above suggests that the acceptance probability is ultimately determined by the overlap between the energy distributions of two replicas. The efficiency of the algorithm depends on the benefits provided by sampling at high-\/temperature. Therefore, an efficient diffusion in temperature space is required and configurational sampling is still limited by entropic barriers. Finally, P\+T scales poorly with system size. In fact, a sufficient overlap between the potential energy distributions of neighboring temperatures is required in order to obtain a significant diffusion in temperature. Therefore, the number of temperatures needed to cover a given temperature range scales as the square root of the number of degrees of freedom, making this approach prohibitively expensive for large systems.

P\+T can be easily combined with metadynamics \cite{bussi_xc}. In the resulting P\+T\+Meta\+D algorithm (16), N replicas performed in parallel a metadynamics simulation at different temperatures, using the same set of C\+Vs. The P\+T acceptance probability must be modified in order to account for the presence of a bias potential\+:

\[ \Delta_{i,j}^{PTMetaD} = \Delta_{i,j}^{PT} + \frac{1}{k_B T_i} \left [ V_G^{i}(s(R_i),t) - V_G^{i}(s(R_j),t) \right ] + \frac{1}{k_B T_j} \left [ V_G^{j}(s(R_j),t) - V_G^{j}(s(R_i),t) \right ], \]

where $ V_G^{i} $ and $ V_G^{j} $ are the bias potentials acting on the i-\/th and j-\/th replicas, respectively.

P\+T\+Meta\+D is particularly effective because it compensates for some of the weaknesses of each method alone. The effect of neglecting a slow degree of freedom in the choice of the metadynamics C\+Vs is alleviated by P\+T, which allows the system to cross moderately high free-\/energy barriers on all degrees of freedom. On the other hand, the metadynamics bias potential allows crossing higher barriers on a few selected C\+Vs, in such a way that the sampling efficiency of P\+T\+Meta\+D is greater than that of P\+T alone.

P\+T\+Meta\+D still suffers from the poor scaling of computational resources with system size. This issue may be circumvented by including the potential energy of the system among the set of well-\/tempered metadynamics C\+Vs. The well-\/tempered metadynamics bias leads to the sampling of a well-\/defined distribution called Well-\/\+Tempered Ensemble (W\+T\+E) \cite{Bonomi:2009p17935}. In this ensemble, the average energy remains close to the canonical value but its fluctuations are enhanced in a tunable way, thus improving sampling. In the so-\/called P\+T\+Meta\+D-\/\+W\+T\+E scheme \cite{ct300297t}, each replica diffusion in temperature space is enhanced by the increased energy fluctuations at all temperatures.\hypertarget{belfast-7_belfast-7-lo}{}\subsection{Learning Outcomes}\label{belfast-7_belfast-7-lo}
Once this tutorial is completed students will know how to\+:


\begin{DoxyItemize}
\item run a P\+T simulation
\item analyze the output of the P\+T simulation and detect problems
\item run a P\+T\+Meta\+D simulation
\item run a P\+T and P\+T\+Meta\+D in the W\+T\+E
\end{DoxyItemize}\hypertarget{belfast-7_belfast-7-resources}{}\subsection{Resources}\label{belfast-7_belfast-7-resources}
The \href{tutorial-resources/belfast-7.tar.gz}{\tt tarball } for this project contains the following directories\+:


\begin{DoxyItemize}
\item Exercise\+\_\+1\+: run a P\+T simulation using 2 replicas and analyze the output
\item Exercise\+\_\+2\+: run a P\+T simulation using 4 replicas and analyze the output
\item Exercise\+\_\+3\+: run a P\+T\+Meta\+D simulation
\item Exercise\+\_\+4\+: run a P\+T, P\+T-\/\+W\+T\+E and P\+T\+Meta\+D-\/\+W\+T\+E simulations
\end{DoxyItemize}

Each directory contains a T\+O\+P\+O subdirectory where topology and configuration files for Gromacs are stored.\hypertarget{belfast-7_belfast-7-instructions}{}\subsection{Instructions}\label{belfast-7_belfast-7-instructions}
\hypertarget{belfast-7_belfast-7-system}{}\subsubsection{The model system}\label{belfast-7_belfast-7-system}
Here we use as model systems alanine dipeptide in vacuum and water with A\+M\+B\+E\+R99\+S\+B-\/\+I\+L\+D\+N all-\/atom force field.\hypertarget{belfast-7_belfast-7-exercise-1}{}\subsubsection{Exercise 1. Setup and run a P\+T simulation, part I}\label{belfast-7_belfast-7-exercise-1}
In this exercise, we will run a P\+T simulation of alanine dipeptide in vacuum, using only two replicas, one at 300\+K, the other at 305\+K. During this simulation, we will monitor the time evolution of the two dihedral angles phi and psi. In order to do that, we need the following P\+L\+U\+M\+E\+D input file (plumed.\+dat).

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17

# monitor the two variables
PRINT STRIDE=10 ARG=phi,psi FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

To submit this simulation with Gromacs, we need the following command line.

\begin{DoxyVerb}mpirun -np 2 mdrun_mpi -s TOPO/topol -plumed -multi 2 -replex 100
\end{DoxyVerb}


This command will execute two M\+P\+I processess in parallel, using the topology files topol0.\+tpr and topol1.\+tpr stored in the T\+O\+P\+O subdirectory. These two binary files have been created using the usual Gromacs procedure (see Gromacs manual for further details) and setting the temperature of the two simulations at 300\+K and 305\+K in the configuration files. An exchange between the configurations of the two simulations will be attempted every 100 steps.

When Gromacs is executed using the -\/multi option and P\+L\+U\+M\+E\+D is activated, the output files produced by P\+L\+U\+M\+E\+D will be renamed and a suffix indicating the replica id will be appended. We will start inspecting the output file C\+O\+L\+V\+A\+R.\+0, which reports the time evolution of the C\+Vs at 300\+K.

\label{belfast-7_belfast-7-pt-fig}%
\hypertarget{belfast-7_belfast-7-pt-fig}{}%
 The plot above suggests that during the P\+T simulation the system is capable to access both the relevant basins in the free-\/energy surface. This seems to suggest that our simulation is converged. We can use the C\+O\+L\+V\+A\+R.\+0 and C\+O\+L\+V\+A\+R.\+1 along with the tool \hyperlink{sum_hills}{sum\+\_\+hills} to estimate the free energy as a function of the C\+V phi. We will do this a function of simulation time to assess convergence more quantitatively, using the following command line\+:

\begin{DoxyVerb}plumed sum_hills --histo COLVAR.0 --idw phi --sigma 0.2 --kt 2.5 --outhisto fes_ --stride 1000
\end{DoxyVerb}


As we did in our previous tutorial, we can now use the script analyze\+\_\+\+F\+E\+S.\+sh to calculate the free-\/energy difference between basin A (-\/3.\+0$<$phi$<$-\/1.\+0) and basin B (0.\+5$<$phi$<$1.\+5), as a function of simulation time.

\label{belfast-7_belfast-7-ptfes-fig}%
\hypertarget{belfast-7_belfast-7-ptfes-fig}{}%
 The estimate of the free-\/energy difference between these two basins seems to be converged. This consideration, along with the observation that the system is exploring all the relevant free-\/energy basins, might lead us to declare convergence and to state that the difference in free energy between basin A and B is roughly 0 k\+Joule/mol. Unfortunately, in doing so we would make a big mistake.

In P\+T simulations we have to be a little bit more careful, and examine the time evolution of each replica diffusing in temperature space, before concluding that our simulation is converged. In order to do that, we need to reconstruct the continuos trajectories of the replicas in temperature from the two files (typically traj0.\+trr and traj1.\+trr) which contain the discontinuous trajectories of the system at 300\+K and 305\+K. To demux the trajectories, we need to use the following command line\+:

\begin{DoxyVerb}demux.pl md0.log
\end{DoxyVerb}


which will create two files, called replica\+\_\+temp.\+xvg and replica\+\_\+index.\+xvg. We can plot the first file, which reports the temperature index of each configuration at a given time of the simulation. This file is extremely useful, because it allows us monitoring the replicas diffusion in temperature. As we discussed in \hyperlink{belfast-7_belfast-7-theory}{Summary of theory}, in order for the P\+T algorithm to be effective, we need an efficient diffusion of the replicas in temperature space. In this case, both replicas are rapidly accessing the highest temperature, so there seems not to be any problem on this side.

We can use the second file to reconstruct the continuous trajectories of each replica in temperature\+:

\begin{DoxyVerb}trjcat_mpi -f traj0.trr traj1.trr -demux replica_index.xvg
\end{DoxyVerb}


and the following P\+L\+U\+M\+E\+D input file (plumed\+\_\+demux.\+dat) to recalculate the value of the C\+Vs on the demuxed trajectories, typically called 0\+\_\+trajout.\+xtc and 1\+\_\+trajout.\+xtc.

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17

# monitor the two variables
PRINT STRIDE=1 ARG=phi,psi FILE=COLVAR_DEMUX\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

For the analysis of the demuxed trajectories, we can use the -\/rerun option of Gromacs, as follows\+:

\begin{DoxyVerb}# rerun Gromacs on replica 0 trajectory
mdrun_mpi -s TOPO/topol0.tpr -plumed plumed_demux.dat -rerun 0_trajout.xtc
# rename the output
mv COLVAR_DEMUX COLVAR_DEMUX.0
# rerun Gromacs on replica 1 trajectory
mdrun_mpi -s TOPO/topol1.tpr -plumed plumed_demux.dat -rerun 1_trajout.xtc
# rename the output
mv COLVAR_DEMUX COLVAR_DEMUX.1
\end{DoxyVerb}


We can now plot the time evolution of the two C\+Vs in the two demuxed trajectories.

\label{belfast-7_belfast-7-ptdemux-fig}%
\hypertarget{belfast-7_belfast-7-ptdemux-fig}{}%
 This plot shows clearly that each replica is sampling only one of the two basins of the free-\/energy landscape, and it is never able to cross the high barrier that separates them. This means that what we considered an exhaustive exploration of the free energy landscape at 300\+K (Figure \hyperlink{belfast-7_belfast-7-pt-fig}{belfast-\/7-\/pt-\/fig}), was instead caused by an efficient exchange of configurations between replicas that were trapped in different free-\/energy basins. The results of the present simulation were then influenced by the initial conformations of the two replicas. If we had initialized both of them in the same basin, we would have never observed \char`\"{}transitions\char`\"{} to the other basin at 300\+K.

To declare convergence of a P\+T simulation, it is thus mandatory to examine the behavior of the replicas diffusing in temperature and check that these are exploring all the relevant basins. Another good practice is repeating the P\+T simulation starting from different initial conformations, and check that the results are consistent.\hypertarget{belfast-7_belfast-7-exercise-2}{}\subsubsection{Exercise 2. Setup and run a P\+T simulation, part I\+I}\label{belfast-7_belfast-7-exercise-2}
We will now repeat the previous exercise, but with a different setup. The problem with the previous exercise was that replicas were never able to interconvert between the two metastable states of alanine dipeptide in vacuum. The reason was that the highest temperature used (305\+K) was too low to accelerate barrier crossing in the time scale of the simulation. We will now use 4 replicas at the following temperatures\+: 300\+K, 400\+K, 600\+K, and 1000\+K.

We can use the same P\+L\+U\+M\+E\+D input file described above (plumed.\+dat), and execute Gromacs using the following command line\+:

\begin{DoxyVerb} mpirun -np 4 mdrun_mpi -s TOPO/topol -plumed -multi 4 -replex 100
\end{DoxyVerb}


At the end of the simulation, we first monitor the diffusion in temperature space of each replica. We need to create the replica\+\_\+temp.\+xvg and replica\+\_\+index.\+xvg\+:

\begin{DoxyVerb}demux.pl md0.log
\end{DoxyVerb}


and plot the content of replica\+\_\+temp.\+xvg. Here how it should look for Replica 0\+:

\label{belfast-7_belfast-7-pt2temp-fig}%
\hypertarget{belfast-7_belfast-7-pt2temp-fig}{}%
 From this analysis, we can conclude that replicas are diffusing effectively in temperature. Now, we need to monitor the space sampled by each replica while diffusing in temperature space and verify that they are interconverting between the different basins of the free-\/energy landscape. The demux is carried out as in the previous example\+:

\begin{DoxyVerb}trjcat_mpi -f traj0.trr traj1.trr traj2.trr traj3.trr -demux replica_index.xvg
\end{DoxyVerb}


and so is the analysis of the demuxed trajectories using the -\/rerun option of Gromacs and the plumed\+\_\+demux.\+dat input file. Here is the space sampled by two of the four replicas\+:

\label{belfast-7_belfast-7-pt2demux-fig}%
\hypertarget{belfast-7_belfast-7-pt2demux-fig}{}%
 It is clear that in this case replicas are able to interconvert between the two metastable states, while efficiently diffusing in temperature. We can then calculate the free-\/energy difference between basin A and B as a function of simulation time at 300\+K\+:

\label{belfast-7_belfast-7-pt2fes-fig}%
\hypertarget{belfast-7_belfast-7-pt2fes-fig}{}%
 and conclude that in this case the P\+T simulation is converged.\hypertarget{belfast-7_belfast-7-exercise-3}{}\subsubsection{Exercise 3. Setup and run a P\+T\+Meta\+D simulation}\label{belfast-7_belfast-7-exercise-3}
In this exercise we will learn how to combine P\+T with metadynamics. We will use the setup of the previous exercise, and run a P\+T simulations with 4 replicas at the following temperatures\+: 300\+K, 400\+K, 600\+K, and 1000\+K. Each simulation will perform a well-\/tempered metadynamics calculation, using the dihedral psi alone as C\+V and a biasfactor equal to 10 (see \hyperlink{belfast-6_belfast-6-exercise-5}{Exercise 5. Setup and run a well-\/tempered metadynamics simulation, part I\+I}).

Previously, we prepared a single P\+L\+U\+M\+E\+D input file to run a P\+T simulation. This was enough, since in that case the same task was performed at all temperatures. Here instead we need to have a slightly different P\+L\+U\+M\+E\+D input file for each simulation, since we need to use the keyword T\+E\+M\+P to specify the temperature on the line of the \hyperlink{METAD}{M\+E\+T\+A\+D} directory. We will thus prepare 4 input files, called plumed.\+dat.\+0, plumed.\+dat.\+1, plumed.\+dat.\+2, and plumed.\+dat.\+3, with a different value for the keyword T\+E\+M\+P. Here how plumed.\+dat.\+3 should look like\+:

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate metadynamics in psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad.
# Well-tempered metadynamics is activated,
# and the biasfactor is set to 10.0
#
metad: METAD ARG=psi PACE=500 HEIGHT=1.2 SIGMA=0.35 FILE=HILLS BIASFACTOR=10.0 TEMP=1000.0

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The P\+T\+Meta\+D simulation is executed in the same way as the P\+T\+:

\begin{DoxyVerb} mpirun -np 4 mdrun_mpi -s TOPO/topol -plumed -multi 4 -replex 100
\end{DoxyVerb}


and it will produce one C\+O\+L\+V\+A\+R and H\+I\+L\+L\+S file per temperature (C\+O\+L\+V\+A\+R.\+0, H\+I\+L\+L\+S.\+0, ...). The analysis of the results requires what we have learned in the previous exercise for the P\+T case (analysis of the replica diffusion in temperature and demuxing of each replica trajectory), and the post-\/processing of a well-\/tempered metadynamics simulation (F\+E\+S calculation using \hyperlink{sum_hills}{sum\+\_\+hills} and convergence analysis).

Since in the previous tutorial we performed the same well-\/tempered metadynamics simulation without the use of P\+T (see \hyperlink{belfast-6_belfast-6-exercise-5}{Exercise 5. Setup and run a well-\/tempered metadynamics simulation, part I\+I}), here we can focus on the differences with the P\+T\+Meta\+D simulation. Let's compare the behavior of the biased variable psi in the two simulations\+:

\label{belfast-7_belfast-7-ptmetadh-fig}%
\hypertarget{belfast-7_belfast-7-ptmetadh-fig}{}%
 In well-\/tempered metadynamics (left panel), the biased variable psi looked stuck early in the simulation (t=3 ns). The reason was the transition of the other hidden degree of freedom phi from one free-\/energy basin to the other. In the P\+T\+Meta\+D case (right panel), this seems not to happen. To better appreciate the difference, we can plot the time evolution of the hidden degree of freedom phi in the two cases.

\label{belfast-7_belfast-7-ptmetadhidd-fig}%
\hypertarget{belfast-7_belfast-7-ptmetadhidd-fig}{}%
 Thanks to the excursions at high temperature, in the P\+T\+Meta\+D simulation the transition of the C\+V phi between the two basins is accelerate. As a result, the convergence of the reconstructed free energy in psi will be accelerated. This simple exercise demonstrates how P\+T\+Meta\+D can be used to cure a bad choice of metadynamics C\+Vs and the neglecting of slow degrees of freedom.\hypertarget{belfast-7_belfast-7-exercise-4}{}\subsubsection{Exercise 4. The Well-\/\+Tempered Ensemble}\label{belfast-7_belfast-7-exercise-4}
In this exercise we will learn how to run a P\+T-\/\+W\+T\+E and P\+T\+Meta\+D-\/\+W\+T\+E simulations of alanine dipeptide in water. We will start by running a short P\+T simulation using 4 replicas in the temperature range between 300\+K and 400\+K. We will use a geometric distribution of temperatures, which is valid under the assumption that the specific heat of the system is constant across temperatures. Replicas will thus be simulated at T=300, 330.\+2, 363.\+4, and 400\+K. In this simulation, we will just monitor the two dihedral angles and the total energy of the system, by preparing the following P\+L\+U\+M\+E\+D input file (plumed\+\_\+\+P\+T.\+dat)\+:

\begin{DoxyVerb}# set up three variables for Phi and Psi dihedral angles
# and total energy
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
ene: ENERGY

# monitor the three variables
PRINT STRIDE=10 ARG=phi,psi,ene FILE=COLVAR_PT\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{ENERGY}{E\+N\+E\+R\+G\+Y}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

As usual, the simulation is run for 400ps using the following command\+:

\begin{DoxyVerb} mpirun -np 4 mdrun_mpi -s TOPO/topol -plumed plumed_PT.dat -multi 4 -replex 100
\end{DoxyVerb}


At the end of the run, we want to analyze the acceptance rate between exchanges. This quantity is reported at the end of the Gromacs output file, typically called md.\+log, and it can be extracted using the following bash command line\+:

\begin{DoxyVerb}grep -A2 "Repl  average probabilities" md0.log
\end{DoxyVerb}


From the line above, we will find out that none of the attempted exchanges has been accepted. The reason is that the current setup (4 replicas to cover the temperature range 300-\/400\+K) resulted in a poor overlap of the energy distributions at different temperatures. We can easily realize this by plotting the time series of the total energy in the different replicas\+:

\label{belfast-7_belfast-7-ptalaw-fig}%
\hypertarget{belfast-7_belfast-7-ptalaw-fig}{}%
 To improve the overlap of the potential energy distributions at different temperatures, we enlarge the fluctuations of the energy by sampling the Well-\/\+Tempered Ensemble (W\+T\+E). In order to do that, we need to setup a well-\/tempered metadynamics simulation using energy as C\+V. In W\+T\+E, fluctuations -\/ the standard deviation of the energy time serie measured above -\/ will be enhanced by a factor equal to the square root of the biasfactor. In this exercise, we will enhance fluctuations of a factor of 4, thus we will set the biasfactor equal to 16. We need to prepare 4 P\+L\+U\+M\+E\+D input files (plumed\+\_\+\+P\+T\+W\+T\+E.\+dat.\+0, plumed\+\_\+\+P\+T\+W\+T\+E.\+dat.\+1,...), which will be identical to the following but for the temperature specified in the line of the \hyperlink{METAD}{M\+E\+T\+A\+D} directive\+:

\begin{DoxyVerb}# set up three variables for Phi and Psi dihedral angles
# and total energy
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
ene: ENERGY

# Activate metadynamics in ene
# depositing a Gaussian every 250 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 140 kJoule/mol.
# Well-tempered metadynamics is activated,
# and the biasfactor is set to 16.0
#
wte: METAD ARG=ene PACE=250 HEIGHT=1.2 SIGMA=140.0 FILE=HILLS_PTWTE BIASFACTOR=16.0 TEMP=300.0

# monitor the three variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,ene,wte.bias FILE=COLVAR_PTWTE\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{ENERGY}{E\+N\+E\+R\+G\+Y}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

Here, we use a Gaussian width larger than usual, and of the order of the fluctuations of the potential energy at 300\+K, as calculated from the preliminary P\+T run.

We run the simulation following the usual procedure\+:

\begin{DoxyVerb} mpirun -np 4 mdrun_mpi -s TOPO/topol -plumed plumed_PTWTE.dat -multi 4 -replex 100
\end{DoxyVerb}


If we analyze the average acceptance probability in this run\+:

\begin{DoxyVerb}grep -A2 "Repl  average probabilities" md0.log
\end{DoxyVerb}


we will notice that now on average 18\% of the exchanges are accepted. To monitor the diffusion of each replica in temperature, we can examine the file replica\+\_\+temp.\+xvg created by the following command line\+:

\begin{DoxyVerb}demux.pl md0.log
\end{DoxyVerb}


This analysis assures us that the system is efficiently diffusing in the entire temperature range and no bottlenecks are present.

\label{belfast-7_belfast-7-ptwtediff-fig}%
\hypertarget{belfast-7_belfast-7-ptwtediff-fig}{}%
 Finally, as done in the previous run, we can visualize the time serie of the energy C\+V at all temperatures\+:

\label{belfast-7_belfast-7-ptwteene-fig}%
\hypertarget{belfast-7_belfast-7-ptwteene-fig}{}%
 If we compare this plot with the one obtained in the P\+T run, we can notice that now the enlarged fluctuations caused by the use of W\+T\+E lead to a good overlap between energy distributions at different temperatures, thus increasing the exchange acceptance probability. At this point, we can extend our P\+T-\/\+W\+T\+E simulation and for example converge the free energy as a function of the dihedral angles phi and psi. Alternatively, we can accelerate sampling of the phi and psi dihedrals by combining P\+T-\/\+W\+T\+E with a metadynamics simulation using phi and psi as C\+Vs (P\+T\+Meta\+D-\/\+W\+T\+E \cite{ct300297t}). This can be achieved by preparing 4 P\+L\+U\+M\+E\+D input files (plumed\+\_\+\+P\+T\+Meta\+D\+W\+T\+E.\+dat.\+0, plumed\+\_\+\+P\+T\+Meta\+D\+W\+T\+E.\+dat.\+1,...), which will be identical to the following but for the temperature specified in the two lines containing the \hyperlink{METAD}{M\+E\+T\+A\+D} directives\+:

\begin{DoxyVerb}# reload WTE bias
RESTART

# set up three variables for Phi and Psi dihedral angles
# and total energy
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
ene: ENERGY

# Activate metadynamics in ene
# Old Gaussians will be reloaded to perform
# the second metadynamics run in WTE.
#
wte: METAD ARG=ene PACE=99999999 HEIGHT=1.2 SIGMA=140.0 FILE=HILLS_PTWTE BIASFACTOR=16.0 TEMP=300.0

# Activate metadynamics in phi and psi
# depositing a Gaussian every 500 time steps,
# with height equal to 1.2 kJoule/mol,
# and width 0.35 rad for both CVs.
# Well-tempered metadynamics is activated,
# and the biasfactor is set to 6.0
#
metad: METAD ARG=phi,psi PACE=500 HEIGHT=1.2 SIGMA=0.35,0.35 FILE=HILLS_PTMetaDWTE BIASFACTOR=6.0 TEMP=300.0

# monitor the three variables, the wte and metadynamics bias potentials
PRINT STRIDE=10 ARG=phi,psi,ene,wte.bias,metad.bias FILE=COLVAR_PTMetaDWTE\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{ENERGY}{E\+N\+E\+R\+G\+Y}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

These scripts activate two metadynamics simulations. One will use the energy as C\+V and will reload the Gaussians deposited during the preliminary P\+T-\/\+W\+T\+E run. No additional Gaussians on this variable will be deposited during the P\+T\+Meta\+D-\/\+W\+T\+E simulation, due to the large deposition stride. A second metadynamics simulation will be activated on the dihedral angles. Please note the different parameters and biasfactors in the two metadynamics runs.

The simulation is carried out using the usual procedure\+:

\begin{DoxyVerb} mpirun -np 4 mdrun_mpi -s TOPO/topol -plumed plumed_PTMetaDWTE.dat -multi 4 -replex 100
\end{DoxyVerb}
 \hypertarget{belfast-8}{}\section{Belfast tutorial\+: Replica exchange I\+I and Multiple walkers}\label{belfast-8}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
The aim of this tutorial is to introduce the users to the use of Bias-\/\+Exchange Metadynamics. We will go through the writing of the input files for B\+E\+M\+E\+T\+A for a simple case of three peptide and we will use M\+E\+T\+A\+G\+U\+I to to analyse them. We will compare the results of W\+T-\/\+B\+E\+M\+E\+T\+A and S\+T\+A\+N\+D\+A\+R\+D-\/\+B\+E\+M\+E\+T\+A with four independent runs on the four Collective Variables. Finally we will use a simplified version of B\+E\+M\+E\+T\+A that is Multiple Walkers Metadynamics.\hypertarget{belfast-8_belfast-8-lo}{}\subsection{Learning Outcomes}\label{belfast-8_belfast-8-lo}
Once this tutorial is completed students will\+:


\begin{DoxyItemize}
\item Know how to run a Bias-\/\+Exchange simulation using P\+L\+U\+M\+E\+D and G\+R\+O\+M\+A\+C\+S
\item Know how to analyse the results of B\+E\+M\+E\+T\+A with the help of M\+E\+T\+A\+G\+U\+I
\item Know how to run a Multiple Walker simulation
\end{DoxyItemize}\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
The \href{tutorial-resources/belfast-8.tgz}{\tt tarball } for this project contains the following files\+:


\begin{DoxyItemize}
\item system folder\+: a starting structure for Val-\/\+Ile-\/\+Leu system
\item W\+T\+B\+X\+: a run of Well-\/\+Tempered Bias-\/\+Exchange Metadynamics ready for the analysis
\end{DoxyItemize}\hypertarget{belfast-10_Instructions}{}\subsection{Instructions}\label{belfast-10_Instructions}
\hypertarget{belfast-8_bemeta}{}\subsubsection{Bias-\/\+Exchange Metadynamics}\label{belfast-8_bemeta}
In all variants of metadynamics the free-\/energy landscape of the system is reconstructed by gradually filling the local minima with gaussian hills. The dimensionality of the landscape is equal to the number of C\+Vs which are biased, and typically a number of C\+Vs smaller than three is employed. The reason for this is that qualitatively, if the C\+Vs are not correlated among them, the simulation time required to fill the free-\/energy landscape grows exponentially with the number of C\+Vs. This limitation can be severe when studying complex transformations or reactions in which more than say three relevant C\+Vs can be identified.

A possible technique to overcome this limitation is parallel-\/tempering metadynamics, \hyperlink{belfast-7}{Belfast tutorial\+: Replica exchange I}. A different solution is performing a bias-\/exchange simulation\+: in this approach a relatively large number N of C\+Vs is chosen to describe the possible transformations of the system (e.\+g., to study the conformations of a peptide one may consider all the dihedral angles between amino acids). Then, N metadynamics simulations (replicas) are run on the same system at the same temperature, biasing a different C\+V in each replica.

Normally, in these conditions, each bias profile would converge very slowly to the equilibrium free-\/energy, due to hysteresis. Instead, in the bias-\/exchange approach every fixed number of steps (say 10,000) an exchange is attempted between a randomly selected pair of replicas $ a $ and $ b $. The probability to accept the exchange is given by a Metropolis rule\+:

\[ \min\left( 1, \exp \left[ \beta ( V_G^a(x^a,t)+V_G^b(x^b,t)-V_G^a(x^b,t)-V_G^b(x^a,t) ) \right] \right) \]

where $ x^{a} $ and $ x^{b} $ are the coordinates of replicas $a $ and $ b $ and $ V_{G}^{a(b)}\left(x,t\right) $ is the metadynamics potential acting on the replica $ a $( $ b $). Each trajectory evolves through the high dimensional free energy landscape in the space of the C\+Vs sequentially biased by different metadynamics potentials acting on one C\+V at each time. The results of the simulation are N one-\/dimensional projections of the free energy.

In the following example, a bias-\/exchange simulation is performed on a V\+I\+L peptide (zwitterionic form, in vacuum with $ \epsilon=80 $, force field amber03), using the four backbone dihedral angles as C\+Vs.

Four replicas of the system are employed, each one biased on a different C\+V, thus four similar Plumed input files are prepared as follows\+:


\begin{DoxyItemize}
\item a common input file in which all the collective variables are defined\+:
\end{DoxyItemize}

\begin{DoxyVerb}MOLINFO STRUCTURE=VIL.pdb
RANDOM_EXCHANGES

cv1: TORSION ATOMS=@psi-1
cv2: TORSION ATOMS=@phi-2
cv3: TORSION ATOMS=@psi-2
cv4: TORSION ATOMS=@phi-3
\end{DoxyVerb}


N\+O\+T\+E\+:
\begin{DoxyEnumerate}
\item By using \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} we can use shortcut to select atoms for dihedral angles (currently @phi, @psi, @omega and @chi1 are available).
\item We use cv\# as labels in order to make the output compatible with M\+E\+T\+A\+G\+U\+I.
\item \hyperlink{RANDOM_EXCHANGES}{R\+A\+N\+D\+O\+M\+\_\+\+E\+X\+C\+H\+A\+N\+G\+E\+S} generates random exchanges list that are sent back to G\+R\+O\+M\+A\+C\+S.
\end{DoxyEnumerate}
\begin{DoxyItemize}
\item four additional input files that \hyperlink{INCLUDE}{I\+N\+C\+L\+U\+D\+E} the common input and define the four \hyperlink{METAD}{M\+E\+T\+A\+D} along the four C\+Vs, respectively.
\end{DoxyItemize}

\begin{DoxyVerb}INCLUDE FILE=plumed-common.dat
be: METAD ARG=cv1 HEIGHT=0.2 SIGMA=0.2 PACE=100 GRID_MIN=-pi GRID_MAX=pi GRID_BIN=200 
PRINT ARG=cv1,cv2,cv3,cv4 STRIDE=1000 FILE=COLVAR
\end{DoxyVerb}


N\+O\+T\+E\+:
\begin{DoxyEnumerate}
\item in C\+O\+L\+V\+A\+R we \hyperlink{PRINT}{P\+R\+I\+N\+T} only the four collective variables, always in the same order in such a way that C\+O\+L\+V\+A\+Rs are compatible with M\+E\+T\+A\+G\+U\+I
\item if you want to print additional information, like the \hyperlink{METAD}{M\+E\+T\+A\+D} bias it is possibile to use additional \hyperlink{PRINT}{P\+R\+I\+N\+T} keyword
\end{DoxyEnumerate}

\begin{DoxyVerb}PRINT ARG=cv1,be.bias STRIDE=xxx FILE=BIAS
\end{DoxyVerb}


The four replicas start from the same G\+R\+O\+M\+A\+C\+S topology file replicated four times\+: topol0.\+tpr, topol1.\+tpr, topol2.\+tpr, topol3.\+tpr. Finally, G\+R\+O\+M\+A\+C\+S is launched as a parallel run on 4 cores, with one replica per core, with the command

\begin{DoxyVerb}mpirun -np 4 mdrun_mpi -s topol -plumed plumed -multi 4 -replex 2000 >& log &
\end{DoxyVerb}


where -\/replex 2000 indicates that every 2000 molecular-\/dynamics steps all replicas are randomly paired (e.\+g. 0-\/2 and 1-\/3) and exchanges are attempted between each pair (as printed in the G\+R\+O\+M\+A\+C\+S $\ast$.log files).

The same simulation can be run using W\+E\+L\+L\+T\+E\+M\+P\+E\+R\+E\+D metadynamics.\hypertarget{belfast-8_bxcon}{}\subsubsection{Convergence of the Simulations}\label{belfast-8_bxcon}
In the resources for this tutorial you can find the results for a 40ns long Well-\/\+Tempered Bias Exchange simulation. First of all we can try to assess the convergence of the simulations by looking at the profiles. In the \char`\"{}convergence\char`\"{} folder there is a script that calculates the free energy from the H\+I\+L\+L\+S.\+0 file at incresing simulation lengths (i.\+e. every more 0.\+8 ns of simulation). The scripts also generate two measures of the evolution of the profiles in time\+:


\begin{DoxyEnumerate}
\item time-\/diff.\+H\+I\+L\+L\+S.\+0\+: where it is stored the average deviation between two successive profiles
\item K\+L.\+H\+I\+L\+L\+S.\+0\+: where it is stored the average deviation between profiles correctly weigheted for the free energy of the profiles themselves (Symmetrized Kullback-\/\+Lieber divergence)
\end{DoxyEnumerate}

From both plots one can deduce that after 8 ns the profiles don't change significantly thus suggesting that averaging over the range 8-\/40ns should result in a accurate profile (we will test this using metagui). Another test is that of looking at the fluctuations of the profiles in a time window instead of looking at successive profiles\+:

\label{belfast-8_belfast-8-convergence-fig}%
\hypertarget{belfast-8_belfast-8-convergence-fig}{}%
\hypertarget{belfast-8_metagui}{}\subsubsection{Bias-\/\+Exchange Analysis with M\+E\+T\+A\+G\+U\+I}\label{belfast-8_metagui}
In principle Bias-\/\+Exchange Metadynamics can give as a results only N 1\+D free energy profiles. But the information contained in all the replicas can be used to recover multidensional free energy surfaces in $>$=N dimensions. A simple way to perform this analysis is to use M\+E\+T\+A\+G\+U\+I. M\+E\+T\+A\+G\+U\+I performs the following operations\+:


\begin{DoxyEnumerate}
\item Clusterizes the trajectories on a multidimensional G\+R\+I\+D defined by at least the biased coordinates.
\item By using the 1\+D free energy profiles and the clustering assigns a free energy to the cluster using a W\+H\+A\+M procedure.
\item Lets the user visualize the clusters.
\item Approximates the kinetics among clusters.
\end{DoxyEnumerate}

M\+E\+T\+A\+G\+U\+I (Biarnes et. al) is a plugin for V\+M\+D that implements the approch developed by Marinelli et. al 2009. It can be downloaded from the P\+L\+U\+M\+E\+D website.

In order for the colvar and hills file to be compatible with M\+E\+T\+A\+G\+U\+I their header must be formatted as following\+:

C\+O\+L\+V\+A\+R.\#\+: \begin{DoxyVerb}#! FIELDS time cv1 cv2 cv3 cv4
#! ACTIVE 1 1 A
#! ..
...
\end{DoxyVerb}


N\+O\+T\+E\+:
\begin{DoxyEnumerate}
\item the C\+O\+L\+V\+A\+R.\# files should contain A\+L\+L the collective variables employed (all those biased in at least one replica plus those additionaly analysed). They M\+U\+S\+T be named cv1 ... cv\+N.
\item the C\+O\+L\+V\+A\+R.\# files must be synchronised with the trajectories, this means that for each frame in the trajectory at time t there must be a line in each colvar at time t and viceversa. The best option is usually to analyse the trajectories a posteriori using plumed driver.
\item a keyword \#! A\+C\+T\+I\+V\+E N\+B\+I\+A\+S\+E\+D\+C\+V B\+I\+A\+S\+E\+D\+C\+V L\+A\+B\+E\+L is needed, where N\+B\+I\+A\+S\+E\+D\+C\+V is the number of biased cv in that replica (not overall), B\+I\+A\+S\+E\+D\+C\+V is the index of the biased cv in that replica (i.\+e. 1 for the first replica and so on); L\+A\+B\+E\+L is a letter that identify the replica (usually is simply A for the first, B for the second and so on) this is usufull if two replicas are biasing the same collective variable\+:
\end{DoxyEnumerate}

\begin{DoxyVerb}COLVAR.0:
#! FIELDS time cv1 cv2 cv3
#! ACTIVE 1 1 A
#! ..
...
COLVAR.1:
#! FIELDS time cv1 cv2 cv3
#! ACTIVE 1 2 B
#! ..
...
COLVAR.2:
#! FIELDS time cv1 cv2 cv3
#! ACTIVE 1 2 C
#! ..
...
COLVAR.3:
#! FIELDS time cv1 cv2 cv3
#! ACTIVE 0 
#! ..
...\end{DoxyVerb}


In the above case Replica 0 biases cv1; replicas 1 and 2 biases cv2 while replica 3 is a neutral (unbiased) replica. cv3 is unbiased in all the replicas.

The A\+C\+T\+I\+V\+E keyword must be the F\+I\+R\+S\+T L\+I\+N\+E in the H\+I\+L\+L\+S.\# files\+:

H\+I\+L\+L\+S.\#\+: \begin{DoxyVerb}#! ACTIVE 1 1 A
#! FIELDS time cv1 sigma_cv1 height biasf
#! ..
...
\end{DoxyVerb}


The above notes hold for the H\+I\+L\+L\+S files as well. In the folder metagui the script check\+\_\+for\+\_\+metagui.\+sh checks if the header of your file is compatible with M\+E\+T\+A\+G\+U\+I, but remember that this is not enough! Synchronisation of C\+O\+L\+V\+A\+R and trajectory files is also needed. H\+I\+L\+L\+S files can be written with a different frequency but times must be consistent.

N\+O\+T\+E\+: It is important to copy H\+I\+L\+L\+S files in the metagui folder.

\begin{DoxyVerb}./check_for_metagui.sh ../COLVAR.0
\end{DoxyVerb}


will tell you that the A\+C\+T\+I\+V\+E keyword is missing, you need to modify all the header B\+E\+F\+O\+R\+E proceeding with the tutorial!!

In the metagui folder there is a metagui.\+input file\+:

\begin{DoxyVerb}WHAM_EXE        wham_bemeta.x
BASINS_EXE      kinetic_basins.x
KT 2.4900
HILLS_FILE   HILLS.0  
HILLS_FILE   HILLS.1  
HILLS_FILE   HILLS.2  
HILLS_FILE   HILLS.3  
GRO_FILE     VIL.pdb 
COLVAR_FILE COLVAR.0 ../traj0.xtc "psi-1"
COLVAR_FILE COLVAR.1 ../traj1.xtc "phi-2"
COLVAR_FILE COLVAR.2 ../traj2.xtc "psi-2"
COLVAR_FILE COLVAR.3 ../traj3.xtc "phi-3"
TRAJ_SKIP 10
CVGRID 1  -3.1415 3.1415 15 PERIODIC
CVGRID 2  -3.1415 3.1415 15 PERIODIC
CVGRID 3  -3.1415 3.1415 15 PERIODIC
CVGRID 4  -3.1415 3.1415 15 PERIODIC
ACTIVE 4 1 2 3 4  
T_CLUSTER 0.
T_FILL    8000.
DELTA 4
GCORR 1
TR_N_EXP 5
\end{DoxyVerb}


where are defined the temperature in energy units, the place where to find C\+O\+L\+V\+A\+R, H\+I\+L\+L\+S and trajectory files. A reference gro or pdb file is needed to load the trajectories. The definition of the ranges and the number of bins for the available collective variables.

Now let's start with the analysis\+:


\begin{DoxyEnumerate}
\item run V\+M\+D and load metagui
\item in metagui load the metagui.\+input file \hyperlink{belfast-8_belfast-8-mg1-fig}{belfast-\/8-\/mg1-\/fig}
\item In the left section of the interface \char`\"{}load all\char`\"{} the trajectories
\item Find the Microstates
\end{DoxyEnumerate}

In order to visualise the microstate it is convenient to align all the structures using the V\+M\+D R\+M\+S\+D Trajectory tool that can be found in Extensions-\/$>$Analysis.

One or more microstates can be visualised by selecting them and clicking show.

You can sort the microstates using the column name tabs, for example by clicking on size the microstates will be ordered from the larger to the smaller. If you look at the largest one it is possible to observe that by using the four selected collective variables the backbone conformation of the peptide is well defined while the sidechains can populate different rotameric states.

The equilibrium time in the analysis panel should be such that by averaging over the two halves of the remind of the simulation the profiles are the same (i.\+e the profile averaged between Teq and Teq+(Ttot-\/\+Teq)/2 should be the same of that averaged from Teq+(Ttot-\/\+Teq)/2 and Ttot). By clicking on C\+O\+M\+P\+U\+T\+E F\+R\+E\+E E\+N\+E\+R\+G\+I\+E\+S, the program will first generate the 1\+D free energy profiles from the H\+I\+L\+L\+S files and then run the W\+H\+A\+M analysis on the microstates. Once the analysis is done it is possible to visually check the convergence of the 1\+D profiles one by one by clicking on the K bottons next to the H\+I\+L\+L\+S.\# files. The B\+L\+U\+E and the R\+E\+D profiles are the two profiles just defined, while the G\+R\+E\+E\+N is the average of the two. Now it is possible for example to sort the microstates as a function of the free energy and save them by dumping the structures for further analysis.

\label{belfast-8_belfast-8-mg1-fig}%
\hypertarget{belfast-8_belfast-8-mg1-fig}{}%


If you look in the metagui folder you will see a lot of files, some of them can be very usefull\+:

metagui/\+M\+I\+C\+R\+O\+S\+T\+A\+T\+E\+S\+: is the content of the microstates list table metagui/\+W\+H\+A\+M\+\_\+\+R\+U\+N/\+V\+G\+\_\+\+H\+I\+L\+L\+S.\#\+: are the opposite of the free energies calculated from the hills files metagui/\+W\+H\+A\+M\+\_\+\+R\+U\+N/$\ast$.gnp\+: are gnuplot input files to plot the V\+G\+\_\+\+H\+I\+L\+L\+S.\# files (i.\+e. gnuplot -\/$>$ load \char`\"{}convergence..\char`\"{}) metagui/\+W\+H\+A\+M\+\_\+\+R\+U\+N/\+F\+E\+S\+: is the result of the W\+H\+A\+M, for each cluster there is its free energy and the error estimate from W\+H\+A\+M

\begin{DoxyVerb}gnuplot> plot [0:40]'FES' u 2:3
\end{DoxyVerb}


plots the microstate error in the free energy estimate as a function of the microstates free energy. Finally in the folder metagui/\+F\+E\+S there is script to integrate the multidimensional free energy contained in the M\+I\+C\+R\+O\+S\+T\+A\+T\+E\+S files to a 2\+D F\+E\+S as a function of two of the used C\+V. To use it is enough to copy the M\+I\+C\+R\+O\+S\+T\+A\+T\+E\+S file in F\+E\+S\+:

\begin{DoxyVerb}cp MICROSTATES FES/FES.4D
\end{DoxyVerb}


and edit the script to select the two columns of M\+I\+C\+R\+O\+S\+T\+A\+T\+E\+S on which show the integrated F\+E\+S.\hypertarget{belfast-8_mw}{}\subsubsection{Multiple Walker Metadynamics}\label{belfast-8_mw}
Multiple Walker metadynamics is the simplest way to parallelise a Meta\+D calculation\+: multiple simulation of the same system are run in parallel using metadynamics on the same set of collective variables. The deposited bias is shared among the replicas in such a way that the history dependent potential depends on the whole history.

We can use the same common input file defined above and then we can define four metadynamics bias in a similar way of what was done above for bias-\/exchange but now all the biases are defined on the same collective variables\+:

\begin{DoxyVerb}plumed.dat.#
INCLUDE FILE=plumed-common.dat

METAD ...
LABEL=mw 
ARG=cv2,cv3 
SIGMA=0.3,0.3 
HEIGHT=0.2 
PACE=100
BIASFACTOR=8
TEMP=300 
GRID_MIN=-pi,-pi 
GRID_MAX=pi,pi 
GRID_BIN=200,200
WALKERS_MPI
... METAD

PRINT ARG=cv1,cv2,cv3,cv4 STRIDE=1000 FILE=COLVAR
\end{DoxyVerb}


and the simulation can be run in a similar way without doing exchanges\+:

\begin{DoxyVerb}mpirun -np 4 mdrun_mpi -s topol -plumed plumed -multi 4  >& log &
\end{DoxyVerb}


alternatively Multiple Walkers can be run as independent simulations sharing via the file system the biasing potential, this is usefull because it provides a parallelisation that does not need a parallel code. In this case the walkers read with a given frequency the gaussians deposited by the others and add them to their own \hyperlink{METAD}{M\+E\+T\+A\+D}.\hypertarget{belfast-9_refer}{}\subsection{Reference}\label{belfast-9_refer}
This tutorial is freely inspired to the work of Biarnes et al.

More materials can be found in


\begin{DoxyEnumerate}
\item Marinelli, F., Pietrucci, F., Laio, A. \& Piana, S. A kinetic model of trp-\/cage folding from multiple biased molecular dynamics simulations. P\+Lo\+S Comput. Biol. 5, e1000452 (2009).
\item Biarnés, X., Pietrucci, F., Marinelli, F. \& Laio, A. M\+E\+T\+A\+G\+U\+I. A V\+M\+D interface for analyzing metadynamics and molecular dynamics simulations. Comput. Phys. Commun. 183, 203–211 (2012).
\item Baftizadeh, F., Cossio, P., Pietrucci, F. \& Laio, A. Protein folding and ligand-\/enzyme binding from bias-\/exchange metadynamics simulations. Current Physical Chemistry 2, 79–91 (2012).
\item Granata, D., Camilloni, C., Vendruscolo, M. \& Laio, A. Characterization of the free-\/energy landscapes of proteins by N\+M\+R-\/guided metadynamics. Proc. Natl. Acad. Sci. U.\+S.\+A. 110, 6817–6822 (2013).
\item Raiteri, P., Laio, A., Gervasio, F. L., Micheletti, C. \& Parrinello, M. Efficient reconstruction of complex free energy landscapes by multiple walkers metadynamics. J. Phys. Chem. B 110, 3533–3539 (2006). 
\end{DoxyEnumerate}\hypertarget{belfast-9}{}\section{Belfast tutorial\+: N\+M\+R constraints}\label{belfast-9}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
This tutorial is about the use of experimental data, in particular N\+M\+R data, either as collective variables or as replica-\/averaged restraints in M\+D simulations. While the first is a just a simple extension of what we have been already doing in previous tutorials, the latter is an approach that can be used to increase the quality of a force-\/field in describing the properties of a specific system.\hypertarget{belfast-9_belfast-9-lo}{}\subsection{Learning Outcomes}\label{belfast-9_belfast-9-lo}
Once this tutorial is completed students will\+:
\begin{DoxyItemize}
\item know why and how to use experimental data to define a collective variable
\item know why and how to use experimental data as replica-\/averaged restraints in M\+D simulations
\end{DoxyItemize}\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
The \href{tutorial-resources/belfast-9a.tar.gz}{\tt tarball } for this project contains the following\+:
\begin{DoxyItemize}
\item system\+: the files use to generate the topol?.tpr files of the first and second example
\item first\+: an example on the use of chemical shifts as a collective variable
\item second\+: an example on the use of chemical shifts as replica-\/averaged restraints
\item third\+: an example on the use of R\+D\+Cs (calculated with the theta-\/method) as replica-\/averaged restrains
\end{DoxyItemize}\hypertarget{belfast-10_Instructions}{}\subsection{Instructions}\label{belfast-10_Instructions}
\hypertarget{belfast-9_expdata}{}\subsubsection{Experimental data as Collective Variables}\label{belfast-9_expdata}
In the former tutorials it has been often discussed the possibility of measuring a distance with respect to a structure representing some kind of state for a system, i.\+e. \hyperlink{belfast-5}{Belfast tutorial\+: Out of equilibrium dynamics}. An alternative possibility is to use as a reference a set of experimental data that represent a state and measure the current deviation from the set. In plumed there are currently implemented the following N\+M\+R experimental observables\+: Chemical Shifts (only for proteins) \hyperlink{CS2BACKBONE}{C\+S2\+B\+A\+C\+K\+B\+O\+N\+E} and \hyperlink{CH3SHIFTS}{C\+H3\+S\+H\+I\+F\+T\+S}, \hyperlink{NOE}{N\+O\+E} distances and Residual Dipolar couplings \hyperlink{RDC}{R\+D\+C}. In addition \hyperlink{NOE}{N\+O\+E} collective variable can be also used for P\+R\+E distances and 3\+J Couplings can be implemented using \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} and \hyperlink{MATHEVAL}{M\+A\+T\+H\+E\+V\+A\+L}. Among the above listed collective variables those based on chemical shifts make use of an external library, A\+L\+M\+O\+S\+T, that must be downloaded and compiled separately. In addition plumed must be configured in such a way to link A\+L\+M\+O\+S\+T. Detailed instructions on how to compile P\+L\+U\+M\+E\+D with A\+L\+M\+O\+S\+T can be found in \hyperlink{CS2BACKBONE}{C\+S2\+B\+A\+C\+K\+B\+O\+N\+E}.

In the following we will write the C\+S2\+B\+A\+C\+K\+B\+O\+N\+E collective variable that has been used in Gratana et al. (2013).

\begin{DoxyVerb}prot: GROUP ATOMS=1-862
WHOLEMOLECULES ENTITY0=prot

cs: CS2BACKBONE ATOMS=prot DATA=data FF=a03_gromacs.mdb NRES=56 FLAT=1.0 WRITE_CS=50 

PRINT ARG=cs FILE=COLVAR STRIDE=100

ENDPLUMED
\end{DoxyVerb}


In this case the chemical shifts are those measured for the native state of the protein and can be used, together with other C\+Vs and Bias-\/\+Exchange Metadynanics, to guide the system back and forth from the native structure. The experimental chemical shifts are in six files inside the \char`\"{}data/\char`\"{} folder (see first example in the resources tarball), one file for each nucleus. A 0 chemical shift is used where a chemical shift doesn't exist (i.\+e. C\+B of G\+L\+Y) or where it has not been assigned. Additionally the data folder contains\+:


\begin{DoxyItemize}
\item camshift.\+db\+: this file is a parameter file for camshift, it is a standard file needed to calculate the chemical shifts from a structure
\item a03\+\_\+gromacs.\+mdb\+: this is a Amber force field in A\+L\+M\+O\+S\+T format and it is used to map the atom names from plumed and almost (in this case we are using amber for our simulation)
\item template.\+pdb\+: this is a pdb file for the protein we are simulating (i.\+e. editconf -\/f conf.\+gro -\/o template.\+pdb) where atoms are ordered in the same way in which are included in the main code and again it is used to map the atom in plumed with those in almost.
\end{DoxyItemize}

This example can be executed as

\begin{DoxyVerb}mdrun_mpi -s topol -plumed plumed
\end{DoxyVerb}
\hypertarget{belfast-9_replica}{}\subsubsection{Replica-\/\+Averaged Restrained Simulations}\label{belfast-9_replica}
N\+M\+R data, as all the equilibrium experimental data, are the result of a measure over an ensemble of structures and over time. In principle a \char`\"{}perfect\char`\"{} molecular dynamics simulations, that is a simulations with a perfect force-\/field and a perfect sampling can predict the outcome of an experiments in a quantitative way. Actually in most of the cases obtaining a qualitative agreement is already a fortunate outcome. In order to increase the accuracy of a force field in a system dependent manner it is possible to add to the force-\/field an additional term based on the agreement with a set of experimental data. This agreement is not enforced as a simple restraint because this would mean to ask the system to be always in agreement with all the experimental data at the same time, instead the restraint is applied over an A\+V\+E\+R\+A\+G\+E\+D C\+O\+L\+L\+E\+C\+T\+I\+V\+E V\+A\+R\+I\+A\+B\+L\+E where the average is performed over multiple identical simulations. In this way the is not a single replica that must be in agreement with the experimental data but they should be in agreement on average. It has been shown that this approach is equivalent in solving the problem of finding a modified version of the force field that will reproduce the provided set of experimental data withouth any additional assumption on the data themselves.

Currently E\+N\+S\+E\+M\+B\+L\+E A\+V\+E\+R\+A\+G\+I\+N\+G of a collective variable can be performed only using the N\+M\+R variables (\hyperlink{CS2BACKBONE}{C\+S2\+B\+A\+C\+K\+B\+O\+N\+E}, \hyperlink{CH3SHIFTS}{C\+H3\+S\+H\+I\+F\+T\+S}, \hyperlink{NOE}{N\+O\+E} and \hyperlink{RDC}{R\+D\+C}).

The second example included in the resources show how the amber force field can be improved in the case of protein domain G\+B3 using the native state chemical shifts a replica-\/averaged restraint. By the fact that replica-\/averaging needs the use of multiple replica simulated in parallel in the same conditions it is easily complemented with B\+I\+A\+S-\/\+E\+X\+C\+H\+A\+N\+G\+E or M\+U\+L\+T\+I\+P\+L\+E W\+A\+L\+K\+E\+R metadynamics to enhance the sampling.

\begin{DoxyVerb}prot: GROUP ATOMS=1-862
WHOLEMOLECULES ENTITY0=prot

cs: CS2BACKBONE ATOMS=prot DATA=data FF=a03_gromacs.mdb NRES=56 FLAT=0.0 WRITE_CS=500 ENSEMBLE

cse: RESTRAINT ARG=cs AT=0. KAPPA=0. SLOPE=24

PRINT ARG=cs FILE=COLVAR STRIDE=10

ENDPLUMED
\end{DoxyVerb}


with respect to the case in which chemical shifts are used to define a standard collective variable, in this case the keyword E\+N\+S\+E\+M\+B\+L\+E tells plumed to calculate all the chemical shifts from the replicas (i.\+e. 4 replicas) average them and only after the averaging calculate the difference with respect to the experimental ones. On this difference that is the A\+V\+E\+R\+A\+G\+E\+D Collective Variable it is possible to apply a linear \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T} (because the variable is already a sum of squared differences) that is the new term we are adding to the underlying force field.

This example can be executed as \begin{DoxyVerb}mpiexec -np 4 mdrun_mpi -s topol -plumed plumed -multi 4
\end{DoxyVerb}


The third example show how \hyperlink{RDC}{R\+D\+C} (calculated with the theta-\/methods) can be employed in the same way, in this case to describe the native state of Ubiquitin. In particular it is possible to observe how the R\+D\+C averaged restraint applied on the correlation between the calculated and experimental N-\/\+H and C\+A-\/\+H\+A R\+D\+Cs result in the increase of the correlation of the R\+D\+Cs for other bonds already on a very short time scale.

\begin{DoxyVerb}RDC ...
ENSEMBLE
CORRELATION
GYROM=-72.5388
SCALE=0.001060 
ATOMS1=20,21 COUPLING1=8.17
ATOMS2=37,38 COUPLING2=-8.271
ATOMS3=56,57 COUPLING3=-10.489
ATOMS4=76,77 COUPLING4=-9.871
#continue....
\end{DoxyVerb}


In this input the first four N-\/\+H R\+D\+Cs are defined.

This example can be executed as \begin{DoxyVerb}mpiexec -np 8 mdrun_mpi -s topol -plumed plumed -multi 8
\end{DoxyVerb}
\hypertarget{belfast-9_refer}{}\subsection{Reference}\label{belfast-9_refer}

\begin{DoxyEnumerate}
\item Granata, D., Camilloni, C., Vendruscolo, M. \& Laio, A. Characterization of the free-\/energy landscapes of proteins by N\+M\+R-\/guided metadynamics. Proc. Natl. Acad. Sci. U.\+S.\+A. 110, 6817–6822 (2013).
\item Cavalli, A., Camilloni, C. \& Vendruscolo, M. Molecular dynamics simulations with replica-\/averaged structural restraints generate structural ensembles according to the maximum entropy principle. J. Chem. Phys. 138, 094112 (2013).
\item Camilloni, C., Cavalli, A. \& Vendruscolo, M. Replica-\/\+Averaged Metadynamics. Journal of Chemical Theory … 9, 5610–5617 (2013).
\item Roux, B. \& Weare, J. On the statistical equivalence of restrained-\/ensemble simulations with the maximum entropy method. J. Chem. Phys. 138, 084107 (2013).
\item Boomsma, W., Lindorff-\/\+Larsen, K. \& Ferkinghoff-\/\+Borg, J. Combining Experiments and Simulations Using the Maximum Entropy Principle. P\+Lo\+S Comput. Biol. 10, e1003406 (2014).
\item Camilloni, C. \& Vendruscolo M. A Tensor-\/\+Free Method for the Structural and Dynamical Refinement of Proteins using Residual Dipolar Couplings. J. P\+H\+Y\+S. C\+H\+E\+M. B X\+X\+X (2014). 
\end{DoxyEnumerate}\hypertarget{belfast-10}{}\section{Belfast tutorial\+: Steinhardt Parameters}\label{belfast-10}
\hypertarget{belfast-10_Aims}{}\subsection{Aims}\label{belfast-10_Aims}
This tutorial is concerned with the collective variables that we use to study order/disorder transitions such as the transition between the solid and liquid phase. In this tutorial we will look at the transition from solid to liquid as this is easier to study using simulation. The C\+Vs we will intorduce can, and have, been used to study the transition from liquid to solid. More information can be found in the further reading section.\hypertarget{belfast-10_belfast-10-lo}{}\subsection{Learning Outcomes}\label{belfast-10_belfast-10-lo}
Once this tutorial is completed students will\+:


\begin{DoxyItemize}
\item Know of the existence of the Steinhardt Parameters and know how to create plumed input files for calculating them.
\item Appreciate that the Steinhardt Parameter for a particular atom is a vector quantity and that you can thus measure local order in a material by taking dot products of the Steinhardt Parameter vectors of adjacent atoms. Students will know how to calculate such quantities using plumed.
\end{DoxyItemize}\hypertarget{belfast-10_Resources}{}\subsection{Resources}\label{belfast-10_Resources}
The \href{tutorial-resources/belfast-9b.tar.gz}{\tt tarball } for this project contains the following files\+:


\begin{DoxyItemize}
\item in \+: An input file for the simplemd code that forms part of plumed
\item input.\+xyz \+: A configuration file for Lennard-\/\+Jones solid with an fcc solid structure
\end{DoxyItemize}\hypertarget{belfast-10_Instructions}{}\subsection{Instructions}\label{belfast-10_Instructions}
\hypertarget{belfast-10_Simplemd}{}\subsubsection{Simplemd}\label{belfast-10_Simplemd}
For this tutorial we will be using the M\+D code that is part of plumed -\/ simplemd. This code allows us to do the simulations of \href{http://en.wikipedia.org/wiki/Lennard-Jones_potential}{\tt Lennard-\/\+Jones } that we require here but not much else. We will thus start this tutorial by doing some simulations with this code. You should have two files from the tarball, the first is called input.\+xyz and is basically an xyz file containing the posisitions of the atoms. The second meanwhile is called in and is the input to simplemd. If you open the file the contents should look something like this\+:

\begin{DoxyVerb}inputfile input.xyz
outputfile output.xyz
temperature 0.2
tstep 0.005
friction 1
forcecutoff 2.5
listcutoff  3.0
nstep 50000
nconfig 100 trajectory.xyz
nstat   10 energies.dat
\end{DoxyVerb}


Having run some molecular dynamics simulations in the past it should be pretty obvious what each line of the file does. One thing that might be a little strange is the units. Simplemd works with Lennard-\/\+Jones units so energy is in units of $\epsilon$ and length is in units of $\sigma$. This means that temperature is in units of $\frac{k_B T}{\epsilon}$, which is why the temperature in the above file is apparently so low.

Use simplemd to run 50000 step calculations at 0.\+2, 0.\+8 and 1.\+2 $\frac{k_B T}{\epsilon}$. (N.\+B. You will need an empty file called plumed.\+dat in order to run these calculations.) Visualise the trajectory output by each of your simulations using V\+M\+D. Please be aware that simplemd does not wrap the atoms into the cell box automatically. If you are at the tutorial we have resolved this problem by making it so that if you press w when you load all the atoms they will be wrapped into the box. At what temperatures did the simulations melt? What then is the melting point of the Lennard-\/\+Jones potential at this density?\hypertarget{belfast-10_cvs}{}\subsubsection{Finding collective variables}\label{belfast-10_cvs}
At the end of the previous section you were able to make very sophisticated judegements about whether or not the arrangment of atoms in your system was solid-\/like or liquid-\/like by simply looking at the configuration. The aim in the rest of this tutorial is to see if we can derive collective variables that are able to make an equally sophisticated judgement. For our first attempt lets use a C\+V which we have encoutered elsewhere, the \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R}.

Create a plumed input file that calculates the average \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R} of the atoms in your system and outputs it to a file every 100 steps. You will need to use the \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R} and \hyperlink{PRINT}{P\+R\+I\+N\+T} actions. The switching function that tells plumed whether or not two atoms are bonded should be of type R\+A\+T\+I\+O\+N\+A\+L and should have its $d_0$ parameter equal to 1.\+3 its $r_0$ parameter equal to 0.\+2 and its $n$ and $m$ parameters set equal to 6 and 12 repsectively.

Rerun the simpled simulations described at the end of the previous section. Is the average coordination number good at differentiaing between solid and liquid configurations? Given your knowledge of physics/chemistry is this result surprising?\hypertarget{belfast-10_steinhardt}{}\subsubsection{Steinhard parameter}\label{belfast-10_steinhardt}
The solid and liquid phases of a material are both relatively dense so the result at the end of the last section -\/ the fact that the coordination number is not particularly good at differentiating between them -\/ should not be that much of a surprise. As you will have learnt early on in your scientific education when solids melt the atoms rearrange themselves in a much less ordered fashion. The bonds between them do not necessarily break it is just that, whereas in a the solid the bonds were all at pretty well defined angles to each other, in a liquid the spread of bond angles is considerably larger. To detect the transition from solid to liquid what we need then is a coordinate that is able to recognise whether or not the geometry in the coordination spheres around each of the atoms in the system are the same or different. If these geometries are the same the system is in a solid-\/like configuration, whereas if they are different the system is liquid-\/like. The Steinhardt parameters \hyperlink{Q3}{Q3}, \hyperlink{Q4}{Q4} and \hyperlink{Q6}{Q6} are coordinates that allow us to examine the coordination spheres of atoms in precisely this way. The way in which these coordinates are able to do this is explained in the \href{https://www.youtube.com/watch?v=ou0uKgK35lE}{\tt video }.

Repeat the calculations from the end of the previous section but edit the plumed.\+dat file so that the average \hyperlink{Q6}{Q6} parameter is calculated and printed as well as the average \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R}. In the Steinhardt parameter implementation in plumed the set of atoms in the coordination sphere of a particular atom are defined using a continuous switching function. For this calculation you should use a R\+A\+T\+I\+O\+N\+A\+L switching funciton with parameters $d_0$ equal to 1.\+3, $r_0$ equal to 0.\+2 and $n$ and $m$ set equal to 6 and 12 repsectively. Is the average Q6 parameter able to differentiate between the solid and liquid states?\hypertarget{belfast-10_lvsg}{}\subsubsection{Local versus Global}\label{belfast-10_lvsg}
At the end of the previous section you showed that the average Q6 parameter for a system of atoms is able to tell you whether or not that collection of atoms is in a solid-\/like or liquid-\/like configuration. In this section we will now ask the question -\/ can the Steinhardt parameter always, unambiously tell us whether particular tagged atoms are in a solid-\/like region of the material or whether they are in a liquid-\/like region of the material? This is an important question that might come up if we are looking at nucleation of a solid from the melt. Our question in this context reads -\/ how do we unambigously identify those atoms that are in the crystalline nucleus? A similar question would also come up when studying an interface between the solid and liquid phases. In this guise we would be asking about the extent of interface; namely, how far from the interface do we have to go before we can start thinking of the atoms as just another atom in the solid/liquid phase?

With these questions in mind our aim is to look at the distribution of values for the Q6 parameters of atoms in our system of Lennard-\/\+Jones have. If Q6 is able to unambigously tell us whether or not an atom is in a solid-\/like pose or a liquid-\/like pose then there should be no overlap in the distributions obtained for the solid and liquid phases. If there is overlap, however, then we cannot use these coordinates for the applications described in the previous paragraph. To calculate these distributions you will need to run two simulations -\/ one at 0.\+2 $\frac{k_B T}{\epsilon}$ and one at 1.\+2 $\frac{k_Bt}{\epsilon}$. For these simulation you will need plumed.\+dat files that calculate and output the distribution of Steinhardt parameters. In writing the plumed input for these calcualtions you will need to use the \hyperlink{PRINT}{P\+R\+I\+N\+T} command and the \hyperlink{Q6}{Q6} command. In your Q6 instructions you will need to use the H\+I\+S\+T\+O\+G\+R\+A\+M keyword -\/ my recommendation would be to calculate a histogram consisting of 20 bins over a range starting at 0.\+0 and finishing at 1.\+0. Set the S\+M\+E\+A\+R parameter equal to 0.\+1. Do the distributions of Q6 parameters for the solid and liquid phase overlap? N.\+B. You can load the output from these simulations using G\+I\+S\+M\+O and the all cvs button from the bar at the bottom.\hypertarget{belfast-10_links}{}\subsubsection{Local Steinhardt parameters}\label{belfast-10_links}
Hopefully you saw that the distribution of Q6 parameters for the solid and liquid phase overlap at the end of the previous section. Again this is not so surprising -\/ as you go from solid to liquid the distribution of the geometries of the coordination spheres widens. The system is now able to arrange the atoms in the coordination spheres in a much wider variety of different poses. Importantly, however, the fact that an atom is in a liquid does not preclude it from having a very-\/ordered, solid-\/like coordination environment. As such in the liquid state we will find the occasional atom with a high value for the Q6 parameter. Consequently, an ordred coordination environment does not always differentiate solid-\/like atoms from liquid-\/like atoms. The major difference is the liquid the ordered atoms will always be isolated. That is to say in the liquid atoms with an ordered coordination will always be surrounded by atoms with a disordered coordination sphere. By contrast in the solid each ordered atom will be surrounded by further ordered atoms. This observation forms the basis of the local Steinhardt parameters and the locally averaged Steinhardt parameters that are explained in this \href{https://www.youtube.com/watch?v=JBtIs5qYIPE&feature=youtu.be}{\tt video }.

Lets use plumed to calculate the distribution of \hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6} parameters in the solid and liquid phases. Repeat the calculations from the previous section but now use the H\+I\+S\+T\+O\+G\+R\+A\+M keyword to calculate the distribution of \hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6} parameters. For the switching function in the \hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6} parameter instruction use a R\+A\+T\+I\+O\+N\+A\+L function with $d_0$ equal to 1.\+3, $r_0$ equal to 0.\+2 and $n$ and $m$ set equal to 6 and 12 repsectively. For the H\+I\+S\+T\+O\+G\+R\+A\+M use 20 bins starting at 0.\+0 and finishing at 1.\+0. Se the S\+M\+E\+A\+R paraemter equal to 0.\+1. Do the distributions of \hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6} parameter for the solid and liquid phases overlap?

Lectner and Dellago have designed an alternative to the \hyperlink{LOCAL_Q6}{L\+O\+C\+A\+L\+\_\+\+Q6} parameter that is based on taking the \hyperlink{LOCAL_AVERAGE}{L\+O\+C\+A\+L\+\_\+\+A\+V\+E\+R\+A\+G\+E} of the Q6 parameter arround a central atom. This quantity can be calcualted using plumed. If you have time try to use the manual to work out how.\hypertarget{belfast-10_further}{}\subsection{Further Reading}\label{belfast-10_further}
There is a substantial literature on simulation of nucleation. Some papers are listed below but this list is far from exhaustive.


\begin{DoxyItemize}
\item F. Trudu, D. Donadio and M. Parrinello \href{http://journals.aps.org/prl/abstract/10.1103/PhysRevLett.97.105701}{\tt Freezing of a Lennard-\/\+Jones Fluid\+: From Nucleation to Spinodal Regime }, Phys. Rev. Lett. 97 105701 (2006)
\item D. Quigley and P. M. Rodger \href{http://dx.doi.org/10.1080/08927020802647280}{\tt A metadynamics-\/based approach to sampling crystallization events }, Mol. Simul. 2009
\item W. Lechner and C. Dellago \href{http://scitation.aip.org/content/aip/journal/jcp/129/11/10.1063/1.2977970}{\tt Accurate determination of crystal structures based on averaged local bond order parameters } J. Chem. Phys 129 114707 (2008) 
\end{DoxyItemize}\hypertarget{mindist}{}\section{Calculating a minimum distance}\label{mindist}
To calculate and print the minimum distance between two groups of atoms you use the following commands

\begin{DoxyVerb}d1: DISTANCES GROUPA=1-10 GROUPB=11-20 MIN={BETA=500.} 
PRINT ARG=d1.min FILE=colvar STRIDE=10
\end{DoxyVerb}
 (see \hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S} and \hyperlink{PRINT}{P\+R\+I\+N\+T})

In order to ensure differentiability the minimum is calculated using the following function\+:

\[ s = \frac{\beta}{ \log \sum_i \exp\left( \frac{\beta}{s_i} \right) } \]

where $\beta$ is a user specified parameter.

This input is used rather than a separate M\+I\+N\+D\+I\+S\+T colvar so that the same routine and the same input style can be used to calculate minimum coordinatetion numbers (see \hyperlink{COORDINATIONNUMBER}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N\+N\+U\+M\+B\+E\+R}), minimum angles (see \hyperlink{ANGLES}{A\+N\+G\+L\+E\+S}) and many other variables.

This new way of calculating mindist is part of plumed 2's multicolvar functionality. These special actions allow you to calculate multiple functions of a distribution of simple collective variables. As an example you can calculate the number of distances less than 1.\+0, the minimum distance, the number of distances more than 2.\+0 and the number of distances between 1.\+0 and 2.\+0 by using the following command\+:

\begin{DoxyVerb}DISTANCES ...
 GROUPA=1-10 GROUPB=11-20 
 LESS_THAN={RATIONAL R_0=1.0} 
 MORE_THAN={RATIONAL R_0=2.0} 
 BETWEEN={GAUSSIAN LOWER=1.0 UPPER=2.0} 
 MIN={BETA=500.}
... DISTANCES
PRINT ARG=d1.lessthan,d1.morethan,d1.between,d1.min FILE=colvar STRIDE=10
\end{DoxyVerb}
 (see \hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S} and \hyperlink{PRINT}{P\+R\+I\+N\+T})

A calculation performed this way is fast because the expensive part of the calculation -\/ the calculation of all the distances -\/ is only done once per step. Furthermore, it can be made faster by using the T\+O\+L keyword to discard those distance that make only a small contributions to the final values together with the N\+L\+\_\+\+S\+T\+R\+I\+D\+E keyword, which ensures that the distances that make only a small contribution to the final values aren't calculated at every step. \hypertarget{moving}{}\section{Moving from Plumed 1 to Plumed 2}\label{moving}
Syntax in P\+L\+U\+M\+E\+D 2 has been completely redesigned. The main difference is that whereas in P\+L\+U\+M\+E\+D 1 lines could be inserted in any order, in P\+L\+U\+M\+E\+D 2 the order of the lines matters. This is due to a major change in the internal architecture of P\+L\+U\+M\+E\+D. In version 2, commands (or \char`\"{}actions\char`\"{}) are executed in the order they are found in the input file. Because of this, you must e.\+g. first compute a collective variable and then print it later. More information can be found in the Section about \hyperlink{_syntax}{Getting started}.

Another very important change is in the way groups are used, discussed below. Finally, many features appear under a different name in the new version.\hypertarget{moving_moving-Groups}{}\subsection{Groups}\label{moving_moving-Groups}
In Plumed 1 groups (lists) were used for two tasks\+:
\begin{DoxyItemize}
\item To provide centers of masses to collective variables such as distances, angles, etc. This is now done by defining virtual atoms using either \hyperlink{CENTER}{C\+E\+N\+T\+E\+R} or \hyperlink{COM}{C\+O\+M}
\item To provide lists of atoms to collective variables such as coordination, gyration radius, etc. This is now done directly in the line that defines the collective variable.
\end{DoxyItemize}

If you would still like to use groups you can use the \hyperlink{GROUP}{G\+R\+O\+U\+P} commands. Whenever the label for a \hyperlink{GROUP}{G\+R\+O\+U\+P} action appears in the input it is replaced by the list of atoms that were specified in the \hyperlink{GROUP}{G\+R\+O\+U\+P}.\hypertarget{moving_moving-Directives}{}\subsection{Directives}\label{moving_moving-Directives}
What follows is a list of all the documented directives of Plumed 1 together with their plumed 2 equivalents. Be aware that the input syntaxes for these directives are not totally equivalent. You should read the documentation for the Plumed 2 Action.

\begin{TabularC}{2}
\hline
H\+I\+L\+L\+S  &\hyperlink{METAD}{M\+E\+T\+A\+D}   \\\cline{1-2}
W\+E\+L\+L\+T\+E\+M\+P\+E\+R\+E\+D  &\hyperlink{METAD}{M\+E\+T\+A\+D} with B\+I\+A\+S\+F\+A\+C\+T\+O\+R   \\\cline{1-2}
G\+R\+I\+D  &\hyperlink{METAD}{M\+E\+T\+A\+D} with G\+R\+I\+D\+\_\+\+M\+I\+N, G\+R\+I\+D\+\_\+\+M\+A\+X, and G\+R\+I\+D\+\_\+\+B\+I\+N   \\\cline{1-2}
W\+R\+I\+T\+E\+\_\+\+G\+R\+I\+D  &\hyperlink{METAD}{M\+E\+T\+A\+D} with G\+R\+I\+D\+\_\+\+W\+F\+I\+L\+E, G\+R\+I\+D\+\_\+\+W\+S\+T\+R\+I\+D\+E   \\\cline{1-2}
R\+E\+A\+D\+\_\+\+G\+R\+I\+D  &{\bfseries  currently missing }   \\\cline{1-2}
M\+U\+L\+T\+I\+P\+L\+E\+\_\+\+W\+A\+L\+K\+E\+R\+S  &\hyperlink{METAD}{M\+E\+T\+A\+D} with options W\+A\+L\+K\+E\+R\+S\+\_\+\+I\+D, W\+A\+L\+K\+E\+R\+S\+\_\+\+N, W\+A\+L\+K\+E\+R\+S\+\_\+\+D\+I\+R, and W\+A\+L\+K\+E\+R\+S\+\_\+\+R\+S\+T\+R\+I\+D\+E   \\\cline{1-2}
N\+O\+H\+I\+L\+L\+S  &not needed (collective variables are not biased by default)   \\\cline{1-2}
I\+N\+T\+E\+R\+V\+A\+L  &\hyperlink{METAD}{M\+E\+T\+A\+D} with I\+N\+T\+E\+R\+V\+A\+L   \\\cline{1-2}
I\+N\+V\+E\+R\+T  &{\bfseries  currently missing }   \\\cline{1-2}
P\+T\+M\+E\+T\+A\+D  &not needed (replica exchange detected from M\+D engine)   \\\cline{1-2}
B\+I\+A\+S\+X\+M\+D  &not needed (replica exchange detected from M\+D engine); one should anyway use \hyperlink{RANDOM_EXCHANGES}{R\+A\+N\+D\+O\+M\+\_\+\+E\+X\+C\+H\+A\+N\+G\+E\+S} to get the normal behavior   \\\cline{1-2}
U\+M\+B\+R\+E\+L\+L\+A  &\hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T}   \\\cline{1-2}
S\+T\+E\+E\+R  &\hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T}   \\\cline{1-2}
S\+T\+E\+E\+R\+P\+L\+A\+N  &\hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T}   \\\cline{1-2}
A\+B\+M\+D  &\hyperlink{ABMD}{A\+B\+M\+D}   \\\cline{1-2}
U\+W\+A\+L\+L  &\hyperlink{UPPER_WALLS}{U\+P\+P\+E\+R\+\_\+\+W\+A\+L\+L\+S}   \\\cline{1-2}
L\+W\+A\+L\+L  &\hyperlink{LOWER_WALLS}{L\+O\+W\+E\+R\+\_\+\+W\+A\+L\+L\+S}   \\\cline{1-2}
E\+X\+T\+E\+R\+N\+A\+L  &\hyperlink{EXTERNAL}{E\+X\+T\+E\+R\+N\+A\+L}   \\\cline{1-2}
C\+O\+M\+M\+I\+T\+M\+E\+N\+T  &\hyperlink{COMMITTOR}{C\+O\+M\+M\+I\+T\+T\+O\+R}   \\\cline{1-2}
P\+R\+O\+J\+\_\+\+G\+R\+A\+D  &\hyperlink{DUMPPROJECTIONS}{D\+U\+M\+P\+P\+R\+O\+J\+E\+C\+T\+I\+O\+N\+S}   \\\cline{1-2}
D\+A\+F\+E\+D  &{\bfseries  currently missing}   \\\cline{1-2}
D\+I\+S\+T\+A\+N\+C\+E  &\hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}   \\\cline{1-2}
P\+O\+S\+I\+T\+I\+O\+N  &\hyperlink{POSITION}{P\+O\+S\+I\+T\+I\+O\+N}   \\\cline{1-2}
M\+I\+N\+D\+I\+S\+T  &\hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S} with keyword M\+I\+N (See also \hyperlink{mindist}{Calculating a minimum distance})   \\\cline{1-2}
A\+N\+G\+L\+E  &\hyperlink{ANGLE}{A\+N\+G\+L\+E}   \\\cline{1-2}
T\+O\+R\+S\+I\+O\+N  &\hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}   \\\cline{1-2}
C\+O\+O\+R\+D  &\hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N}   \\\cline{1-2}
H\+B\+O\+N\+D  &{\bfseries  currently missing }, can be emulated with \hyperlink{COORDINATION}{C\+O\+O\+R\+D\+I\+N\+A\+T\+I\+O\+N}   \\\cline{1-2}
W\+A\+T\+E\+R\+B\+R\+I\+D\+G\+E  &\hyperlink{BRIDGE}{B\+R\+I\+D\+G\+E}   \\\cline{1-2}
R\+G\+Y\+R  &\hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N}   \\\cline{1-2}
D\+I\+P\+O\+L\+E  &\hyperlink{DIPOLE}{D\+I\+P\+O\+L\+E}   \\\cline{1-2}
D\+I\+H\+C\+O\+R  &\hyperlink{DIHCOR}{D\+I\+H\+C\+O\+R}   \\\cline{1-2}
A\+L\+P\+H\+A\+B\+E\+T\+A  &\hyperlink{ALPHABETA}{A\+L\+P\+H\+A\+B\+E\+T\+A}   \\\cline{1-2}
A\+L\+P\+H\+A\+R\+M\+S\+D  &\hyperlink{ALPHARMSD}{A\+L\+P\+H\+A\+R\+M\+S\+D}   \\\cline{1-2}
A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D  &\hyperlink{ANTIBETARMSD}{A\+N\+T\+I\+B\+E\+T\+A\+R\+M\+S\+D}   \\\cline{1-2}
P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D  &\hyperlink{PARABETARMSD}{P\+A\+R\+A\+B\+E\+T\+A\+R\+M\+S\+D}   \\\cline{1-2}
E\+L\+S\+T\+P\+O\+T  &{\bfseries  currently missing }   \\\cline{1-2}
P\+U\+C\+K\+E\+R\+I\+N\+G  &{\bfseries  currently missing }   \\\cline{1-2}
S\+\_\+\+P\+A\+T\+H  &\hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D}, s component   \\\cline{1-2}
Z\+\_\+\+P\+A\+T\+H  &\hyperlink{PATHMSD}{P\+A\+T\+H\+M\+S\+D}, z component   \\\cline{1-2}
T\+A\+R\+G\+E\+T\+E\+D  &\hyperlink{RMSD}{R\+M\+S\+D}   \\\cline{1-2}
E\+N\+E\+R\+G\+Y  &\hyperlink{ENERGY}{E\+N\+E\+R\+G\+Y}   \\\cline{1-2}
H\+E\+L\+I\+X  &{\bfseries  currently missing }   \\\cline{1-2}
P\+C\+A  &{\bfseries  currently missing }   \\\cline{1-2}
S\+P\+R\+I\+N\+T  &\hyperlink{SPRINT}{S\+P\+R\+I\+N\+T}   \\\cline{1-2}
R\+D\+F  &\hyperlink{DISTANCES}{D\+I\+S\+T\+A\+N\+C\+E\+S}, used in combination with H\+I\+S\+T\+O\+G\+R\+A\+M / B\+E\+T\+W\+E\+E\+N keyword   \\\cline{1-2}
A\+D\+F  &\hyperlink{ANGLES}{A\+N\+G\+L\+E\+S}, used in combination with H\+I\+S\+T\+O\+G\+R\+A\+M / B\+E\+T\+W\+E\+E\+N keyword   \\\cline{1-2}
P\+O\+L\+Y  &\hyperlink{COMBINE}{C\+O\+M\+B\+I\+N\+E}   \\\cline{1-2}
F\+U\+N\+C\+T\+I\+O\+N  &\hyperlink{MATHEVAL}{M\+A\+T\+H\+E\+V\+A\+L}   \\\cline{1-2}
A\+L\+I\+G\+N\+\_\+\+A\+T\+O\+M\+S  &\hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}   \\\cline{1-2}
\end{TabularC}
\hypertarget{munster}{}\section{Munster tutorial}\label{munster}
\begin{DoxyAuthor}{Authors}
Max Bonomi and Giovanni Bussi, stealing a lot of material from other tutorials. Richard Cunha is acknowledged for beta-\/testing this tutorial. 
\end{DoxyAuthor}
\begin{DoxyDate}{Date}
March 11, 2015
\end{DoxyDate}
This document describes the P\+L\+U\+M\+E\+D tutorial held in Munster, March 2015. The aim of this tutorial is to learn how to use P\+L\+U\+M\+E\+D to analyze molecular dynamics simulations on the fly, to analyze existing trajectories, and to perform enhanced sampling. Although the presented input files are correct, the users are invited to {\bfseries refer to the literature to understand how the parameters of enhanced sampling methods should be chosen in a real application.}

Users are also encouraged to follow the links to the full P\+L\+U\+M\+E\+D reference documentation and to wander around in the manual to discover the many available features and to do the other, more complete, tutorials. Here we are going to present only a very narrow selection of methods.

We here use P\+L\+U\+M\+E\+D 2.\+1 syntax and we explicitly note if some syntax is expected to change in P\+L\+U\+M\+E\+D 2.\+2, which will be released later in 2015. All the tests here are performed on a toy system, alanine dipeptide, simulated using the A\+M\+B\+E\+R99\+S\+B force field. We provide both a setup that includes explicit water, which is more realistic but slower, and a setup in gas phase, which is much faster. Simulations are made using G\+R\+O\+M\+A\+C\+S 4.\+6.\+7, which is here assumed to be already patched with P\+L\+U\+M\+E\+D and properly installed. However, these examples could be easily converted to other M\+D software.

All the gromacs input files and analsys scripts are provided in this \href{tutorial-resources/munster.tar.gz}{\tt tarball }.

Users are expected to write P\+L\+U\+M\+E\+D input files based on the instructions below.\hypertarget{munster_munster-toymodel}{}\subsection{Alanine dipeptide\+: our toy model}\label{munster_munster-toymodel}
In this tutorial we will play with alanine dipeptide (see Fig. \hyperlink{munster_munster-1-ala-fig}{munster-\/1-\/ala-\/fig}). This rather simple molecule is useful to make benchmark that are around for data analysis and free energy methods. It is a nice example since it presents two metastable states separated by a high (free) energy barrier. Here metastable states are intended as states which have a relatively low free energy compared to adjacent conformations. It is conventional use to show the two states in terms of Ramachandran dihedral angles, which are denoted with $ \Phi $ and $ \Psi $ in Fig. \hyperlink{munster_munster-1-transition-fig}{munster-\/1-\/transition-\/fig} .

\label{munster_munster-1-ala-fig}%
\hypertarget{munster_munster-1-ala-fig}{}%
 \label{munster_munster-1-transition-fig}%
\hypertarget{munster_munster-1-transition-fig}{}%
 \hypertarget{munster_munster-monitor}{}\subsection{Monitoring collective variables}\label{munster_munster-monitor}
The main goal of P\+L\+U\+M\+E\+D is to compute collective variables, which are complex descriptors than can be used to analyze a conformational change or a chemical reaction. This can be done either on the fly, that is during molecular dynamics, or a posteriori, using P\+L\+U\+M\+E\+D as a post-\/processing tool. In both cases one should create an input file with a specific P\+L\+U\+M\+E\+D syntax. A sample input file is below\+:

\begin{DoxyVerb}# compute distance between atoms 1 and 10
d: DISTANCE ATOMS=1,10
# create a virtual atom in the center between atoms 20 and 30
center: CENTER ATOMS=20,30
# compute torsional angle between atoms 1,10,20 and center
phi: TORSION ATOMS=1,10,20,center
# compute some function of previously computed variables
d2: MATHEVAL ARG=phi FUNC=cos(x) PERIODIC=NO
# print both of them every 10 step
PRINT ARG=d,phi,d2 STRIDE=10
\end{DoxyVerb}
 (see \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E}, \hyperlink{CENTER}{C\+E\+N\+T\+E\+R}, \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{MATHEVAL}{M\+A\+T\+H\+E\+V\+A\+L}, and \hyperlink{PRINT}{P\+R\+I\+N\+T})

P\+L\+U\+M\+E\+D works using k\+J/nm/ps as energy/length/time units. This can be personalized using \hyperlink{UNITS}{U\+N\+I\+T\+S}. Notice that variables should be given a name (in the example above, {\ttfamily d}, {\ttfamily phi}, and {\ttfamily d2}), which is then used to refer to these variables. Lists of atoms should be provided as comma separated numbers, with no space. Virtual atoms can be created and assigned a name for later use. You can find more information on the P\+L\+U\+M\+E\+D syntax at \hyperlink{_syntax}{Getting started} page of the manual. The complete documentation for all the supported collective variables can be found at the \hyperlink{colvarintro}{Collective variables} page.\hypertarget{munster_munster-monitor-of}{}\subsubsection{Analyze on the fly}\label{munster_munster-monitor-of}
Here we will run a plain M\+D on alanine dipeptide and compute two torsional angles on the fly. G\+R\+O\+M\+A\+C\+S needs a .tpr file, which is a binary file containing initial positions as well as force-\/field parameters. We also provide .gro, .mdp, and .top files, that can be modified and used to generate a new .tpr file. For this tutorial, it is sufficient to use the provided .tpr files. You will find several tpr files, namely\+:
\begin{DoxyItemize}
\item topol\+Awat.\+tpr -\/ setup in water, initialized in state A
\item topol\+Bwat.\+tpr -\/ setup in water, initialized in state B
\item topol\+A.\+tpr -\/ setup in vacuum, initialized in state A
\item topol\+B.\+tpr -\/ setup in vacuum, initialized in state B
\end{DoxyItemize}

Gromacs md can be run using on the command line\+: \begin{DoxyVerb}> mdrun_mpi -s topolA.tpr -nsteps 10000
\end{DoxyVerb}
 The nsteps flags can be used to change the number of timesteps and topol\+A.\+tpr is the name of the tpr file. While running, gromacs will produce an md.\+log file, with log information, and a traj.\+xtc file, with a binary trajectory. The trajectory can be visualized with V\+M\+D using a command such as \begin{DoxyVerb}> vmd confout.gro tra.xtc
\end{DoxyVerb}


To run a simulation with gromacs+plumed you just need to add a -\/plumed flag \begin{DoxyVerb}> mdrun_mpi -s topolA.tpr -nsteps 10000 -plumed plumed.dat
\end{DoxyVerb}
 Here plumed.\+dat is the name of the plumed input file. Notice that P\+L\+U\+M\+E\+D will write information in the md.\+log that could be useful to verify if the simulation has been set up properly.\hypertarget{munster_munster-exercise-0}{}\paragraph{Exercise 0}\label{munster_munster-exercise-0}
In this exercise, we will run a plain molecular dynamics simulation and monitor the $\Phi$ and $\Psi$ dihedral angles on the fly. Using the following P\+L\+U\+M\+E\+D input file you can monitor $\Phi$ and $\Psi$ angles during the M\+D simulation \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
PRINT ARG=phi,psi STRIDE=100 FILE=colvar
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} and \hyperlink{PRINT}{P\+R\+I\+N\+T})

Notice that P\+L\+U\+M\+E\+D is going to compute the collective variables only when necessary, that is, in this case, every 100 steps. This is not very relevant for simple variables such as torsional angles, but provides a significant speedup when using expensive collective variables.

P\+L\+U\+M\+E\+D will write a textual file named {\ttfamily colvar} containing three columns\+: physical time, $\Phi$ and $\Psi$. Results can be plotted using gnuplot\+: \begin{DoxyVerb}> gnuplot
# this shows phi as a function of time
gnuplot> plot "colvar" u 2
# this shows psi as a function of time
gnuplot> plot "colvar" u 3
# this shows psi as a function of phi
gnuplot> plot "colvar" u 2:3
\end{DoxyVerb}


Now try to do the same using the two different initial configurations that we provided ({\ttfamily topol\+A.\+tpr} and {\ttfamily topol\+B.\+tpr}). You can try both setup (water and vacuum). Results from 200ps (100000 steps) trajectories in vacuum are shown in Figure \hyperlink{munster_munster-ala-traj}{munster-\/ala-\/traj}.

\label{munster_munster-ala-traj}%
\hypertarget{munster_munster-ala-traj}{}%
 Notice that the result depends heavily on the starting structure. For the simulation in vacuum, the two free-\/energy minima are separated by a large barrier and, in such a short simulation, the system cannot cross it. In water the barrier is smaller and you might see some crossing. Also notice that the two clouds are well separated, indicating that these two collective variables are good enough to properly distinguish among the two minima.

As a final comment, notice that if you run twice the same calculation in the same directory, you might overwrite the resulting files. G\+R\+O\+M\+A\+C\+S takes automatic backup of the output files, and P\+L\+U\+M\+E\+D does it as well. In case you are restarting a simulation, you can add the keyword \hyperlink{RESTART}{R\+E\+S\+T\+A\+R\+T} at the beginning of the P\+L\+U\+M\+E\+D input file. This will tell P\+L\+U\+M\+E\+D to {\itshape append} files instead of taking a backup copy.

 \hypertarget{munster_munster-monitor-dr}{}\subsubsection{Analyze using the driver}\label{munster_munster-monitor-dr}
Imagine you already made a simulation, with or without P\+L\+U\+M\+E\+D. You might want to compute the collective variables a posteriori, from the trajectory file. You can do this by using the plumed executable on the command line. Type \begin{DoxyVerb}> plumed driver --help
\end{DoxyVerb}
 to have an idea of the possible options. See \hyperlink{driver}{driver} for the full documentation.

Here we will use the driver the compute $\Phi$ and $\Psi$ on the already generated trajectory. Let's assume the trajectory is named {\ttfamily traj.\+xtc}. You should prepare an P\+L\+U\+M\+E\+D input file named {\ttfamily analysis.\+dat} as\+: \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
PRINT ARG=phi,psi FILE=analysis
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} and \hyperlink{PRINT}{P\+R\+I\+N\+T}) Notice that typically when using the driver we do not provide a S\+T\+R\+I\+D\+E keyword to P\+R\+I\+N\+T. This implies \char`\"{}print at every step\char`\"{} which, analyzing a trajectory, means \char`\"{}print for all the available snapshots\char`\"{}. Then, you can use the following command\+: \begin{DoxyVerb}> plumed driver --mf_xtc traj.xtc --plumed analysis.dat
\end{DoxyVerb}
 Notice that P\+L\+U\+M\+E\+D has no way to now the value of physical time from the trajectory. If you want physical time to be printed in the {\ttfamily analysis} file you should give more information to the driver, e.\+g.\+: \begin{DoxyVerb}> plumed driver --mf_xtc traj.xtc --plumed analysis.dat --timestep 0.002 --trajectory-stride 1000
\end{DoxyVerb}
 (see \hyperlink{driver}{driver})

In this case we inform the driver that the {\ttfamily traj.\+xtc} file was produced in a run with a timestep of 0.\+002 ps and saving a snapshop every 1000 timesteps.

You might want to analyze a different collective variable, such as the gyration radius. The gyration radius tells how extended is the molecules in space. You can do it with the following plumed input file

\begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17

heavy: GROUP ATOMS=1,5,6,7,9,11,15,16,17,19
gyr: GYRATION ATOMS=heavy

# the same could have been achieved with
# gyr: GYRATION ATOMS=1,5,6,7,9,11,15,16,17,19

PRINT ARG=phi,psi,gyr FILE=analyze
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N}, \hyperlink{GROUP}{G\+R\+O\+U\+P}, and \hyperlink{PRINT}{P\+R\+I\+N\+T})

Now try to compute the time series of the gyration radius.



 \hypertarget{munster_munster-monitor-pbc}{}\subsubsection{Periodic boundaries and explicit water}\label{munster_munster-monitor-pbc}
In case you are running the simulation in water, you might see that at some point this variable shows some crazy jump. The reason is that the trajectory containes coordinates where molecules are broken across periodic-\/boundary conditions This happens with G\+R\+O\+M\+A\+C\+S and some other M\+D code. These codes typically have tools to process trajecories and restore whole molecules. This trick is ok for the a-\/posteriori analysis we are trying now, but cannot used when one needs to compute a collective variable on-\/the-\/fly or, as we will see later, one wants to add a bias to that collective variable. For this reason, we implemented a workaround in P\+L\+U\+M\+E\+D, that is the molecule should be made whole using the W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E command. What this command is doing is making a loop over all the atoms (in the order they are provided) and set the coordinates of each of them in the periodic image which is as close as possible to the coordinates of the preceeding atom. In most cases it is sufficient to list all atoms of a molecule in order. Look in the W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E page to get more information. Here this will be enough\+: \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17

# notice the 1-22 syntax, a shortcut for a list 1,2,3,...,22
WHOLEMOLECULES ENTITY0=1-22

heavy: GROUP ATOMS=1,5,6,7,9,11,15,16,17,19
gyr: GYRATION ATOMS=heavy

PRINT ARG=phi,psi,gyr FILE=analyze
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}, \hyperlink{GROUP}{G\+R\+O\+U\+P}, \hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N}, and \hyperlink{PRINT}{P\+R\+I\+N\+T})

This is a very important issue that should be kept in mind when using P\+L\+U\+M\+E\+D. Notice that starting with version 2.\+2 P\+L\+U\+M\+E\+D will make molecules used in \hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N} (as well as in other variables) whole automatically, so that this extra command will not be necessary.

Notice that you can instruct P\+L\+U\+M\+E\+D to dump on a file not only the collective variables (as we are doing with \hyperlink{PRINT}{P\+R\+I\+N\+T}) but also the atomic positions. This is a very good way to understand what \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S} is actually doing. Try the following input

\begin{DoxyVerb}MOLINFO STRUCTURE=../TOPO/reference.pdb
DUMPATOMS FILE=test1.gro ATOMS=1-22
WHOLEMOLECULES ENTITY0=1-22
DUMPATOMS FILE=test2.gro ATOMS=1-22
\end{DoxyVerb}
 (see \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O}, \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S}, and \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}).

\hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S} writes on a gro file the coordinates of the alanine dipeptide atoms. Here P\+L\+U\+M\+E\+D will produce two files, one with coordinates {\itshape before} the application of \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}, and one with coordinates after$\ast$ the application of \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}. You can load both trajectories in V\+M\+D to see the difference. The \hyperlink{MOLINFO}{M\+O\+L\+I\+N\+F\+O} command is here used to provide atom names to P\+L\+U\+M\+E\+D so that the resulting gro file looks nicer in V\+M\+D.

Notice that P\+L\+U\+M\+E\+D has several commands that manipulate atomic coordinates. One example is \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}, that fixes problems with periodic boundary conditions. \hyperlink{COM}{C\+O\+M} and \hyperlink{CENTER}{C\+E\+N\+T\+E\+R} add new atoms, and \hyperlink{FIT_TO_TEMPLATE}{F\+I\+T\+\_\+\+T\+O\+\_\+\+T\+E\+M\+P\+L\+A\+T\+E} can actually move atoms from their original position to align them on a template. \hyperlink{DUMPATOMS}{D\+U\+M\+P\+A\+T\+O\+M\+S} is this very useful to check what these commands are doing and for using the P\+L\+U\+M\+E\+D \hyperlink{driver}{driver} to manipulate M\+D trajectories.



 \hypertarget{munster_munster-monitor-an}{}\subsubsection{Other analysis tools}\label{munster_munster-monitor-an}
P\+L\+U\+M\+E\+D also allows you to make some analyis on the collective variables you are calculating. For example, you can compute a histogram with an input like this one \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
heavy: GROUP ATOMS=1,5,6,7,9,11,15,16,17,19
gyr: GYRATION ATOMS=heavy
PRINT ARG=phi,psi,gyr FILE=analyze
HISTOGRAM ...
  ARG=gyr
  USE_ALL_DATA
  KERNEL=discrete
  GRID_MIN=0
  GRID_MAX=1
  GRID_BIN=50
  GRID_WFILE=histogram
... HISTOGRAM
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{WHOLEMOLECULES}{W\+H\+O\+L\+E\+M\+O\+L\+E\+C\+U\+L\+E\+S}, \hyperlink{GROUP}{G\+R\+O\+U\+P}, \hyperlink{GYRATION}{G\+Y\+R\+A\+T\+I\+O\+N}, \hyperlink{PRINT}{P\+R\+I\+N\+T}, and \hyperlink{HISTOGRAM}{H\+I\+S\+T\+O\+G\+R\+A\+M})

An histogram with 50 bins will be performed on the gyration radius. Try to compute the histogram for the $\Phi$ and $\Psi$ angles.

P\+L\+U\+M\+E\+D can do much more than a histogram, more information on analysis can be found at the page \hyperlink{_analysis}{Analysis}

Notice that the plumed driver can also be used directly from V\+M\+D taking advantage of the P\+L\+U\+M\+E\+D collective variable tool developed by Toni Giorgino (\href{http://multiscalelab.org/utilities/PlumedGUI}{\tt http\+://multiscalelab.\+org/utilities/\+Plumed\+G\+U\+I}). Just open a recent version of V\+M\+D and go to Extensions/\+Analysis/\+Collective Variable Analsys (P\+L\+U\+M\+E\+D). This graphical interface can also be used to quickly build P\+L\+U\+M\+E\+D input files based on template lines.

\hypertarget{munster_munster-biasing}{}\subsection{Biasing collective variables}\label{munster_munster-biasing}
We have seen that P\+L\+U\+M\+E\+D can be used to compute collective variables. However, P\+L\+U\+M\+E\+D is most often use to add forces on the collective variables. To this aim, we have implemented a variety of possible biases acting on collective variables. A bias works in a manner conceptually similar to the \hyperlink{PRINT}{P\+R\+I\+N\+T} command, taking as argument one or more collective variables. However, here the S\+T\+R\+I\+D\+E is usually omitted (that is equivalent to setting it to 1), which means that forces are applied at every timestep. In P\+L\+U\+M\+E\+D 2.\+2 you will be able to change the S\+T\+R\+I\+D\+E also for bias potentials, but that's another story. In the following we will see how to apply harmonic restraints and how to build an adaptive bias potential with metadynamics. The complete documentation for all the biasing methods available in P\+L\+U\+M\+E\+D can be found at the \hyperlink{_bias}{Bias} page.\hypertarget{munster_munster-biasing-me}{}\subsubsection{Metadynamics}\label{munster_munster-biasing-me}
 \hypertarget{munster_munster-biasing-me-theory}{}\paragraph{Summary of theory}\label{munster_munster-biasing-me-theory}
In metadynamics, an external history-\/dependent bias potential is constructed in the space of a few selected degrees of freedom $ \vec{s}({q}) $, generally called collective variables (C\+Vs) \cite{metad}. This potential is built as a sum of Gaussians deposited along the trajectory in the C\+Vs space\+:

\[ V(\vec{s},t) = \sum_{ k \tau < t} W(k \tau) \exp\left( -\sum_{i=1}^{d} \frac{(s_i-s_i({q}(k \tau)))^2}{2\sigma_i^2} \right). \]

where $ \tau $ is the Gaussian deposition stride, $ \sigma_i $ the width of the Gaussian for the ith C\+V, and $ W(k \tau) $ the height of the Gaussian. The effect of the metadynamics bias potential is to push the system away from local minima into visiting new regions of the phase space. Furthermore, in the long time limit, the bias potential converges to minus the free energy as a function of the C\+Vs\+:

\[ V(\vec{s},t\rightarrow \infty) = -F(\vec{s}) + C. \]

In standard metadynamics, Gaussians of constant height are added for the entire course of a simulation. As a result, the system is eventually pushed to explore high free-\/energy regions and the estimate of the free energy calculated from the bias potential oscillates around the real value. In well-\/tempered metadynamics \cite{Barducci:2008}, the height of the Gaussian is decreased with simulation time according to\+:

\[ W (k \tau ) = W_0 \exp \left( -\frac{V(\vec{s}({q}(k \tau)),k \tau)}{k_B\Delta T} \right ), \]

where $ W_0 $ is an initial Gaussian height, $ \Delta T $ an input parameter with the dimension of a temperature, and $ k_B $ the Boltzmann constant. With this rescaling of the Gaussian height, the bias potential smoothly converges in the long time limit, but it does not fully compensate the underlying free energy\+:

\[ V(\vec{s},t\rightarrow \infty) = -\frac{\Delta T}{T+\Delta T}F(\vec{s}) + C. \]

where $ T $ is the temperature of the system. In the long time limit, the C\+Vs thus sample an ensemble at a temperature $ T+\Delta T $ which is higher than the system temperature $ T $. The parameter $ \Delta T $ can be chosen to regulate the extent of free-\/energy exploration\+: $ \Delta T = 0$ corresponds to standard molecular dynamics, $ \Delta T \rightarrow \infty $ to standard metadynamics. In well-\/tempered metadynamics literature and in P\+L\+U\+M\+E\+D, you will often encounter the term \char`\"{}biasfactor\char`\"{} which is the ratio between the temperature of the C\+Vs ( $ T+\Delta T $) and the system temperature ( $ T $)\+:

\[ \gamma = \frac{T+\Delta T}{T}. \]

The biasfactor should thus be carefully chosen in order for the relevant free-\/energy barriers to be crossed efficiently in the time scale of the simulation.

Additional information can be found in the several review papers on metadynamics \cite{gerv-laio09review} \cite{WCMS:WCMS31} \cite{WCMS:WCMS1103}.



If you do not know exactly where you would like your collective variables to go, and just know (or suspect) that some variables have large free-\/energy barriers that hinder some conformational rearrangement or some chemical reaction, you can bias them using metadynamics. In this way, a time dependent, adaptive potential will be constructed that tends to disfavor visited configurations in the collective-\/variable space. The bias is usually built as a sum of Gaussian deposited in the already visited states.\hypertarget{munster_munster-exercise-1}{}\paragraph{Exercise 1}\label{munster_munster-exercise-1}
Now run a metadynamics simulation with the following input \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
METAD ARG=phi,psi HEIGHT=1.0 BIASFACTOR=10 SIGMA=0.35,0.35 PACE=100 GRID_MIN=-pi,-pi GRID_MAX=pi,pi
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N} and \hyperlink{METAD}{M\+E\+T\+A\+D}) Thus, a single M\+E\+T\+A\+D line will contain all the metadynamics related options, such as Gaussian height ({\ttfamily H\+E\+I\+G\+H\+T}, here in k\+J/mol), stride ({\ttfamily P\+A\+C\+E}, here in number of time steps), bias factor ({\ttfamily B\+I\+A\+S\+F\+A\+C\+T\+O\+R}, here indicates that we are going to effectively boost the temperature of the collective variables by a factor 10), and width ({\ttfamily S\+I\+G\+M\+A}, an array with same size as the number of collective variables).

There are two additional keywords that are optional, namely G\+R\+I\+D\+\_\+\+M\+I\+N and G\+R\+I\+D\+\_\+\+M\+A\+X. These keywords sets the range of the collective variables and tell P\+L\+U\+M\+E\+D to keep the bias potential stored on a grid. This affects speed but, in principle, not the accuracy of the calculation. You can try to remove those keywords and see the difference.

Now, run a metadynamics simulations and check the explored collective variable space. Results from a 200ps (100000 steps) trajectory in vacuum are shown in Figure \hyperlink{munster_munster-ala-traj-metad}{munster-\/ala-\/traj-\/metad}.

\label{munster_munster-ala-traj-metad}%
\hypertarget{munster_munster-ala-traj-metad}{}%
 As you can see, exploration is greatly enhanced. Notice that the explored ensemble can be tuned using the biasfactor $\gamma$. Larger $\gamma$ implies that the system will explore states with higher free energy. As a rule of thumb, if you expect a barrier of the order of $\Delta G^*$, a reasonable choice for the biasfactor is $\gamma\approx\frac{\Delta G}{2k_BT}$.

Finally, notice that \hyperlink{METAD}{M\+E\+T\+A\+D} potential depends on the previously visited trajectories. As such, when you restart a previous simulation, it should read the previously deposited H\+I\+L\+L\+S file. This is automatically triggered by the \hyperlink{RESTART}{R\+E\+S\+T\+A\+R\+T} keyword.\hypertarget{munster_munster-exercise-2}{}\paragraph{Exercise 2}\label{munster_munster-exercise-2}
In this exercise, we will run a well-\/tempered metadynamics simulation on alanine dipeptide in vacuum, using as C\+V the backbone dihedral angle phi. In order to run this simulation we need to prepare the P\+L\+U\+M\+E\+D input file (plumed.\+dat) as follows.

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate well-tempered metadynamics in phi depositing 
# a Gaussian every 500 time steps, with initial height equal 
# to 1.2 kJoule/mol, biasfactor equal to 10.0, and width to 0.35 rad

METAD ...
LABEL=metad
ARG=phi
PACE=500
HEIGHT=1.2
SIGMA=0.35
FILE=HILLS
BIASFACTOR=10.0
TEMP=300.0
GRID_MIN=-pi
GRID_MAX=pi
GRID_SPACING=0.1
... METAD

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

The syntax for the command \hyperlink{METAD}{M\+E\+T\+A\+D} is simple. The directive is followed by a keyword A\+R\+G followed by the labels of the C\+Vs on which the metadynamics potential will act. The keyword P\+A\+C\+E determines the stride of Gaussian deposition in number of time steps, while the keyword H\+E\+I\+G\+H\+T specifies the height of the Gaussian in k\+Joule/mol. For each C\+Vs, one has to specified the width of the Gaussian by using the keyword S\+I\+G\+M\+A. Gaussian will be written to the file indicated by the keyword F\+I\+L\+E.

The bias potential will be stored on a grid, whose boundaries are specified by the keywords G\+R\+I\+D\+\_\+\+M\+I\+N and G\+R\+I\+D\+\_\+\+M\+A\+X. Notice that you should provide either the number of bins for every collective variable (G\+R\+I\+D\+\_\+\+B\+I\+N) or the desired grid spacing (G\+R\+I\+D\+\_\+\+S\+P\+A\+C\+I\+N\+G). In case you provide both P\+L\+U\+M\+E\+D will use the most conservative choice (highest number of bins) for each dimension. In case you do not provide any information about bin size (neither G\+R\+I\+D\+\_\+\+B\+I\+N nor G\+R\+I\+D\+\_\+\+S\+P\+A\+C\+I\+N\+G) and if Gaussian width is fixed P\+L\+U\+M\+E\+D will use 1/5 of the Gaussian width as grid spacing. This default choice should be reasonable for most applications.

Once the P\+L\+U\+M\+E\+D input file is prepared, one has to run Gromacs with the option to activate P\+L\+U\+M\+E\+D and read the input file\+:

\begin{DoxyVerb}mdrun_mpi -s ../TOPO/topolA.tpr -plumed plumed.dat -nsteps 5000000
\end{DoxyVerb}


During the metadynamics simulation, P\+L\+U\+M\+E\+D will create two files, named C\+O\+L\+V\+A\+R and H\+I\+L\+L\+S. The C\+O\+L\+V\+A\+R file contains all the information specified by the P\+R\+I\+N\+T command, in this case the value of the C\+Vs every 10 steps of simulation, along with the current value of the metadynamics bias potential. We can use C\+O\+L\+V\+A\+R to visualize the behavior of the C\+V during the simulation\+:

\label{munster_munster-metad-phi-fig}%
\hypertarget{munster_munster-metad-phi-fig}{}%
 By inspecting Figure \hyperlink{munster_munster-metad-phi-fig}{munster-\/metad-\/phi-\/fig}, we can see that the system is initialized in one of the two metastable states of alanine dipeptide. After a while (t=0.\+1 ns), the system is pushed by the metadynamics bias potential to visit the other local minimum. As the simulation continues, the bias potential fills the underlying free-\/energy landscape, and the system is able to diffuse in the entire phase space.

The H\+I\+L\+L\+S file contains a list of the Gaussians deposited along the simulation. If we give a look at the header of this file, we can find relevant information about its content\+:

\begin{DoxyVerb}#! FIELDS time phi psi sigma_phi sigma_psi height biasf
#! SET multivariate false
#! SET min_phi -pi
#! SET max_phi pi
#! SET min_psi -pi
#! SET max_psi pi
\end{DoxyVerb}


The line starting with F\+I\+E\+L\+D\+S tells us what is displayed in the various columns of the H\+I\+L\+L\+S file\+: the time of the simulation, the value of phi and psi, the width of the Gaussian in phi and psi, the height of the Gaussian, and the biasfactor. We can use the H\+I\+L\+L\+S file to visualize the decrease of the Gaussian height during the simulation, according to the well-\/tempered recipe\+:

\label{munster_munster-metad-phihills-fig}%
\hypertarget{munster_munster-metad-phihills-fig}{}%
 If we look carefully at the scale of the y-\/axis, we will notice that in the beginning the value of the Gaussian height is higher than the initial height specified in the input file, which should be 1.\+2 k\+Joule/mol. In fact, this column reports the height of the Gaussian rescaled by the pre-\/factor that in well-\/tempered metadynamics relates the bias potential to the free energy. In this way, when we will use \hyperlink{sum_hills}{sum\+\_\+hills}, the sum of the Gaussians deposited will directly provide the free-\/energy, without further rescaling needed (see below).

One can estimate the free energy as a function of the metadynamics C\+Vs directly from the metadynamics bias potential. In order to do so, the utility \hyperlink{sum_hills}{sum\+\_\+hills} should be used to sum the Gaussians deposited during the simulation and stored in the H\+I\+L\+L\+S file. To calculate the free energy as a function of phi, it is sufficient to use the following command line\+:

\begin{DoxyVerb}plumed sum_hills --hills HILLS
\end{DoxyVerb}


The command above generates a file called fes.\+dat in which the free-\/energy surface as function of phi is calculated on a regular grid. One can modify the default name for the free energy file, as well as the boundaries and bin size of the grid, by using the following options of \hyperlink{sum_hills}{sum\+\_\+hills} \+:

\begin{DoxyVerb}--outfile - specify the outputfile for sumhills
--min - the lower bounds for the grid
--max - the upper bounds for the grid
--bin - the number of bins for the grid
--spacing - grid spacing, alternative to the number of bins
\end{DoxyVerb}


The result should look like this\+:

\label{munster_munster-metad-phifes-fig}%
\hypertarget{munster_munster-metad-phifes-fig}{}%
 To assess the convergence of a metadynamics simulation, one can calculate the estimate of the free energy as a function of simulation time. At convergence, the reconstructed profiles should be similar. The option --stride should be used to give an estimate of the free energy every N Gaussians deposited, and the option --mintozero can be used to align the profiles by setting the global minimum to zero. If we use the following command line\+:

\begin{DoxyVerb}plumed sum_hills --hills HILLS --stride 100 --mintozero
\end{DoxyVerb}


one free energy is calculated every 100 Gaussians deposited, and the global minimum is set to zero in all profiles. The resulting plot should look like the following\+:

\label{munster_munster-metad-phifest-fig}%
\hypertarget{munster_munster-metad-phifest-fig}{}%
 To assess the convergence of the simulation more quantitatively, we can calculate the free-\/energy difference between the two local minima of the free energy along phi as a function of simulation time. We can use following script to integrate the multiple free-\/energy profiles in the two basins defined by the following intervals in phi space\+: basin A, -\/3$<$phi$<$-\/1, basin B, 0.\+5$<$phi$<$1.\+5.

\begin{DoxyVerb}# number of free-energy profiles
nfes= # put here the number of profiles
# minimum of basin A
minA=-3
# maximum of basin A
maxA=1
# minimum of basin B
minB=0.5
# maximum of basin B
maxB=1.5
# temperature in energy units
kbt=2.5

for((i=0;i<nfes;i++))
do
 # calculate free-energy of basin A
 A=`awk 'BEGIN{tot=0.0}{if($1!="#!" && $1>min && $1<max)tot+=exp(-$2/kbt)}END{print -kbt*log(tot)}' min=${minA} max=${maxA} kbt=${kbt} fes_${i}.dat`
 # and basin B
 B=`awk 'BEGIN{tot=0.0}{if($1!="#!" && $1>min && $1<max)tot+=exp(-$2/kbt)}END{print -kbt*log(tot)}' min=${minB} max=${maxB} kbt=${kbt} fes_${i}.dat`
 # calculate difference
 Delta=$(echo "${A} - ${B}" | bc -l)
 # print it
 echo $i $Delta
done\end{DoxyVerb}


notice that nfes should be set to the number of profiles (free-\/energy estimates at different times of the simulation) generated by the option --stride of \hyperlink{sum_hills}{sum\+\_\+hills}.

\label{munster_munster-metad-phifes-difft-fig}%
\hypertarget{munster_munster-metad-phifes-difft-fig}{}%
 This analysis, along with the observation of the diffusive behavior in the C\+Vs space, suggest that the simulation is converged.\hypertarget{munster_munster-exercise-3}{}\paragraph{Exercise 3}\label{munster_munster-exercise-3}
In this exercise, we will run a well-\/tempered metadynamics simulation on alanine dipeptide in vacuum, using as C\+V the backbone dihedral angle psi. In order to run this simulation we need to prepare the P\+L\+U\+M\+E\+D input file (plumed.\+dat) as follows.

\begin{DoxyVerb}# set up two variables for Phi and Psi dihedral angles 
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Activate well-tempered metadynamics in psi depositing 
# a Gaussian every 500 time steps, with initial height equal 
# to 1.2 kJoule/mol, biasfactor equal to 10.0, and width to 0.35 rad

METAD ...
LABEL=metad
ARG=psi
PACE=500
HEIGHT=1.2
SIGMA=0.35
FILE=HILLS
BIASFACTOR=10.0
TEMP=300.0
GRID_MIN=-pi
GRID_MAX=pi
GRID_SPACING=0.1
... METAD

# monitor the two variables and the metadynamics bias potential
PRINT STRIDE=10 ARG=phi,psi,metad.bias FILE=COLVAR\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{METAD}{M\+E\+T\+A\+D}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

Once the P\+L\+U\+M\+E\+D input file is prepared, one has to run Gromacs with the option to activate P\+L\+U\+M\+E\+D and read the input file\+:

\begin{DoxyVerb}mdrun_mpi -s ../TOPO/topolA.tpr -plumed plumed.dat -nsteps 5000000
\end{DoxyVerb}


As we did in the previous exercise, we can use C\+O\+L\+V\+A\+R to visualize the behavior of the C\+V during the simulation. Here we will plot at the same time the evolution of the metadynamics C\+V psi and of the other dihedral phi\+:

\label{munster_munster-metad-psi-phi-fig}%
\hypertarget{munster_munster-metad-psi-phi-fig}{}%
 By inspecting Figure \hyperlink{munster_munster-metad-psi-phi-fig}{munster-\/metad-\/psi-\/phi-\/fig}, we notice that something different happened compared to the previous exercise. At first the behavior of psi looks diffusive in the entire C\+V space. However, around t=1 ns, psi seems trapped in a region of the C\+V space in which it was previously diffusing without problems. The reason is that the non-\/biased C\+V phi after a while has jumped into a different local minima. Since phi is not directly biased, one has to wait for this (slow) degree of freedom to equilibrate before the free energy along psi can converge. Try to repeat the analysis done in the previous exercise (calculate the estimate of the free energy as a function of time and monitor the free-\/energy difference between basins) to assess the convergence of this metadynamics simulation.\hypertarget{munster_munster-biasing-re}{}\subsubsection{Restraints}\label{munster_munster-biasing-re}
 \hypertarget{munster_munster-biased-sampling-theory}{}\paragraph{Biased sampling theory}\label{munster_munster-biased-sampling-theory}
A system at temperature $ T$ samples conformations from the canonical ensemble\+: \[ P(q)\propto e^{-\frac{U(q)}{k_BT}} \]. Here $ q $ are the microscopic coordinates and $ k_B $ is the Boltzmann constant. Since $ q $ is a highly dimensional vector, it is often convenient to analyze it in terms of a few collective variables (see \hyperlink{belfast-1}{Belfast tutorial\+: Analyzing C\+Vs} , \hyperlink{belfast-2}{Belfast tutorial\+: Adaptive variables I} , and \hyperlink{belfast-3}{Belfast tutorial\+: Adaptive variables I\+I} ). The probability distribution for a C\+V $ s$ is \[ P(s)\propto \int dq e^{-\frac{U(q)}{k_BT}} \delta(s-s(q)) \] This probability can be expressed in energy units as a free energy landscape $ F(s) $\+: \[ F(s)=-k_B T \log P(s) \].

Now we would like to modify the potential by adding a term that depends on the C\+V only. That is, instead of using $ U(q) $, we use $ U(q)+V(s(q))$. There are several reasons why one would like to introduce this potential. One is to avoid that the system samples some un-\/desired portion of the conformational space. As an example, imagine that you want to study dissociation of a complex of two molecules. If you perform a very long simulation you will be able to see association and dissociation. However, the typical time required for association will depend on the size of the simulation box. It could be thus convenient to limit the exploration to conformations where the distance between the two molecules is lower than a given threshold. This could be done by adding a bias potential on the distance between the two molecules. Another example is the simulation of a portion of a large molecule taken out from its initial context. The fragment alone could be unstable, and one might want to add additional potentials to keep the fragment in place. This could be done by adding a bias potential on some measure of the distance from the experimental structure (e.\+g. on root-\/mean-\/square deviation).

Whatever C\+V we decide to bias, it is very important to recognize which is the effect of this bias and, if necessary, remove it a posteriori. The biased distribution of the C\+V will be \[ P'(s)\propto \int dq e^{-\frac{U(q)+V(s(q))}{k_BT}} \delta(s-s(q))\propto e^{-\frac{V(s(q))}{k_BT}}P(s) \] and the biased free energy landscape \[ F'(s)=-k_B T \log P'(s)=F(s)+V(s)+C \] Thus, the effect of a bias potential on the free energy is additive. Also notice the presence of an undetermined constant $ C $. This constant is irrelevant for what concerns free-\/energy differences and barriers, but will be important later when we will learn the weighted-\/histogram method. Obviously the last equation can be inverted so as to obtain the original, unbiased free-\/energy landscape from the biased one just subtracting the bias potential \[ F(s)=F'(s)-V(s)+C \]

Additionally, one might be interested in recovering the distribution of an arbitrary observable. E.\+g., one could add a bias on the distance between two molecules and be willing to compute the unbiased distribution of some torsional angle. In this case there is no straightforward relationship that can be used, and one has to go back to the relationship between the microscopic probabilities\+: \[ P(q)\propto P'(q) e^{\frac{V(s(q))}{k_BT}} \] The consequence of this expression is that one can obtained any kind of unbiased information from a biased simulation just by weighting every sampled conformation with a weight \[ w\propto e^{\frac{V(s(q))}{k_BT}} \] That is, frames that have been explored in spite of a high (disfavoring) bias potential $ V $ will be counted more than frames that has been explored with a less disfavoring bias potential.



 \hypertarget{munster_munster-umbrella-sampling-theory}{}\paragraph{Umbrella sampling theory}\label{munster_munster-umbrella-sampling-theory}
Often in interesting cases the free-\/energy landscape has several local minima. If these minima have free-\/energy differences that are on the order of a few times $k_BT$ they might all be relevant. However, if they are separated by a high saddle point in the free-\/energy landscape (i.\+e. a low probability region) than the transition between one and the other will take a lot of time and these minima will correspond to metastable states. The transition between one minimum and the other could require a time scale which is out of reach for molecular dynamics. In these situations, one could take inspiration from catalysis and try to favor in a controlled manner the conformations corresponding to the transition state.

Imagine that you know since the beginning the shape of the free-\/energy landscape $ F(s) $ as a function of one C\+V $ s $. If you perform a molecular dynamics simulation using a bias potential which is exactly equal to $ -F(s) $, the biased free-\/energy landscape will be flat and barrierless. This potential acts as an \char`\"{}umbrella\char`\"{} that helps you to safely cross the transition state in spite of its high free energy.

It is however difficult to have an a priori guess of the free-\/energy landscape. We will see later how adaptive techniques such as metadynamics (\hyperlink{belfast-6}{Belfast tutorial\+: Metadynamics}) can be used to this aim. Because of this reason, umbrella sampling is often used in a slightly different manner.

Imagine that you do not know the exact height of the free-\/energy barrier but you have an idea of where the barrier is located. You could try to just favor the sampling of the transition state by adding a harmonic restraint on the C\+V, e.\+g. in the form \[ V(s)=\frac{k}{2} (s-s_0)^2 \]. The sampled distribution will be \[ P'(q)\propto P(q) e^{\frac{-k(s(q)-s_0)^2}{2k_BT}} \] For large values of $ k $, only points close to $ s_0 $ will be explored. It is thus clear how one can force the system to explore only a predefined region of the space adding such a restraint. By combining simulations performed with different values of $ s_0 $, one could obtain a continuous set of simulations going from one minimum to the other crossing the transition state. In the next section we will see how to combine the information from these simulations.



If you want to just bring a collective variables to a specific value, you can use a simple restraint. Let's imagine that we want to force the $\Phi$ angle to visit a region close to $\Phi=\pi/2$. We can do it adding a restraint in $\Phi$, with the following input \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
res: RESTRAINT ARG=phi AT=0.5pi KAPPA=5
PRINT ARG=phi,psi,res.bias
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

Notice that here we are printing a quantity named {\ttfamily res.\+bias}. We do this because \hyperlink{RESTRAINT}{R\+E\+S\+T\+R\+A\+I\+N\+T} does not define a single value (that here would be theoretically named {\ttfamily res}) but a structure with several components. All biasing methods (including \hyperlink{METAD}{M\+E\+T\+A\+D}) do so, as well as many collective variables (see e.\+g. \hyperlink{DISTANCE}{D\+I\+S\+T\+A\+N\+C\+E} used with C\+O\+M\+P\+O\+N\+E\+N\+T\+S keyword). Printing the bias allows one to know how much a given snapshop was penalized. Also notice that P\+L\+U\+M\+E\+D understands numbers in the for {\ttfamily \{number\}pi}. This is convenient when using torsions, since they are expressed in radians.

Now you can plot your trajectory with gnuplot and see the effect of K\+A\+P\+P\+A. You can also try different values of K\+A\+P\+P\+A. The stiffer the restraint, the less the collective variable will fluctuate. However, notice that a too large kappa could make the M\+D integrator unstable.

 \hypertarget{munster_munster-biasing-moving}{}\subsubsection{Moving restraints}\label{munster_munster-biasing-moving}
A restraint can also be modified as a function of time. For example, if you want to bring the system from one minimum to the other, you can use a moving restraint on $\Phi$\+: \begin{DoxyVerb}phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
# notice that a long line can be splitted with this syntax
MOVINGRESTRAINT ...
# also notice that a LABEL keyword can be used and is equivalent
# to adding the name at the beginning of the line with colon, as we did so far
  LABEL=res
  ARG=phi
  STEP0=0 AT0=-0.5pi KAPPA0=5
  STEP1=10000 AT0=0.5pi 
...
PRINT ARG=phi,psi,res.work,res.phi_cntr FILE=colvar
\end{DoxyVerb}
 (see \hyperlink{TORSION}{T\+O\+R\+S\+I\+O\+N}, \hyperlink{MOVINGRESTRAINT}{M\+O\+V\+I\+N\+G\+R\+E\+S\+T\+R\+A\+I\+N\+T}, and \hyperlink{PRINT}{P\+R\+I\+N\+T}).

Notice that here we are plotting a few new components, namely {\ttfamily work} and {\ttfamily phi\+\_\+cntr}. The former gives the work performed in pulling the restraint, and the latter the position of the restraint. Notice that if pulling is slow enough one can compute free energy profile from the work. You can plot the putative free-\/energy landscape with \begin{DoxyVerb}> gnuplot
# column 5 is res.phi_cntr
# column 4 is res.work
gnuplot> p "colvar" u 5:4
\end{DoxyVerb}


\hypertarget{munster_munster-multi}{}\subsubsection{Using multiple replicas}\label{munster_munster-multi}
\begin{DoxyWarning}{Warning}
Notice that multireplica simulations with P\+L\+U\+M\+E\+D are fully supported with G\+R\+O\+M\+A\+C\+S, but only partly supported with other M\+D engines.
\end{DoxyWarning}
Some free-\/energy methods are intrinsically parallel and requires running several simultaneous simulations. This can be done with gromacs using the multi replica framework. That is, if you have 4 tpr files named topol0.\+tpr, topol1.\+tpr, topol2.\+tpr, topol3.\+tpr you can run 4 simultaneous simulations. \begin{DoxyVerb}> mpirun -np 4 mdrun_mpi -s topol.tpr -plumed plumed.dat -multi 4 -nsteps 500000
\end{DoxyVerb}
 Each of the 4 replicas will open a different topol file, and G\+R\+O\+M\+A\+C\+S will take care of adding the replica number before the .tpr suffix. P\+L\+U\+M\+E\+D deals with the extra number in a slightly different way. In this case, for example, P\+L\+U\+M\+E\+D first look for a file named {\ttfamily plumed.\+dat.\+X}, where X is the number of the replica. In case the file is not found, then P\+L\+U\+M\+E\+D looks for {\ttfamily plumed.\+dat}. If also this is not found, P\+L\+U\+M\+E\+D will complain. As a consequence, if all the replicas should use the same input file it is sufficient to put a single {\ttfamily plumed.\+dat} file, but one has also the flexibility of using separate files named {\ttfamily plumed.\+dat.\+0}, {\ttfamily plumed.\+dat.\+1} etc. Finally, notice that the way P\+L\+U\+M\+E\+D adds suffixes will change in version 2.\+2, and names will be {\ttfamily plumed.\+0.\+dat} etc.

Also notice that providing the flag {\ttfamily -\/replex} one can instruct gromacs to perform a replica exchange simulation. Namely, from time to time gromacs will try to swap coordinates among neighboring replicas and accept of reject the exchange with a Monte Carlo procedure which also takes into account the bias potentials acting on the replicas, even if different bias potentials are used in different replicas. That is, P\+L\+U\+M\+E\+D allows to easily implement many forms of Hamiltonian replica exchange.\hypertarget{munster_munster-multi-wham}{}\subsubsection{Using multiple restraints with replica exchange}\label{munster_munster-multi-wham}
 \hypertarget{munster_munster-wham-theory}{}\paragraph{Weighted histogram analysis method theory}\label{munster_munster-wham-theory}
Let's now consider multiple simulations performed with restraints located in different positions. In particular, we will consider the $i$-\/th bias potential as $V_i$. The probability to observe a given value of the collective variable $s$ is\+: \[ P_i({s})=\frac{P({s})e^{-\frac{V_i({s})}{k_BT}}}{\int ds' P({s}') e^{-\frac{V_i({s}')}{k_BT}}}= \frac{P({s})e^{-\frac{V_i({s})}{k_BT}}}{Z_i} \] where \[ Z_i=\sum_{q}e^{-\left(U(q)+V_i(q)\right)} \] The likelyhood for the observation of a sequence of snapshots $q_i(t)$ (where $i$ is the index of the trajectory and $t$ is time) is just the product of the probability of each of the snapshots. We use here the minus-\/logarithm of the likelihood (so that the product is converted to a sum) that can be written as \[ \mathcal{L}=-\sum_i \int dt \log P_i({s}_i(t))= \sum_i \int dt \left( -\log P({s}_i(t)) +\frac{V_i({s}_i(t))}{k_BT} +\log Z_i \right) \] One can then maximize the likelyhood by setting $\frac{\delta \mathcal{L}}{\delta P({\bf s})}=0$. After some boring algebra the following expression can be obtained \[ 0=\sum_{i}\int dt\left(-\frac{\delta_{{\bf s}_{i}(t),{\bf s}}}{P({\bf s})}+\frac{e^{-\frac{V_{i}({\bf s})}{k_{B}T}}}{Z_{i}}\right) \] In this equation we aim at finding $P(s)$. However, also the list of normalization factors $Z_i$ is unknown, and they should be found selfconsistently. Thus one can find the solution as \[ P({\bf s})\propto \frac{N({\bf s})}{\sum_i\int dt\frac{e^{-\frac{V_{i}({\bf s})}{k_{B}T}}}{Z_{i}} } \] where $Z$ is selfconsistently determined as \[ Z_i\propto\int ds' P({\bf s}') e^{-\frac{V_i({\bf s}')}{k_BT}} \]

These are the W\+H\+A\+M equations that are traditionally solved to derive the unbiased probability $P(s)$ by the combination of multiple restrained simulations. To make a slightly more general implementation, one can compute the weights that should be assigned to each snapshot, that turn out to be\+: \[ w_i(t)\propto \frac{1}{\sum_j\int dt\frac{e^{-\beta V_{j}({\bf s}_i(t))}}{Z_{j}} } \] The normalization factors can in turn be found from the weights as \[ Z_i\propto\frac{\sum_j \int dt e^{-\beta V_i({\bf s}_j(t))} w_j(t)}{ \sum_j \int dt w_j(t)} \]

This allows to straighforwardly compute averages related to other, non-\/biased degrees of freedom, and it is thus a bit more flexible. It is sufficient to precompute this factors $w$ and use them to weight every single frame in the trajectory.

\hypertarget{munster_munster-exercise-4}{}\paragraph{Exercise 4}\label{munster_munster-exercise-4}
In this exercise we will run multiple restraint simulations and learn how to reweight and combine data with W\+H\+A\+M to obtain free-\/energy profiles. We start with running in a replica-\/exchange scheme 32 simulations with a restraint on phi in different positions, ranging from -\/3 to 3. We will instruct gromacs to attempt an exchange between different simulations every 1000 steps.

\begin{DoxyVerb}nrep=32
dx=`echo "6.0 / ( $nrep - 1 )" | bc -l`

for((i=0;i<nrep;i++))
do
# center of the restraint
AT=`echo "$i * $dx - 3.0" | bc -l`

cat >plumed.dat.$i << EOF
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
#
# Impose an umbrella potential on phi
# with a spring constant of 200 kjoule/mol
# and centered in phi=AT
#
restraint-phi: RESTRAINT ARG=phi KAPPA=200.0 AT=$AT
# monitor the two variables and the bias potential
PRINT STRIDE=100 ARG=phi,psi,restraint-phi.bias FILE=COLVAR
EOF

# we initialize some replicas in A and some in B:
if((i%2==0)); then
  cp ../TOPO/topolA.tpr topol$i.tpr
else
  cp ../TOPO/topolB.tpr topol$i.tpr
fi
done

# run REM
mpirun -np $nrep mdrun_mpi -plumed plumed.dat -s topol.tpr -multi $nrep -replex 1000 -nsteps 500000
\end{DoxyVerb}


To be able to combine data from all the simulations, it is necessary to have an overlap between statistics collected in two adjacent umbrellas. Have a look at the plot of (phi,psi) for the different simulations to understand what is happening.

\label{munster_munster-usrem-phi-all}%
\hypertarget{munster_munster-usrem-phi-all}{}%
 An often misunderstood fact about W\+H\+A\+M is that data of the different trajectories can be mixed and it is not necessary to keep track of which restraint was used to produce every single frame. Let's get the concatenated trajectory

\begin{DoxyVerb}trjcat_mpi -cat -f traj*.xtc -o alltraj.xtc
\end{DoxyVerb}


Now we should compute the value of each of the bias potentials on the entire (concatenated) trajectory.

\begin{DoxyVerb}nrep=32
dx=`echo "6.0 / ( $nrep - 1 )" | bc -l`

for i in `seq 0 $(( $nrep - 1 ))`
do
# center of the restraint
AT=`echo "$i * $dx - 3.0" | bc -l`

cat >plumed.dat << EOF
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17
restraint-phi: RESTRAINT ARG=phi KAPPA=200.0 AT=$AT

# monitor the two variables and the bias potential
PRINT STRIDE=100 ARG=phi,psi,restraint-phi.bias FILE=ALLCOLVAR.$i
EOF

plumed driver --mf_xtc alltraj.xtc --trajectory-stride=10 --plumed plumed.dat

done
\end{DoxyVerb}


It is very important that this script is consistent with the one used to generate the multiple simulations above. Now, single files named A\+L\+L\+C\+O\+L\+V\+A\+R.\+X\+X will contain on the fourth column the value of the bias centered in a given position, computed on the entire concatenated trajectory.

Next step is to compute the weights self-\/consistently solving the W\+H\+A\+M equations, using the python script \char`\"{}wham.\+py\char`\"{} contained in the S\+C\+R\+I\+P\+T\+S directory. To use this code\+:

\begin{DoxyVerb}../SCRIPTS/wham.sh ALLCOLVAR.*
\end{DoxyVerb}


This script will produce several files. Let's visualize \char`\"{}phi\+\_\+fes.\+dat\char`\"{}, which contains the free energy as a function of phi, and compare this with the result previously obtained with metadynamics.

\label{munster_munster-usrem-phi-fes}%
\hypertarget{munster_munster-usrem-phi-fes}{}%
\hypertarget{munster_munster-exercise-5}{}\paragraph{Exercise 5}\label{munster_munster-exercise-5}
In the previous exercise, we use multiple restraint simulations to calculate the free energy as a function of the dihedral phi. The resulting free energy was in excellent agreement with our previous metadynamics simulation. In this exercise we will repeat the same procedure for the dihedral psi. At the end of the steps defined above, we can plot the free energy \char`\"{}psi\+\_\+fes.\+dat\char`\"{} and compare it with the reference profile calculated from a metadynamics simulations using both phi and psi as C\+Vs.

\label{munster_munster-usrem-psi-fes}%
\hypertarget{munster_munster-usrem-psi-fes}{}%
 We can easily spot from the plot above that something went wrong in this multiple restraint simulations, despite we used the very same approach we adopted for the phi dihedral. The problem here is that psi is a \char`\"{}bad\char`\"{} collective variable, and the system is not able to equilibrate the missing slow degree of freedom phi in the short time scale of the umbrella simulation (1 ns). In the metadynamics exercise in which we biased only psi, we detect problems by observing the behavior of the C\+V as a function of simulation time. How can we detect problems in multiple restraint simulations? This is slightly more complicated, but running this kind of simulation in a replica-\/exchange scheme offers a convenient way to detect problems.

The first thing we need to do is to demux the replica-\/exchange trajectories and reconstruct the continous trajectories of the replicas across the different restraint potentials. In order to do so, we can use the following script\+:

\begin{DoxyVerb}demux.pl md0.log
trjcat_mpi -f traj*.xtc  -demux replica_index.xvg
\end{DoxyVerb}


This commands will generate 32 continous trajectories, named X\+X\+\_\+trajout.\+xtc. We will use the driver to calculate the value of the C\+Vs phi and psi on these trajectories.

\begin{DoxyVerb}nrep=32

for i in `seq 0 $(( $nrep - 1 ))`
do

cat >plumed.dat << EOF
phi: TORSION ATOMS=5,7,9,15
psi: TORSION ATOMS=7,9,15,17

# monitor the two variables
PRINT STRIDE=100 ARG=phi,psi FILE=COLVARDEMUX.$i
EOF

plumed driver --mf_xtc ${i}_trajout.xtc --trajectory-stride=10 --plumed plumed.dat

done
\end{DoxyVerb}


The C\+O\+L\+V\+A\+R\+D\+E\+M\+U\+X.\+X\+X files will contain the value of the C\+Vs on the demuxed trajectory. If we visualize these files we will notice that replicas sample the C\+Vs space differently. In order for each umbrella to equilibrate the slow degrees of freedom phi, the continuous replicas must be ergodic and thus sample the same distribution in phi and psi.

\label{munster_munster-usrem-psi-demux}%
\hypertarget{munster_munster-usrem-psi-demux}{}%
